<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-07-13" />

<title>Tumor-intrinsic mutations</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Tumor-intrinsic mutations</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-07-13</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190613code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190613)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190613code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190613)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreee2b04164414d529fe354c85445de58e0329486c7targetblanke2b0416a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/e2b04164414d529fe354c85445de58e0329486c7" target="_blank">e2b0416</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreee2b04164414d529fe354c85445de58e0329486c7targetblanke2b0416a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    NCI_mutations_vs_deconvolution.csv
    Ignored:    data/.DS_Store
    Ignored:    data/1-s2.0-S1074761316304320-mmc2.xlsx
    Ignored:    data/12864_2005_498_MOESM2_ESM.xls
    Ignored:    data/41591_2018_45_MOESM5_ESM.xlsx
    Ignored:    data/CCLE_sample_info_file_2012-10-18.txt
    Ignored:    data/NIH_PDL1_amp_cases.csv
    Ignored:    data/NIH_PDL1_nonamp_cases.csv
    Ignored:    data/PDL1dlbcl.rnaseq.kallisto.raw.txi.coding.cpm1.TMM.logCPM.sm24.csv
    Ignored:    data/PanImmune_GeneSet_Definitions.xlsx
    Ignored:    data/aliquot.tsv
    Ignored:    data/azizi_curated.xlsx
    Ignored:    data/c2_human_immune_genesets.gmt
    Ignored:    data/dlbcl_total_immune_gset_v1_results.rds
    Ignored:    data/ds_JCO.18.01583-2.xlsx
    Ignored:    data/duke_study_driver_mutations_top150_maf.csv
    Ignored:    data/final_analysis_set.maf
    Ignored:    data/gdc_clinical_data.json
    Ignored:    data/gdc_files_and_case_ids.json
    Ignored:    data/gencode.v22.annotation.gtf
    Ignored:    data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Ignored:    data/genesets/
    Ignored:    data/gset_ids_complete.rds
    Ignored:    data/histo_metadata_dlbcl_landscape.csv
    Ignored:    data/journal.pone.0088309.s001.XLS
    Ignored:    data/mmc1.xlsx
    Ignored:    data/nejmoa1801445_appendix_2.xlsx
    Ignored:    data/reactome_imm_signaling.tsv
    Ignored:    data/sample.tsv
    Ignored:    data/sig.rda
    Ignored:    data/sig_genes.txt
    Ignored:    data/temp.dgelist_edger.rds
    Ignored:    data/temp.dgelist_limma.rds
    Ignored:    data/uchi_metadata.csv
    Ignored:    logs/
    Ignored:    output/.DS_Store
    Ignored:    output/NCI_mutations_vs_deconvolution.csv
    Ignored:    output/clust1_vs_others.txt
    Ignored:    output/clust1_vs_others_dge.rnk
    Ignored:    output/clust2_vs_others_dge.rnk
    Ignored:    output/clust3_vs_others_dge.rnk
    Ignored:    output/clust4_vs_others_dge.rnk
    Ignored:    output/cluster1_dge.rds
    Ignored:    output/cluster2_dge.rds
    Ignored:    output/cluster3_dge.rds
    Ignored:    output/cluster4_dge.rds
    Ignored:    output/combined_clin_data.xlsx
    Ignored:    output/combined_dlbcl_expr_matrix.csv
    Ignored:    output/combined_es_gsva2_4cluster.rds
    Ignored:    output/combined_es_gsva4_4cluster.rds
    Ignored:    output/combined_expressionset.rds
    Ignored:    output/combined_tpm_matrix.csv
    Ignored:    output/counts_expressionset_for_limma.rds
    Ignored:    output/dlbcl_expr_matrix.csv
    Ignored:    output/dlbcl_total_immune_gset_v2_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v3_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v4_results.rds
    Ignored:    output/duke_expression_set_cleaned_log.rds
    Ignored:    output/duke_expressionset.rds
    Ignored:    output/duke_expressionset_pext.rds
    Ignored:    output/duke_extended_pheno_data.csv
    Ignored:    output/expr_matrix.csv
    Ignored:    output/gsea_res_cluster1.csv
    Ignored:    output/gset_ids_complete_2.rds
    Ignored:    output/gset_ids_complete_3.rds
    Ignored:    output/gset_ids_complete_4.rds
    Ignored:    output/gsva_v2_4clust_assign.rds
    Ignored:    output/gsva_v3_4clust_assign.rds
    Ignored:    output/gsva_v4_4clust_assign.rds
    Ignored:    output/limma_voom_elist_1.rds
    Ignored:    output/merged_raw_counts.csv
    Ignored:    output/nci_dlbcl_annotation.csv
    Ignored:    output/nci_dlbcl_unprocessed_counts.csv
    Ignored:    output/nci_expressionset.rds
    Ignored:    output/nci_expressionset_pext.rds
    Ignored:    output/nci_extended_pheno_data.csv
    Ignored:    output/networkanal3_clust1.txt
    Ignored:    output/networkanal3_clust2.txt
    Ignored:    output/networkanal3_clust3.txt
    Ignored:    output/networkanal3_clust4.txt
    Ignored:    output/total_immune_gset_v1.rds
    Ignored:    output/tpm_cyt_expressionset.rds
    Ignored:    output/tpm_expressionset.rds
    Ignored:    output/uchi_cluster_assignments.csv
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.pvals.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019_RAW.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.pvals.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819_RAW.txt
    Ignored:    output/xcell_input_matrix.csv

Untracked files:
    Untracked:  NCI_mutations_vs_deconvolution.xls
    Untracked:  analysis/correlation_clusters.Rmd
    Untracked:  analysis/mif.Rmd
    Untracked:  analysis/new_signature_matrix.Rmd
    Untracked:  analysis/table1.Rmd
    Untracked:  analysis/visualizing_gsva_scores_5.Rmd
    Untracked:  batch1_cell_counts.pdf
    Untracked:  data/cibersort/
    Untracked:  data/gistic_results_nci/
    Untracked:  es_10k_hgnc/
    Untracked:  output/ASH_ORAL_figures.png
    Untracked:  output/ASH_ORAL_figures/
    Untracked:  output/Rplot.pdf
    Untracked:  output/Rplot01.pdf
    Untracked:  output/batch1_cell_seg_df.rds
    Untracked:  output/batch1_summary_cell_seg_df.rds
    Untracked:  output/colored_hclust_plot_by_group.pdf
    Untracked:  output/colored_hclust_plot_by_samples.pdf
    Untracked:  output/combined_counts_dge.rds
    Untracked:  output/combined_expr_matrix_for_signature_unprocessed.rmd
    Untracked:  output/dgelist_pheno_data.rds
    Untracked:  output/dlbcl_final_signature_matrix.tsv
    Untracked:  output/dlbcl_gsva5_4cluster_hcpc.rds
    Untracked:  output/dlbcl_gsva5_4cluster_pca.rds
    Untracked:  output/dlbcl_uchi_batch1_mixture.tsv
    Untracked:  output/duke_dlbcl_mixture.tsv
    Untracked:  output/gse_69030_probes.tsv
    Untracked:  output/hclust_labeled_by_cell_type.pdf
    Untracked:  output/mif_pilot_data_plot.pdf
    Untracked:  output/nci_dlbcl_mixture.tsv
    Untracked:  output/phenodata_for_signature_matrix.rmd
    Untracked:  output/reference_matrix_file.tsv
    Untracked:  output/table1.html
    Untracked:  output/uchi_deconvolution_batch_1.rds
    Untracked:  output/uchi_pheno_clust
    Untracked:  output/var_contribution_axis2.pdf
    Untracked:  output/venn.png
    Untracked:  output/venn.png.2020-02-23_11-17-47.log
    Untracked:  output/venn.png.2020-02-23_11-17-56.log
    Untracked:  output/venn.png.2020-02-23_11-20-44.log
    Untracked:  preprocessed_data_for_signature_matrix_es.rds

Unstaged changes:
    Modified:   .gitignore
    Deleted:    Rplot.pdf
    Deleted:    Rplot001.png
    Deleted:    Rplot002.png
    Deleted:    Rplot01.pdf
    Modified:   analysis/cyt.Rmd
    Modified:   analysis/deconvolution.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/predictor.Rmd
    Modified:   analysis/sparse_clustering.Rmd
    Modified:   analysis/visualizing_gsva_scores_4.Rmd
    Deleted:    data/DLBCL_histo/JG55/HE/14-1464/14-1464_Scan1_annotations.xml.lock
    Modified:   output/design_for_limma.rds
    Modified:   output/dgelist_for_limma.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/e2b04164414d529fe354c85445de58e0329486c7/analysis/assign_mutations.Rmd" target="_blank">e2b0416</a>
</td>
<td>
mleukam
</td>
<td>
2020-05-09
</td>
<td>
wflow_publish(c(“analysis/about.Rmd”, “analysis/abstract_figure.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/a78a45cffe9d483012683d912ca3ab8f19d2000b/analysis/assign_mutations.Rmd" target="_blank">a78a45c</a>
</td>
<td>
mleukam
</td>
<td>
2019-10-12
</td>
<td>
archives dump from local machine - near final gsva v4 cluster model
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/ea9871093a36139d500684b2618f89164552dc76/analysis/assign_mutations.Rmd" target="_blank">ea98710</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-19
</td>
<td>
record keeping push - no html made yet
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/6eb3aa2331bda38f2c5361c0d7105cc737d189fd/docs/assign_mutations.html" target="_blank">6eb3aa2</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/d372d0ac0400710cfcfe3f4ee00a2f7709c7d07d/analysis/assign_mutations.Rmd" target="_blank">d372d0a</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-15
</td>
<td>
new notebooks, first commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Clear the workspace</p>
<pre class="r"><code>rm(list = ls())</code></pre>
<pre class="r"><code>knitr::opts_chunk$set(eval = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(maftools)
library(tidyverse)
library(RColorBrewer)
library(knitr)
library(ggpubr)
library(ggsci)
library(Biobase)
library(BiocGenerics)
library(parallel)</code></pre>
</div>
<div id="read-in-and-clean-data" class="section level2">
<h2>Read in and clean data</h2>
<p>Read in NCI mutation data</p>
<pre class="r"><code># read in significance rank list from MutSig2CV
mutsig_siggenes &lt;- read_delim(&quot;data/sig_genes.txt&quot;, &quot;\t&quot;)

# read in final analysis MAF from MutSig2CV
final_analysis_set &lt;- read_delim(
  &quot;data/final_analysis_set.maf&quot;, &quot;\t&quot;)

# repair nonstandard variant classification
final_analysis_set$Variant_Classification &lt;- str_replace(final_analysis_set$Variant_Classification, &quot;Start_Codon_Del&quot;, &quot;Translation_Start_Site&quot;)

final_analysis_set$Variant_Classification &lt;- str_replace(final_analysis_set$Variant_Classification, &quot;Stop_Codon_Ins&quot;, &quot;Nonstop_Mutation&quot;)

final_analysis_set &lt;- as.data.frame(final_analysis_set)

nci_mut_by_sample &lt;- final_analysis_set %&gt;%
  dplyr::select(sample_id = Tumor_Sample_Barcode, Hugo_Symbol) %&gt;%
  mutate(Hugo_Symbol = ifelse(Hugo_Symbol == &quot;MLL2&quot;, &quot;KMT2D&quot;, Hugo_Symbol)) %&gt;%
  as_tibble() %&gt;%
  print()</code></pre>
<p>Read in Duke mutation data</p>
<pre class="r"><code>duke_mut_raw &lt;- read_csv(&quot;data/duke_study_driver_mutations_top150_maf.csv&quot;)

duke_mut_by_sample &lt;- duke_mut_raw %&gt;%
  dplyr::select(Tumor_Sample_Barcode, Hugo_Symbol) %&gt;%
  tidyr::separate(Tumor_Sample_Barcode, 
                  into = c(NA, NA, &quot;sample_id&quot;), 
                  sep = &quot;_&quot;)  %&gt;%
  mutate(sample_id = paste0(&quot;duke_&quot;, sample_id)) %&gt;%
  arrange(sample_id) %&gt;%
  as_tibble() %&gt;%
  mutate(Hugo_Symbol = ifelse(Hugo_Symbol == &quot;MLL2&quot;, &quot;KMT2D&quot;, Hugo_Symbol)) %&gt;%
  print()</code></pre>
<div id="combine-mutation-data" class="section level4">
<h4>Combine mutation data</h4>
<pre class="r"><code>combined_mut &lt;- nci_mut_by_sample %&gt;%
  bind_rows(duke_mut_by_sample) %&gt;%
  print()

# remove duplicate mutations in the same patient and same gene
unique_mut &lt;- combined_mut %&gt;%
  distinct(sample_id, Hugo_Symbol, .keep_all = TRUE) %&gt;%
  print()

nrow(combined_mut)
nrow(unique_mut)</code></pre>
</div>
<div id="read-in-deconvolution-data" class="section level4">
<h4>Read in deconvolution data</h4>
<pre class="r"><code>nci_decon &lt;- read_csv(&quot;data/cibersort/NCI/abs_cell_number_estimate_022320/CIBERSORTx_Job2_Results.csv&quot;)

duke_decon &lt;- read_csv(&quot;data/cibersort/Duke/abs_cell_number_estimate_022320/CIBERSORTx_Job3_Results.csv&quot;)

combined_decon &lt;- nci_decon %&gt;%
  bind_rows(duke_decon) %&gt;%
  print()</code></pre>
<p>Read in cluster assignment and combine</p>
<pre class="r"><code>gsva4_es &lt;- readRDS(&quot;output/combined_es_gsva4_4cluster.rds&quot;)

clust_assgn &lt;- pData(gsva4_es) %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  dplyr::select(sample_id, clust) %&gt;%
  as_tibble() %&gt;%
  print()

# get the total n
clust_assgn %&gt;% distinct(sample_id) %&gt;% nrow()

mut_clust &lt;- unique_mut %&gt;%
  left_join(clust_assgn) %&gt;%
  print()</code></pre>
</div>
<div id="mutation-counts-by-cluster" class="section level4">
<h4>Mutation counts by cluster</h4>
<p>Get mutation counts per gene for each cluster and combine back to a single table</p>
<pre class="r"><code># get number of cases in the cluster
mut_clust %&gt;% dplyr::filter(clust == &quot;1&quot;) %&gt;% distinct(sample_id) %&gt;% nrow()
mut_clust %&gt;% dplyr::filter(clust == &quot;2&quot;) %&gt;% distinct(sample_id) %&gt;% nrow()
mut_clust %&gt;% dplyr::filter(clust == &quot;3&quot;) %&gt;% distinct(sample_id) %&gt;% nrow()
mut_clust %&gt;% dplyr::filter(clust == &quot;4&quot;) %&gt;% distinct(sample_id) %&gt;% nrow()

# get counts by gene for clusters and compare each cluster for significance against pooled counts from each other cluster

# first get mutation counts for each gene per cluster and create column for counts of unmutated genes
cluster_1 &lt;- mut_clust %&gt;%
  dplyr::filter(clust == &quot;1&quot;) %&gt;%
  dplyr::select(-clust) %&gt;%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %&gt;%
  count(Hugo_Symbol) %&gt;%
  dplyr::rename(c1_mut = n) %&gt;%
  mutate(c1_nonmut = 352 - c1_mut) %&gt;%
  print()

cluster_2 &lt;- mut_clust %&gt;%
  dplyr::filter(clust == &quot;2&quot;) %&gt;%
  dplyr::select(-clust) %&gt;%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %&gt;%
  count(Hugo_Symbol) %&gt;%
  dplyr::rename(c2_mut = n) %&gt;%
  mutate(c2_nonmut = 277 - c2_mut) %&gt;%
  print() 

cluster_3 &lt;- mut_clust %&gt;%
  dplyr::filter(clust == &quot;3&quot;) %&gt;%
  dplyr::select(-clust) %&gt;%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %&gt;%
  count(Hugo_Symbol) %&gt;%
  dplyr::rename(c3_mut = n) %&gt;%
  mutate(c3_nonmut = 222 - c3_mut) %&gt;%
  print() 

cluster_4 &lt;- mut_clust %&gt;%
  dplyr::filter(clust == &quot;4&quot;) %&gt;%
  dplyr::select(-clust) %&gt;%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %&gt;%
  count(Hugo_Symbol) %&gt;%
  dplyr::rename(c4_mut = n) %&gt;%
  mutate(c4_nonmut = 317 - c4_mut) %&gt;%
  print() 

mut_counts_by_cluster &lt;- cluster_1 %&gt;%
  full_join(cluster_2) %&gt;%
  full_join(cluster_3) %&gt;%
  full_join(cluster_4) %&gt;%
  replace_na(list(c1_mut = 0, c2_mut = 0, c3_mut = 0, c4_mut = 0)) %&gt;%
  replace_na(list(c1_nonmut = 0, c2_nonmut = 0, c3_nonmut = 0, c4_nonmut = 0)) %&gt;%
  arrange(desc(c4_mut)) %&gt;%
  print()

mut_counts_matrix &lt;- as.data.frame(mut_counts_by_cluster) %&gt;%
  column_to_rownames(var = &quot;Hugo_Symbol&quot;) %&gt;%
  as.matrix()

mut_counts_matrix[1:5, ]
mut_counts &lt;- mut_counts_matrix[, c(1,3,5,7)]
  
# filter for genes without any meaningful number of total mutations
# filter out genes that have a mutant allele fraction &lt; 0.05
cutoff &lt;- 1189 * 0.05

mut_counts_filtered &lt;- mut_counts_by_cluster %&gt;%
  mutate(mut_totals = c1_mut + c2_mut + c3_mut + c4_mut) %&gt;%
  filter(mut_totals &gt; cutoff) %&gt;%
  print()</code></pre>
</div>
</div>
<div id="test-pairwise-significance" class="section level2">
<h2>Test pairwise significance</h2>
<p>Significance testing per for each cluster vs pool of other clusters</p>
<p><a href="https://stackoverflow.com/questions/41507230/fisher-test-on-multiple-data-entries-and-extract-results-in-separate-table/41512086#41512086?newreg=fbae75b1bdbc4796b925c53ca1e2f132" class="uri">https://stackoverflow.com/questions/41507230/fisher-test-on-multiple-data-entries-and-extract-results-in-separate-table/41512086#41512086?newreg=fbae75b1bdbc4796b925c53ca1e2f132</a></p>
<p><a href="https://blog.rstudio.com/2016/02/02/tidyr-0-4-0/" class="uri">https://blog.rstudio.com/2016/02/02/tidyr-0-4-0/</a></p>
<p><a href="https://jennybc.github.io/purrr-tutorial/index.html" class="uri">https://jennybc.github.io/purrr-tutorial/index.html</a></p>
<pre class="r"><code>clust1_vs_pool &lt;- mut_counts_filtered %&gt;%
  mutate(pool_mut = c2_mut + c3_mut + c4_mut) %&gt;%
  mutate(pool_nonmut = c2_nonmut + c3_nonmut + c4_nonmut) %&gt;%
  dplyr::select(Hugo_Symbol, c1_mut, c1_nonmut, pool_mut, pool_nonmut) %&gt;%
  print()

sig &lt;- clust1_vs_pool %&gt;%
  nest(-Hugo_Symbol) %&gt;%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %&gt;% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %&gt;%
  mutate(stats = map(fisher, ~broom::glance(.x))) %&gt;%
  unnest(stats) %&gt;%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %&gt;%
  arrange(p.value)

clust_1_summary &lt;- sig %&gt;%
  left_join(clust1_vs_pool) %&gt;%
  mutate(p_adj = p.adjust(p.value, method = &quot;BH&quot;)) %&gt;%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %&gt;%
  mutate(c1_rate = c1_mut / (c1_mut + c1_nonmut)) %&gt;%
  print()

clust_1_plotdata &lt;- clust_1_summary %&gt;%
  mutate(pool_rate = pool_mut / pool_nonmut) %&gt;%
  select(Hugo_Symbol, c1_rate, pool_rate, p_adj) %&gt;%
  pivot_longer(cols = c(c1_rate, pool_rate),
               names_to = &quot;c1_vs_pool&quot;, 
               values_to = &quot;mut_rate&quot;) %&gt;%
  dplyr::filter(p_adj &lt; 0.26)
clust_1_plotdata

clust_1_plot &lt;- ggplot(data = clust_1_plotdata,
                       aes(x = Hugo_Symbol, y = mut_rate, group = c1_vs_pool, fill = c1_vs_pool)) +
  geom_col(position = &quot;dodge&quot;)
clust_1_plot</code></pre>
<pre class="r"><code>clust2_vs_pool &lt;- mut_counts_filtered %&gt;%
  mutate(pool_mut = c1_mut + c3_mut + c4_mut) %&gt;%
  mutate(pool_nonmut = c1_nonmut + c3_nonmut + c4_nonmut) %&gt;%
  dplyr::select(Hugo_Symbol, c2_mut, c2_nonmut, pool_mut, pool_nonmut) %&gt;%
  print()

sig &lt;- clust2_vs_pool %&gt;%
  nest(-Hugo_Symbol) %&gt;%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %&gt;% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %&gt;%
  mutate(stats = map(fisher, ~broom::glance(.x))) %&gt;%
  unnest(stats) %&gt;%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %&gt;%
  arrange(p.value)

clust_2_summary &lt;- sig %&gt;%
  left_join(clust2_vs_pool) %&gt;%
  mutate(p_adj = p.adjust(p.value, method = &quot;BH&quot;)) %&gt;%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %&gt;%
  print()</code></pre>
<pre class="r"><code>clust3_vs_pool &lt;- mut_counts_filtered %&gt;%
  mutate(pool_mut = c1_mut + c2_mut + c4_mut) %&gt;%
  mutate(pool_nonmut = c1_nonmut + c2_nonmut + c4_nonmut) %&gt;%
  dplyr::select(Hugo_Symbol, c3_mut, c3_nonmut, pool_mut, pool_nonmut) %&gt;%
  print()

sig &lt;- clust3_vs_pool %&gt;%
  nest(-Hugo_Symbol) %&gt;%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %&gt;% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %&gt;%
  mutate(stats = map(fisher, ~broom::glance(.x))) %&gt;%
  unnest(stats) %&gt;%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %&gt;%
  arrange(p.value)

clust_3_summary &lt;- sig %&gt;%
  left_join(clust3_vs_pool) %&gt;%
  mutate(p_adj = p.adjust(p.value, method = &quot;BH&quot;)) %&gt;%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %&gt;%
  print()</code></pre>
<pre class="r"><code>clust4_vs_pool &lt;- mut_counts_filtered %&gt;%
  mutate(pool_mut = c1_mut + c2_mut + c3_mut) %&gt;%
  mutate(pool_nonmut = c1_nonmut + c2_nonmut + c3_nonmut) %&gt;%
  dplyr::select(Hugo_Symbol, c4_mut, c4_nonmut, pool_mut, pool_nonmut) %&gt;%
  print()

sig &lt;- clust4_vs_pool %&gt;%
  nest(-Hugo_Symbol) %&gt;%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %&gt;% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %&gt;%
  mutate(stats = map(fisher, ~broom::glance(.x))) %&gt;%
  unnest(stats) %&gt;%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %&gt;%
  arrange(p.value)

clust_4_summary &lt;- sig %&gt;%
  left_join(clust4_vs_pool) %&gt;%
  mutate(p_adj = p.adjust(p.value, method = &quot;BH&quot;)) %&gt;%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %&gt;%
  print()</code></pre>
</div>
<div id="test-pairwise-mutations-of-pre-selected-genes" class="section level2">
<h2>Test pairwise mutations of pre-selected genes</h2>
<pre class="r"><code># EZH2 across all clusters

# get column of mutants by cluster
ezh2_mut &lt;- mut_counts_by_cluster %&gt;% 
  dplyr::filter(Hugo_Symbol %in% c(&quot;EZH2&quot;)) %&gt;%
  dplyr::select(c1 = c1_mut, c2 = c2_mut, c3 = c3_mut, c4 = c4_mut, -Hugo_Symbol) %&gt;%
  as.matrix() %&gt;%
  t() %&gt;%
  print()

colnames(ezh2_mut) &lt;- &quot;ezh2_mut&quot;
ezh2_mut

# get column of non-mutants by cluster
ezh2_nonmut &lt;- mut_counts_by_cluster %&gt;% 
  dplyr::filter(Hugo_Symbol %in% c(&quot;EZH2&quot;)) %&gt;%
  dplyr::select(c1 = c1_nonmut, c2 = c2_nonmut, c3 = c3_nonmut, c4 = c4_nonmut, -Hugo_Symbol) %&gt;%
  as.matrix() %&gt;%
  t() %&gt;%
  print()

colnames(ezh2_nonmut) &lt;- &quot;ezh2_nonmut&quot;
ezh2_nonmut

# combine
ezh2 &lt;- cbind(ezh2_mut, ezh2_nonmut)
ezh2

# test significance
fisher.test(ezh2, simulate.p.value = TRUE)

# TNFAIP3 across all clusters

# get column of mutants by cluster
tnfaip3_mut &lt;- mut_counts_by_cluster %&gt;% 
  dplyr::filter(Hugo_Symbol %in% c(&quot;TNFAIP3&quot;)) %&gt;%
  dplyr::select(c1 = c1_mut, c2 = c2_mut, c3 = c3_mut, c4 = c4_mut, -Hugo_Symbol) %&gt;%
  as.matrix() %&gt;%
  t() %&gt;%
  print()

colnames(tnfaip3_mut) &lt;- &quot;tnfaip3_mut&quot;
tnfaip3_mut

# get column of non-mutants by cluster
tnfaip3_nonmut &lt;- mut_counts_by_cluster %&gt;% 
  dplyr::filter(Hugo_Symbol %in% c(&quot;TNFAIP3&quot;)) %&gt;%
  dplyr::select(c1 = c1_nonmut, c2 = c2_nonmut, c3 = c3_nonmut, c4 = c4_nonmut, -Hugo_Symbol) %&gt;%
  as.matrix() %&gt;%
  t() %&gt;%
  print()

colnames(tnfaip3_nonmut) &lt;- &quot;tnfaip3_nonmut&quot;
tnfaip3_nonmut

# combine
tnfaip3 &lt;- cbind(tnfaip3_mut, tnfaip3_nonmut)
tnfaip3

# test significance
fisher.test(tnfaip3, simulate.p.value = TRUE)

# get column of mutants by cluster
ezh2_mut &lt;- mut_counts_by_cluster %&gt;% 
  dplyr::filter(Hugo_Symbol %in% c(&quot;EZH2&quot;)) %&gt;%
  dplyr::select(c1 = c1_mut, c2 = c2_mut, c3 = c3_mut, c4 = c4_mut, -Hugo_Symbol) %&gt;%
  as.matrix() %&gt;%
  t() %&gt;%
  print()

colnames(ezh2_mut) &lt;- &quot;ezh2_mut&quot;
ezh2_mut

# get column of non-mutants by cluster
ezh2_nonmut &lt;- mut_counts_by_cluster %&gt;% 
  dplyr::filter(Hugo_Symbol %in% c(&quot;EZH2&quot;)) %&gt;%
  dplyr::select(c1 = c1_nonmut, c2 = c2_nonmut, c3 = c3_nonmut, c4 = c4_nonmut, -Hugo_Symbol) %&gt;%
  as.matrix() %&gt;%
  t() %&gt;%
  print()

colnames(ezh2_nonmut) &lt;- &quot;ezh2_nonmut&quot;
ezh2_nonmut

# combine
ezh2 &lt;- cbind(ezh2_mut, ezh2_nonmut)
ezh2

# test significance
fisher.test(ezh2, simulate.p.value = TRUE)</code></pre>
<pre class="r"><code># Test significance in hottest and coldest clusters compared to each other
clust4_vs_clust1 &lt;- mut_counts_filtered %&gt;%
  dplyr::select(Hugo_Symbol, c4_mut, c4_nonmut, c1_mut, c1_nonmut) %&gt;%
  print()

sig &lt;- clust4_vs_clust1 %&gt;%
  nest(-Hugo_Symbol) %&gt;%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %&gt;% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %&gt;%
  mutate(stats = map(fisher, ~broom::glance(.x))) %&gt;%
  unnest(stats) %&gt;%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %&gt;%
  arrange(p.value)

clust4_clust1_summary &lt;- sig %&gt;%
  left_join(clust4_vs_clust1) %&gt;%
  mutate(p_adj = p.adjust(p.value, method = &quot;BH&quot;)) %&gt;%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %&gt;%
  arrange(p_adj) %&gt;%
  print()

clust_14_plotdata &lt;- clust4_clust1_summary %&gt;%
  mutate(c4_rate = c4_mut / (c4_nonmut + c4_mut)) %&gt;%
  mutate(c1_rate = c1_mut / (c1_nonmut + c1_mut) ) %&gt;%
  mutate(difference = (c4_rate - c1_rate)^2) %&gt;%
  arrange(desc(difference)) %&gt;%
  slice(1:10) %&gt;%
  select(Hugo_Symbol, c1_rate, c4_rate, difference) %&gt;%
  mutate(Hugo_Symbol = as_factor(Hugo_Symbol)) %&gt;%
  mutate(Hugo_Symbol = fct_reorder(Hugo_Symbol, difference, .desc = TRUE)) %&gt;%
  pivot_longer(cols = c(c1_rate, c4_rate),
               names_to = &quot;c1_vs_c4&quot;, 
               values_to = &quot;mut_rate&quot;) %&gt;%
  mutate(c1_vs_c4 = as.factor(c1_vs_c4)) %&gt;%
  mutate(c1_vs_c4 = fct_recode(c1_vs_c4, &quot;Inflamed&quot; = &quot;c4_rate&quot;,
                               &quot;Cold&quot; = &quot;c1_rate&quot;)) %&gt;%
  mutate(c1_vs_c4 = fct_relevel(c1_vs_c4, &quot;Cold&quot;, &quot;Inflamed&quot;)) %&gt;%
  print()

clust_14_plot &lt;- ggplot(data = clust_14_plotdata,
                       aes(x = Hugo_Symbol, 
                           y = mut_rate, 
                           group = c1_vs_c4, 
                           fill = c1_vs_c4)) +
  geom_col(position = &quot;dodge&quot;) +
  theme_classic2() +
  scale_fill_d3(name = &quot;Cluster&quot;) +
  xlab(&quot;&quot;) +
  ylab(&quot;Mutation Rate&quot;)
clust_14_plot

# Test significance in intermedite clusters compared to each other
clust2_vs_clust3 &lt;- mut_counts_filtered %&gt;%
  dplyr::select(Hugo_Symbol, c2_mut, c2_nonmut, c3_mut, c3_nonmut) %&gt;%
  print()

sig &lt;- clust2_vs_clust3 %&gt;%
  nest(-Hugo_Symbol) %&gt;%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %&gt;% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %&gt;%
  mutate(stats = map(fisher, ~broom::glance(.x))) %&gt;%
  unnest(stats) %&gt;%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %&gt;%
  arrange(p.value)

clust2_clust3_summary &lt;- sig %&gt;%
  left_join(clust2_vs_clust3) %&gt;%
  mutate(p_adj = p.adjust(p.value, method = &quot;BH&quot;)) %&gt;%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %&gt;%
  print()

clust_23_plotdata &lt;- clust2_clust3_summary %&gt;%
  mutate(c2_rate = c2_mut / (c2_nonmut + c2_mut) ) %&gt;%
  mutate(c3_rate = c3_mut / (c3_nonmut + c3_mut) ) %&gt;%
  mutate(difference = (c3_rate - c2_rate)^2) %&gt;%
  arrange(desc(difference)) %&gt;%
  slice(1:10) %&gt;%
  select(Hugo_Symbol, c2_rate, c3_rate, difference) %&gt;%
  mutate(Hugo_Symbol = as_factor(Hugo_Symbol)) %&gt;%
  mutate(Hugo_Symbol = fct_reorder(Hugo_Symbol, difference, .desc = TRUE)) %&gt;%
  pivot_longer(cols = c(c2_rate, c3_rate),
               names_to = &quot;c2_vs_c3&quot;, 
               values_to = &quot;mut_rate&quot;) %&gt;%
  mutate(c2_vs_c3 = as.factor(c2_vs_c3)) %&gt;%
  mutate(c2_vs_c3 = fct_recode(c2_vs_c3, &quot;Intermediate-T&quot; = &quot;c2_rate&quot;,
                               &quot;Intermediate-M&quot; = &quot;c3_rate&quot;)) %&gt;%
  mutate(c2_vs_c3 = fct_relevel(c2_vs_c3, &quot;Intermediate-T&quot;, &quot;Intermediate-M&quot;)) %&gt;%
  print()

clust_23_plot &lt;- ggplot(data = clust_23_plotdata,
                       aes(x = Hugo_Symbol, 
                           y = mut_rate, 
                           group = c2_vs_c3, 
                           fill = c2_vs_c3)) +
  geom_col(position = &quot;dodge&quot;) +
  theme_classic2() +
  scale_fill_d3(name = &quot;Cluster&quot;) +
  xlab(&quot;&quot;) +
  ylab(&quot;Mutation Rate&quot;)
clust_23_plot</code></pre>
</div>
<div id="gprofiler-input" class="section level2">
<h2>g:Profiler input</h2>
<p>Get list of genes mutated over allele fraction of 0.05 for each individual cluster for input into g:profiler ordered enrichment test</p>
<p>Rank by pairwise p-value and select only those significant at p &lt; 0.05 in univariate analysis for further investigation. Rank by mutant allele frequency to use with g:Profiler “ordered query”</p>
<pre class="r"><code>cluster_1_maf &lt;- mut_clust %&gt;%
  dplyr::filter(clust == &quot;1&quot;) %&gt;%
  dplyr::select(-clust) %&gt;%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %&gt;%
  count(Hugo_Symbol) %&gt;%
  dplyr::rename(c1_mut = n) %&gt;%
  mutate(MAF = c1_mut/352) %&gt;%
  arrange(desc(MAF)) %&gt;%
  left_join(mut_counts_by_cluster) %&gt;%
  # filter out MAF with at least 1% MAF
  dplyr::filter(MAF &gt; 0.01) %&gt;%
  print()
  
clust1_vs_pool &lt;- cluster_1_maf %&gt;%
  mutate(pool_mut = c2_mut + c3_mut + c4_mut) %&gt;%
  mutate(pool_nonmut = c2_nonmut + c3_nonmut + c4_nonmut) %&gt;%
  dplyr::select(Hugo_Symbol, c1_mut, c1_nonmut, pool_mut, pool_nonmut) %&gt;%
  print()

clust_1_ranked &lt;- clust1_vs_pool %&gt;%
  nest(data = c(c1_mut, c1_nonmut, pool_mut, pool_nonmut)) %&gt;%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %&gt;% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %&gt;%
  mutate(stats = map(fisher, ~broom::glance(.x))) %&gt;%
  unnest(stats) %&gt;%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %&gt;%
  # dplyr::filter(p.value &lt; 0.05) %&gt;%
  left_join(cluster_1_maf) %&gt;%
  arrange(desc(MAF)) %&gt;%
  dplyr::select(Hugo_Symbol) %&gt;%
  print()

#-----------------------------------
cluster_2_maf &lt;- mut_clust %&gt;%
  dplyr::filter(clust == &quot;2&quot;) %&gt;%
  dplyr::select(-clust) %&gt;%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %&gt;%
  count(Hugo_Symbol) %&gt;%
  dplyr::rename(c2_mut = n) %&gt;%
  mutate(MAF = c2_mut/277) %&gt;%
  arrange(desc(MAF)) %&gt;%
  left_join(mut_counts_by_cluster) %&gt;%
  # filter out MAF with at least 1% MAF
  dplyr::filter(MAF &gt; 0.01) %&gt;%
  print()
  
clust2_vs_pool &lt;- cluster_2_maf %&gt;%
  mutate(pool_mut = c1_mut + c3_mut + c4_mut) %&gt;%
  mutate(pool_nonmut = c1_nonmut + c3_nonmut + c4_nonmut) %&gt;%
  dplyr::select(Hugo_Symbol, c2_mut, c2_nonmut, pool_mut, pool_nonmut) %&gt;%
  print()

clust_2_ranked &lt;- clust2_vs_pool %&gt;%
  nest(data = c(c2_mut, c2_nonmut, pool_mut, pool_nonmut)) %&gt;%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %&gt;% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %&gt;%
  mutate(stats = map(fisher, ~broom::glance(.x))) %&gt;%
  unnest(stats) %&gt;%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %&gt;%
  # dplyr::filter(p.value &lt; 0.05) %&gt;%
  left_join(cluster_2_maf) %&gt;%
  arrange(desc(MAF)) %&gt;%
  dplyr::select(Hugo_Symbol) %&gt;%
  print()

#-----------------------------------
cluster_3_maf &lt;- mut_clust %&gt;%
  dplyr::filter(clust == &quot;3&quot;) %&gt;%
  dplyr::select(-clust) %&gt;%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %&gt;%
  count(Hugo_Symbol) %&gt;%
  dplyr::rename(c3_mut = n) %&gt;%
  mutate(MAF = c3_mut/222) %&gt;%
  arrange(desc(MAF)) %&gt;%
  left_join(mut_counts_by_cluster) %&gt;%
  # filter out MAF with at least 1% MAF
  dplyr::filter(MAF &gt; 0.01) %&gt;%
  print()
  
clust3_vs_pool &lt;- cluster_3_maf %&gt;%
  mutate(pool_mut = c2_mut + c1_mut + c4_mut) %&gt;%
  mutate(pool_nonmut = c2_nonmut + c1_nonmut + c4_nonmut) %&gt;%
  dplyr::select(Hugo_Symbol, c3_mut, c3_nonmut, pool_mut, pool_nonmut) %&gt;%
  print()

clust_3_ranked &lt;- clust3_vs_pool %&gt;%
  nest(data = c(c3_mut, c3_nonmut, pool_mut, pool_nonmut)) %&gt;%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %&gt;% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %&gt;%
  mutate(stats = map(fisher, ~broom::glance(.x))) %&gt;%
  unnest(stats) %&gt;%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %&gt;%
  # dplyr::filter(p.value &lt; 0.05) %&gt;%
  left_join(cluster_3_maf) %&gt;%
  arrange(desc(MAF)) %&gt;%
  dplyr::select(Hugo_Symbol) %&gt;%
  print()

#-----------------------------------
cluster_4_maf &lt;- mut_clust %&gt;%
  dplyr::filter(clust == &quot;4&quot;) %&gt;%
  dplyr::select(-clust) %&gt;%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %&gt;%
  count(Hugo_Symbol) %&gt;%
  dplyr::rename(c4_mut = n) %&gt;%
  mutate(MAF = c4_mut/317) %&gt;%
  arrange(desc(MAF)) %&gt;%
  left_join(mut_counts_by_cluster) %&gt;%
  # filter out MAF with at least 1% MAF
  dplyr::filter(MAF &gt; 0.01) %&gt;%
  print()
  
clust4_vs_pool &lt;- cluster_4_maf %&gt;%
  mutate(pool_mut = c2_mut + c1_mut + c3_mut) %&gt;%
  mutate(pool_nonmut = c2_nonmut + c1_nonmut + c3_nonmut) %&gt;%
  dplyr::select(Hugo_Symbol, c4_mut, c4_nonmut, pool_mut, pool_nonmut) %&gt;%
  print()

clust_4_ranked &lt;- clust4_vs_pool %&gt;%
  nest(data = c(c4_mut, c4_nonmut, pool_mut, pool_nonmut)) %&gt;%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %&gt;% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %&gt;%
  mutate(stats = map(fisher, ~broom::glance(.x))) %&gt;%
  unnest(stats) %&gt;%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %&gt;%
  # dplyr::filter(p.value &lt; 0.05) %&gt;%
  left_join(cluster_4_maf) %&gt;%
  arrange(desc(MAF)) %&gt;%
  dplyr::select(Hugo_Symbol) %&gt;%
  print()

#-----------------------------------

mutations &lt;- list(c1 = pull(clust_1_ranked, Hugo_Symbol), c2 = pull(clust_2_ranked, Hugo_Symbol), c3 = pull(clust_3_ranked, Hugo_Symbol), c4 = pull(clust_4_ranked, Hugo_Symbol))

library(gprofiler2)
gostres &lt;- gost(query = mutations,
                organism = &quot;hsapiens&quot;, 
                ordered_query = TRUE, 
                multi_query = TRUE, 
                significant = TRUE, 
                exclude_iea = FALSE, 
                measure_underrepresentation = FALSE, 
                evcodes = FALSE, 
                user_threshold = 0.05, 
                correction_method = &quot;g_SCS&quot;, 
                domain_scope = &quot;annotated&quot;, 
                custom_bg = NULL, 
                numeric_ns = &quot;&quot;, 
                sources = NULL, 
                as_short_link = TRUE)
gostres</code></pre>
</div>
<div id="display-mutation-density-versus-pca-from-gsva-v4v5" class="section level2">
<h2>Display mutation density versus PCA from GSVA v4/v5</h2>
<pre class="r"><code>dlbcl_pca &lt;- readRDS(&quot;output/dlbcl_gsva5_4cluster_pca.rds&quot;)
dlbcl_hcpc &lt;- readRDS(&quot;output/dlbcl_gsva5_4cluster_hcpc.rds&quot;)

hcpc_clust &lt;- dlbcl_hcpc$data.clust %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  dplyr::select(sample_id, clust)

# get coordinates for each sample and merge with total mutation data
# merge with cluster assignment
pca_ind_coord &lt;- dlbcl_pca$ind$coord %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  left_join(unique_mut) %&gt;%
  left_join(hcpc_clust) %&gt;%
  print()

pca_ind_coord %&gt;%
  group_by(Hugo_Symbol) %&gt;%
  summarize(mean_dim1 = mean(Dim.1), n = n()) %&gt;%
  dplyr::filter(n &gt; 25) %&gt;%
  arrange(desc(mean_dim1))

pca_ind_coord %&gt;%
  group_by(Hugo_Symbol) %&gt;%
  summarize(mean_dim2 = mean(Dim.2), n = n()) %&gt;%
  dplyr::filter(n &gt; 25) %&gt;%
  arrange(desc(mean_dim2))</code></pre>
<p>MYD88</p>
<pre class="r"><code>pca_ind_coord %&gt;%
  dplyr::filter(Hugo_Symbol == &quot;MYD88&quot;) %&gt;%
  ggplot(aes(x = Dim.1, y = Dim.2)) + 
    geom_bin2d(bins = 8) +
    scale_fill_continuous(type = &quot;viridis&quot;) +
    theme_classic2()

pca_ind_coord %&gt;%
  dplyr::filter(Hugo_Symbol == &quot;MYD88&quot;) %&gt;%
  ggplot(aes(x = Dim.1)) +
  geom_density() +
  geom_vline(xintercept = 0) +
  geom_vline(aes(xintercept = mean(Dim.1)), 
             color = &quot;blue&quot;, 
             linetype = &quot;dashed&quot;, 
             size = 1)</code></pre>
<p>MYC</p>
<pre class="r"><code>pca_ind_coord %&gt;%
  dplyr::filter(Hugo_Symbol == &quot;MYC&quot;) %&gt;%
  ggplot(aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 12) +
    scale_fill_continuous(type = &quot;viridis&quot;) +
    theme_classic2()

pca_ind_coord %&gt;%
  dplyr::filter(Hugo_Symbol == &quot;EZH2&quot;) %&gt;%
  ggplot(aes(x = Dim.1)) +
  geom_density() +
  geom_vline(xintercept = 0) +
  geom_vline(aes(xintercept = mean(Dim.1)), 
             color = &quot;blue&quot;, 
             linetype = &quot;dashed&quot;, 
             size = 1) +
  theme_classic2() +
  ylab(&quot;Mutation frequency&quot;)</code></pre>
<p>TP53</p>
<pre class="r"><code>tp53_ind_coord &lt;- pca_ind_coord %&gt;%
  dplyr::filter(Hugo_Symbol == &quot;TP53&quot;)

ggplot(data = tp53_ind_coord,
       aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 12) +
    scale_fill_continuous(type = &quot;viridis&quot;) +
    theme_classic2()

ggplot(data = tp53_ind_coord,
       aes(x = Dim.1)) +
  geom_density() +
  geom_vline(xintercept = 0) +
  geom_vline(data = tp53_ind_coord,
             aes(xintercept = mean(Dim.1)), 
             color = &quot;blue&quot;, 
             linetype = &quot;dashed&quot;, 
             size = 1) +
  theme_classic2() +
  ylab(&quot;Mutation frequency&quot;) +
  geom_density(data = tp53_ind_coord, aes(x = Dim.2)) +
  geom_vline(data = tp53_ind_coord,
             aes(xintercept = mean(Dim.1)), 
             color = &quot;red&quot;, 
             linetype = &quot;dashed&quot;, 
             size = 1)</code></pre>
<p>SOCS1</p>
<pre class="r"><code>socs1_ind_coord &lt;- pca_ind_coord %&gt;%
  dplyr::filter(Hugo_Symbol == &quot;SOCS1&quot;)

foxo1_ind_coord &lt;- pca_ind_coord %&gt;%
  dplyr::filter(Hugo_Symbol == &quot;FOXO1&quot;)

ggplot(data = socs1_ind_coord,
       aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 12) +
    scale_fill_continuous(type = &quot;viridis&quot;) +
    theme_classic2()

ggplot(data = foxo1_ind_coord,
       aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 10) +
    scale_fill_continuous(type = &quot;viridis&quot;) +
    theme_classic2()

ggplot(data = socs1_ind_coord,
       aes(x = Dim.1)) +
  geom_density(color = &quot;blue&quot;) +
  geom_vline(xintercept = 0) +
  geom_vline(data = socs1_ind_coord,
             aes(xintercept = mean(Dim.1)), 
             color = &quot;blue&quot;, 
             linetype = &quot;dashed&quot;, 
             size = 1) +
  theme_classic2() +
  ylab(&quot;Mutation frequency&quot;) +
  geom_density(data = foxo1_ind_coord, aes(x = Dim.1), color = &quot;red&quot;) +
  geom_vline(data = foxo1_ind_coord,
             aes(xintercept = mean(Dim.1)), 
             color = &quot;red&quot;, 
             linetype = &quot;dashed&quot;, 
             size = 1)

jak1_ind_coord &lt;- pca_ind_coord %&gt;%
  dplyr::filter(Hugo_Symbol == &quot;JAK1&quot;)

s1pr2_ind_coord &lt;- pca_ind_coord %&gt;%
  dplyr::filter(Hugo_Symbol == &quot;S1PR2&quot;)

ggplot(data = jak1_ind_coord,
       aes(x = Dim.2)) +
  geom_density(color = &quot;red&quot;) +
  geom_vline(xintercept = 0) +
  geom_vline(data = jak1_ind_coord,
             aes(xintercept = mean(Dim.2)), 
             color = &quot;red&quot;, 
             linetype = &quot;dashed&quot;, 
             size = 1) +
  theme_classic2() +
  ylab(&quot;Mutation frequency&quot;) +
  geom_density(data = s1pr2_ind_coord, aes(x = Dim.2), color = &quot;blue&quot;) +
  geom_vline(data = s1pr2_ind_coord,
             aes(xintercept = mean(Dim.2)), 
             color = &quot;blue&quot;, 
             linetype = &quot;dashed&quot;, 
             size = 1)

ggplot(data = s1pr2_ind_coord,
       aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 5) +
    scale_fill_continuous(type = &quot;viridis&quot;) +
    theme_classic2() +
    geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, color = &quot;grey&quot;, size = 1)

ggplot(data = jak1_ind_coord,
       aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 5) +
    scale_fill_continuous(type = &quot;viridis&quot;) +
    theme_classic2() +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, color = &quot;grey&quot;, size = 1)</code></pre>
<p>SOCS1, oncogene or tumor suppressor: <a href="https://www.sciencedirect.com/science/article/abs/pii/S1043466616300059" class="uri">https://www.sciencedirect.com/science/article/abs/pii/S1043466616300059</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
