<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-07-15" />

<title>Differential Gene Expression by Cluster</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential Gene Expression by Cluster</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-07-15</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190613code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190613)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190613code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190613)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreee2b04164414d529fe354c85445de58e0329486c7targetblanke2b0416a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/e2b04164414d529fe354c85445de58e0329486c7" target="_blank">e2b0416</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreee2b04164414d529fe354c85445de58e0329486c7targetblanke2b0416a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    NCI_mutations_vs_deconvolution.csv
    Ignored:    data/.DS_Store
    Ignored:    data/1-s2.0-S1074761316304320-mmc2.xlsx
    Ignored:    data/12864_2005_498_MOESM2_ESM.xls
    Ignored:    data/41591_2018_45_MOESM5_ESM.xlsx
    Ignored:    data/CCLE_sample_info_file_2012-10-18.txt
    Ignored:    data/NIH_PDL1_amp_cases.csv
    Ignored:    data/NIH_PDL1_nonamp_cases.csv
    Ignored:    data/PDL1dlbcl.rnaseq.kallisto.raw.txi.coding.cpm1.TMM.logCPM.sm24.csv
    Ignored:    data/PanImmune_GeneSet_Definitions.xlsx
    Ignored:    data/aliquot.tsv
    Ignored:    data/azizi_curated.xlsx
    Ignored:    data/c2_human_immune_genesets.gmt
    Ignored:    data/dlbcl_total_immune_gset_v1_results.rds
    Ignored:    data/ds_JCO.18.01583-2.xlsx
    Ignored:    data/duke_study_driver_mutations_top150_maf.csv
    Ignored:    data/final_analysis_set.maf
    Ignored:    data/gdc_clinical_data.json
    Ignored:    data/gdc_files_and_case_ids.json
    Ignored:    data/gencode.v22.annotation.gtf
    Ignored:    data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Ignored:    data/genesets/
    Ignored:    data/gset_ids_complete.rds
    Ignored:    data/histo_metadata_dlbcl_landscape.csv
    Ignored:    data/journal.pone.0088309.s001.XLS
    Ignored:    data/mmc1.xlsx
    Ignored:    data/nejmoa1801445_appendix_2.xlsx
    Ignored:    data/reactome_imm_signaling.tsv
    Ignored:    data/sample.tsv
    Ignored:    data/sig.rda
    Ignored:    data/sig_genes.txt
    Ignored:    data/temp.dgelist_edger.rds
    Ignored:    data/temp.dgelist_limma.rds
    Ignored:    data/uchi_metadata.csv
    Ignored:    logs/
    Ignored:    output/.DS_Store
    Ignored:    output/NCI_mutations_vs_deconvolution.csv
    Ignored:    output/clust1_vs_others.txt
    Ignored:    output/clust1_vs_others_dge.rnk
    Ignored:    output/clust2_vs_others_dge.rnk
    Ignored:    output/clust3_vs_others_dge.rnk
    Ignored:    output/clust4_vs_others_dge.rnk
    Ignored:    output/cluster1_dge.rds
    Ignored:    output/cluster2_dge.rds
    Ignored:    output/cluster3_dge.rds
    Ignored:    output/cluster4_dge.rds
    Ignored:    output/combined_clin_data.xlsx
    Ignored:    output/combined_dlbcl_expr_matrix.csv
    Ignored:    output/combined_es_gsva2_4cluster.rds
    Ignored:    output/combined_es_gsva4_4cluster.rds
    Ignored:    output/combined_expressionset.rds
    Ignored:    output/combined_tpm_matrix.csv
    Ignored:    output/counts_expressionset_for_limma.rds
    Ignored:    output/dlbcl_expr_matrix.csv
    Ignored:    output/dlbcl_total_immune_gset_v2_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v3_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v4_results.rds
    Ignored:    output/duke_expression_set_cleaned_log.rds
    Ignored:    output/duke_expressionset.rds
    Ignored:    output/duke_expressionset_pext.rds
    Ignored:    output/duke_extended_pheno_data.csv
    Ignored:    output/expr_matrix.csv
    Ignored:    output/gsea_res_cluster1.csv
    Ignored:    output/gset_ids_complete_2.rds
    Ignored:    output/gset_ids_complete_3.rds
    Ignored:    output/gset_ids_complete_4.rds
    Ignored:    output/gsva_v2_4clust_assign.rds
    Ignored:    output/gsva_v3_4clust_assign.rds
    Ignored:    output/gsva_v4_4clust_assign.rds
    Ignored:    output/limma_voom_elist_1.rds
    Ignored:    output/merged_raw_counts.csv
    Ignored:    output/nci_dlbcl_annotation.csv
    Ignored:    output/nci_dlbcl_unprocessed_counts.csv
    Ignored:    output/nci_expressionset.rds
    Ignored:    output/nci_expressionset_pext.rds
    Ignored:    output/nci_extended_pheno_data.csv
    Ignored:    output/networkanal3_clust1.txt
    Ignored:    output/networkanal3_clust2.txt
    Ignored:    output/networkanal3_clust3.txt
    Ignored:    output/networkanal3_clust4.txt
    Ignored:    output/total_immune_gset_v1.rds
    Ignored:    output/tpm_cyt_expressionset.rds
    Ignored:    output/tpm_expressionset.rds
    Ignored:    output/uchi_cluster_assignments.csv
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.pvals.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019_RAW.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.pvals.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819_RAW.txt
    Ignored:    output/xcell_input_matrix.csv

Untracked files:
    Untracked:  NCI_mutations_vs_deconvolution.xls
    Untracked:  analysis/correlation_clusters.Rmd
    Untracked:  analysis/mif.Rmd
    Untracked:  analysis/new_signature_matrix.Rmd
    Untracked:  analysis/table1.Rmd
    Untracked:  analysis/visualizing_gsva_scores_5.Rmd
    Untracked:  batch1_cell_counts.pdf
    Untracked:  data/cibersort/
    Untracked:  data/gistic_results_nci/
    Untracked:  es_10k_hgnc/
    Untracked:  output/ASH_ORAL_figures.png
    Untracked:  output/ASH_ORAL_figures/
    Untracked:  output/Rplot.pdf
    Untracked:  output/Rplot01.pdf
    Untracked:  output/batch1_cell_seg_df.rds
    Untracked:  output/batch1_summary_cell_seg_df.rds
    Untracked:  output/colored_hclust_plot_by_group.pdf
    Untracked:  output/colored_hclust_plot_by_samples.pdf
    Untracked:  output/combined_counts_dge.rds
    Untracked:  output/combined_expr_matrix_for_signature_unprocessed.rmd
    Untracked:  output/dgelist_pheno_data.rds
    Untracked:  output/dlbcl_final_signature_matrix.tsv
    Untracked:  output/dlbcl_gsva5_4cluster_hcpc.rds
    Untracked:  output/dlbcl_gsva5_4cluster_pca.rds
    Untracked:  output/dlbcl_uchi_batch1_mixture.tsv
    Untracked:  output/duke_dlbcl_mixture.tsv
    Untracked:  output/gse_69030_probes.tsv
    Untracked:  output/hclust_labeled_by_cell_type.pdf
    Untracked:  output/mif_pilot_data_plot.pdf
    Untracked:  output/nci_dlbcl_mixture.tsv
    Untracked:  output/phenodata_for_signature_matrix.rmd
    Untracked:  output/reference_matrix_file.tsv
    Untracked:  output/table1.html
    Untracked:  output/uchi_deconvolution_batch_1.rds
    Untracked:  output/uchi_pheno_clust
    Untracked:  output/var_contribution_axis2.pdf
    Untracked:  output/venn.png
    Untracked:  output/venn.png.2020-02-23_11-17-47.log
    Untracked:  output/venn.png.2020-02-23_11-17-56.log
    Untracked:  output/venn.png.2020-02-23_11-20-44.log
    Untracked:  preprocessed_data_for_signature_matrix_es.rds

Unstaged changes:
    Modified:   .gitignore
    Deleted:    Rplot.pdf
    Deleted:    Rplot001.png
    Deleted:    Rplot002.png
    Deleted:    Rplot01.pdf
    Modified:   analysis/cyt.Rmd
    Modified:   analysis/deconvolution.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/predictor.Rmd
    Modified:   analysis/sparse_clustering.Rmd
    Modified:   analysis/visualizing_gsva_scores_4.Rmd
    Deleted:    data/DLBCL_histo/JG55/HE/14-1464/14-1464_Scan1_annotations.xml.lock
    Modified:   output/design_for_limma.rds
    Modified:   output/dgelist_for_limma.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/e2b04164414d529fe354c85445de58e0329486c7/analysis/cluster_dge.Rmd" target="_blank">e2b0416</a>
</td>
<td>
mleukam
</td>
<td>
2020-05-09
</td>
<td>
wflow_publish(c(“analysis/about.Rmd”, “analysis/abstract_figure.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/a78a45cffe9d483012683d912ca3ab8f19d2000b/analysis/cluster_dge.Rmd" target="_blank">a78a45c</a>
</td>
<td>
mleukam
</td>
<td>
2019-10-12
</td>
<td>
archives dump from local machine - near final gsva v4 cluster model
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/2960d8b54653f6823b6df05cc0a3cc6f023486cd/analysis/cluster_dge.Rmd" target="_blank">2960d8b</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-18
</td>
<td>
record keeping updates - did not yet build the pages for these notebooks
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/01fb3c107e629cfc19b059394907946e892ab03a/analysis/cluster_dge.Rmd" target="_blank">01fb3c1</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-16
</td>
<td>
updates of analysis pages without building html pages - for record keeping and safekeeping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/6eb3aa2331bda38f2c5361c0d7105cc737d189fd/docs/cluster_dge.html" target="_blank">6eb3aa2</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/d372d0ac0400710cfcfe3f4ee00a2f7709c7d07d/analysis/cluster_dge.Rmd" target="_blank">d372d0a</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-15
</td>
<td>
new notebooks, first commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Clear the workspace</p>
<pre class="r"><code>rm(list = ls())</code></pre>
<pre class="r"><code>knitr::opts_chunk$set(eval = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(&quot;tidyverse&quot;)
library(&quot;Biobase&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggdark&quot;)
library(&quot;ggsci&quot;)
library(&quot;viridis&quot;)
library(&quot;sva&quot;)
library(ggpubr)</code></pre>
<p>Read in data</p>
<pre class="r"><code># Combined duke+nci es with gsva4, 4-cluster model phenodata and log-transformed, normalized read counts
combined_es &lt;- readRDS(&quot;output/combined_es_gsva4_4cluster.rds&quot;)</code></pre>
<p>Setup for limma voom linear modeling following limma user’s manual <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf" class="uri">https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf</a></p>
<p>Chapter 15 RNAseq Data</p>
<p>Recommended steps:</p>
<ol style="list-style-type: decimal">
<li>Start with raw counts (HTSeq or Kallisto for example)</li>
<li>Filter for expression</li>
<li>TMM normalize</li>
<li>Do not log transform</li>
<li>Pass batches (sequencing center) to limma voom with precision weights for batch effect modeling</li>
</ol>
</div>
<div id="nci" class="section level2">
<h2>NCI</h2>
<p>Read in raw counts</p>
<pre class="r"><code>total_counts &lt;- read_csv(&quot;output/nci_dlbcl_unprocessed_counts.csv&quot;)

# read in conversion table for gene IDs
gencode_gtf &lt;- read_tsv(&quot;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&quot;)</code></pre>
<div id="filter-for-protein-coding-genes" class="section level4">
<h4>Filter for protein coding genes</h4>
<pre class="r"><code># filter for protein coding genes
total_counts_prcode &lt;- total_counts %&gt;%
  dplyr::rename(gene_id = gene) %&gt;%
  left_join(gencode_gtf, by = &quot;gene_id&quot;) %&gt;%
  dplyr::filter(gene_type == &quot;protein_coding&quot;) %&gt;%
  dplyr::select(-gene_name, -gene_type, -gene_status, -level, -havana_gene) %&gt;%
  dplyr::select(gene_id, everything())
total_counts_prcode[1:5, 1:5]

nrow(total_counts)
nrow(total_counts_prcode)</code></pre>
</div>
<div id="density-plots-before-expression-filtering" class="section level4">
<h4>Density plots before expression filtering</h4>
<pre class="r"><code>tidy_counts &lt;- total_counts_prcode %&gt;% 
  gather(key = &quot;sampleID&quot;, value = &quot;intensity&quot;, -gene_id) 

dplot1 &lt;- ggplot(tidy_counts, aes(intensity)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;) +
  xlim(-5, 100) +
  ggtitle(&quot;Density plot of counts for NCI DLBCL&quot;) +
  theme_classic2()
dplot1

dplot2 &lt;- ggplot(tidy_counts, aes(log(intensity))) +
  geom_density() +
  theme(legend.position = &quot;none&quot;) +
  xlim(-20, 20) +
  ggtitle(&quot;Density plot of log(counts) for NCI DLBCL&quot;) +
  theme_classic2()
dplot2</code></pre>
</div>
<div id="filter-by-expression-levels" class="section level4">
<h4>Filter by expression levels</h4>
<p>Will use CPM for filtering</p>
<pre class="r"><code># move gene names to rownames
totcounts_prcode_matrix &lt;- total_counts_prcode %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;gene_id&quot;)
totcounts_prcode_matrix[1:5, 1:5]
dim(totcounts_prcode_matrix)

cpm_prcode_matrix &lt;- cpm(totcounts_prcode_matrix, log = FALSE)

# apply hard cutoffs 
# filter out genes that have counts &lt;= 10 in at least half of cases
# count expression cutoff: 10
cutoff &lt;- 1
# set filter fraction
min_sm_frac = 0.5
filter_frac = min_sm_frac * ncol(cpm_prcode_matrix)
filter_frac
total_stats &lt;- data.frame(
  total = apply(cpm_prcode_matrix, 
                1, 
               function(x) {sum(x &gt; cutoff, na.rm = TRUE) } ))
# get indices that meet criteria
keep &lt;- which(total_stats$total &gt;= filter_frac)

# use indices to filter COUNTS MATRIX (not cpm) and check results
dim(totcounts_prcode_matrix)
total_nci_filtered = totcounts_prcode_matrix[keep,]
dim(total_nci_filtered)
total_nci_filtered[1:5, 1:5]</code></pre>
</div>
<div id="density-plots-after-expression-filtering" class="section level4">
<h4>Density plots after expression filtering</h4>
<pre class="r"><code>tidy_counts &lt;- total_nci_filtered %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  gather(key = &quot;sampleID&quot;, value = &quot;counts&quot;, -gene_id) 

dplot1 &lt;- ggplot(tidy_counts, aes(counts)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;) +
  xlim(-5, 100) +
  ggtitle(&quot;Density plot of filtered counts for NCI DLBCL&quot;) +
  theme_classic2()
dplot1

dplot2 &lt;- ggplot(tidy_counts, aes(log(counts))) +
  geom_density() +
  theme(legend.position = &quot;none&quot;) +
  xlim(-20, 20) +
  ggtitle(&quot;Density plot of filtered log(counts) for NCI DLBCL&quot;) +
  dark_theme_gray()
dplot2</code></pre>
</div>
</div>
<div id="duke" class="section level2">
<h2>Duke</h2>
<p>Read in raw counts for Duke samples as dgelist</p>
<pre class="r"><code>dgelist &lt;- readRDS(&quot;data/temp.dgelist_edger.rds&quot;)
str(dgelist)</code></pre>
<p>Convert GENCODE v29 to v22 gene ids to match GDC standard</p>
<pre class="r"><code># get expression matrix
expr_duke &lt;- dgelist[[1]]
expr_duke[1:5, 1:5]
col__names &lt;- colnames(expr_duke)
col_names_rep &lt;- str_split(col__names, pattern = &quot;\\.&quot;, simplify = TRUE) %&gt;% 
  as_tibble() %&gt;%
  pull(V1) %&gt;%
  paste0(&quot;duke_&quot;, .) %&gt;%
  print()
colnames(expr_duke) &lt;- col_names_rep
expr_duke[1:5, 1:5]

# gencode v29 was used to generate rownames of expr_duke matrix
# GDC standard is gencode v22, which is source of NCI data
# will convert to gencode v22 to match rows of NCI data
geneID_split &lt;- str_split(gencode_gtf$gene_id, pattern = &quot;\\.&quot;, simplify = TRUE) %&gt;% 
  as_tibble() %&gt;%
  dplyr::select(geneID_split = V1)

gencode_gtf_split &lt;- gencode_gtf %&gt;% 
  bind_cols(geneID_split) %&gt;%
  dplyr::select(gene_id_v22 = gene_id, geneID_split, everything()) %&gt;%
  print()

row__names &lt;- rownames(expr_duke) %&gt;%
  enframe() %&gt;%
  dplyr::select(geneID_split = value) %&gt;%
  print()

# check how many gene ids are missing in v22 that are named in v29
setdiff(row__names$geneID_split, gencode_gtf_split$geneID_split)
# approx 1900 (previous checks show none after filtering for protein coding genes)

# merge gene name data with expression matrix
expr_duke_df &lt;- expr_duke %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id_v29&quot;) %&gt;%
  as_tibble() %&gt;%
  mutate(geneID_split = str_replace(gene_id_v29, pattern = &quot;\\.*&quot;, &quot;&quot;)) %&gt;%
  dplyr::select(gene_id_v29, geneID_split, everything()) %&gt;%
  left_join(gencode_gtf_split) %&gt;%
  dplyr::select(gene_id_v29, gene_id_v22, geneID_split, gene_type, everything()) %&gt;%
  dplyr::select(-gene_status, -gene_name, -level, -havana_gene) %&gt;%
  print()

# check how many gene ids are missing from version 22  
summary(is.na(expr_duke_df$gene_id_v22))

# check the gene types in the expression matrix
summary(as.factor(expr_duke_df$gene_type))</code></pre>
<div id="filter-for-protein-coding" class="section level4">
<h4>Filter for protein coding</h4>
<pre class="r"><code># filter for protein coding genes only
expr_duke_filtered &lt;- expr_duke_df %&gt;% 
  dplyr::filter(gene_type == &quot;protein_coding&quot;)
  
# any missing gene ids also protein coding genes?
summary(is.na(expr_duke_filtered$gene_id_v22))
# NO

expr_duke_protcoding &lt;- expr_duke_filtered %&gt;%
  dplyr::select(-gene_id_v29, -geneID_split, -gene_type) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;gene_id_v22&quot;) %&gt;%
  as.matrix()

str(expr_duke_protcoding)
expr_duke_protcoding[1:5, 1:5]</code></pre>
</div>
<div id="subset-cases-based-on-protein-coding-library-coverage" class="section level4">
<h4>Subset cases based on protein coding library coverage</h4>
<p>Following recommendation in methods section of Reddy et al Cell 2017 (original data source). They reported including only those libraries containing at least 12,000 protein coding genes.</p>
<pre class="r"><code>expr_duke_protcoding[1:5, 1:5]

gene_coverage_threshold &lt;- 12000

ncol(expr_duke_protcoding)
expr_duke_subset &lt;- expr_duke_protcoding[, colSums(expr_duke_protcoding != 0) &gt; gene_coverage_threshold]

expr_duke_subset[1:5, 1:5]
ncol(expr_duke_subset)</code></pre>
</div>
<div id="initial-density-plots" class="section level4">
<h4>Initial density plots</h4>
<pre class="r"><code># Density plots
# tidy data
tidy_counts &lt;- t(expr_duke_subset) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  gather(key = &quot;gene_id&quot;, value = &quot;counts&quot;, -sample_id) %&gt;%
  print()

# plots
dplot_duke &lt;- ggplot(tidy_counts, aes(counts)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;) +
  dark_theme_gray()
dplot_duke +
  xlim(-5, 100)

# plot with log and small offset
dplot_duke_log &lt;- ggplot(tidy_counts, aes(log(counts + 0.5))) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;) +
  dark_theme_gray()
dplot_duke_log +
  xlim(-10, 10)</code></pre>
</div>
<div id="filtering-by-expression" class="section level4">
<h4>Filtering by expression</h4>
<p>Will use CPM for filtering</p>
<pre class="r"><code>cpm_duke_subset &lt;- cpm(expr_duke_subset, log = FALSE)

# apply hard cutoffs 
# filter out genes that have counts &lt;= 10 in at least half of cases
# count expression cutoff: 10
cutoff &lt;- 1
# set filter fraction
min_sm_frac = 0.5
filter_frac = min_sm_frac * ncol(cpm_duke_subset)
filter_frac
total_stats &lt;- data.frame(
  total = apply(cpm_duke_subset, 
                1, 
               function(x) {sum(x &gt; cutoff, na.rm = TRUE) } ))
# get indices that meet criteria
keep &lt;- which(total_stats$total &gt;= filter_frac)

# use indices to filter COUNTS MATRIX (not cpm) and check results
dim(expr_duke_subset)
total_duke_filtered &lt;- expr_duke_subset[keep,]
dim(total_duke_filtered)
total_duke_filtered[1:5, 1:5]</code></pre>
</div>
<div id="post-filtering-density-plots" class="section level4">
<h4>Post-filtering density plots</h4>
<pre class="r"><code># Density plots
# tidy data
tidy_filtered &lt;- t(total_duke_filtered) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  gather(key = &quot;gene_id&quot;, value = &quot;counts&quot;, -sample_id) %&gt;%
  print()

# plots
dplot_duke_2 &lt;- ggplot(tidy_filtered, aes(counts)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;) +
  dark_theme_gray() +
  ggtitle(&quot;Density plot of Duke raw counts after expression filtering&quot;)
dplot_duke_2 +
  xlim(-5, 100)

# plot with log and small offset
dplot_duke_log_2 &lt;- ggplot(tidy_filtered, aes(log(counts + 0.5))) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;) +
  dark_theme_gray() +
  ggtitle(&quot;Density plot of Duke log(counts) after expression filtering&quot;)
dplot_duke_log_2 +
  xlim(-10, 10)</code></pre>
</div>
</div>
<div id="merge" class="section level2">
<h2>Merge</h2>
<div id="combine-expression-data" class="section level4">
<h4>Combine expression data</h4>
<pre class="r"><code>duke_exprs &lt;- total_duke_filtered %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  dplyr::select(gene_id, everything()) %&gt;%
  print()

nci_exprs &lt;- total_nci_filtered %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  dplyr::select(gene_id, everything()) %&gt;%
  print()

# filter rownames to only keep intersecting rows
duke_genes &lt;- duke_exprs %&gt;% pull(gene_id)
length(duke_genes)
nci_genes &lt;- nci_exprs %&gt;% pull(gene_id)
length(nci_genes)

combined_genes &lt;- intersect(duke_genes, nci_genes)
length(combined_genes)
head(combined_genes)

duke_exprs_filt &lt;- duke_exprs %&gt;%
  dplyr::filter(gene_id %in% combined_genes)
nrow(duke_exprs_filt)

nci_exprs_filt &lt;- nci_exprs %&gt;%
  dplyr::filter(gene_id %in% combined_genes)
nrow(nci_exprs_filt)

combined_exprs_df &lt;- nci_exprs_filt %&gt;%
  left_join(duke_exprs_filt, by = &quot;gene_id&quot;) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;gene_id&quot;)
combined_exprs_df[1:5, 1:5]</code></pre>
</div>
<div id="normalize-gene-expression-distributions" class="section level4">
<h4>Normalize gene expression distributions</h4>
<p>Normalisation by the method of trimmed mean of M-values is performed using the calcNormFactors function in edgeR. The normalisation factors calculated here are used as a scaling factor for the library sizes.</p>
<pre class="r"><code># will do below on the dge
# get normalization factors
# norm_factors &lt;- calcNormFactors(combined_exprs_df, method = &quot;TMM&quot;)

# apply factor to each column - not necessary for limma
# norm_exprs &lt;- map2_dfc(combined_exprs_df, norm_factors, `*`)
# norm_exprs &lt;- as.data.frame(norm_exprs)
# rownames(norm_exprs) &lt;- rownames(combined_exprs)

combined_exprs &lt;- combined_exprs_df %&gt;% as.matrix()
combined_exprs[1:5, 1:5]</code></pre>
</div>
<div id="reformat-pheno-data-to-match" class="section level4">
<h4>Reformat pheno data to match</h4>
<pre class="r"><code>pheno_data_tbl &lt;- pData(combined_es) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  dplyr::select(sample_id, 127:198) %&gt;%
  dplyr::select(-alpha_var) %&gt;%
  as_tibble() %&gt;%
  print()

samplenames &lt;- pheno_data_tbl %&gt;% pull(sample_id)
length(samplenames)
columnnames &lt;- colnames(combined_exprs)
length(columnnames)
identical(samplenames, columnnames)

pheno_data &lt;- pheno_data_tbl %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;sample_id&quot;)</code></pre>
</div>
</div>
<div id="dge-by-cluster" class="section level2">
<h2>DGE by cluster</h2>
<p>The contrast matrix for the differential gene expression will be to compare each cluster to all other clusters)</p>
<p>Contrast matrix bioC forum question: <a href="https://support.bioconductor.org/p/26251/" class="uri">https://support.bioconductor.org/p/26251/</a></p>
<p>General contrast coding: <a href="https://stats.idre.ucla.edu/r/library/r-library-contrast-coding-systems-for-categorical-variables/#DEVIATION" class="uri">https://stats.idre.ucla.edu/r/library/r-library-contrast-coding-systems-for-categorical-variables/#DEVIATION</a></p>
<p>Contrast coding for linear models in R (detailed instructions on custom matrix): <a href="https://stats.stackexchange.com/questions/52252/how-to-specify-a-contrast-matrix-in-r-for-the-difference-between-one-level-and" class="uri">https://stats.stackexchange.com/questions/52252/how-to-specify-a-contrast-matrix-in-r-for-the-difference-between-one-level-and</a></p>
<div id="limma" class="section level4">
<h4>Limma</h4>
<pre class="r"><code># create DGEList
dge &lt;- DGEList(counts = combined_exprs)

# apply TMM normalization to DGE
dge &lt;- calcNormFactors(dge)

# set up design matrix
# include sequencing center as a batch variable to the model
rnames &lt;- colnames(combined_es)
groups &lt;- pData(combined_es) %&gt;% 
  pull(clust) %&gt;%
  paste0(&quot;cluster_&quot;, .) %&gt;%
  as.factor()
batch &lt;- pData(combined_es) %&gt;%
  pull(source) %&gt;%
  as.factor
design &lt;- model.matrix(~ 0 + groups + batch)
rownames(design) &lt;- rnames
head(design)

# write out DGElist and design
saveRDS(dge, &quot;output/dgelist_for_limma.rds&quot;)
saveRDS(pheno_data, &quot;output/dgelist_pheno_data.rds&quot;)
saveRDS(design, &quot;output/design_for_limma.rds&quot;)</code></pre>
<p>Limma voom with quality weights - run on a cluster</p>
<pre class="r"><code>## See code folder in github repo for full script
# read in data
dge &lt;- readRDS(&quot;/gpfs/data/kline-lab/inputs/dgelist_for_limma.rds&quot;)
design &lt;- readRDS(&quot;/gpfs/data/kline-lab/inputs/design_for_limma.rds&quot;)

# voom transformation is applied to the normalized and filtered DGEList object
v &lt;- voomWithQualityWeights(dge, 
                            design, 
                            plot = TRUE, 
                            normalize = &quot;quantile&quot;)

# fit linear model and estimate DGE
fit &lt;- lmFit(v, design)

# write out results
saveRDS &lt;- saveRDS(v, &quot;/gpfs/data/kline-lab/tcga_macs/output/limma_voom_elist_1.rds&quot;)

saveRDS &lt;- saveRDS(fit, &quot;/gpfs/data/kline-lab/tcga_macs/output/limma_voom_marraylmfit.rds&quot;)</code></pre>
<p>Read in results</p>
<pre class="r"><code>v &lt;- readRDS(&quot;output/limma_voom_elist_1.rds&quot;)
fit &lt;- readRDS(&quot;output/limma_voom_marraylmfit.rds&quot;)</code></pre>
<p>DGE Cluster 1</p>
<pre class="r"><code># set up contrast matrix
contr_matrix_c1 &lt;- makeContrasts(&quot;clust_1_vs_others&quot; = groupscluster_1 - (groupscluster_2 + groupscluster_3 + groupscluster_4) / 3, 
  levels = colnames(design))
contr_matrix_c1

# use contrast matrix and eBayes to estimate SEM and pvalues
fit_c1 &lt;- contrasts.fit(fit, contr_matrix_c1)
fit_c1 &lt;- eBayes(fit_c1)

cluster1_dge &lt;- topTable(fit_c1, 
                         adjust.method = &quot;BH&quot;, 
                         sort.by = &quot;none&quot;, 
                         number = Inf)

top50_cluster1_dge_up &lt;- cluster1_dge %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  left_join(gencode_gtf) %&gt;%
  arrange(desc(logFC)) %&gt;%
  dplyr::slice(1:50) %&gt;%
  dplyr::select(gene_name, logFC, adj.P.Val) %&gt;%
  print()

top50_cluster1_dge_down &lt;- cluster1_dge %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  left_join(gencode_gtf) %&gt;%
  arrange(logFC) %&gt;%
  dplyr::slice(1:50) %&gt;%
  dplyr::select(gene_name, logFC, adj.P.Val) %&gt;%
  print()</code></pre>
<p>DGE Cluster 2</p>
<pre class="r"><code># set up contrast matrix
contr_matrix_c2 &lt;- makeContrasts(&quot;clust_1_vs_others&quot; = groupscluster_2 - (groupscluster_1 + groupscluster_3 + groupscluster_4) / 3, 
  levels = colnames(design))
contr_matrix_c2

# use contrast matrix and eBayes to estimate SEM and pvalues
fit_c2 &lt;- contrasts.fit(fit, contr_matrix_c2)
fit_c2 &lt;- eBayes(fit_c2)

cluster2_dge &lt;- topTable(fit_c2, 
                         adjust.method = &quot;BH&quot;, 
                         sort.by = &quot;none&quot;, 
                         number = Inf)

top50_cluster2_dge_up &lt;- cluster2_dge %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  left_join(gencode_gtf) %&gt;%
  arrange(desc(logFC)) %&gt;%
  dplyr::slice(1:50) %&gt;%
  dplyr::select(gene_name, logFC, adj.P.Val) %&gt;%
  print()

top50_cluster2_dge_down &lt;- cluster2_dge %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  left_join(gencode_gtf) %&gt;%
  arrange(logFC) %&gt;%
  dplyr::slice(1:50) %&gt;%
  dplyr::select(gene_name, logFC, adj.P.Val) %&gt;%
  print()</code></pre>
<p>DGE Cluster 3</p>
<pre class="r"><code># set up contrast matrix
contr_matrix_c3 &lt;- makeContrasts(&quot;clust_1_vs_others&quot; = groupscluster_3 - (groupscluster_1 + groupscluster_2 + groupscluster_4) / 3, 
  levels = colnames(design))
contr_matrix_c3

# use contrast matrix and eBayes to estimate SEM and pvalues
fit_c3 &lt;- contrasts.fit(fit, contr_matrix_c3)
fit_c3 &lt;- eBayes(fit_c3)

cluster3_dge &lt;- topTable(fit_c3, 
                         adjust.method = &quot;BH&quot;, 
                         sort.by = &quot;none&quot;, 
                         number = Inf)

top50_cluster3_dge_up &lt;- cluster3_dge %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  left_join(gencode_gtf) %&gt;%
  arrange(desc(logFC)) %&gt;%
  dplyr::slice(1:50) %&gt;%
  dplyr::select(gene_name, logFC, adj.P.Val) %&gt;%
  print()

top50_cluster3_dge_down &lt;- cluster3_dge %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  left_join(gencode_gtf) %&gt;%
  arrange(logFC) %&gt;%
  dplyr::slice(1:50) %&gt;%
  dplyr::select(gene_name, logFC, adj.P.Val) %&gt;%
  print()</code></pre>
<p>DGE Cluster 4</p>
<pre class="r"><code># set up contrast matrix
contr_matrix_c4 &lt;- makeContrasts(&quot;clust_1_vs_others&quot; = groupscluster_4 - (groupscluster_1 + groupscluster_2 + groupscluster_3) / 3, 
  levels = colnames(design))
contr_matrix_c4

# use contrast matrix and eBayes to estimate SEM and pvalues
fit_c4 &lt;- contrasts.fit(fit, contr_matrix_c4)
fit_c4 &lt;- eBayes(fit_c4)

cluster4_dge &lt;- topTable(fit_c4, 
                         adjust.method = &quot;BH&quot;, 
                         sort.by = &quot;none&quot;, 
                         number = Inf)

top50_cluster4_dge_up &lt;- cluster4_dge %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  left_join(gencode_gtf) %&gt;%
  arrange(desc(logFC)) %&gt;%
  dplyr::slice(1:50) %&gt;%
  dplyr::select(gene_name, logFC, adj.P.Val) %&gt;%
  print()

top50_cluster4_dge_down &lt;- cluster4_dge %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  left_join(gencode_gtf) %&gt;%
  arrange(logFC) %&gt;%
  dplyr::slice(1:50) %&gt;%
  dplyr::select(gene_name, logFC, adj.P.Val) %&gt;%
  print()</code></pre>
</div>
</div>
<div id="visualizations" class="section level2">
<h2>Visualizations</h2>
<pre class="r"><code>library(EnhancedVolcano)
## Change to gene names
cluster4_dge_names &lt;- cluster4_dge %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  left_join(gencode_gtf, by = &quot;gene_id&quot;) %&gt;%
  dplyr::distinct(gene_name, .keep_all = TRUE) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;gene_name&quot;) %&gt;%
  dplyr::select(-gene_id, -gene_type, -gene_status, -level, -havana_gene) 

head(cluster4_dge_names)

## Plot
EnhancedVolcano(
  toptable = cluster4_dge_names,
  lab = rownames(cluster4_dge_names),
  x = &quot;logFC&quot;,
  y = &quot;P.Value&quot;,
  pCutoff = 0.05,
  FCcutoff = 1.5, 
  legendPosition = &quot;none&quot;,
  title = &quot;DGE: Inflamed cluster vs all others&quot;,
  transcriptPointSize = 1.5
  
)</code></pre>
</div>
<div id="write-out-data" class="section level2">
<h2>Write out data</h2>
<pre class="r"><code>saveRDS(counts_es, &quot;output/counts_expressionset_for_limma.rds&quot;)</code></pre>
<pre class="r"><code>saveRDS(cluster1_dge, &quot;output/cluster1_dge.rds&quot;)
saveRDS(cluster2_dge, &quot;output/cluster2_dge.rds&quot;)
saveRDS(cluster3_dge, &quot;output/cluster3_dge.rds&quot;)
saveRDS(cluster4_dge, &quot;output/cluster4_dge.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
