<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-07-16" />

<title>Pathway analysis of differentially expressed genes by cluster</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Pathway analysis of differentially expressed genes by cluster</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-07-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190613code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190613)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190613code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190613)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreee2b04164414d529fe354c85445de58e0329486c7targetblanke2b0416a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/e2b04164414d529fe354c85445de58e0329486c7" target="_blank">e2b0416</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreee2b04164414d529fe354c85445de58e0329486c7targetblanke2b0416a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    NCI_mutations_vs_deconvolution.csv
    Ignored:    data/.DS_Store
    Ignored:    data/1-s2.0-S1074761316304320-mmc2.xlsx
    Ignored:    data/12864_2005_498_MOESM2_ESM.xls
    Ignored:    data/41591_2018_45_MOESM5_ESM.xlsx
    Ignored:    data/CCLE_sample_info_file_2012-10-18.txt
    Ignored:    data/NIH_PDL1_amp_cases.csv
    Ignored:    data/NIH_PDL1_nonamp_cases.csv
    Ignored:    data/PDL1dlbcl.rnaseq.kallisto.raw.txi.coding.cpm1.TMM.logCPM.sm24.csv
    Ignored:    data/PanImmune_GeneSet_Definitions.xlsx
    Ignored:    data/aliquot.tsv
    Ignored:    data/azizi_curated.xlsx
    Ignored:    data/c2_human_immune_genesets.gmt
    Ignored:    data/dlbcl_total_immune_gset_v1_results.rds
    Ignored:    data/ds_JCO.18.01583-2.xlsx
    Ignored:    data/duke_study_driver_mutations_top150_maf.csv
    Ignored:    data/final_analysis_set.maf
    Ignored:    data/gdc_clinical_data.json
    Ignored:    data/gdc_files_and_case_ids.json
    Ignored:    data/gencode.v22.annotation.gtf
    Ignored:    data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Ignored:    data/genesets/
    Ignored:    data/gset_ids_complete.rds
    Ignored:    data/histo_metadata_dlbcl_landscape.csv
    Ignored:    data/journal.pone.0088309.s001.XLS
    Ignored:    data/mmc1.xlsx
    Ignored:    data/nejmoa1801445_appendix_2.xlsx
    Ignored:    data/reactome_imm_signaling.tsv
    Ignored:    data/sample.tsv
    Ignored:    data/sig.rda
    Ignored:    data/sig_genes.txt
    Ignored:    data/temp.dgelist_edger.rds
    Ignored:    data/temp.dgelist_limma.rds
    Ignored:    data/uchi_metadata.csv
    Ignored:    logs/
    Ignored:    output/.DS_Store
    Ignored:    output/NCI_mutations_vs_deconvolution.csv
    Ignored:    output/clust1_vs_others.txt
    Ignored:    output/clust1_vs_others_dge.rnk
    Ignored:    output/clust2_vs_others_dge.rnk
    Ignored:    output/clust3_vs_others_dge.rnk
    Ignored:    output/clust4_vs_others_dge.rnk
    Ignored:    output/cluster1_dge.rds
    Ignored:    output/cluster2_dge.rds
    Ignored:    output/cluster3_dge.rds
    Ignored:    output/cluster4_dge.rds
    Ignored:    output/combined_clin_data.xlsx
    Ignored:    output/combined_dlbcl_expr_matrix.csv
    Ignored:    output/combined_es_gsva2_4cluster.rds
    Ignored:    output/combined_es_gsva4_4cluster.rds
    Ignored:    output/combined_expressionset.rds
    Ignored:    output/combined_tpm_matrix.csv
    Ignored:    output/counts_expressionset_for_limma.rds
    Ignored:    output/dlbcl_expr_matrix.csv
    Ignored:    output/dlbcl_total_immune_gset_v2_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v3_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v4_results.rds
    Ignored:    output/duke_expression_set_cleaned_log.rds
    Ignored:    output/duke_expressionset.rds
    Ignored:    output/duke_expressionset_pext.rds
    Ignored:    output/duke_extended_pheno_data.csv
    Ignored:    output/expr_matrix.csv
    Ignored:    output/gsea_res_cluster1.csv
    Ignored:    output/gset_ids_complete_2.rds
    Ignored:    output/gset_ids_complete_3.rds
    Ignored:    output/gset_ids_complete_4.rds
    Ignored:    output/gsva_v2_4clust_assign.rds
    Ignored:    output/gsva_v3_4clust_assign.rds
    Ignored:    output/gsva_v4_4clust_assign.rds
    Ignored:    output/limma_voom_elist_1.rds
    Ignored:    output/merged_raw_counts.csv
    Ignored:    output/nci_dlbcl_annotation.csv
    Ignored:    output/nci_dlbcl_unprocessed_counts.csv
    Ignored:    output/nci_expressionset.rds
    Ignored:    output/nci_expressionset_pext.rds
    Ignored:    output/nci_extended_pheno_data.csv
    Ignored:    output/networkanal3_clust1.txt
    Ignored:    output/networkanal3_clust2.txt
    Ignored:    output/networkanal3_clust3.txt
    Ignored:    output/networkanal3_clust4.txt
    Ignored:    output/total_immune_gset_v1.rds
    Ignored:    output/tpm_cyt_expressionset.rds
    Ignored:    output/tpm_expressionset.rds
    Ignored:    output/uchi_cluster_assignments.csv
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.pvals.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019_RAW.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.pvals.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819_RAW.txt
    Ignored:    output/xcell_input_matrix.csv

Untracked files:
    Untracked:  NCI_mutations_vs_deconvolution.xls
    Untracked:  analysis/correlation_clusters.Rmd
    Untracked:  analysis/mif.Rmd
    Untracked:  analysis/new_signature_matrix.Rmd
    Untracked:  analysis/table1.Rmd
    Untracked:  analysis/visualizing_gsva_scores_5.Rmd
    Untracked:  batch1_cell_counts.pdf
    Untracked:  data/cibersort/
    Untracked:  data/gistic_results_nci/
    Untracked:  es_10k_hgnc/
    Untracked:  output/ASH_ORAL_figures.png
    Untracked:  output/ASH_ORAL_figures/
    Untracked:  output/Rplot.pdf
    Untracked:  output/Rplot01.pdf
    Untracked:  output/batch1_cell_seg_df.rds
    Untracked:  output/batch1_summary_cell_seg_df.rds
    Untracked:  output/colored_hclust_plot_by_group.pdf
    Untracked:  output/colored_hclust_plot_by_samples.pdf
    Untracked:  output/combined_counts_dge.rds
    Untracked:  output/combined_expr_matrix_for_signature_unprocessed.rmd
    Untracked:  output/dgelist_pheno_data.rds
    Untracked:  output/dlbcl_final_signature_matrix.tsv
    Untracked:  output/dlbcl_gsva5_4cluster_hcpc.rds
    Untracked:  output/dlbcl_gsva5_4cluster_pca.rds
    Untracked:  output/dlbcl_uchi_batch1_mixture.tsv
    Untracked:  output/duke_dlbcl_mixture.tsv
    Untracked:  output/gse_69030_probes.tsv
    Untracked:  output/hclust_labeled_by_cell_type.pdf
    Untracked:  output/mif_pilot_data_plot.pdf
    Untracked:  output/nci_dlbcl_mixture.tsv
    Untracked:  output/phenodata_for_signature_matrix.rmd
    Untracked:  output/reference_matrix_file.tsv
    Untracked:  output/table1.html
    Untracked:  output/uchi_deconvolution_batch_1.rds
    Untracked:  output/uchi_pheno_clust
    Untracked:  output/var_contribution_axis2.pdf
    Untracked:  output/venn.png
    Untracked:  output/venn.png.2020-02-23_11-17-47.log
    Untracked:  output/venn.png.2020-02-23_11-17-56.log
    Untracked:  output/venn.png.2020-02-23_11-20-44.log
    Untracked:  preprocessed_data_for_signature_matrix_es.rds

Unstaged changes:
    Modified:   .gitignore
    Deleted:    Rplot.pdf
    Deleted:    Rplot001.png
    Deleted:    Rplot002.png
    Deleted:    Rplot01.pdf
    Modified:   analysis/cyt.Rmd
    Modified:   analysis/deconvolution.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/predictor.Rmd
    Modified:   analysis/sparse_clustering.Rmd
    Modified:   analysis/visualizing_gsva_scores_4.Rmd
    Deleted:    data/DLBCL_histo/JG55/HE/14-1464/14-1464_Scan1_annotations.xml.lock
    Modified:   output/design_for_limma.rds
    Modified:   output/dgelist_for_limma.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/e2b04164414d529fe354c85445de58e0329486c7/analysis/dge_pathways.Rmd" target="_blank">e2b0416</a>
</td>
<td>
mleukam
</td>
<td>
2020-05-09
</td>
<td>
wflow_publish(c(“analysis/about.Rmd”, “analysis/abstract_figure.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/a78a45cffe9d483012683d912ca3ab8f19d2000b/analysis/dge_pathways.Rmd" target="_blank">a78a45c</a>
</td>
<td>
mleukam
</td>
<td>
2019-10-12
</td>
<td>
archives dump from local machine - near final gsva v4 cluster model
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/2960d8b54653f6823b6df05cc0a3cc6f023486cd/analysis/dge_pathways.Rmd" target="_blank">2960d8b</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-18
</td>
<td>
record keeping updates - did not yet build the pages for these notebooks
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/01fb3c107e629cfc19b059394907946e892ab03a/analysis/dge_pathways.Rmd" target="_blank">01fb3c1</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-16
</td>
<td>
updates of analysis pages without building html pages - for record keeping and safekeeping
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level1">
<h1>Setup</h1>
<p>Clear the workspace</p>
<pre class="r"><code>rm(list = ls())</code></pre>
<pre class="r"><code>knitr::opts_chunk$set(eval = FALSE)</code></pre>
<p>Load basic packages</p>
<pre class="r"><code>library(&quot;tidyverse&quot;)
library(&quot;fgsea&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(&quot;msigdbr&quot;)</code></pre>
<p>Read in data</p>
<pre class="r"><code># toptables
cluster1_dge &lt;- readRDS(&quot;output/cluster1_dge.rds&quot;)
cluster2_dge &lt;- readRDS(&quot;output/cluster2_dge.rds&quot;)
cluster3_dge &lt;- readRDS(&quot;output/cluster3_dge.rds&quot;)
cluster4_dge &lt;- readRDS(&quot;output/cluster4_dge.rds&quot;)

ttable_list &lt;- list(cluster1_dge, cluster2_dge, cluster3_dge, cluster4_dge)
names(ttable_list) &lt;- c(&quot;cluster1_dge&quot;, &quot;cluster2_dge&quot;, &quot;cluster3_dge&quot;, &quot;cluster4_dge&quot;)

# lookup table for gene names
# read in conversion table for gene IDs
gencode_gtf &lt;- read_tsv(&quot;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&quot;)</code></pre>
</div>
<div id="go-enrichment" class="section level1">
<h1>GO Enrichment</h1>
<div id="go-with-topgo" class="section level4">
<h4>GO with TopGO</h4>
<p><a href="https://bioconductor.org/packages/release/bioc/vignettes/topGO/inst/doc/topGO.pdf" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/topGO/inst/doc/topGO.pdf</a></p>
</div>
<div id="go-with-clusterprofiler" class="section level4">
<h4>GO with clusterProfiler</h4>
<p>Vignettes and tutorials from the author used to generate this analysis: <a href="https://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler/" class="uri">https://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler/</a> <a href="https://yulab-smu.github.io/clusterProfiler-book/chapter5.html" class="uri">https://yulab-smu.github.io/clusterProfiler-book/chapter5.html</a></p>
<p>Load packages</p>
<pre class="r"><code>library(&quot;clusterProfiler&quot;)</code></pre>
<p>Prepare data - input for enrichGO test is simple vector of gene names Will pass gene names that meet cutoff criteria for adj.p.val and logFC from DGE experiment.</p>
<p>Add Entrez identifiers to gencode lookup table</p>
<pre class="r"><code>gencode_symbols &lt;- gencode_gtf %&gt;%
  pull(gene_name)
  
entrez &lt;- mapIds(org.Hs.eg.db, gencode_symbols, &#39;ENTREZID&#39;, &#39;SYMBOL&#39;) %&gt;%
  enframe() %&gt;%
  dplyr::rename(gene_name = name, entrez_id = value) %&gt;%
  print()

gencode_ent &lt;- gencode_gtf %&gt;%
  left_join(entrez) %&gt;%
  print()</code></pre>
<p>Decide which genes to use for enrichment</p>
<pre class="r"><code># set adjusted pval cutoff
p_cut &lt;- 0.001

# see range of fold-changes
summary(ttable_list[[1]]$logFC)
summary(ttable_list[[2]]$logFC)
summary(ttable_list[[3]]$logFC)
summary(ttable_list[[4]]$logFC)

# set logFC cutoff at average of 3rd quartile fold-change (upregulated genes only)
l_cut &lt;- (0.23247 + 0.178593 + 0.1258525 + 0.126851) / 4

# get gene list of entrez IDs that pass cutoff for each cluster from DGE
go_gene_list &lt;- ttable_list %&gt;%
  map(function(df){rownames_to_column(df, var = &quot;gene_id&quot;)}) %&gt;%
  map(as_tibble) %&gt;%
  map(function(df){df %&gt;% 
      dplyr::filter(adj.P.Val &lt; p_cut) %&gt;%
      dplyr::filter(logFC &gt; l_cut)}) %&gt;%
  map(function(df){left_join(df, gencode_ent)}) %&gt;%
  map(function(df){pull(df, entrez_id)})

str(go_gene_list)

# get gene universe list (same for all clusters)
go_gene_universe &lt;- ttable_list[[1]] %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  left_join(gencode_ent) %&gt;%
  pull(entrez_id)
str(go_gene_universe)</code></pre>
<pre class="r"><code># get enriched/overrepresented GO lists for cluster 1 DGE
ego1 &lt;- enrichGO(gene = go_gene_list[[1]],
                 OrgDb = org.Hs.eg.db,
                 ont = &quot;BP&quot;,
                 pvalueCutoff = 0.05,
                 pAdjustMethod = &quot;BH&quot;,
                 universe = go_gene_universe,
                 qvalueCutoff = 0.1)
# http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/

# convert to gene names
ego1 &lt;- setReadable(ego1, OrgDb = org.Hs.eg.db)

# plot
dotplot(ego1, showCategory = 30)

ego1_simp &lt;- clusterProfiler::simplify(ego1, 
                      cutoff = 0.7,
                      by = &quot;p.adjust&quot;,
                      select_fun = min)</code></pre>
<pre class="r"><code># get enriched/overrepresented GO lists for cluster 2 DGE
ego2 &lt;- enrichGO(gene = go_gene_list[[2]],
                 OrgDb = org.Hs.eg.db,
                 ont = &quot;BP&quot;,
                 pvalueCutoff = 0.05,
                 pAdjustMethod = &quot;BH&quot;,
                 universe = go_gene_universe,
                 qvalueCutoff = 0.1)
# http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/

# convert to gene names
ego2 &lt;- setReadable(ego2, OrgDb = org.Hs.eg.db)

# plot
dotplot(ego2, showCategory = 30)

ego2_simp &lt;- clusterProfiler::simplify(ego2, 
                      cutoff = 0.7,
                      by = &quot;p.adjust&quot;,
                      select_fun = min)</code></pre>
<pre class="r"><code># get enriched/overrepresented GO lists for cluster 3 DGE
ego2 &lt;- enrichGO(gene = go_gene_list[[3]],
                 OrgDb = org.Hs.eg.db,
                 ont = &quot;BP&quot;,
                 pvalueCutoff = 0.05,
                 pAdjustMethod = &quot;BH&quot;,
                 universe = go_gene_universe,
                 qvalueCutoff = 0.1)
# http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/

# convert to gene names
ego2 &lt;- setReadable(ego1, OrgDb = org.Hs.eg.db)

# plot
dotplot(ego2, showCategory = 30)

ego2_simp &lt;- clusterProfiler::simplify(ego2, 
                      cutoff = 0.7,
                      by = &quot;p.adjust&quot;,
                      select_fun = min)</code></pre>
</div>
</div>
<div id="fgsea" class="section level1">
<h1>FGSEA</h1>
<div id="rank-lists" class="section level4">
<h4>Rank lists</h4>
<p>Filtering by FC and adjusted P-value</p>
<pre class="r"><code># Adjusted p-value cutoff
p_cut &lt;- 0.01

# filter for p value
nrow(ttable_list[[1]])
ttable_df_list &lt;- ttable_list %&gt;%
  map(function(df){rownames_to_column(df, var = &quot;gene_id&quot;)}) %&gt;%
  map(as_tibble) %&gt;%
  map(function(df){df %&gt;% 
      dplyr::filter(adj.P.Val &lt; p_cut)})

# how many significantly different genes left?
nrow(ttable_df_list[[1]])
nrow(ttable_df_list[[2]])
nrow(ttable_df_list[[3]])
nrow(ttable_df_list[[4]])</code></pre>
<p><a href="http://www.bioconductor.org/packages/release/bioc/vignettes/fgsea/inst/doc/fgsea-tutorial.html" class="uri">http://www.bioconductor.org/packages/release/bioc/vignettes/fgsea/inst/doc/fgsea-tutorial.html</a></p>
<p>Create ranked lists following this tutorial: <a href="https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2018/RNASeq2018/html/06_Gene_set_testing.nb.html" class="uri">https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2018/RNASeq2018/html/06_Gene_set_testing.nb.html</a></p>
<p>Downloading gene sets from MSigDB: <a href="https://cran.r-project.org/web/packages/msigdbr/vignettes/msigdbr-intro.html" class="uri">https://cran.r-project.org/web/packages/msigdbr/vignettes/msigdbr-intro.html</a></p>
<pre class="r"><code># define function for creating ranked lists
ranklist &lt;- function(df){
  ranks &lt;- df %&gt;%
    rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
    left_join(gencode_ent) %&gt;%
    dplyr::select(entrez_id, logFC) %&gt;%
    mutate(logFC = as.double(logFC)) %&gt;%
    drop_na(entrez_id)
  rank_vector &lt;- ranks %&gt;% pull(logFC)
  names(rank_vector) &lt;- ranks %&gt;% pull(entrez_id)
  rank_vector
}

# apply to list of top tables
rtable_list &lt;- map(ttable_list, ranklist)
rtable_list[[1]]
str(rtable_list)</code></pre>
<pre class="r"><code># get gene sets from MSigDB
m_df = msigdbr(species = &quot;Homo sapiens&quot;)

# format  gene sets as a list of character vectors of entrez gene names
gs_df &lt;- m_df %&gt;%
  mutate(gs_idname = paste0(gs_cat, &quot;.&quot;, gs_id, &quot;.&quot;, gs_name)) %&gt;%
  dplyr::select(gs_idname, entrez_gene) %&gt;%
  nest(data = entrez_gene) %&gt;%
  arrange(desc(gs_idname))

gs_list &lt;- map(gs_df$data, pull) %&gt;%
  map(as.character)
names(gs_list) &lt;- gs_df$gs_idname
str(gs_list)

# start with just the hallmark gene sets for testing
hall_gs_subset &lt;- gs_df %&gt;%
  dplyr::filter(grepl(&quot;^H&quot;, gs_idname))

hall_gs_list &lt;- map(hall_gs_subset$data, pull) %&gt;%
  map(as.character)
names(hall_gs_list) &lt;- hall_gs_subset$gs_idname
str(hall_gs_list)</code></pre>
</div>
<div id="gsea" class="section level4">
<h4>GSEA</h4>
<pre class="r"><code># define function to run hallmark gsea on each toptable
library(BiocParallel)

fgsea_hallmark &lt;- function(ranklist){
  temp &lt;- fgsea(pathways = hall_gs_list, 
                                  stats = ranklist,
                                  minSize = 15, 
                                  maxSize = 500,
                                  nperm = 10000)
  
  collapsedPathways &lt;- collapsePathways(temp[order(pval)][padj &lt; 0.01],
                                        hall_gs_list, 
                                        ranklist)
  mainPathways &lt;- temp[pathway %in% collapsedPathways$mainPathways][
                         order(-NES)]
  mainPathways
}

# apply function to list of cluster toptables
hallmark_gsea_results &lt;- map(rtable_list, fgsea_hallmark)

# review results
# review results
map(hallmark_gsea_results, function(df){as_tibble(df) %&gt;% 
    dplyr::filter(padj &lt; 0.01) %&gt;% 
    dplyr::select(&quot;pathway&quot;, &quot;ES&quot;, &quot;padj&quot;) %&gt;% 
    arrange(desc(ES))})</code></pre>
<pre class="r"><code># define function to run gsea on each toptable
fgsea_batch &lt;- function(ranklist){
  temp &lt;- fgsea(pathways = gs_list, 
                                  stats = ranklist,
                                  minSize = 15, 
                                  maxSize = 500,
                                  nperm = 10000)
  
  collapsedPathways &lt;- collapsePathways(temp[order(pval)][padj &lt; 0.01],
                                        gs_list, 
                                        ranklist)
  mainPathways &lt;- temp[pathway %in% collapsedPathways$mainPathways][
                         order(-NES)]
  mainPathways
}

# apply function to list of cluster toptables
gsea_results &lt;- map(rtable_list, fgsea_batch)

# review results
map(gsea_results, function(df){as_tibble(df) %&gt;% 
    dplyr::select(&quot;pathway&quot;, &quot;ES&quot;, &quot;padj&quot;) %&gt;% 
    arrange(desc(ES))})</code></pre>
<p>Write out results for use in Cytoscape</p>
<pre class="r"><code>gsea_cluster1 &lt;- gsea_results[[1]] %&gt;% as_tibble() 
lead &lt;- map_chr(gsea_cluster1$leadingEdge, function(list){
  paste(list, collapse = &quot;,&quot;)
  })
gsea_cluster1$leadingEdge &lt;- lead
gsea_cluster1

write_csv(gsea_cluster1, &quot;output/gsea_res_cluster1.csv&quot;)</code></pre>
<p>Example code for collapsing pathways: <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/fgsea/inst/doc/fgsea-tutorial.html" class="uri">http://www.bioconductor.org/packages/release/bioc/vignettes/fgsea/inst/doc/fgsea-tutorial.html</a></p>
<p>Will instead handle the analysis and visualization of results with the EnrichmentMap App in Cytoscape 3.7.</p>
<p><a href="https://enrichmentmap.readthedocs.io/en/latest/" class="uri">https://enrichmentmap.readthedocs.io/en/latest/</a></p>
<p>Another option for functional annotation: <a href="http://genescf.kandurilab.org/documentation.php" class="uri">http://genescf.kandurilab.org/documentation.php</a> <a href="http://genescf.kandurilab.org/ftp/GeneSCF-Documentation_v1.1.pdf" class="uri">http://genescf.kandurilab.org/ftp/GeneSCF-Documentation_v1.1.pdf</a></p>
<p>A third option for network analysis: <a href="https://www.networkanalyst.ca/NetworkAnalyst/resources/tutorials/t2a_single_list.pdf" class="uri">https://www.networkanalyst.ca/NetworkAnalyst/resources/tutorials/t2a_single_list.pdf</a></p>
<p>Prepare data for GSEA desktop app</p>
<p>RNK file format: <a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats" class="uri">https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats</a></p>
<pre class="r"><code># Column names need to start with `#` or just don&#39;t have any
# Recommend Hugo Gene symbdols
enframe(rtable_list[[1]]) %&gt;% 
  dplyr::rename(entrez_id = name) %&gt;%
  left_join(gencode_ent) %&gt;%
  drop_na(gene_name) %&gt;%
  dplyr::select(`#HGNC_symbol` = gene_name, logFC = value) %&gt;%
  write_delim(&quot;output/clust1_vs_others_dge.rnk&quot;, delim = &quot;\t&quot;, col_names = FALSE)

enframe(rtable_list[[2]]) %&gt;% 
  dplyr::rename(entrez_id = name) %&gt;%
  left_join(gencode_ent) %&gt;%
  drop_na(gene_name) %&gt;%
  dplyr::select(`#HGNC_symbol` = gene_name, logFC = value) %&gt;%
  write_delim(&quot;output/clust2_vs_others_dge.rnk&quot;, delim = &quot;\t&quot;, col_names = FALSE)

enframe(rtable_list[[3]]) %&gt;% 
  dplyr::rename(entrez_id = name) %&gt;%
  left_join(gencode_ent) %&gt;%
  drop_na(gene_name) %&gt;%
  dplyr::select(`#HGNC_symbol` = gene_name, logFC = value) %&gt;%
  write_delim(&quot;output/clust3_vs_others_dge.rnk&quot;, delim = &quot;\t&quot;, col_names = FALSE)

enframe(rtable_list[[4]]) %&gt;% 
  dplyr::rename(entrez_id = name) %&gt;%
  left_join(gencode_ent) %&gt;%
  drop_na(gene_name) %&gt;%
  dplyr::select(`#HGNC_symbol` = gene_name, logFC = value) %&gt;%
  write_delim(&quot;output/clust4_vs_others_dge.rnk&quot;, delim = &quot;\t&quot;, col_names = FALSE)</code></pre>
</div>
<div id="hallmark-gene-sets" class="section level4">
<h4>Hallmark gene sets</h4>
<pre class="r"><code># Get hallmark gene sets from MSigDb (human)
msigdbr_show_species()
m_df = msigdbr(species = &quot;Homo sapiens&quot;, category = &quot;H&quot;)
m_df

# format for fgsea
gs_list_hallmark = m_df %&gt;% 
  split(x = .$gene_symbol, f = .$gs_name)
str(gs_list_hallmark)

# Get all human gene sets for relevant biological processes
# Include Hallmark and curated C2 biological process sets
# Exclude chromosome position gene sets
# Exclude miRNA / regulatory motif gene sets
# Include GO terms (C5)
# Include computational gene sets from cancer microarray data (C4)
# Include microarray &#39;oncogenic&#39; signatures
# Include immunologic signatures, manual curation from human and mouse studies (C7)
m_df_total = msigdbr(species = &quot;Homo sapiens&quot;) %&gt;%
  dplyr::filter(gs_cat %in% c(&quot;H&quot;, &quot;C2&quot;, &quot;C4&quot;, &quot;C5&quot;, &quot;C7&quot;))
m_df_total

# get number of gene sets in each category
summary(as.factor(m_df_total$gs_cat))

# format for fgsea
gs_list_total = m_df_total %&gt;% 
  split(x = .$gene_symbol, f = .$gs_name)
str(gs_list_total)</code></pre>
<pre class="r"><code># read in toptables
cluster1_dge &lt;- readRDS(&quot;output/cluster1_dge.rds&quot;)
cluster2_dge &lt;- readRDS(&quot;output/cluster2_dge.rds&quot;)
cluster3_dge &lt;- readRDS(&quot;output/cluster3_dge.rds&quot;)
cluster4_dge &lt;- readRDS(&quot;output/cluster4_dge.rds&quot;)</code></pre>
<p>Source for GSEA file prep (supplemental method 1): <a href="https://www.nature.com/articles/s41596-018-0103-9" class="uri">https://www.nature.com/articles/s41596-018-0103-9</a> Reimand et al “Pathway enrichment analysis and visualization of omics data using g:Profiler, GSEA, Cytoscape and EnrichmentMap” Nature Methods Jan 2019</p>
<p>Create a two-column rank (.RNK) file of all gene IDs and corresponding scores to for GSEA pre-ranked analysis. To run GSEA in pre-ranked mode, you need a two column RNK file with gene/protein/probe name (column 1) and the associated score (column 2). The first column should contain the same type of gene IDs used in the pathway gene-set (GMT) file. GSEA will look for enrichment in the set of most differentially expressed genes at the top of the list as well as those at the bottom of the list. Genes at the top of the list are more highly expressed in class A of samples while genes at the bottom are highly expressed in class B. A score can be computed by multiplying direction (sign) of fold change and logarithm of p-value for each gene.</p>
<pre class="r"><code># change gene ids to gene symbol
cluster1_dge_names &lt;- cluster1_dge %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  left_join(gencode_gtf, by = &quot;gene_id&quot;) %&gt;%
  dplyr::distinct(gene_name, .keep_all = TRUE) %&gt;%
  dplyr::select(gene_name, everything()) %&gt;%
  dplyr::select(-gene_id, -gene_type, -gene_status, -level, -havana_gene) 
cluster1_dge_names

cluster2_dge_names &lt;- cluster2_dge %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  left_join(gencode_gtf, by = &quot;gene_id&quot;) %&gt;%
  dplyr::distinct(gene_name, .keep_all = TRUE) %&gt;%
  dplyr::select(gene_name, everything()) %&gt;%
  dplyr::select(-gene_id, -gene_type, -gene_status, -level, -havana_gene) 
cluster2_dge_names

cluster3_dge_names &lt;- cluster3_dge %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  left_join(gencode_gtf, by = &quot;gene_id&quot;) %&gt;%
  dplyr::distinct(gene_name, .keep_all = TRUE) %&gt;%
  dplyr::select(gene_name, everything()) %&gt;%
  dplyr::select(-gene_id, -gene_type, -gene_status, -level, -havana_gene) 
cluster3_dge_names

cluster4_dge_names &lt;- cluster4_dge %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  left_join(gencode_gtf, by = &quot;gene_id&quot;) %&gt;%
  dplyr::distinct(gene_name, .keep_all = TRUE) %&gt;%
  dplyr::select(gene_name, everything()) %&gt;%
  dplyr::select(-gene_id, -gene_type, -gene_status, -level, -havana_gene) 
cluster4_dge_names

# calculate ranks
ranks_clust1 &lt;- sign(cluster1_dge_names$logFC) * -log10(cluster1_dge_names$P.Value)
names(ranks_clust1) &lt;- cluster1_dge_names$gene_name
ranks_clust1 &lt;- sort(ranks_clust1, decreasing = TRUE)
head(ranks_clust1)
tail(ranks_clust1)
summary(duplicated(names(ranks_clust1)))

ranks_clust2 &lt;- sign(cluster2_dge_names$logFC) * -log10(cluster2_dge_names$P.Value)
names(ranks_clust2) &lt;- cluster2_dge_names$gene_name
ranks_clust2 &lt;- sort(ranks_clust2, decreasing = TRUE)
head(ranks_clust2)
tail(ranks_clust2)
summary(duplicated(names(ranks_clust2)))

ranks_clust3 &lt;- sign(cluster3_dge_names$logFC) * -log10(cluster3_dge_names$P.Value)
names(ranks_clust3) &lt;- cluster3_dge_names$gene_name
ranks_clust3 &lt;- sort(ranks_clust3, decreasing = TRUE)
head(ranks_clust3)
tail(ranks_clust3)
summary(duplicated(names(ranks_clust3)))

ranks_clust4 &lt;- sign(cluster4_dge_names$logFC) * -log10(cluster3_dge_names$P.Value)
names(ranks_clust4) &lt;- cluster4_dge_names$gene_name
ranks_clust4 &lt;- sort(ranks_clust4, decreasing = TRUE)
head(ranks_clust4)
tail(ranks_clust4)
summary(duplicated(names(ranks_clust4)))</code></pre>
</div>
<div id="hallmark-gsea" class="section level4">
<h4>Hallmark GSEA</h4>
<p>Source: <a href="http://bioconductor.org/packages/devel/bioc/vignettes/fgsea/inst/doc/fgsea-tutorial.html" class="uri">http://bioconductor.org/packages/devel/bioc/vignettes/fgsea/inst/doc/fgsea-tutorial.html</a></p>
<p>Purrr functions</p>
<pre class="r"><code>library(BiocParallel)

fgsea_hallmark &lt;- function(ranks){
  temp &lt;- fgsea(pathways = gs_list_hallmark, 
                                  stats = ranks,
                                  minSize = 15, 
                                  maxSize = 500,
                                  nperm = 10000)
  
  collapsedPathways &lt;- collapsePathways(temp[order(pval)][padj &lt; 0.01],
                                        gs_list_hallmark, 
                                        ranks)
  mainPathways &lt;- temp[pathway %in% collapsedPathways$mainPathways][
                         order(-NES)]
  mainPathways
}

# test function
ranks &lt;- ranks_clust1
fgsea_hallmark(ranks)[1:5, 1:5]

# make list of rank stat vectors
ranklist &lt;- list(ranks_clust1,
                 ranks_clust2,
                 ranks_clust3, 
                 ranks_clust4)
                 

# apply function to list
hallmark_results &lt;- map(ranklist, fgsea_hallmark)

# display function
mainpaths &lt;- function(df){
  pathways &lt;- df %&gt;% pull(pathway)
  pathways
}

# apply display function to results
map(hallmark_results, mainpaths)

# plots
pathways1 &lt;- hallmark_results[[1]] %&gt;% arrange(desc(abs(NES))) %&gt;% pull(pathway)
ranks = ranklist[[1]]
plot1 &lt;- plotGseaTable(gs_list_hallmark[pathways1], 
                       ranks, 
                       hallmark_results[[1]])</code></pre>
</div>
<div id="larger-collection-of-gene-sets" class="section level4">
<h4>Larger collection of gene sets</h4>
<pre class="r"><code>fgsea_total &lt;- function(ranks){
  temp &lt;- fgsea(pathways = gs_list_total, 
                                  stats = ranks,
                                  minSize = 15, 
                                  maxSize = 500,
                                  nperm = 10000)
  
  collapsedPathways &lt;- collapsePathways(temp[order(pval)][padj &lt; 0.01],
                                        gs_list_total, 
                                        ranks)
  mainPathways &lt;- temp[pathway %in% collapsedPathways$mainPathways][
                         order(abs(NES))]
  mainPathways
}

# test function
ranks &lt;- ranks_clust1
fgsea_total(ranks)[1:5, 1:5]

# make list of rank stat vectors
ranklist &lt;- list(ranks_clust1,
                 ranks_clust2,
                 ranks_clust3, 
                 ranks_clust4)

# apply function to list
total_results &lt;- map(ranklist, fgsea_total)

# display function
mainpaths &lt;- function(df){
  pathways &lt;- df %&gt;% pull(pathway)
  pathways
}

# apply display function to results
map(total_results, mainpaths)

# plots
pathways1 &lt;- total_results[[1]] %&gt;% 
  arrange(abs(NES)) %&gt;% 
  dplyr::slice(1:20) %&gt;% 
  pull(pathway)
ranks &lt;- ranklist[[1]]
plot1 &lt;- plotGseaTable(gs_list_total[pathways1], 
                       ranks, 
                       total_results[[1]])</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
