<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-06-26" />

<title>dukedata</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">dukedata</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-06-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190613code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190613)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190613code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190613)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreee2b04164414d529fe354c85445de58e0329486c7targetblanke2b0416a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/e2b04164414d529fe354c85445de58e0329486c7" target="_blank">e2b0416</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreee2b04164414d529fe354c85445de58e0329486c7targetblanke2b0416a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    NCI_mutations_vs_deconvolution.csv
    Ignored:    data/.DS_Store
    Ignored:    data/1-s2.0-S1074761316304320-mmc2.xlsx
    Ignored:    data/12864_2005_498_MOESM2_ESM.xls
    Ignored:    data/41591_2018_45_MOESM5_ESM.xlsx
    Ignored:    data/CCLE_sample_info_file_2012-10-18.txt
    Ignored:    data/NIH_PDL1_amp_cases.csv
    Ignored:    data/NIH_PDL1_nonamp_cases.csv
    Ignored:    data/PDL1dlbcl.rnaseq.kallisto.raw.txi.coding.cpm1.TMM.logCPM.sm24.csv
    Ignored:    data/PanImmune_GeneSet_Definitions.xlsx
    Ignored:    data/aliquot.tsv
    Ignored:    data/azizi_curated.xlsx
    Ignored:    data/c2_human_immune_genesets.gmt
    Ignored:    data/dlbcl_total_immune_gset_v1_results.rds
    Ignored:    data/ds_JCO.18.01583-2.xlsx
    Ignored:    data/duke_study_driver_mutations_top150_maf.csv
    Ignored:    data/final_analysis_set.maf
    Ignored:    data/gdc_clinical_data.json
    Ignored:    data/gdc_files_and_case_ids.json
    Ignored:    data/gencode.v22.annotation.gtf
    Ignored:    data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Ignored:    data/genesets/
    Ignored:    data/gset_ids_complete.rds
    Ignored:    data/histo_metadata_dlbcl_landscape.csv
    Ignored:    data/journal.pone.0088309.s001.XLS
    Ignored:    data/mmc1.xlsx
    Ignored:    data/nejmoa1801445_appendix_2.xlsx
    Ignored:    data/reactome_imm_signaling.tsv
    Ignored:    data/sample.tsv
    Ignored:    data/sig.rda
    Ignored:    data/sig_genes.txt
    Ignored:    data/temp.dgelist_edger.rds
    Ignored:    data/temp.dgelist_limma.rds
    Ignored:    data/uchi_metadata.csv
    Ignored:    logs/
    Ignored:    output/.DS_Store
    Ignored:    output/NCI_mutations_vs_deconvolution.csv
    Ignored:    output/clust1_vs_others.txt
    Ignored:    output/clust1_vs_others_dge.rnk
    Ignored:    output/clust2_vs_others_dge.rnk
    Ignored:    output/clust3_vs_others_dge.rnk
    Ignored:    output/clust4_vs_others_dge.rnk
    Ignored:    output/cluster1_dge.rds
    Ignored:    output/cluster2_dge.rds
    Ignored:    output/cluster3_dge.rds
    Ignored:    output/cluster4_dge.rds
    Ignored:    output/combined_clin_data.xlsx
    Ignored:    output/combined_dlbcl_expr_matrix.csv
    Ignored:    output/combined_es_gsva2_4cluster.rds
    Ignored:    output/combined_es_gsva4_4cluster.rds
    Ignored:    output/combined_expressionset.rds
    Ignored:    output/combined_tpm_matrix.csv
    Ignored:    output/counts_expressionset_for_limma.rds
    Ignored:    output/dlbcl_expr_matrix.csv
    Ignored:    output/dlbcl_total_immune_gset_v2_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v3_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v4_results.rds
    Ignored:    output/duke_expression_set_cleaned_log.rds
    Ignored:    output/duke_expressionset.rds
    Ignored:    output/duke_expressionset_pext.rds
    Ignored:    output/duke_extended_pheno_data.csv
    Ignored:    output/expr_matrix.csv
    Ignored:    output/gsea_res_cluster1.csv
    Ignored:    output/gset_ids_complete_2.rds
    Ignored:    output/gset_ids_complete_3.rds
    Ignored:    output/gset_ids_complete_4.rds
    Ignored:    output/gsva_v2_4clust_assign.rds
    Ignored:    output/gsva_v3_4clust_assign.rds
    Ignored:    output/gsva_v4_4clust_assign.rds
    Ignored:    output/limma_voom_elist_1.rds
    Ignored:    output/merged_raw_counts.csv
    Ignored:    output/nci_dlbcl_annotation.csv
    Ignored:    output/nci_dlbcl_unprocessed_counts.csv
    Ignored:    output/nci_expressionset.rds
    Ignored:    output/nci_expressionset_pext.rds
    Ignored:    output/nci_extended_pheno_data.csv
    Ignored:    output/networkanal3_clust1.txt
    Ignored:    output/networkanal3_clust2.txt
    Ignored:    output/networkanal3_clust3.txt
    Ignored:    output/networkanal3_clust4.txt
    Ignored:    output/total_immune_gset_v1.rds
    Ignored:    output/tpm_cyt_expressionset.rds
    Ignored:    output/tpm_expressionset.rds
    Ignored:    output/uchi_cluster_assignments.csv
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.pvals.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019_RAW.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.pvals.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819_RAW.txt
    Ignored:    output/xcell_input_matrix.csv

Untracked files:
    Untracked:  NCI_mutations_vs_deconvolution.xls
    Untracked:  analysis/correlation_clusters.Rmd
    Untracked:  analysis/mif.Rmd
    Untracked:  analysis/new_signature_matrix.Rmd
    Untracked:  analysis/table1.Rmd
    Untracked:  analysis/visualizing_gsva_scores_5.Rmd
    Untracked:  batch1_cell_counts.pdf
    Untracked:  data/cibersort/
    Untracked:  data/gistic_results_nci/
    Untracked:  es_10k_hgnc/
    Untracked:  output/ASH_ORAL_figures.png
    Untracked:  output/ASH_ORAL_figures/
    Untracked:  output/Rplot.pdf
    Untracked:  output/Rplot01.pdf
    Untracked:  output/batch1_cell_seg_df.rds
    Untracked:  output/batch1_summary_cell_seg_df.rds
    Untracked:  output/colored_hclust_plot_by_group.pdf
    Untracked:  output/colored_hclust_plot_by_samples.pdf
    Untracked:  output/combined_counts_dge.rds
    Untracked:  output/combined_expr_matrix_for_signature_unprocessed.rmd
    Untracked:  output/dgelist_pheno_data.rds
    Untracked:  output/dlbcl_final_signature_matrix.tsv
    Untracked:  output/dlbcl_gsva5_4cluster_hcpc.rds
    Untracked:  output/dlbcl_gsva5_4cluster_pca.rds
    Untracked:  output/dlbcl_uchi_batch1_mixture.tsv
    Untracked:  output/duke_dlbcl_mixture.tsv
    Untracked:  output/gse_69030_probes.tsv
    Untracked:  output/hclust_labeled_by_cell_type.pdf
    Untracked:  output/mif_pilot_data_plot.pdf
    Untracked:  output/nci_dlbcl_mixture.tsv
    Untracked:  output/phenodata_for_signature_matrix.rmd
    Untracked:  output/reference_matrix_file.tsv
    Untracked:  output/table1.html
    Untracked:  output/uchi_deconvolution_batch_1.rds
    Untracked:  output/uchi_pheno_clust
    Untracked:  output/var_contribution_axis2.pdf
    Untracked:  output/venn.png
    Untracked:  output/venn.png.2020-02-23_11-17-47.log
    Untracked:  output/venn.png.2020-02-23_11-17-56.log
    Untracked:  output/venn.png.2020-02-23_11-20-44.log
    Untracked:  preprocessed_data_for_signature_matrix_es.rds

Unstaged changes:
    Modified:   .gitignore
    Deleted:    Rplot.pdf
    Deleted:    Rplot001.png
    Deleted:    Rplot002.png
    Deleted:    Rplot01.pdf
    Modified:   analysis/cyt.Rmd
    Modified:   analysis/deconvolution.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/predictor.Rmd
    Modified:   analysis/sparse_clustering.Rmd
    Modified:   analysis/visualizing_gsva_scores_4.Rmd
    Deleted:    data/DLBCL_histo/JG55/HE/14-1464/14-1464_Scan1_annotations.xml.lock
    Modified:   output/design_for_limma.rds
    Modified:   output/dgelist_for_limma.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/e2b04164414d529fe354c85445de58e0329486c7/analysis/dukedata.Rmd" target="_blank">e2b0416</a>
</td>
<td>
mleukam
</td>
<td>
2020-05-09
</td>
<td>
wflow_publish(c(“analysis/about.Rmd”, “analysis/abstract_figure.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/f1ed7b84631b39254e3501bf48f0c8c4a881e54f/docs/dukedata.html" target="_blank">f1ed7b8</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/fe5d4fd9041e164bdf73f0c225e86c757de78434/analysis/dukedata.Rmd" target="_blank">fe5d4fd</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-14
</td>
<td>
fixed duke raw counts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/5a57ec091786bd8b0362d31767fd966d90be89cd/docs/dukedata.html" target="_blank">5a57ec0</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/42da9d5893be09775a582da78b8bec151d4ea9bf/analysis/dukedata.Rmd" target="_blank">42da9d5</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-28
</td>
<td>
major overhaul of Duke count processing, updates of NCI pheno data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/3cee826038da6f1bd23b82e184f1f26e99fe574d/docs/dukedata.html" target="_blank">3cee826</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/107e89bfb9707ced9be9d531835f5d656f622f5d/analysis/dukedata.Rmd" target="_blank">107e89b</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-27
</td>
<td>
working notebook for cleaning and normalizing duke expression data with some
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-sources" class="section level2">
<h2>Data sources</h2>
<ul>
<li>Original publication: Reddy et al. Cell. 2017 Oct 5;171(2):481-494 Link: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28985567" class="uri">https://www.ncbi.nlm.nih.gov/pubmed/28985567</a></li>
<li>776 out of a total of 1001 samples underwent RNA sequencing:</li>
</ul>
<blockquote>
<p>RNaseq libraries were hybridized to Human All Exon V6 + 3′ UTR capture baits (Agilent, Cat. #5190-9306) as described previously (Cieslik et al., 2015). Each capture pool contained 24 libraries at 50 ng each for a total of 1200 ng of DNA. The pools were lyophilized using the speed vacuum technique. The libraries were first denatured and blocked at 95 C for 5 minutes, followed by holding at 65 C. The capture baits were mixed with RNase OUT and incubated for 2 minutes at 65 C. Hybridization buffer was mixed with the baits, followed by addition of the denatured libraries. The hybridization reaction was held at 65 C for 24 hours, followed by washing and amplification. These sequencing libraries were assessed for quality with the Bioanalyzer 2100 (Agilent) using the Agilent DNA 1000 Kit per manufacturer’s instructions. Libraries of high quality (e.g., with size distribution between 120-400bp and sufficient quantity for sequencing) were subjected to high throughput sequencing using the Illumina HiSeq 2500 platform per manufacturer’s instructions using HiSeq V4 125 PE, to generate an average of 10 million reads per sample.</p>
</blockquote>
<ul>
<li>Raw sequences were downloaded using from the European Genome/Phenome Archive (EGA) using IBM Aspera file transfer software.</li>
</ul>
<p>The following processing steps were carried out on the Gardner Supercomputing Cluster due to system requirements:</p>
<ul>
<li>RNAseq BAM files deconstructed to FASTQ</li>
<li>FastQC completed for RNAseq files</li>
<li>Read counts for RNAseq quantified with Kallisto (reverse strand)</li>
<li>Read counts combined with tximport into a single data structure</li>
<li>Counts and metadata extracted from txi structure as a EdgeR “dgelist” object</li>
<li>Dgelist object transferred off cluster for local processing in this notebook</li>
</ul>
<p>Parameters set for downstream analysis with limma voom following examples here: <a href="https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html</a></p>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Clear workspace</p>
<pre class="r"><code>rm(list = ls())</code></pre>
<pre class="r"><code>knitr::opts_chunk$set(eval = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(&quot;tidyverse&quot;)
library(&quot;edgeR&quot;)
library(&quot;limma&quot;)
library(&quot;Biobase&quot;)</code></pre>
<p>Batch script from Gardner HPC Cluster</p>
<pre class="r"><code>#!/usr/bin/env Rscript

# requires R version 3.5.0

# load packages
library(&quot;dplyr&quot;)
library(&quot;readr&quot;)
library(&quot;tximport&quot;)
library(&quot;GenomicFeatures&quot;)
library(&quot;AnnotationDbi&quot;)
library(&quot;edgeR&quot;)

# import input files
TxDb.gencode.v29.primary &lt;- loadDb(&quot;/gpfs/data/kline-lab/ref/Gencode/TxDb.gencode.v29.primary.sqlite&quot;)
kallisto.files &lt;- read_table(&quot;/gpfs/data/kline-lab/inputs/abundance_tsv.tsv&quot;, col_names=FALSE)
metadata &lt;- read_csv(&quot;/gpfs/data/kline-lab/inputs/rna_meta_table.csv&quot;, col_names=FALSE)

# select columns to make tx2gene
k &lt;- keys(TxDb.gencode.v29.primary, keytype = &quot;TXNAME&quot;)
tx2gene &lt;- AnnotationDbi::select(TxDb.gencode.v29.primary, k, &quot;GENEID&quot;, &quot;TXNAME&quot;)

# convert filenames and samples to vector
kallisto.files &lt;- dplyr::pull(kallisto.files[,1])
sample.names &lt;- dplyr::pull(metadata[,1])
names(kallisto.files) &lt;- sample.names

# how to work around ignoreAfterBar
# https://www.biostars.org/p/322297/

# import reabdundance files for use with edger
txi.edger &lt;- tximport(kallisto.files, 
    type = &quot;kallisto&quot;, 
    tx2gene = tx2gene, 
    ignoreAfterBar = TRUE)

# save tximport objects for later use
saveRDS(txi.edger, &quot;/gpfs/data/kline-lab/inputs/txi.edger.rds&quot;)</code></pre>
<p>Cluster interactive node:</p>
<pre class="r"><code>## load packages
library(tximport)
library(edgeR)
library(limma)
library(tidyverse)
library(biomaRt)

## read in tximport objects and metadata
## this takes a while
txi_edger &lt;- readRDS(&quot;/gpfs/data/kline-lab/inputs/txi.edger.rds&quot;)
metadata &lt;- read_csv(&quot;/gpfs/data/kline-lab/inputs/rna_meta_table.csv&quot;,col_names=FALSE)

## add counts by gene to DGEList as matrix
dgelist_edger &lt;- DGEList(txi_edger$counts)

## add readgroup information to DGEList as data.frame
lane &lt;- as.factor(metadata$X4) 
dgelist_edger$samples$lane &lt;- lane 

## add gene symbol information to DGEList
## first remove version number from gene names in dgelist$counts
genes_edger &lt;- sapply(strsplit((rownames(dgelist_edger$counts)), &quot;.&quot;, fixed = TRUE), function(x) x[1])
rownames(dgelist_edger$counts) &lt;- genes_edger

## then pull gene symbols for each gene name
mart &lt;- useDataset(&quot;hsapiens_gene_ensembl&quot;, useMart(&quot;ensembl&quot;))  

symbol_edger &lt;- getBM(filters = &quot;ensembl_gene_id&quot;, attributes = c(&quot;ensembl_gene_id&quot;, &quot;hgnc_symbol&quot;,&quot;chromosome_name&quot;, &quot;gene_biotype&quot;), values = genes_edger, mart = mart)

## reorder and adjust length to match my genes
symbol_edger &lt;- symbol_edger[match(genes_edger, symbol_edger$ensembl_gene_id),]

## add data.frame of gene symbols to DGElist
dgelist_edger$genes &lt;- symbol_edger

saveRDS(dgelist_edger, &quot;/gpfs/data/kline-lab/inputs/temp.dgelist_edger.rds&quot;)</code></pre>
<p>Move output file to local machine and load into R notebook session</p>
<pre class="r"><code>dgelist &lt;- readRDS(&quot;data/temp.dgelist_edger.rds&quot;)
str(dgelist)

# read in conversion table for gene IDs
gencode_gtf &lt;- read_tsv(&quot;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&quot;)</code></pre>
</div>
<div id="data-cleaning" class="section level2">
<h2>Data cleaning</h2>
<p>Convert GENCODE v29 to v22 gene ids to match GDC standard</p>
<pre class="r"><code># get expression matrix
expr_duke &lt;- dgelist[[1]]
expr_duke[1:5, 1:5]
col__names &lt;- colnames(expr_duke)
col_names_rep &lt;- str_split(col__names, pattern = &quot;\\.&quot;, simplify = TRUE) %&gt;% 
  as_tibble() %&gt;%
  pull(V1) %&gt;%
  paste0(&quot;duke_&quot;, .) %&gt;%
  print()
colnames(expr_duke) &lt;- col_names_rep
expr_duke[1:5, 1:5]

# gencode v29 was used to generate rownames of expr_duke matrix
# GDC standard is gencode v22, which is source of NCI data
# will convert to gencode v22 to match rows of NCI data
geneID_split &lt;- str_split(gencode_gtf$gene_id, pattern = &quot;\\.&quot;, simplify = TRUE) %&gt;%
  as_tibble() %&gt;%
  dplyr::select(geneID_split = V1)

gencode_gtf_split &lt;- gencode_gtf %&gt;% 
  bind_cols(geneID_split) %&gt;%
  dplyr::select(gene_id_v22 = gene_id, gene_name, geneID_split, everything()) %&gt;%
  print()

row__names &lt;- rownames(expr_duke) %&gt;%
  enframe() %&gt;%
  dplyr::select(geneID_split = value) %&gt;%
  print()

# check how many gene ids are missing in v22 that are named in v29
setdiff(row__names$geneID_split, gencode_gtf_split$geneID_split)
# approx 1900 (previous checks show none after filtering for protein coding genes)

# merge gene name data with expression matrix
expr_duke_df &lt;- expr_duke %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id_v29&quot;) %&gt;%
  as_tibble() %&gt;%
  mutate(geneID_split = str_replace(gene_id_v29, pattern = &quot;\\.*&quot;, &quot;&quot;)) %&gt;%
  dplyr::select(gene_id_v29, geneID_split, everything()) %&gt;%
  left_join(gencode_gtf_split) %&gt;%
  dplyr::select(gene_id_v29, gene_id_v22, geneID_split, gene_type, everything()) %&gt;%
  dplyr::select(-gene_status, -gene_name, -level, -havana_gene) %&gt;%
  print()

# same but for hugo gene names
expr_duke_hgnc &lt;- expr_duke %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id_v29&quot;) %&gt;%
  as_tibble() %&gt;%
  mutate(geneID_split = str_replace(gene_id_v29, pattern = &quot;\\.*&quot;, &quot;&quot;)) %&gt;%
  dplyr::select(gene_id_v29, geneID_split, everything()) %&gt;%
  left_join(gencode_gtf_split) %&gt;%
  dplyr::select(gene_name, geneID_split, gene_type, everything()) %&gt;%
  dplyr::select(-gene_status, -gene_id_v29, -gene_id_v22, -level, -havana_gene) %&gt;%
  print()

# check how many gene ids are missing from version 22  
summary(is.na(expr_duke_hgnc$gene_name))

# check the gene types in the expression matrix
summary(as.factor(expr_duke_df$gene_type))</code></pre>
<div id="filter-for-protein-coding" class="section level4">
<h4>Filter for protein coding</h4>
<pre class="r"><code># filter for protein coding genes only
expr_duke_filtered &lt;- expr_duke_df %&gt;% 
  dplyr::filter(gene_type == &quot;protein_coding&quot;)

expr_duke_hgnc_filtered &lt;- expr_duke_hgnc %&gt;% 
  dplyr::filter(gene_type == &quot;protein_coding&quot;)
  
# any missing gene ids also protein coding genes?
summary(is.na(expr_duke_filtered$gene_id_v22))
# NO

expr_duke_protcoding &lt;- expr_duke_filtered %&gt;%
  dplyr::select(-gene_id_v29, -geneID_split, -gene_type) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;gene_id_v22&quot;) %&gt;%
  as.matrix()
str(expr_duke_protcoding)
expr_duke_protcoding[1:5, 1:5]

expr_duke_hgnc_protcoding &lt;- expr_duke_hgnc_filtered %&gt;%
  dplyr::select(-geneID_split, -gene_type, -gene_name) %&gt;%
  as.matrix()
rownames(expr_duke_hgnc_protcoding) &lt;- expr_duke_hgnc_filtered$gene_name
expr_duke_hgnc_protcoding_ave &lt;- avereps(expr_duke_hgnc_protcoding)
str(expr_duke_hgnc_protcoding_ave)
expr_duke_hgnc_protcoding_ave[1:5, 1:5]</code></pre>
</div>
<div id="subset-cases-based-on-protein-coding-library-coverage" class="section level4">
<h4>Subset cases based on protein coding library coverage</h4>
<pre class="r"><code>expr_duke_protcoding[1:5, 1:5]

expr_duke_hgnc_protcoding_ave[1:5, 1:5]

gene_coverage_threshold &lt;- 12000

ncol(expr_duke_protcoding)
expr_duke_subset &lt;- expr_duke_protcoding[, colSums(expr_duke_protcoding != 0) &gt; gene_coverage_threshold]
expr_duke_subset[1:5, 1:5]
ncol(expr_duke_subset)

ncol(expr_duke_hgnc_protcoding_ave)
expr_duke_hgnc_subset &lt;- expr_duke_hgnc_protcoding_ave[, colSums(expr_duke_hgnc_protcoding_ave != 0) &gt; gene_coverage_threshold]
expr_duke_hgnc_subset[1:5, 1:5]
ncol(expr_duke_hgnc_subset)</code></pre>
</div>
<div id="convert-to-cpm" class="section level4">
<h4>Convert to CPM</h4>
<pre class="r"><code># correct for library size with cpm using EdgeR
expr_duke_cpm &lt;- cpm(expr_duke_subset, log = FALSE)
str(expr_duke_cpm)
expr_duke_cpm[1:5, 1:5]

expr_duke_hgnc_cpm &lt;- cpm(expr_duke_hgnc_subset, log = FALSE)
str(expr_duke_hgnc_cpm)
expr_duke_hgnc_cpm[1:5, 1:5]</code></pre>
<p>Write out mixture file for deconvolution</p>
<pre class="r"><code># TMM normalization
hgnc_norm_factors &lt;- calcNormFactors(expr_duke_hgnc_cpm, method = &quot;TMM&quot;)
summary(hgnc_norm_factors)

# convert expression matrix to dataframe
expr_duke_hgnc_cpm_df &lt;- as.data.frame(expr_duke_hgnc_cpm)

# apply factor to each column
total_cpm_norm_duke &lt;- map2_dfc(expr_duke_hgnc_cpm_df, hgnc_norm_factors, `*`)
total_cpm_norm_duke &lt;- as.data.frame(expr_duke_hgnc_cpm_df)
total_cpm_norm_duke[1:5, 1:5]
str(total_cpm_norm_duke)
duke_dlbcl_mixture &lt;- total_cpm_norm_duke %&gt;%
  rownames_to_column(var = &quot;GeneSymbol&quot;) %&gt;%
  as_tibble() %&gt;%
  print()

write_delim(duke_dlbcl_mixture, 
            &quot;output/duke_dlbcl_mixture.tsv&quot;,
            delim = &quot;\t&quot;,
            col_names = TRUE)</code></pre>
</div>
<div id="initial-density-plots" class="section level4">
<h4>Initial density plots</h4>
<pre class="r"><code># Density plots
# tidy data
tidy_cpm &lt;- t(expr_duke_cpm) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  gather(key = &quot;gene_id&quot;, value = &quot;intensity&quot;, -sample_id) %&gt;%
  print()

# plots
dplot_duke &lt;- ggplot(tidy_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;)
dplot_duke +
  xlim(-5, 20)

# plot with log and small offset
dplot_duke_log &lt;- ggplot(tidy_cpm, aes(log(intensity + 0.5))) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;)
dplot_duke_log +
  xlim(-10, 10)</code></pre>
</div>
<div id="filtering-by-expression" class="section level4">
<h4>Filtering by expression</h4>
<pre class="r"><code># apply hard cutoffs
# cpm expression cutoff: 1
cutoff &lt;- 1
# must be expressed in at least: half
min_sm_frac &lt;- 0.5
filter_frac &lt;- min_sm_frac * ncol(expr_duke_cpm)
 filter_frac
 total_cpm_stats &lt;- data.frame(
   total = apply(expr_duke_cpm, 1, function(x){
     sum(x &gt; cutoff, na.rm = TRUE) 
     } ))
keep &lt;- which(total_cpm_stats$total &gt;= filter_frac)

# convert to data frame for subsetting
expr_duke_cpm_df &lt;- expr_duke_cpm %&gt;% as.data.frame()

# apply soft cutoffs
# decided not to use
#expr_duke_cpm_df &lt;- as.data.frame(expr_duke_cpm)
# set CPM threshold based on library size
#cpm_thres &lt;- 10 / min(colSums(expr_duke)) * 10^6
# apply CPM threshold to at least half of all samples
#keep &lt;- rowSums(expr_duke_cpm_df &gt; cpm_thres) &gt;= #as.integer(ncol(expr_duke_cpm_df) / 2)

# check results
dim(expr_duke_cpm_df)
expr_duke_cpm_filtered &lt;- expr_duke_cpm_df[keep,]
dim(expr_duke_cpm_filtered)</code></pre>
</div>
<div id="post-filtering-density-plots" class="section level4">
<h4>Post-filtering density plots</h4>
<pre class="r"><code># Density plots
# tidy data
tidy_cpm_filtered &lt;- t(expr_duke_cpm_filtered) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  gather(key = &quot;gene_id&quot;, value = &quot;intensity&quot;, -sample_id) %&gt;%
  print()

# plots
dplot_duke_2 &lt;- ggplot(tidy_cpm_filtered, aes(intensity)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;)
dplot_duke_2 +
  xlim(-5, 20)

# plot with log and small offset
dplot_duke_log_2 &lt;- ggplot(tidy_cpm_filtered, aes(log(intensity + 0.5))) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;)
dplot_duke_log_2 +
  xlim(-10, 10)</code></pre>
</div>
<div id="normalize-gene-expression-distributions" class="section level4">
<h4>Normalize gene expression distributions</h4>
<p>Normalization by the method of trimmed mean of M-values (TMM) is performed using the calcNormFactors function in edgeR. The normalisation factors calculated here are used as a scaling factor for the library sizes.</p>
<pre class="r"><code># get normalization factors
norm_factors &lt;- calcNormFactors(expr_duke_cpm_filtered, method = &quot;TMM&quot;)

# convert expression matrix to dataframe
expr_duke_cpm_df &lt;- as.data.frame(expr_duke_cpm_filtered)

# apply factor to each column
duke_cpm_norm &lt;- map2_dfc(expr_duke_cpm_df, norm_factors, `*`)
duke_cpm_norm &lt;- as.data.frame(duke_cpm_norm)
rownames(duke_cpm_norm) &lt;- rownames(expr_duke_cpm_filtered)
duke_cpm_norm[1:5, 1:5]</code></pre>
</div>
<div id="log-transformation" class="section level4">
<h4>Log transformation</h4>
<pre class="r"><code># introduce offset to prevent -Inf
offset_duke_cpm_norm &lt;- duke_cpm_norm + 0.5
offset_duke_cpm_norm[1:5, 1:5]

# log transformation
duke_log_cpm_filtered_norm &lt;- log2(offset_duke_cpm_norm)
duke_log_cpm_filtered_norm[1:5, 1:5]</code></pre>
<p>Final density plot</p>
<pre class="r"><code>duke_log_cpm_filtered_tbl &lt;- duke_log_cpm_filtered_norm %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  as_tibble()

tidy_log_cpm &lt;- duke_log_cpm_filtered_tbl %&gt;% 
  gather(key = &quot;sampleID&quot;, value = &quot;intensity&quot;, -gene)

dplot3 &lt;- ggplot(tidy_log_cpm, aes(intensity)) +
  geom_density() +
  theme(legend.position = &quot;none&quot;) +
  xlim(-20, 20) +
  ggtitle(&quot;Final density plot of log(CPM) for Duke DLBCL&quot;)
dplot3</code></pre>
</div>
</div>
<div id="prepare-phenotype-data" class="section level2">
<h2>Prepare phenotype data</h2>
<pre class="r"><code>pheno_data &lt;- dgelist[[2]] 
pheno_data_split &lt;- as.data.frame(pheno_data) %&gt;%
  rownames_to_column(var = &quot;sample_long&quot;) %&gt;%
  as_tibble()

sample_id &lt;- str_split(pheno_data_split$sample_long, 
                       pattern = &quot;\\.&quot;, 
                       simplify = TRUE) %&gt;% 
  as_tibble() %&gt;%
  pull(V1) %&gt;%
  paste0(&quot;duke_&quot;, .) %&gt;%
  print()

sample_id_df &lt;- sample_id %&gt;%
  enframe() %&gt;%
  dplyr::select(sample_id = value, -name)
  
pheno_data_joined &lt;- pheno_data_split %&gt;%
  bind_cols(sample_id_df) %&gt;%
  dplyr::select(sample_id, everything()) %&gt;%
  dplyr::select(-group, -norm.factors, -sample_long) %&gt;%
  dplyr::rename(lib_size = lib.size) %&gt;%
  mutate(lib_size = as.integer(lib_size)) %&gt;%
  print()

# select rows based on columns in expression matrix
pheno_data_filtered &lt;- pheno_data_joined %&gt;%
  dplyr::filter(sample_id %in% colnames(duke_log_cpm_filtered_norm))
nrow(pheno_data_filtered)

# Ensure that samples in expression matrix matches pheno data
sample_exprs &lt;- colnames(duke_log_cpm_filtered_norm)
sample_pheno &lt;- pheno_data_filtered %&gt;% pull(sample_id)
identical(sample_exprs, sample_pheno)</code></pre>
</div>
<div id="write-out-files-for-later-use" class="section level2">
<h2>Write out files for later use</h2>
<p>Following ExpressionSet vignette in Biobase package</p>
<pre class="r"><code># make expression set
# first need variable annotation
v_metadata &lt;- data.frame(
  labelDescription = c(&quot;original library size&quot;,&quot;Sequencing lane for batch correction&quot;),
  row.names = c(&quot;lib_size&quot;, &quot;lane&quot;))

# make annotated phenoData
pheno_data_df &lt;- as.data.frame(pheno_data_filtered) %&gt;%
  column_to_rownames(var = &quot;sample_id&quot;)
  
phenoData &lt;- new(&quot;AnnotatedDataFrame&quot;, 
                 data = pheno_data_df, 
                 varMetadata = v_metadata)
phenoData

# convert expression data to matrix format
duke_exprs &lt;- duke_log_cpm_filtered_norm %&gt;% as.matrix()

# create global annotation
annotation &lt;- as.character(&quot;RNAseq counts derived from raw sequence data published by the Dave lab at Duke in Reddy et al Cell 2017. Quantified with Kallisto with reverse strand parameter, collected with tximport, filtered for only protein-coding genes, selected for cases with expression in &gt;12,000 genes, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed.&quot;)

# assemble expressionset
duke_es &lt;- ExpressionSet(
  assayData = duke_exprs,
  phenoData = phenoData,
  annotation = annotation)
duke_es</code></pre>
<pre class="r"><code>saveRDS(duke_es, &quot;output/duke_expressionset.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
