<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-06-13" />

<title>Download NCI data from GDC</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Download NCI data from GDC</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-06-13</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190613code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190613)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190613code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190613)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreee2b04164414d529fe354c85445de58e0329486c7targetblanke2b0416a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/e2b04164414d529fe354c85445de58e0329486c7" target="_blank">e2b0416</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreee2b04164414d529fe354c85445de58e0329486c7targetblanke2b0416a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    NCI_mutations_vs_deconvolution.csv
    Ignored:    data/.DS_Store
    Ignored:    data/1-s2.0-S1074761316304320-mmc2.xlsx
    Ignored:    data/12864_2005_498_MOESM2_ESM.xls
    Ignored:    data/41591_2018_45_MOESM5_ESM.xlsx
    Ignored:    data/CCLE_sample_info_file_2012-10-18.txt
    Ignored:    data/NIH_PDL1_amp_cases.csv
    Ignored:    data/NIH_PDL1_nonamp_cases.csv
    Ignored:    data/PDL1dlbcl.rnaseq.kallisto.raw.txi.coding.cpm1.TMM.logCPM.sm24.csv
    Ignored:    data/PanImmune_GeneSet_Definitions.xlsx
    Ignored:    data/aliquot.tsv
    Ignored:    data/azizi_curated.xlsx
    Ignored:    data/c2_human_immune_genesets.gmt
    Ignored:    data/dlbcl_total_immune_gset_v1_results.rds
    Ignored:    data/ds_JCO.18.01583-2.xlsx
    Ignored:    data/duke_study_driver_mutations_top150_maf.csv
    Ignored:    data/final_analysis_set.maf
    Ignored:    data/gdc_clinical_data.json
    Ignored:    data/gdc_files_and_case_ids.json
    Ignored:    data/gencode.v22.annotation.gtf
    Ignored:    data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Ignored:    data/genesets/
    Ignored:    data/gset_ids_complete.rds
    Ignored:    data/histo_metadata_dlbcl_landscape.csv
    Ignored:    data/journal.pone.0088309.s001.XLS
    Ignored:    data/mmc1.xlsx
    Ignored:    data/nejmoa1801445_appendix_2.xlsx
    Ignored:    data/reactome_imm_signaling.tsv
    Ignored:    data/sample.tsv
    Ignored:    data/sig.rda
    Ignored:    data/sig_genes.txt
    Ignored:    data/temp.dgelist_edger.rds
    Ignored:    data/temp.dgelist_limma.rds
    Ignored:    data/uchi_metadata.csv
    Ignored:    logs/
    Ignored:    output/.DS_Store
    Ignored:    output/NCI_mutations_vs_deconvolution.csv
    Ignored:    output/clust1_vs_others.txt
    Ignored:    output/clust1_vs_others_dge.rnk
    Ignored:    output/clust2_vs_others_dge.rnk
    Ignored:    output/clust3_vs_others_dge.rnk
    Ignored:    output/clust4_vs_others_dge.rnk
    Ignored:    output/cluster1_dge.rds
    Ignored:    output/cluster2_dge.rds
    Ignored:    output/cluster3_dge.rds
    Ignored:    output/cluster4_dge.rds
    Ignored:    output/combined_clin_data.xlsx
    Ignored:    output/combined_dlbcl_expr_matrix.csv
    Ignored:    output/combined_es_gsva2_4cluster.rds
    Ignored:    output/combined_es_gsva4_4cluster.rds
    Ignored:    output/combined_expressionset.rds
    Ignored:    output/combined_tpm_matrix.csv
    Ignored:    output/counts_expressionset_for_limma.rds
    Ignored:    output/dlbcl_expr_matrix.csv
    Ignored:    output/dlbcl_total_immune_gset_v2_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v3_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v4_results.rds
    Ignored:    output/duke_expression_set_cleaned_log.rds
    Ignored:    output/duke_expressionset.rds
    Ignored:    output/duke_expressionset_pext.rds
    Ignored:    output/duke_extended_pheno_data.csv
    Ignored:    output/expr_matrix.csv
    Ignored:    output/gsea_res_cluster1.csv
    Ignored:    output/gset_ids_complete_2.rds
    Ignored:    output/gset_ids_complete_3.rds
    Ignored:    output/gset_ids_complete_4.rds
    Ignored:    output/gsva_v2_4clust_assign.rds
    Ignored:    output/gsva_v3_4clust_assign.rds
    Ignored:    output/gsva_v4_4clust_assign.rds
    Ignored:    output/limma_voom_elist_1.rds
    Ignored:    output/merged_raw_counts.csv
    Ignored:    output/nci_dlbcl_annotation.csv
    Ignored:    output/nci_dlbcl_unprocessed_counts.csv
    Ignored:    output/nci_expressionset.rds
    Ignored:    output/nci_expressionset_pext.rds
    Ignored:    output/nci_extended_pheno_data.csv
    Ignored:    output/networkanal3_clust1.txt
    Ignored:    output/networkanal3_clust2.txt
    Ignored:    output/networkanal3_clust3.txt
    Ignored:    output/networkanal3_clust4.txt
    Ignored:    output/total_immune_gset_v1.rds
    Ignored:    output/tpm_cyt_expressionset.rds
    Ignored:    output/tpm_expressionset.rds
    Ignored:    output/uchi_cluster_assignments.csv
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.pvals.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019_RAW.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.pvals.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819_RAW.txt
    Ignored:    output/xcell_input_matrix.csv

Untracked files:
    Untracked:  NCI_mutations_vs_deconvolution.xls
    Untracked:  analysis/correlation_clusters.Rmd
    Untracked:  analysis/mif.Rmd
    Untracked:  analysis/new_signature_matrix.Rmd
    Untracked:  analysis/table1.Rmd
    Untracked:  analysis/visualizing_gsva_scores_5.Rmd
    Untracked:  batch1_cell_counts.pdf
    Untracked:  data/cibersort/
    Untracked:  data/gistic_results_nci/
    Untracked:  es_10k_hgnc/
    Untracked:  output/ASH_ORAL_figures.png
    Untracked:  output/ASH_ORAL_figures/
    Untracked:  output/Rplot.pdf
    Untracked:  output/Rplot01.pdf
    Untracked:  output/batch1_cell_seg_df.rds
    Untracked:  output/batch1_summary_cell_seg_df.rds
    Untracked:  output/colored_hclust_plot_by_group.pdf
    Untracked:  output/colored_hclust_plot_by_samples.pdf
    Untracked:  output/combined_counts_dge.rds
    Untracked:  output/combined_expr_matrix_for_signature_unprocessed.rmd
    Untracked:  output/dgelist_pheno_data.rds
    Untracked:  output/dlbcl_final_signature_matrix.tsv
    Untracked:  output/dlbcl_gsva5_4cluster_hcpc.rds
    Untracked:  output/dlbcl_gsva5_4cluster_pca.rds
    Untracked:  output/dlbcl_uchi_batch1_mixture.tsv
    Untracked:  output/duke_dlbcl_mixture.tsv
    Untracked:  output/gse_69030_probes.tsv
    Untracked:  output/hclust_labeled_by_cell_type.pdf
    Untracked:  output/mif_pilot_data_plot.pdf
    Untracked:  output/nci_dlbcl_mixture.tsv
    Untracked:  output/phenodata_for_signature_matrix.rmd
    Untracked:  output/reference_matrix_file.tsv
    Untracked:  output/table1.html
    Untracked:  output/uchi_deconvolution_batch_1.rds
    Untracked:  output/uchi_pheno_clust
    Untracked:  output/var_contribution_axis2.pdf
    Untracked:  output/venn.png
    Untracked:  output/venn.png.2020-02-23_11-17-47.log
    Untracked:  output/venn.png.2020-02-23_11-17-56.log
    Untracked:  output/venn.png.2020-02-23_11-20-44.log
    Untracked:  preprocessed_data_for_signature_matrix_es.rds

Unstaged changes:
    Modified:   .gitignore
    Deleted:    Rplot.pdf
    Deleted:    Rplot001.png
    Deleted:    Rplot002.png
    Deleted:    Rplot01.pdf
    Modified:   analysis/cyt.Rmd
    Modified:   analysis/deconvolution.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/predictor.Rmd
    Modified:   analysis/sparse_clustering.Rmd
    Modified:   analysis/visualizing_gsva_scores_4.Rmd
    Deleted:    data/DLBCL_histo/JG55/HE/14-1464/14-1464_Scan1_annotations.xml.lock
    Modified:   output/design_for_limma.rds
    Modified:   output/dgelist_for_limma.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/2a36551d1138124b9046b6a9d9117e2210bac86d/docs/gdc_download.html" target="_blank">2a36551</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/b979bc2c52f04f06681aabe8a567d360529e5a0b/analysis/gdc_download.Rmd" target="_blank">b979bc2</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-17
</td>
<td>
full pheno data added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/5f4a79befc3e839f1de95d289af56cac9869017e/docs/gdc_download.html" target="_blank">5f4a79b</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/c4a6b640bcaec15be232ae3a7aa3cabda0876e23/analysis/gdc_download.Rmd" target="_blank">c4a6b64</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
working gdc download
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/f207953e76a00cb16d1b82cfca6df07c8a6d2c88/docs/gdc_download.html" target="_blank">f207953</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/19bc0e483b8e6bed9956c9356ad409c96f85be74/docs/gdc_download.html" target="_blank">19bc0e4</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/979795d6d254203eb1bfd76c9fd2bcfeed1493a0/analysis/gdc_download.Rmd" target="_blank">979795d</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
working gdc download
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/3b64bb0ce77fdf1b1a1222107d41a87c4d0d7961/docs/gdc_download.html" target="_blank">3b64bb0</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/f0014c731bfad11e07491ff5c229b89de8f77feb/analysis/gdc_download.Rmd" target="_blank">f0014c7</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
start of analysis workbook for GDC download
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/d0188f04264d71d0a567b3f7b3e3e44c494adbd1/docs/gdc_download.html" target="_blank">d0188f0</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/b71852f0c9fa32d1608dfd0477ba319604461464/analysis/gdc_download.Rmd" target="_blank">b71852f</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
start of analysis workbook for GDC download
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/6f7a1caac2fdb07549ae1f8683109bc26c28c717/docs/gdc_download.html" target="_blank">6f7a1ca</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/61469678861a47a84ebeeca681c8895eecac03f4/analysis/gdc_download.Rmd" target="_blank">6146967</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
start of analysis workbook for GDC download
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Clear workspace</p>
<pre class="r"><code>rm(list = ls())</code></pre>
<p>All of this is recorded for completeness - this notebook is not intended to be run more than once and the global options for evaluation are set to FALSE. Once this notebook was completed, results were stored in a list of dataframes saved with the saveRDS function.</p>
<pre class="r"><code>knitr::opts_chunk$set(eval = FALSE)</code></pre>
<pre class="r"><code>library(&quot;GenomicDataCommons&quot;)
library(&quot;tidyverse&quot;)</code></pre>
<div id="test-connection" class="section level4">
<h4>Test connection</h4>
<pre class="r"><code>GenomicDataCommons::status()

# how many cases available in NCICCR
cases() %&gt;% 
  GenomicDataCommons::filter(~ project.program.name==&#39;NCICCR&#39;) %&gt;%
  facet(&#39;samples.sample_type&#39;) %&gt;% 
  aggregations()</code></pre>
</div>
</div>
<div id="download-raw-count-files" class="section level2">
<h2>Download raw count files</h2>
<pre class="r"><code># Get manifest ids for HTSeq raw counts
nci_manifest &lt;- files() %&gt;%
  GenomicDataCommons::select(available_fields(&#39;files&#39;)) %&gt;%
    GenomicDataCommons::filter(~ cases.project.project_id == &#39;NCICCR-DLBCL&#39; &amp;
             data_type==&#39;Gene Expression Quantification&#39; &amp;
             analysis.workflow_type == &#39;HTSeq - Counts&#39;) %&gt;%
    GenomicDataCommons::select(&#39;file_id&#39;) %&gt;%
    manifest()

# Get uuIDs for each file
file_ids &lt;- nci_manifest %&gt;%
  pull(id)</code></pre>
<pre class="r"><code># download all selected files
library(BiocParallel)
register(MulticoreParam())
destdir = tmpdir()
fnames = gdcdata(file_ids)</code></pre>
<p>Move files out of folders and unzip</p>
<pre class="bash"><code>## done in command line, stored here for documentation only - not part of notebook
## get files out of subfolders and move to parent
# find . -mindepth 2 -type f -print -exec mv {} . \;

## create a folder for counts
# mkdir htseq_counts

## move all sequence files into a folder
# find . -name &#39;*.htseq_counts.txt.gz&#39; -exec mv {} ~/Library/Caches/GenomicDataCommons/htseq_counts \;

## unzip
# find . -name &#39;*.htseq_counts.txt.gz&#39; -exec gunzip {} \;

## move to local repo data folder
# find . -name &#39;*.htseq_counts.txt&#39; -exec mv {} ~/dlbcl_landscape/data/htseq_counts \;
</code></pre>
<p>Check for missing files</p>
<pre class="r"><code># get filenames that were downloaded
downloaded_fnames &lt;- list.files(&quot;data/htseq_counts/&quot;, all.files = FALSE) 

# get filenames from manifest
manifest_fnames &lt;- nci_manifest %&gt;% pull(filename) %&gt;% str_replace(&quot;.gz&quot;, &quot;&quot;)

# find the difference
missing_files &lt;- setdiff(manifest_fnames, downloaded_fnames)
missing_files
# no missing files</code></pre>
</div>
<div id="initial-processing" class="section level2">
<h2>Initial processing</h2>
<p>Combine files into dataframe</p>
<pre class="r"><code># get a vector of filepaths
filepaths &lt;- nci_manifest %&gt;% dplyr::select(filename) %&gt;% 
    mutate(filename = (str_replace(filename, &quot;.gz&quot;, &quot;&quot;))) %&gt;%
    mutate(filepath = (str_c(&quot;data/htseq_counts/&quot;, filename))) %&gt;%
  pull(filepath)</code></pre>
<pre class="r"><code># read in files
nci_dlbcl_df &lt;- map(filepaths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# set column names
cols &lt;- paste0(&quot;nci_dlbcl_&quot;, seq(1, (ncol(nci_dlbcl_df) - 1)))
allcols &lt;- c(&quot;gene&quot;, cols)
colnames(nci_dlbcl_df) &lt;- allcols

# start metadata with manifest
enframe(cols) 
nci_dlbcl_pheno &lt;- add_column(nci_manifest, cols, .before = 1)
nci_dlbcl_pheno &lt;- dplyr::rename(nci_dlbcl_pheno, sample = cols)

# review results
nci_dlbcl_pheno
nci_dlbcl_df</code></pre>
<p>Get additional metadata</p>
<pre class="r"><code># build query
qfiles = files() %&gt;% GenomicDataCommons::filter( ~ cases.project.project_id == &#39;NCICCR-DLBCL&#39; &amp;
                                 type == &#39;gene_expression&#39; &amp;
                                 analysis.workflow_type == &#39;HTSeq - Counts&#39;)

# get results
extended_metadata &lt;- results_all(qfiles)
extended_metadata &lt;- extended_metadata %&gt;% as_tibble() %&gt;%
  dplyr::select(-state, -acl)

# combine with existing phenodata
nci_dlbcl_annotation &lt;- nci_dlbcl_pheno %&gt;%
  left_join(extended_metadata, by = &quot;id&quot;) %&gt;%
  print()</code></pre>
</div>
<div id="phenotype-data" class="section level2">
<h2>Phenotype data</h2>
<p>Get phenotype data from GenomicDataCommons</p>
<pre class="r"><code># get linked htseq count filenames / caseID from GDC web portal as JSON
# saved as &quot;data/gdc_files_and_case_ids.json&quot;
library(jsonlite)
filescases &lt;- fromJSON(&quot;data/gdc_files_and_case_ids.json&quot;, 
                       flatten = TRUE) %&gt;%
  as_tibble() %&gt;%
  print()

# get case ids out of nested table structure
casefiles &lt;- filescases$cases %&gt;% 
  bind_rows() %&gt;% 
  dplyr::select(case_id) %&gt;%
  bind_cols(filescases) %&gt;%
  as_tibble() %&gt;%
  dplyr::select(case_id, file_name) %&gt;%
  print()

# get linked case IDs and case uuIDs from GDC here: https://portal.gdc.cancer.gov/repository?facetTab=cases&amp;filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.program.name%22%2C%22value%22%3A%5B%22NCICCR%22%5D%7D%7D%5D%7D&amp;searchTableTab=cases
# exported as TSV (two tables, aliquot.tsv and sample.tsv)
# moved files to data folder
sample_ids &lt;- read_tsv(&quot;data/sample.tsv&quot;)
str(sample_ids)

phenodata &lt;- casefiles %&gt;%
  left_join(sample_ids) %&gt;%
  dplyr::select(case_submitter_id, everything()) %&gt;%
  print()

# download clinical data from GDC as JSON and import
clinicaldata &lt;- fromJSON(&quot;data/gdc_clinical_data.json&quot;, 
                       flatten = TRUE) %&gt;%
  as_tibble() %&gt;%
  print()

diag_df &lt;- clinicaldata$diagnoses %&gt;%
  bind_rows() %&gt;%
  as_tibble() %&gt;%
  mutate(case_submitter_id = str_replace(submitter_id,
                                         &quot;-diagnosis&quot;,
                                         &quot;&quot;)) %&gt;%
  dplyr::select(case_submitter_id, everything()) %&gt;%
  print()

demo_df &lt;- clinicaldata %&gt;%
  dplyr::select(-diagnoses) %&gt;%
  mutate(case_submitter_id = str_replace(demographic.submitter_id,
                                         &quot;-demographic&quot;,
                                         &quot;&quot;)) %&gt;%
  dplyr::select(case_submitter_id, everything()) %&gt;%
  print()

pheno_data &lt;- phenodata %&gt;%
  left_join(diag_df) %&gt;%
  left_join(demo_df) %&gt;%
  print()

# remove columns with all NA
not_all_na &lt;- function(x) {!all(is.na(x))}
pheno_data &lt;- pheno_data %&gt;% dplyr::select_if(not_all_na)
pheno_data

# add in PDL1-amp calls from James&#39; Blood paper project using NCI data
# table of calls imported to data folder
pdl1_amp_csv &lt;- read_csv(&quot;data/NIH_PDL1_amp_cases.csv&quot;)
pdl1_nonamp_csv &lt;- read_csv(&quot;data/NIH_PDL1_nonamp_cases.csv&quot;)
pdl1_nonamp_cases &lt;- pdl1_nonamp_csv %&gt;% 
  dplyr::select(key) %&gt;% 
  dplyr::rename(cases = key) %&gt;%
  mutate(status = rep(&quot;non-amplified&quot;, length = nrow(pdl1_nonamp_csv))) %&gt;%
  print()
pdl1_amp_cases &lt;- pdl1_amp_csv %&gt;%
  dplyr::rename(cases = key) %&gt;%
  mutate(status = ifelse(value == 1, &quot;low_amplified&quot;, 
                         ifelse(value == 2, &quot;high_amplified&quot;, NA))) %&gt;%
  dplyr::select(cases, status) %&gt;%
  print()
summary(as.factor(pdl1_amp_cases$status))
all_pdl1_status &lt;- bind_rows(pdl1_amp_cases, pdl1_nonamp_cases) %&gt;%
  mutate(pdl1_status = as.factor(status)) %&gt;%
  dplyr::rename(case_submitter_id = cases) %&gt;%
  dplyr::select(case_submitter_id, pdl1_status)
summary(all_pdl1_status$pdl1_status)

# add to pheno_data
pheno_data_pdl1 &lt;- pheno_data %&gt;%
  left_join(all_pdl1_status) %&gt;%
  dplyr::select(case_submitter_id, case_id, pdl1_status, everything()) %&gt;%
  print()
summary(pheno_data_pdl1$pdl1_status)

# pull in temporary sample names
nci_dlbcl_pheno

pheno_data_samples &lt;- pheno_data_pdl1 %&gt;%
  rename(filename = file_name) %&gt;%
  left_join(nci_dlbcl_pheno, by = &quot;filename&quot;) %&gt;%
  dplyr::select(case_submitter_id, sample, everything()) %&gt;%
  print()</code></pre>
<pre class="r"><code># rename the expression matrix columns with meaningful sample names
pheno_data_ids &lt;- pheno_data_samples %&gt;%
  dplyr::select(case_submitter_id, sample)
id_frame &lt;- (colnames(nci_dlbcl_df)) %&gt;%
  enframe() %&gt;%
  dplyr::select(sample = value) %&gt;%
  left_join(pheno_data_ids) %&gt;%
  print()
id_frame$case_submitter_id[1] &lt;- &quot;gene&quot;
id_frame
cnames &lt;- id_frame %&gt;% pull(case_submitter_id)

colnames(nci_dlbcl_df) &lt;- cnames</code></pre>
</div>
<div id="save-for-later-use" class="section level2">
<h2>Save for later use</h2>
<pre class="r"><code>write_csv(nci_dlbcl_df, &quot;output/nci_dlbcl_unprocessed_counts.csv&quot;)
write_csv(pheno_data_samples, &quot;output/nci_dlbcl_annotation.csv&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
