<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Immune Landscape of DLBCL</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Immune Landscape of DLBCL</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreed17eebb851fb7365886b73bbec7b8693904e53betargetblankd17eebba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/d17eebb851fb7365886b73bbec7b8693904e53be" target="_blank">d17eebb</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreed17eebb851fb7365886b73bbec7b8693904e53betargetblankd17eebba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/enrichment.Rmd
    Untracked:  analysis/newgs.Rmd
    Untracked:  code/gsva1.r
    Untracked:  code/gsva1.sh
    Untracked:  code/run_gsva1.pbs
    Untracked:  data/NIH_PDL1_amp_cases.csv
    Untracked:  data/NIH_PDL1_nonamp_cases.csv
    Untracked:  data/aliquot.tsv
    Untracked:  data/c2_human_immune_genesets.gmt
    Untracked:  data/dlbcl_total_immune_gset_v1_results.rds
    Untracked:  data/gdc_clinical_data.json
    Untracked:  data/gdc_files_and_case_ids.json
    Untracked:  data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Untracked:  data/gset_ids_complete.rds
    Untracked:  data/htseq_counts/
    Untracked:  data/mmc1.xlsx
    Untracked:  data/nejmoa1801445_appendix_2.xlsx
    Untracked:  data/sample.tsv
    Untracked:  data/temp.dgelist_limma.rds
    Untracked:  docs/assets/
    Untracked:  output/combined_clin_data.xlsx
    Untracked:  output/dlbcl_expr_matrix.csv
    Untracked:  output/duke_expression_set_cleaned_log.rds
    Untracked:  output/duke_expressionset.rds
    Untracked:  output/duke_expressionset_pext.rds
    Untracked:  output/duke_extended_pheno_data.csv
    Untracked:  output/expr_matrix.csv
    Untracked:  output/nci_dlbcl_annotation.csv
    Untracked:  output/nci_dlbcl_unprocessed_counts.csv
    Untracked:  output/nci_expressionset.rds
    Untracked:  output/nci_expressionset_pext.rds
    Untracked:  output/nci_extended_pheno_data.csv
    Untracked:  output/total_immune_gset_v1.rds

Unstaged changes:
    Modified:   analysis/ext_pheno.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/d17eebb851fb7365886b73bbec7b8693904e53be/analysis/index.Rmd" target="_blank">d17eebb</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-30
</td>
<td>
added images and text to homepage describing first GSVA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/5a57ec091786bd8b0362d31767fd966d90be89cd/docs/index.html" target="_blank">5a57ec0</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/42da9d5893be09775a582da78b8bec151d4ea9bf/analysis/index.Rmd" target="_blank">42da9d5</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-28
</td>
<td>
major overhaul of Duke count processing, updates of NCI pheno data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/afd41f8a903eb6c87869616709a00cac81ecf545/docs/index.html" target="_blank">afd41f8</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/a1df19d8d6eafa38667f58737de22a01c88bbff3/analysis/index.Rmd" target="_blank">a1df19d</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-27
</td>
<td>
updated homepage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/a566468c2e368ab5001aa9d93dffb82bd831349a/docs/index.html" target="_blank">a566468</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/ff61a74b9b7400ba96a6dd2548123f915f1ddcae/analysis/index.Rmd" target="_blank">ff61a74</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-19
</td>
<td>
updated links to notebooks
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/1cfd6533ee179611329cf2e876b07a91722aaaef/docs/index.html" target="_blank">1cfd653</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/d87cb437c131c87fa6ad8f947b608f1b8413a62f/analysis/index.Rmd" target="_blank">d87cb43</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/a3fea9cdc07c65623dfe9a37c68a8610f895e840/docs/index.html" target="_blank">a3fea9c</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/275aead0cc6b8c4f5fdbbec934af4036875148cc/analysis/index.Rmd" target="_blank">275aead</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/4c6b59d40c5973962b3750482dd92624fe1e3304/docs/index.html" target="_blank">4c6b59d</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/6d82275e30ce9994a8784f66a546f90ffc2cb281/docs/index.html" target="_blank">6d82275</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/b42800d1b3a63f1d82fbab3409e7e6f7409af15f/analysis/index.Rmd" target="_blank">b42800d</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
update title
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/02539e99e7c6fbe6aa9e60b77e8e8c69f987ca4f/docs/index.html" target="_blank">02539e9</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/a254509286997206e85255831484905ab3f3862e/analysis/index.Rmd" target="_blank">a254509</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/8582b744481b5df2fdb3929e2950a5f27e5388cb/analysis/index.Rmd" target="_blank">8582b74</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Welcome to the computational lab notebook for our research project focused on profiling the immune microenvironment of DLBCL using gene expression data and multi-channel immunohistochemistry, then linking clusters of similar immune signatures with mutations in tumors.</p>
<p>The over-arching questions are:</p>
<ol style="list-style-type: decimal">
<li>Are there recurrent immune phenotypes in DLBCL?</li>
<li>What is the influence of tumor gene mutations on immune phenotypes in DLBCL?</li>
</ol>
<p>The primary data sources are:</p>
<ol style="list-style-type: decimal">
<li>Duke dataset
<ul>
<li>From <a href="https://www.ncbi.nlm.nih.gov/pubmed/28985567">Reddy et al Cell. 2017 Oct 5;171(2):481</a></li>
<li>Hosted at the European Genome-phenome Archive <a href="https://www.ebi.ac.uk/ega/datasets/EGAD00001003600">EGAD00001003600</a></li>
</ul></li>
<li>NCI dataset
<ul>
<li>From <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=29641966">Schmitz et al N Engl J Med. 2018 Apr 12;378(15):1396</a><br />
</li>
<li>Hosted at the Genomic Data Commons project ID <a href="https://portal.gdc.cancer.gov/projects/NCICCR-DLBCL">NCICCR-DLBCL</a></li>
</ul></li>
</ol>
<div id="aim-1-identify-recurrent-immune-phenotypes-using-rnaseq-data" class="section level2">
<h2>Aim 1: Identify Recurrent Immune Phenotypes using RNAseq Data</h2>
<div id="prepare-rnaseq-data-for-analysis" class="section level3">
<h3>Prepare RNAseq data for analysis</h3>
<ul>
<li><a href="gdc_download.html">Download raw RNAseq counts in NCI lymphoma dataset from GDC</a></li>
<li><a href="preprocessing_count_data.html">Preprocessing NCI counts</a></li>
<li>Pseudoalign and count Duke RNAseq sequence data
<ul>
<li>Run on Gardner HPC cluster</li>
<li>Source code in <code>/code/</code>)</li>
</ul></li>
<li><a href="dukedata.html">Preprocessing Duke counts</a></li>
<li><a href="ext_pheno.html">Expanded phenotype data for Duke and NCI datasets</a></li>
<li><a href="merge.html">Merge expression and associated clinical/metadata, batch correction</a></li>
</ul>
</div>
<div id="gene-set-variation-analysis" class="section level3">
<h3>Gene set variation analysis</h3>
<div id="first-try-at-gene-list-collection" class="section level4">
<h4>First try at gene list collection:</h4>
<p>Gene lists selected from work-in-progress analyis of TCGA macrophage signatures and from immune-related C2 “curated” gene sets from MSigDB. Also included T-cell inflamed signature from prior publications (Riyue Bao). No additional filtering was done after selection of candidate gene lists. GSVA run on expression data from NCI only to see how well it recapitulates known <em>PD-L1</em> amplified immune phenotype.</p>
<ul>
<li><a href="enrichment.html">GSVA</a></li>
<li><a href="clustering_enrichment_scores.html">Clustering GSVA results</a></li>
</ul>
<p>There is poor separation, especially on PCA2, of the known immune phenotype <em>PD-L1</em> amplified in NCI data.</p>
<p><img src="assets/first_gsva_pca_samples.png" /></p>
<p>This is because the gene sets are highly clustered in the same axis on the PCA analysis. There is too much co-variation among the immune gene sets I initially selected, and the overall analysis is not able to find meaningul differences in immune phenotype.</p>
<p><img src="assets/first_gsva_pca_variables.png" /></p>
</div>
<div id="gene-set-collection-refinement" class="section level4">
<h4>Gene set collection refinement:</h4>
<p>We will need better-separating gene sets to proceed.</p>
<ul>
<li><a href="newgs.html">New gene set collection</a></li>
</ul>
</div>
</div>
</div>
<div id="aim-2-link-immune-phenotype-clusters-to-oncogenes" class="section level2">
<h2>Aim 2: Link Immune Phenotype Clusters to Oncogenes</h2>
<div id="mutation-calls" class="section level3">
<h3>Mutation calls</h3>
</div>
</div>
<div id="aim-3-visualize-the-immune-phenotype-with-multi-channel-ihc" class="section level2">
<h2>Aim 3: Visualize the Immune Phenotype with Multi-Channel IHC</h2>
</div>
<div id="sub-aims" class="section level2">
<h2>Sub-aims:</h2>
<div id="sub-aim-1-pd-l1-amplified-cohort" class="section level4">
<h4>Sub-aim 1: PD-L1 amplified cohort</h4>
</div>
<div id="sub-aim-2-additional-gene-signatures-cytolytic-and-double-hit" class="section level4">
<h4>Sub-aim 2: Additional gene signatures: cytolytic and double hit</h4>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
