<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Immune Landscape of DLBCL</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Immune Landscape of DLBCL</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-07-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreeee33393a83e190aee226c5dd3c12adc406d51c8btargetblankee33393a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/ee33393a83e190aee226c5dd3c12adc406d51c8b" target="_blank">ee33393</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetreeee33393a83e190aee226c5dd3c12adc406d51c8btargetblankee33393a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/1-s2.0-S1074761316304320-mmc2.xlsx
    Ignored:    data/12864_2005_498_MOESM2_ESM.xls
    Ignored:    data/41591_2018_45_MOESM5_ESM.xlsx
    Ignored:    data/NIH_PDL1_amp_cases.csv
    Ignored:    data/NIH_PDL1_nonamp_cases.csv
    Ignored:    data/PanImmune_GeneSet_Definitions.xlsx
    Ignored:    data/aliquot.tsv
    Ignored:    data/azizi_curated.xlsx
    Ignored:    data/c2_human_immune_genesets.gmt
    Ignored:    data/dlbcl_total_immune_gset_v1_results.rds
    Ignored:    data/ds_JCO.18.01583-2.xlsx
    Ignored:    data/duke_study_driver_mutations_top150_maf.csv
    Ignored:    data/final_analysis_set.maf
    Ignored:    data/gdc_clinical_data.json
    Ignored:    data/gdc_files_and_case_ids.json
    Ignored:    data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Ignored:    data/genesets/
    Ignored:    data/gset_ids_complete.rds
    Ignored:    data/htseq_counts/
    Ignored:    data/journal.pone.0088309.s001.XLS
    Ignored:    data/mmc1.xlsx
    Ignored:    data/nejmoa1801445_appendix_2.xlsx
    Ignored:    data/reactome_imm_signaling.tsv
    Ignored:    data/sample.tsv
    Ignored:    data/sig.rda
    Ignored:    data/sig_genes.txt
    Ignored:    data/temp.dgelist_edger.rds
    Ignored:    data/temp.dgelist_limma.rds
    Ignored:    docs/.DS_Store
    Ignored:    docs/assets/Rplot01.pdf
    Ignored:    docs/assets/Rplot02.pdf
    Ignored:    docs/assets/Rplot03.pdf
    Ignored:    docs/assets/gsva4_heatmap_rawscores.pdf
    Ignored:    output/clust1_vs_others.txt
    Ignored:    output/clust1_vs_others_dge.rnk
    Ignored:    output/clust2_vs_others_dge.rnk
    Ignored:    output/clust3_vs_others_dge.rnk
    Ignored:    output/clust4_vs_others_dge.rnk
    Ignored:    output/cluster1_dge.rds
    Ignored:    output/cluster2_dge.rds
    Ignored:    output/cluster3_dge.rds
    Ignored:    output/cluster4_dge.rds
    Ignored:    output/combined_clin_data.xlsx
    Ignored:    output/combined_dlbcl_expr_matrix.csv
    Ignored:    output/combined_es_gsva2_4cluster.rds
    Ignored:    output/combined_es_gsva4_4cluster.rds
    Ignored:    output/combined_expressionset.rds
    Ignored:    output/combined_tpm_matrix.csv
    Ignored:    output/counts_expressionset_for_limma.rds
    Ignored:    output/dlbcl_expr_matrix.csv
    Ignored:    output/dlbcl_total_immune_gset_v2_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v3_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v4_results.rds
    Ignored:    output/duke_expression_set_cleaned_log.rds
    Ignored:    output/duke_expressionset.rds
    Ignored:    output/duke_expressionset_pext.rds
    Ignored:    output/duke_extended_pheno_data.csv
    Ignored:    output/expr_matrix.csv
    Ignored:    output/gsea_res_cluster1.csv
    Ignored:    output/gset_ids_complete_2.rds
    Ignored:    output/gset_ids_complete_3.rds
    Ignored:    output/gset_ids_complete_4.rds
    Ignored:    output/gsva_v2_4clust_assign.rds
    Ignored:    output/gsva_v3_4clust_assign.rds
    Ignored:    output/gsva_v4_4clust_assign.rds
    Ignored:    output/mc1_score_matrix.csv
    Ignored:    output/mc2_score_matrix.csv
    Ignored:    output/mc3_score_matrix.csv
    Ignored:    output/mc4_score_matrix.csv
    Ignored:    output/mc5_score_matrix.csv
    Ignored:    output/mc6_score_matrix.csv
    Ignored:    output/mc7_score_matrix.csv
    Ignored:    output/merged_raw_counts.csv
    Ignored:    output/merged_raw_counts_1.csv
    Ignored:    output/merged_raw_counts_2.csv
    Ignored:    output/merged_raw_counts_3.csv
    Ignored:    output/merged_raw_counts_4.csv
    Ignored:    output/merged_raw_counts_5.csv
    Ignored:    output/merged_raw_counts_6.csv
    Ignored:    output/merged_raw_counts_7.csv
    Ignored:    output/nci_dlbcl_annotation.csv
    Ignored:    output/nci_dlbcl_unprocessed_counts.csv
    Ignored:    output/nci_expressionset.rds
    Ignored:    output/nci_expressionset_pext.rds
    Ignored:    output/nci_extended_pheno_data.csv
    Ignored:    output/networkanal3_clust1.txt
    Ignored:    output/networkanal3_clust2.txt
    Ignored:    output/networkanal3_clust3.txt
    Ignored:    output/networkanal3_clust4.txt
    Ignored:    output/total_immune_gset_v1.rds
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.pvals.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019_RAW.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.pvals.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819_RAW.txt
    Ignored:    output/xcell_input_matrix.csv

Untracked files:
    Untracked:  code/limmavoom.r
    Untracked:  code/predictionmodel.r
    Untracked:  data/gencode.v22.annotation.gtf
    Untracked:  output/clust1_mut_ranked_maf05.tsv
    Untracked:  output/clust1_mut_ranked_p05.tsv
    Untracked:  output/design_for_limma.rds
    Untracked:  output/dgelist_for_limma.rds
    Untracked:  output/gencode_v22_exonic_gene_sizes.rds
    Untracked:  output/tpm_expressionset.rds

Unstaged changes:
    Modified:   analysis/assign_mutations.Rmd
    Modified:   analysis/cluster_dge.Rmd
    Modified:   analysis/deconvolution.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/ee33393a83e190aee226c5dd3c12adc406d51c8b/analysis/index.Rmd" target="_blank">ee33393</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-23
</td>
<td>
updated homepage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/cc1e45ead669caefaae22546681be422e0a9db2a/docs/index.html" target="_blank">cc1e45e</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/79273ad25d94576529a78b683d26c0239d0f5c42/analysis/index.Rmd" target="_blank">79273ad</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-21
</td>
<td>
sparse hierarchical clustering
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/e4fde8a869aede82fe7210e8aceeae97922ff4d3/docs/index.html" target="_blank">e4fde8a</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/01e4acfb63eb651b4b0785b5ae5c794d1d7bda67/analysis/index.Rmd" target="_blank">01e4acf</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-15
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/20609e01e9a35c09bb18fd5130704d6f8a7fec3a/docs/index.html" target="_blank">20609e0</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/3f5de36a8c2cf46ebd8aa75a9e3262d8833f8964/analysis/index.Rmd" target="_blank">3f5de36</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-15
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/45fe8f275ba735d518c341d931bf62c5d296d03b/docs/index.html" target="_blank">45fe8f2</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/2f549bf07b22b250dc450889671d6d8933b654fe/docs/index.html" target="_blank">2f549bf</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/803095951a60c73e0fe43e51336f6e014908fbaa/analysis/index.Rmd" target="_blank">8030959</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-03
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/28102e4b3f0566cd16eee8eabc29e4858db98476/docs/index.html" target="_blank">28102e4</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/1453a6cecf91ea4aee54c16b64ff1dfcd2d532a8/analysis/index.Rmd" target="_blank">1453a6c</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-02
</td>
<td>
updating index and titles
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/a8762f34af3540609d07fdabcea42e253c8ee1b9/docs/index.html" target="_blank">a8762f3</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/9d8dc7a36480ae86a82784f20a0e88f7db18b5b3/analysis/index.Rmd" target="_blank">9d8dc7a</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-02
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/8406e0a26287a321a82addbce6db46618a7026a8/docs/index.html" target="_blank">8406e0a</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/d17eebb851fb7365886b73bbec7b8693904e53be/analysis/index.Rmd" target="_blank">d17eebb</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-30
</td>
<td>
added images and text to homepage describing first GSVA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/5a57ec091786bd8b0362d31767fd966d90be89cd/docs/index.html" target="_blank">5a57ec0</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/42da9d5893be09775a582da78b8bec151d4ea9bf/analysis/index.Rmd" target="_blank">42da9d5</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-28
</td>
<td>
major overhaul of Duke count processing, updates of NCI pheno data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/afd41f8a903eb6c87869616709a00cac81ecf545/docs/index.html" target="_blank">afd41f8</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/a1df19d8d6eafa38667f58737de22a01c88bbff3/analysis/index.Rmd" target="_blank">a1df19d</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-27
</td>
<td>
updated homepage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/a566468c2e368ab5001aa9d93dffb82bd831349a/docs/index.html" target="_blank">a566468</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/ff61a74b9b7400ba96a6dd2548123f915f1ddcae/analysis/index.Rmd" target="_blank">ff61a74</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-19
</td>
<td>
updated links to notebooks
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/1cfd6533ee179611329cf2e876b07a91722aaaef/docs/index.html" target="_blank">1cfd653</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/d87cb437c131c87fa6ad8f947b608f1b8413a62f/analysis/index.Rmd" target="_blank">d87cb43</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-14
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/a3fea9cdc07c65623dfe9a37c68a8610f895e840/docs/index.html" target="_blank">a3fea9c</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/275aead0cc6b8c4f5fdbbec934af4036875148cc/analysis/index.Rmd" target="_blank">275aead</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-14
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/4c6b59d40c5973962b3750482dd92624fe1e3304/docs/index.html" target="_blank">4c6b59d</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/6d82275e30ce9994a8784f66a546f90ffc2cb281/docs/index.html" target="_blank">6d82275</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/b42800d1b3a63f1d82fbab3409e7e6f7409af15f/analysis/index.Rmd" target="_blank">b42800d</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-14
</td>
<td>
update title
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/02539e99e7c6fbe6aa9e60b77e8e8c69f987ca4f/docs/index.html" target="_blank">02539e9</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/a254509286997206e85255831484905ab3f3862e/analysis/index.Rmd" target="_blank">a254509</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-14
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/8582b744481b5df2fdb3929e2950a5f27e5388cb/analysis/index.Rmd" target="_blank">8582b74</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-14
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Welcome to the computational lab notebook for our research project focused on profiling the immune microenvironment of DLBCL using gene expression data and multi-channel immunohistochemistry, then linking clusters of similar immune signatures with mutations in tumors.</p>
<p><a href="https://docs.google.com/presentation/d/1sKzira205EXmpdX3D1h6Hy_67gWcaE9-PLBPEcOIE5g/edit?usp=sharing">Slides containing work-in-progress figures</a></p>
</div>
<div id="background" class="section level2">
<h2>Background</h2>
<p>The over-arching questions are:</p>
<ol style="list-style-type: decimal">
<li>Are there recurrent immune phenotypes in DLBCL?</li>
<li>What is the influence of tumor gene mutations on immune phenotypes in DLBCL?</li>
</ol>
<p>The primary data sources are:</p>
<ol style="list-style-type: decimal">
<li>Duke dataset
<ul>
<li>From <a href="https://www.ncbi.nlm.nih.gov/pubmed/28985567">Reddy et al Cell. 2017 Oct 5;171(2):481</a></li>
<li>Hosted at the European Genome-phenome Archive <a href="https://www.ebi.ac.uk/ega/datasets/EGAD00001003600">EGAD00001003600</a></li>
</ul></li>
<li>NCI dataset
<ul>
<li>From <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=29641966">Schmitz et al N Engl J Med. 2018 Apr 12;378(15):1396</a><br />
</li>
<li>Hosted at the Genomic Data Commons project ID <a href="https://portal.gdc.cancer.gov/projects/NCICCR-DLBCL">NCICCR-DLBCL</a></li>
</ul></li>
</ol>
</div>
<div id="aim-1-identify-recurrent-immune-phenotypes-using-rnaseq-data" class="section level2">
<h2>Aim 1: Identify Recurrent Immune Phenotypes using RNAseq Data</h2>
<div id="prepare-rnaseq-data-for-analysis" class="section level3">
<h3>Prepare RNAseq data for analysis</h3>
<p>All steps in data download, cleaning, and analysis are documented in R markdown notebooks listed below or in source code in <code>/code/</code> folder in github repo.</p>
<ul>
<li><a href="gdc_download.html">Download raw RNAseq counts in NCI lymphoma dataset from GDC</a></li>
<li><a href="preprocessing_count_data.html">Preprocessing NCI counts</a></li>
<li>Pseudoalign and count Duke RNAseq sequence data
<ul>
<li>Run on Gardner HPC cluster</li>
<li>Source code in <code>/code/</code>)</li>
</ul></li>
<li><a href="dukedata.html">Preprocessing Duke counts</a></li>
<li><a href="ext_pheno.html">Expanded phenotype data for Duke and NCI datasets</a></li>
<li><a href="merge.html">Merge expression and associated clinical/metadata, batch correction</a></li>
</ul>
</div>
<div id="gene-set-variation-analysis" class="section level3">
<h3>Gene set variation analysis</h3>
<div id="first-try-at-gene-list-collection" class="section level4">
<h4>First try at gene list collection:</h4>
<p>Gene lists selected from work-in-progress analyis of TCGA macrophage signatures and from immune-related C2 “curated” gene sets from MSigDB. Also included T-cell inflamed signature from prior publications (Riyue Bao). No additional filtering was done after selection of candidate 49 gene lists. GSVA run on expression data from NCI only to see how well it recapitulates known <em>PD-L1</em> amplified immune phenotype.</p>
<ul>
<li><a href="enrichment.html">Data prep for GSVA</a></li>
<li>GSVA analysis run on Gardner Supercomputer (HPC) Cluster at UChicago
<ul>
<li>Source code in <code>/code/</code></li>
</ul></li>
<li><a href="clustering_enrichment_scores.html">Visualizing and clustering GSVA results</a></li>
</ul>
<p>There is poor separation, especially on PCA2, of the known immune phenotype <em>PD-L1</em> amplified in NCI data. Contribution of individual gene sets to the PCA axes shows high concentration pointing towards (0,1) on a unit circle and few parallel to the PC2 axis. Will try to find additional gene sets that can separate immune phenotypes better.</p>
</div>
<div id="second-and-third-gene-list-collections" class="section level4">
<h4>Second and third gene list collections:</h4>
<ul>
<li><a href="newgs.html">New gene set collections</a></li>
</ul>
<p>Downloaded and formatted 83 gene sets first published in Thorsson et al. Also found an additional 60 gene sets from recent publications, especially focused on recent single-cell RNAseq publications of tumor-infiltrating immune cells.</p>
<ul>
<li><a href="visualizing_gsva_scores_2.html">Visualizing and clustering GSVA results using recently published gene sets</a></li>
<li><a href="visualizing_gsva_scores_3.html">Visualizing and clustering GSVA results using Thorsson et al gene sets</a></li>
<li><a href="visualizing_gsva_scores_4.html">Visualizing and clustering GSVA results using a combination of recently published and Thorsson et al gene sets</a></li>
</ul>
<p>Both new collections of gene sets perform better than original MSigDB in separating known phenotype PD-L1, and have a broader distribution of vector angles in contributions of each gene set to the principal component axes. A four-cluster model was found to best fit the data in both cases, with 75% identical assignments and 87% Pearson concordance with cluster assignment treated as a numerical value. Several gene sets were removed from Thorsson et al and combined GSVA analyses due to multicollinearity. See <a href="visualizing_gsva_scores_3.html">GSVA v3 notebook</a> and <a href="visualizing_gsva_scores_4.html">GSVA v4 notebook</a> for details.</p>
</div>
</div>
<div id="clustering" class="section level3">
<h3>Clustering</h3>
<p>4-cluster model assignments made using HCPC on the first 10 principal components of GSVA v4 (combination of Thorsson and new gene set) scores. See <a href="(visualizing_gsva_scores_4.html)">notebook</a> for details.</p>
<p>Also tested sparse heirarchical clustering to reduce number of contributing gene sets via L1 (Lasso-like) mechanism of iteratively forcing coefficients of each variable toward 0 while optimizing the gap statistic. This resulted in selection of 35 gene sets which could explain a high fraction of the total variability, unfortunately these were predominantly T-cell-related gene sets which were already overrepresented in the original gene set list due to characteristics of T-cells and larger number of studies focused on characterizing T-cells. The overall clustering was poorer, and was essentially a gradient on these closely related T-cell, IFN-gamma axis. This model was discarded in favor of noise reduction with HCPC of the first 10 principal components as described above.</p>
<ul>
<li><a href="sparse_clustering.html">Sparse hierarchical clustering</a></li>
</ul>
</div>
<div id="describe-the-clusters" class="section level3">
<h3>Describe the clusters</h3>
<div id="differential-gene-expression" class="section level4">
<h4>Differential gene expression</h4>
<ul>
<li><a href="cluster_dge.html">Differential gene expression by cluster</a></li>
<li><a href="enrichment.html">Differential gene set expression by cluster</a></li>
</ul>
</div>
<div id="deconvolution-of-immune-cell-types" class="section level4">
<h4>Deconvolution of immune cell types</h4>
<ul>
<li><a href="deconvolution.html">xCell deconvolution</a></li>
</ul>
</div>
</div>
</div>
<div id="aim-2-link-immune-phenotype-clusters-to-oncogenes" class="section level2">
<h2>Aim 2: Link Immune Phenotype Clusters to Oncogenes</h2>
<div id="mutation-calls" class="section level3">
<h3>Mutation calls</h3>
<ul>
<li>Somatic mutation pipeline run on HPC cluster. Source code and scripts in <a href="https://github.com/mleukam/dlbcl_landscape/tree/master/code">/code/</a></li>
</ul>
<p>Input data: * Duke: somatic mutations in top 150 driver genes (supplemental data from Reddy et al Cell 2017) * NCI: non-silent somatic mutations from all samples (supplemental data from Schmitz et al NEJM 2018)</p>
<p>NCI data assessed for likelihood of driver status using MutSig2CV. Only those genes with q score &lt; 0.1 included in downstream analysis.</p>
<p>Mutation data collapsed to one mutation per gene per patient for statistical tests of proportionality.</p>
<ul>
<li><a href="assign_mutations.Rmd">Identify significantly different mutations by cluster</a></li>
</ul>
</div>
</div>
<div id="aim-3-investigate-primary-samples" class="section level2">
<h2>Aim 3: Investigate Primary Samples</h2>
<div id="derive-predictor" class="section level4">
<h4>Derive predictor</h4>
<p>Derive predictive algorithm using gene expression data for cluster assignment.</p>
<ul>
<li><a href="predictor.html">Prediction algorithm</a></li>
</ul>
</div>
<div id="multi-channel-ihc" class="section level4">
<h4>Multi-Channel IHC</h4>
<div id="antibody-panel" class="section level5">
<h5>Antibody panel:</h5>
<table>
<thead>
<tr class="header">
<th>Marker</th>
<th>Cell type</th>
<th>Company</th>
<th>Product ID</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CD4</td>
<td>Th and Treg</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>CD8</td>
<td>CTL</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>FoxP3</td>
<td>Treg</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>CD68</td>
<td>Macrophage</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>CD11c</td>
<td>DC</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>PAX5</td>
<td>B-cell</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>DAPI</td>
<td>all</td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
