<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-06-27" />

<title>Merge Duke and NCI datasets</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Merge Duke and NCI datasets</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-06-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190613code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190613)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190613code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190613)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetree3b9db7d138cc2bb849acc70f8218adc98120712atargetblank3b9db7da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/3b9db7d138cc2bb849acc70f8218adc98120712a" target="_blank">3b9db7d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetree3b9db7d138cc2bb849acc70f8218adc98120712atargetblank3b9db7da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    NCI_mutations_vs_deconvolution.csv
    Ignored:    data/.DS_Store
    Ignored:    data/1-s2.0-S1074761316304320-mmc2.xlsx
    Ignored:    data/12864_2005_498_MOESM2_ESM.xls
    Ignored:    data/41591_2018_45_MOESM5_ESM.xlsx
    Ignored:    data/CCLE_sample_info_file_2012-10-18.txt
    Ignored:    data/NIH_PDL1_amp_cases.csv
    Ignored:    data/NIH_PDL1_nonamp_cases.csv
    Ignored:    data/PDL1dlbcl.rnaseq.kallisto.raw.txi.coding.cpm1.TMM.logCPM.sm24.csv
    Ignored:    data/PanImmune_GeneSet_Definitions.xlsx
    Ignored:    data/aliquot.tsv
    Ignored:    data/azizi_curated.xlsx
    Ignored:    data/c2_human_immune_genesets.gmt
    Ignored:    data/dlbcl_total_immune_gset_v1_results.rds
    Ignored:    data/ds_JCO.18.01583-2.xlsx
    Ignored:    data/duke_study_driver_mutations_top150_maf.csv
    Ignored:    data/final_analysis_set.maf
    Ignored:    data/gdc_clinical_data.json
    Ignored:    data/gdc_files_and_case_ids.json
    Ignored:    data/gencode.v22.annotation.gtf
    Ignored:    data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Ignored:    data/genesets/
    Ignored:    data/gset_ids_complete.rds
    Ignored:    data/histo_metadata_dlbcl_landscape.csv
    Ignored:    data/journal.pone.0088309.s001.XLS
    Ignored:    data/mmc1.xlsx
    Ignored:    data/nejmoa1801445_appendix_2.xlsx
    Ignored:    data/reactome_imm_signaling.tsv
    Ignored:    data/sample.tsv
    Ignored:    data/sig.rda
    Ignored:    data/sig_genes.txt
    Ignored:    data/temp.dgelist_edger.rds
    Ignored:    data/temp.dgelist_limma.rds
    Ignored:    data/uchi_metadata.csv
    Ignored:    logs/
    Ignored:    output/.DS_Store
    Ignored:    output/NCI_mutations_vs_deconvolution.csv
    Ignored:    output/clust1_vs_others.txt
    Ignored:    output/clust1_vs_others_dge.rnk
    Ignored:    output/clust2_vs_others_dge.rnk
    Ignored:    output/clust3_vs_others_dge.rnk
    Ignored:    output/clust4_vs_others_dge.rnk
    Ignored:    output/cluster1_dge.rds
    Ignored:    output/cluster2_dge.rds
    Ignored:    output/cluster3_dge.rds
    Ignored:    output/cluster4_dge.rds
    Ignored:    output/combined_clin_data.xlsx
    Ignored:    output/combined_dlbcl_expr_matrix.csv
    Ignored:    output/combined_es_gsva2_4cluster.rds
    Ignored:    output/combined_es_gsva4_4cluster.rds
    Ignored:    output/combined_expressionset.rds
    Ignored:    output/combined_tpm_matrix.csv
    Ignored:    output/counts_expressionset_for_limma.rds
    Ignored:    output/dlbcl_expr_matrix.csv
    Ignored:    output/dlbcl_total_immune_gset_v2_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v3_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v4_results.rds
    Ignored:    output/duke_expression_set_cleaned_log.rds
    Ignored:    output/duke_expressionset.rds
    Ignored:    output/duke_expressionset_pext.rds
    Ignored:    output/duke_extended_pheno_data.csv
    Ignored:    output/expr_matrix.csv
    Ignored:    output/gsea_res_cluster1.csv
    Ignored:    output/gset_ids_complete_2.rds
    Ignored:    output/gset_ids_complete_3.rds
    Ignored:    output/gset_ids_complete_4.rds
    Ignored:    output/gsva_v2_4clust_assign.rds
    Ignored:    output/gsva_v3_4clust_assign.rds
    Ignored:    output/gsva_v4_4clust_assign.rds
    Ignored:    output/limma_voom_elist_1.rds
    Ignored:    output/merged_raw_counts.csv
    Ignored:    output/nci_dlbcl_annotation.csv
    Ignored:    output/nci_dlbcl_unprocessed_counts.csv
    Ignored:    output/nci_expressionset.rds
    Ignored:    output/nci_expressionset_pext.rds
    Ignored:    output/nci_extended_pheno_data.csv
    Ignored:    output/networkanal3_clust1.txt
    Ignored:    output/networkanal3_clust2.txt
    Ignored:    output/networkanal3_clust3.txt
    Ignored:    output/networkanal3_clust4.txt
    Ignored:    output/total_immune_gset_v1.rds
    Ignored:    output/tpm_cyt_expressionset.rds
    Ignored:    output/tpm_expressionset.rds
    Ignored:    output/uchi_cluster_assignments.csv
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.pvals.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019_RAW.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.pvals.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819_RAW.txt
    Ignored:    output/xcell_input_matrix.csv

Untracked files:
    Untracked:  NCI_mutations_vs_deconvolution.xls
    Untracked:  analysis/correlation_clusters.Rmd
    Untracked:  batch1_cell_counts.pdf
    Untracked:  data/cibersort/
    Untracked:  data/gistic_results_nci/
    Untracked:  es_10k_hgnc/
    Untracked:  output/ASH_ORAL_figures.png
    Untracked:  output/ASH_ORAL_figures/
    Untracked:  output/Rplot.pdf
    Untracked:  output/Rplot01.pdf
    Untracked:  output/batch1_cell_seg_df.rds
    Untracked:  output/batch1_summary_cell_seg_df.rds
    Untracked:  output/colored_hclust_plot_by_group.pdf
    Untracked:  output/colored_hclust_plot_by_samples.pdf
    Untracked:  output/combined_counts_dge.rds
    Untracked:  output/combined_expr_matrix_for_signature_unprocessed.rmd
    Untracked:  output/dgelist_pheno_data.rds
    Untracked:  output/dlbcl_final_signature_matrix.tsv
    Untracked:  output/dlbcl_gsva5_4cluster_hcpc.rds
    Untracked:  output/dlbcl_gsva5_4cluster_pca.rds
    Untracked:  output/dlbcl_uchi_batch1_mixture.tsv
    Untracked:  output/duke_dlbcl_mixture.tsv
    Untracked:  output/gse_69030_probes.tsv
    Untracked:  output/hclust_labeled_by_cell_type.pdf
    Untracked:  output/mif_pilot_data_plot.pdf
    Untracked:  output/nci_dlbcl_mixture.tsv
    Untracked:  output/phenodata_for_signature_matrix.rmd
    Untracked:  output/reference_matrix_file.tsv
    Untracked:  output/table1.html
    Untracked:  output/uchi_deconvolution_batch_1.rds
    Untracked:  output/uchi_pheno_clust
    Untracked:  output/var_contribution_axis2.pdf
    Untracked:  output/venn.png
    Untracked:  output/venn.png.2020-02-23_11-17-47.log
    Untracked:  output/venn.png.2020-02-23_11-17-56.log
    Untracked:  output/venn.png.2020-02-23_11-20-44.log
    Untracked:  preprocessed_data_for_signature_matrix_es.rds

Unstaged changes:
    Modified:   .gitignore
    Deleted:    Rplot.pdf
    Deleted:    Rplot001.png
    Deleted:    Rplot002.png
    Deleted:    Rplot01.pdf
    Modified:   analysis/cyt.Rmd
    Modified:   analysis/deconvolution.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/newgs.Rmd
    Modified:   analysis/predictor.Rmd
    Modified:   analysis/preprocessing_count_data.Rmd
    Deleted:    data/DLBCL_histo/JG55/HE/14-1464/14-1464_Scan1_annotations.xml.lock
    Modified:   output/design_for_limma.rds
    Modified:   output/dgelist_for_limma.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/3b9db7d138cc2bb849acc70f8218adc98120712a/analysis/merge.Rmd" target="_blank">3b9db7d</a>
</td>
<td>
mleukam
</td>
<td>
2020-05-09
</td>
<td>
wflow_publish(c(“analysis/merge.Rmd”, “analysis/mif.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/f1ed7b84631b39254e3501bf48f0c8c4a881e54f/docs/merge.html" target="_blank">f1ed7b8</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/fe5d4fd9041e164bdf73f0c225e86c757de78434/analysis/merge.Rmd" target="_blank">fe5d4fd</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-14
</td>
<td>
fixed duke raw counts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/f8d62657160144c0f130a04e653dec62c00690e7/docs/merge.html" target="_blank">f8d6265</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/8ec17b0e52437052b3a76c521495fce411c69e01/analysis/merge.Rmd" target="_blank">8ec17b0</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-02
</td>
<td>
new gene sets for take 2 GSVA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/10b3bdd301c7bc263f5a0a0aed8cda5167a2eca9/docs/merge.html" target="_blank">10b3bdd</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/d9718424cb00f80972c6f9255b3db01b0c1f815a/analysis/merge.Rmd" target="_blank">d971842</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-30
</td>
<td>
working notebook for merging expression and phenotype
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/5a57ec091786bd8b0362d31767fd966d90be89cd/docs/merge.html" target="_blank">5a57ec0</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/42da9d5893be09775a582da78b8bec151d4ea9bf/analysis/merge.Rmd" target="_blank">42da9d5</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-28
</td>
<td>
major overhaul of Duke count processing, updates of NCI pheno data
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>The purpose of this notebook is to merge the RNAseq data from NCI (downloaded as raw counts from GDC) and Duke (processed from sequence data) into a single gene expression matrix.</p>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Clear workspace</p>
<pre class="r"><code>rm(list = ls())</code></pre>
<pre class="r"><code>knitr::opts_chunk$set(eval = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(&quot;tidyverse&quot;)
library(&quot;sva&quot;)
library(&quot;Biobase&quot;)
library(&quot;viridis&quot;)
library(&quot;ggdark&quot;)
library(&quot;jcolors&quot;)
library(&quot;ggsci&quot;)
library(&quot;ggpubr&quot;)</code></pre>
<p>Read in expressionsets</p>
<pre class="r"><code>duke_es &lt;- readRDS(&quot;output/duke_expressionset_pext.rds&quot;)

nci_es &lt;- readRDS(&quot;output/nci_expressionset_pext.rds&quot;)</code></pre>
</div>
<div id="batch-correction-duke-expression-data" class="section level2">
<h2>Batch correction Duke expression data</h2>
<p>Following examples and suggestions in vignette here: <a href="https://bioconductor.org/packages/release/bioc/vignettes/sva/inst/doc/sva.pdf" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/sva/inst/doc/sva.pdf</a> and examples here: <a href="http://jtleek.com/genstats/inst/doc/02_13_batch-effects.html" class="uri">http://jtleek.com/genstats/inst/doc/02_13_batch-effects.html</a></p>
<p>Duke expression data did not go through GDC pipeline and may contain batch effects carried over from raw sequence data. Duke pheno data includes lanes, which will be examined for batch effect using SVA ComBat function</p>
<pre class="r"><code>pheno &lt;- pData(duke_es)
edata &lt;- exprs(duke_es)

# There is a single &quot;NA&quot; value in the lane assignments. Will review and decide whether to keep or not for further analysis
which(is.na(pheno$lane))
pheno[225,]

# The sample looks otherwise complete and with a large enough library size compared to others. Will set to &quot;unknown&quot; and keep. Because one batch now will only have one member, will have to adjust means only and not variance (cannot calculate variance from one sample)

# The ComBat function adjusts for known batches using an empirical Bayesian framework. In order to use the function, you must have a known batch variable in your dataset.
batch = pheno$lane %&gt;% as.data.frame() %&gt;%
  dplyr::rename(lane = 1) %&gt;%
  mutate(lane = as.character(lane)) %&gt;%
  replace(is.na(.), &quot;unknown&quot;) %&gt;%
  mutate(lane = as.factor(lane)) %&gt;%
  pull(lane)
summary(batch)
# Note that adjustment variables will be treated as given to the ComBat function. This means if you are trying to adjust for a categorical variable with p different levels, you will need to give ComBat p-1 indicator variables for this covariate.
modlane &lt;- model.matrix(~ lane, data = pheno)

# We need to create a model matrix for the adjustment variables, including the variable of interest. Note that you do not include batch in creating this model matrix - it will be included later in the ComBat function. In this case there are no other adjustment variables so we simply fit an intercept term.
modcombat &lt;- model.matrix(~ 1, data = pheno)

# apply combat to get &quot;cleaned data matrix&quot;
# will use the mean.only=TRUE option for correction for lanes that only adjusts the mean of the batch effects across batches (default adjusts the mean and variance). This option is recommended for cases where milder batch effects are expected (so no need to adjust the variance), or in cases where the variances are expected to be different across batches due to the biology.

# plot the before
batch_df &lt;- enframe(batch) %&gt;%
  dplyr::select(lane = value)
edata_df &lt;- as.data.frame(t(edata)) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  bind_cols(batch_df) %&gt;%
  dplyr::select(sample_id, lane, everything()) %&gt;%
  gather(key = &quot;gene_id&quot;, value = &quot;intensity&quot;, -sample_id, -lane) %&gt;%
  group_by(lane) %&gt;%
  print()

before_box &lt;- ggplot(edata_df, 
                     aes(x = lane,
                         y = intensity, 
                         group = lane)) +
  geom_boxplot(aes(fill = lane)) +
  geom_hline(yintercept = 5, 
             color = &quot;#767676&quot;, 
             linetype = 2) +
  dark_theme_gray() +
  scale_color_manual(&quot;white&quot;) + 
  scale_fill_viridis(discrete = TRUE, 
                     option = &quot;plasma&quot;)

before_box

# pretty minimal differences in mean and distribution

# apply batch correction
combat_edata &lt;- ComBat(dat = edata,
                       batch = batch,
                       mod = modcombat,
                       mean.only = TRUE,
                       par.prior = TRUE, 
                       prior.plots = FALSE)

# plot the after
batch_df &lt;- enframe(batch) %&gt;%
  dplyr::select(lane = value)
combat_edata_df &lt;- as.data.frame(t(combat_edata)) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  bind_cols(batch_df) %&gt;%
  dplyr::select(sample_id, lane, everything()) %&gt;%
  gather(key = &quot;gene_id&quot;, value = &quot;intensity&quot;, -sample_id, -lane) %&gt;%
  group_by(lane) %&gt;%
  print()

after_box &lt;- ggplot(combat_edata_df, aes(x = lane, y = intensity, group = lane)) +
  geom_boxplot(aes(fill = lane)) +
  geom_hline(yintercept = 5, 
             color = &quot;#767676&quot;, 
             linetype = 2) +
  dark_theme_gray() +
  scale_color_manual(&quot;white&quot;) + 
  scale_fill_viridis(discrete = TRUE, 
                     option = &quot;plasma&quot;)
after_box

# even more better :-)</code></pre>
</div>
<div id="combine-expression-data" class="section level2">
<h2>Combine expression data</h2>
<p>Use Duke expression data that is batch-corrected for lanes</p>
<pre class="r"><code>duke_exprs &lt;- combat_edata %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  dplyr::select(gene_id, everything()) %&gt;%
  print()

nci_exprs &lt;- exprs(nci_es) %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  dplyr::select(gene_id, everything()) %&gt;%
  print()

# filter rownames to only keep intersecting rows
duke_genes &lt;- duke_exprs %&gt;% pull(gene_id)
length(duke_genes)
nci_genes &lt;- nci_exprs %&gt;% pull(gene_id)
length(nci_genes)

combined_genes &lt;- intersect(duke_genes, nci_genes)
length(combined_genes)
head(combined_genes)

duke_exprs_filt &lt;- duke_exprs %&gt;%
  dplyr::filter(gene_id %in% combined_genes)
nrow(duke_exprs_filt)

nci_exprs_filt &lt;- nci_exprs %&gt;%
  dplyr::filter(gene_id %in% combined_genes)
nrow(nci_exprs_filt)

combined_exprs &lt;- nci_exprs_filt %&gt;%
  left_join(duke_exprs_filt, by = &quot;gene_id&quot;) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;gene_id&quot;) %&gt;%
  as.matrix()
dim(combined_exprs)
combined_exprs[1:10, 1:6]</code></pre>
</div>
<div id="combine-phenotype-data" class="section level2">
<h2>Combine phenotype data</h2>
<p>Overlapping data (NCI = Duke)</p>
<ul>
<li><code>Gene Expression Subgroup</code> = <code>ABC GCB (RNAseq)</code> – no reformatting</li>
<li><code>IPI range</code> = <code>IPI</code> – NCI dataset split into two numbers and save both high and low IPI. Merge Duke data with low IPI (presumably at diagnosis?)</li>
<li><code>IPI Group</code> = <code>IPI groups</code> – no reformatting</li>
<li><code>Status at Follow_up_ 0 Alive_ 1 Dead</code> = <code>Censored</code> – no reformatting</li>
<li><code>Follow_up Time _yrs</code> = <code>Overall Survival years</code> – no reformatting</li>
<li><code>Gender</code> = <code>Gender</code></li>
<li><code>Age</code> = <code>Age at diagnosis</code> - Duke dataset, remove decimals, remove <code>age_at_diagnosis</code> from NCI dataset</li>
<li><code>ann_arbor_clinical_stage</code> = <code>AnnArbor stage IPI</code> - NCI dataset to be converted to numeric and then converted to indicator for stage 3/4 per IPI</li>
<li><code>ECOG Performance Status</code> = <code>ECOG IPI</code> - NCI dataset to be converted to indicator for ECOG 2+ vs ECOG 0-1 per IPI</li>
<li><code>Number of Extranodal Sites</code> = <code>multiple extranodal IPI</code> - NCI dataset to be converted to indicator for &gt;1 extranodal sites</li>
<li><code>LDH Ratio</code> = <code>LDH IPI</code> - NCI dataset to be converted to indicator for raio &gt;1 per IPI</li>
</ul>
<pre class="r"><code># pull out pheno_data
nci_pheno &lt;- pData(nci_es) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  print()

duke_pheno &lt;- pData(duke_es) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  print()</code></pre>
<div id="format-nci-data-for-merge" class="section level4">
<h4>Format NCI data for merge</h4>
<pre class="r"><code>ipi_highlow &lt;- nci_pheno$`IPI Range` %&gt;%
  str_split_fixed(pattern = &quot;&quot;, n = 2) %&gt;%
  as_tibble() %&gt;%
  dplyr::rename(ipi_low = V1) %&gt;%
  mutate(ipi_high = ifelse(V2 == &quot;&quot;, ipi_low, V2)) %&gt;%
  dplyr::select(-V2) %&gt;%
  print()

nci_pheno_ipi &lt;- nci_pheno %&gt;% 
  bind_cols(ipi_highlow) %&gt;%
  dplyr::select(sample_id, ipi_initial = ipi_low, ipi_high, everything()) %&gt;%
  print()

nci_pheno_formatted &lt;- nci_pheno_ipi %&gt;%
  dplyr::rename(abc_gcb_rna = `Gene Expression Subgroup`,
                ipi_group = `IPI Group`,
                os_status = `Status at Follow_up_ 0 Alive_ 1 Dead`,
                os_years = `Follow_up Time _yrs`,
                gender = Gender, 
                age = Age) %&gt;%
  dplyr::select(sample_id, abc_gcb_rna, ipi_group, os_status, os_years, gender, age, everything()) %&gt;%
  mutate(ann_arbor_stage_3plus = as.factor(ann_arbor_clinical_stage)) %&gt;%
  mutate(ann_arbor_stage_3plus = fct_recode(ann_arbor_stage_3plus,
                                            &quot;1&quot; = &quot;Stage III&quot;,
                                            &quot;1&quot; = &quot;Stage IV&quot;,
                                            &quot;0&quot; = &quot;Stage I&quot;,
                                            &quot;0&quot; = &quot;Stage II&quot;,
                                            &quot;NA&quot; = &quot;Not Reported&quot;)) %&gt;%
  dplyr::select(sample_id, ann_arbor_stage_3plus, everything()) %&gt;%
  mutate(ps_2plus = as.numeric(`ECOG Performance Status`)) %&gt;%
  mutate(ps_2plus = ifelse(ps_2plus &gt; 1, 1, 0),
         ps_2plus = as.factor(ps_2plus)) %&gt;%
  dplyr::select(sample_id, ps_2plus, everything()) %&gt;%
  mutate(extranodal_sites_2plus = as.numeric(`Number of Extranodal Sites`)) %&gt;%
  mutate(extranodal_sites_2plus = ifelse(extranodal_sites_2plus &gt; 1, 1, 0)) %&gt;%
  dplyr::select(sample_id, extranodal_sites_2plus, everything()) %&gt;%
  mutate(extranodal_sites_2plus = as.factor(extranodal_sites_2plus)) %&gt;%
  mutate(ldh_abnormal = ifelse(`LDH Ratio` &gt; 1, 1, 0),
         ldh_abnormal = as.factor(ldh_abnormal),
         ipi_initial = as.factor(ipi_initial)) %&gt;%
  dplyr::select(sample_id, ldh_abnormal, everything()) %&gt;%
  print()</code></pre>
</div>
<div id="format-duke-data-for-merge" class="section level4">
<h4>Format Duke data for merge</h4>
<pre class="r"><code>duke_pheno_formatted &lt;- duke_pheno %&gt;%
  dplyr::rename(abc_gcb_rna = `ABC GCB (RNAseq)`,
                ipi_group = `IPI groups`,
                os_status = `Censored`,
                os_years = `Overall Survival years`,
                gender = Gender) %&gt;%
  mutate(age = (str_split_fixed(`age at diagnosis`, &quot;\\.&quot;, n = 2))[,1]) %&gt;%
  dplyr::select(sample_id, age, abc_gcb_rna, ipi_group, os_status, os_years, gender, everything()) %&gt;%
  dplyr::rename(ann_arbor_stage_3plus = `AnnArbor stage IPI`,
                ps_2plus = `ECOG IPI`,
                extranodal_sites_2plus = `multiple extranodal IPI`,
                ldh_abnormal = `LDH IPI`,
                ipi_initial = IPI) %&gt;%
  mutate(ps_2plus = as.factor(ps_2plus),
         ann_arbor_stage_3plus = as.factor(ann_arbor_stage_3plus),
         extranodal_sites_2plus = as.factor(extranodal_sites_2plus),
         ldh_abnormal = as.factor(ldh_abnormal),
         ipi_initial = as.factor(ipi_initial)) %&gt;%
  mutate(age = as.numeric(age)) %&gt;%
  print()</code></pre>
</div>
<div id="merge-pheno-data" class="section level4">
<h4>Merge pheno data</h4>
<pre class="r"><code>combined_pheno &lt;- nci_pheno_formatted %&gt;%
  bind_rows(duke_pheno_formatted) %&gt;%
  dplyr::select(sample_id, abc_gcb_rna, ipi_group, os_status, os_years, age, gender, ipi_initial, pdl1_status, everything())</code></pre>
</div>
</div>
<div id="remake-expressionset" class="section level2">
<h2>Remake expressionset</h2>
<pre class="r"><code>pheno_data_df &lt;- as.data.frame(combined_pheno) %&gt;%
  column_to_rownames(var = &quot;sample_id&quot;)

v_metadata &lt;- data.frame(
  labelDescription = colnames(pheno_data_df),
  row.names = colnames(pheno_data_df) )
  
phenoData &lt;- new(&quot;AnnotatedDataFrame&quot;, 
                 data = pheno_data_df, 
                 varMetadata = v_metadata)
phenoData

annotation &lt;- as.character(&quot;Duke RNAseq counts derived from raw sequence data published by the Dave lab at Duke in Reddy et al Cell 2017. Quantified with Kallisto with reverse strand parameter, collected with tximport, filtered for only protein-coding genes, selected for cases with expression in &gt;12,000 genes, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed. NCI RNAseq counts originally published by the Staudt lab at NCI in Schmitz et al NEJM 2018. Raw counts downloaded from GDC as HTseq counts, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed.&quot;)

# assemble expressionset
combined_es &lt;- ExpressionSet(
  assayData = combined_exprs,
  phenoData = phenoData,
  annotation = annotation)
combined_es</code></pre>
</div>
<div id="batch-correction-between-dukenci" class="section level2">
<h2>Batch correction between Duke/NCI</h2>
<p>Following documentation as above</p>
<pre class="r"><code># create marker variable for Duke vs NCI in pheno data
pheno_df &lt;- pData(combined_es) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  mutate(source = ifelse(grepl(&quot;duke_&quot;, sample_id),&quot;duke&quot;,&quot;nci&quot;)) %&gt;%
  mutate(source = as.factor(source)) %&gt;%
  dplyr::select(sample_id, source, everything()) %&gt;%
  # also fix gcb_abc column factors
  mutate(abc_gcb_rna = as.character(abc_gcb_rna)) %&gt;%
  mutate(abc_gcb_rna = ifelse(abc_gcb_rna == &quot;Unclass&quot;, &quot;Unclassified&quot;, abc_gcb_rna)) %&gt;%
  mutate(abc_gcb_rna = as.factor(abc_gcb_rna)) %&gt;%
  print()

# convert back to regular dataframe with rownames for downstream processing
pheno_d &lt;- as.data.frame(pheno_df) %&gt;%
  column_to_rownames(var = &quot;sample_id&quot;)

# define batch as source marker
batch = pheno_df$source
summary(batch)

# boxplot by data source
# plot the before
batch_df &lt;- enframe(batch) %&gt;%
  dplyr::select(source = value)
edata_df &lt;- as.data.frame(t(exprs(combined_es))) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  bind_cols(batch_df) %&gt;%
  dplyr::select(sample_id, source, everything()) %&gt;%
  gather(key = &quot;gene_id&quot;, value = &quot;intensity&quot;, -sample_id, -source) %&gt;%
  group_by(source) %&gt;%
  print()

before_box &lt;- ggplot(edata_df, 
                     aes(x = source, 
                         y = intensity, 
                         group = source)) +
  geom_boxplot(aes(fill = source)) +
  geom_hline(yintercept = 5, 
             color = &quot;#767676&quot;, 
             linetype = 2) +
  dark_theme_gray() +
  scale_color_manual(&quot;white&quot;) + 
  scale_fill_viridis(discrete = TRUE, 
                     option = &quot;plasma&quot;)

before_box
# small but notable differences in mean and distribution
# will use the NCI data as the &quot;ground truth&quot; for normalization given higher read counts overall and standardized count pipeline through GDC

# Note that adjustment variables will be treated as given to the ComBat function. This means if you are trying to adjust for a categorical variable with p different levels, you will need to give ComBat p-1 indicator variables for this covariate.
modsource &lt;- model.matrix(~ source, data = pheno_d)

# We need to create a model matrix for the adjustment variables, including the variable of interest. Note that you do not include batch in creating this model matrix - it will be included later in the ComBat function. In this case there are no other adjustment variables so we simply fit an intercept term.
modcombat &lt;- model.matrix(~ 1, data = pheno_d)

# apply batch correction
combined_combat_edata &lt;- ComBat(dat = exprs(combined_es),
                       batch = batch,
                       mod = modcombat,
                       par.prior = TRUE,
                       prior.plots = FALSE,
                       mean.only = FALSE,
                       ref.batch = &quot;nci&quot;)

after_df &lt;- as.data.frame(t(combined_combat_edata)) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  bind_cols(batch_df) %&gt;%
  dplyr::select(sample_id, source, everything()) %&gt;%
  gather(key = &quot;gene_id&quot;, value = &quot;intensity&quot;, -sample_id, -source) %&gt;%
  group_by(source) %&gt;%
  print()

after_box &lt;- ggplot(after_df, 
                    aes(x = source, 
                        y = intensity, 
                        group = source)) +
  geom_boxplot(aes(fill = source)) +
  geom_hline(yintercept = 5, 
             color = &quot;#767676&quot;, 
             linetype = 2) +
  dark_theme_gray() +
  scale_color_manual(&quot;white&quot;) + 
  scale_fill_tron()
after_box</code></pre>
</div>
<div id="final-expressionset" class="section level2">
<h2>Final Expressionset</h2>
<pre class="r"><code># use updated pheno data frame from last chunk
v_metadata &lt;- data.frame(
  labelDescription = colnames(pheno_d),
  row.names = colnames(pheno_d) )
  
phenoData &lt;- new(&quot;AnnotatedDataFrame&quot;, 
                 data = pheno_d, 
                 varMetadata = v_metadata)
phenoData

annotation &lt;- as.character(&quot;Duke RNAseq counts derived from raw sequence data published by the Dave lab at Duke in Reddy et al Cell 2017. Quantified with Kallisto with reverse strand parameter, collected with tximport, filtered for only protein-coding genes, selected for cases with expression in &gt;12,000 genes, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed. Batch corrected using ComBat by sequencing lane. NCI RNAseq counts originally published by the Staudt lab at NCI in Schmitz et al NEJM 2018. Raw counts downloaded from GDC as HTseq counts, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed. Batch corrected for NCI vs Duke source using ComBat&quot;)

# assemble expressionset
combined_es &lt;- ExpressionSet(
  assayData = combined_combat_edata,
  phenoData = phenoData,
  annotation = annotation)
combined_es</code></pre>
</div>
<div id="write-out-for-cluster" class="section level2">
<h2>Write out for cluster</h2>
<pre class="r"><code># write expression data as csv for large file transfer/reading on the cluster
combined_exprs &lt;- exprs(combined_es)
combined_csv &lt;- as.data.frame(combined_exprs) %&gt;% rownames_to_column(var = &quot;gene&quot;) %&gt;% as_tibble() %&gt;% select(gene, everything()) %&gt;% print() 
write_csv(combined_csv, &quot;output/combined_dlbcl_expr_matrix.csv&quot;)</code></pre>
<pre class="r"><code>saveRDS(combined_es, &quot;output/combined_expressionset.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
