<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-06-13" />

<title>Preprocessing NCI Count Data from GDC</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Preprocessing NCI Count Data from GDC</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-06-13</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190613code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190613)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190613code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190613)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetree5ae4f4765761d496ad037a59b6cb502764cd2893targetblank5ae4f47a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/5ae4f4765761d496ad037a59b6cb502764cd2893" target="_blank">5ae4f47</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetree5ae4f4765761d496ad037a59b6cb502764cd2893targetblank5ae4f47a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Untracked:  data/htseq_counts/
    Untracked:  output/expr_matrix.csv
    Untracked:  output/nci_dlbcl_annotation.csv
    Untracked:  output/nci_dlbcl_unprocessed_counts.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/5ae4f4765761d496ad037a59b6cb502764cd2893/analysis/preprocessing_count_data.Rmd" target="_blank">5ae4f47</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
working preprocessing notebook
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Clear workspace</p>
<pre class="r"><code>rm(list = ls())</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(&quot;tidyverse&quot;)</code></pre>
<pre><code>── Attaching packages ──────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.1.1     ✔ purrr   0.3.2
✔ tibble  2.1.3     ✔ dplyr   0.8.1
✔ tidyr   0.8.3     ✔ stringr 1.4.0
✔ readr   1.3.1     ✔ forcats 0.4.0</code></pre>
<pre><code>── Conflicts ─────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(&quot;edgeR&quot;)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(&quot;limma&quot;)
library(&quot;GSVA&quot;)</code></pre>
<p>Read in data</p>
<pre class="r"><code>total_counts &lt;- read_csv(&quot;output/nci_dlbcl_unprocessed_counts.csv&quot;)</code></pre>
<pre><code>Parsed with column specification:
cols(
  .default = col_double(),
  gene = col_character()
)</code></pre>
<pre><code>See spec(...) for full column specifications.</code></pre>
<p>Read in lookup table for features, gencode v22 (used by GDC to label features)</p>
<pre class="r"><code>gencode_gtf &lt;- read_tsv(&quot;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&quot;)</code></pre>
<pre><code>Parsed with column specification:
cols(
  gene_id = col_character(),
  gene_type = col_character(),
  gene_status = col_character(),
  gene_name = col_character(),
  level = col_double(),
  havana_gene = col_character()
)</code></pre>
<pre><code>Warning: 10327 parsing failures.
  row col  expected    actual                                                        file
 5304  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
12445  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
12913  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
13082  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
13310  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
..... ... ......... ......... ...........................................................
See problems(...) for more details.</code></pre>
</div>
<div id="normalization-filtering-and-transformation" class="section level2">
<h2>Normalization, filtering and transformation</h2>
<p>Preprocessing RNAseq count data following methods outlined here: <a href="https://f1000research.com/articles/5-1408/v3" class="uri">https://f1000research.com/articles/5-1408/v3</a></p>
<div id="filter-for-protein-coding-genes" class="section level4">
<h4>Filter for protein coding genes</h4>
<pre class="r"><code># filter for protein coding genes
total_counts_prcode &lt;- total_counts %&gt;%
  dplyr::rename(gene_id = gene) %&gt;%
  left_join(gencode_gtf, by = &quot;gene_id&quot;) %&gt;%
  dplyr::filter(gene_type == &quot;protein_coding&quot;) %&gt;%
  dplyr::select(-gene_name, -gene_type, -gene_status, -level, -havana_gene) %&gt;%
  dplyr::select(gene_id, everything())

nrow(total_counts)</code></pre>
<pre><code>[1] 60488</code></pre>
<pre class="r"><code>nrow(total_counts_prcode)</code></pre>
<pre><code>[1] 19814</code></pre>
</div>
<div id="correct-for-library-size-convert-to-cpm" class="section level4">
<h4>Correct for library size: convert to CPM</h4>
<pre class="r"><code># normalize rows by log cpm using EdgeR
df_data &lt;- total_counts_prcode %&gt;% 
  dplyr::select(-gene_id) %&gt;% as.matrix()
df_names &lt;- total_counts_prcode %&gt;% dplyr::select(gene_id) 
out_data &lt;- cpm(df_data, log = FALSE) %&gt;% as_tibble()
total_counts_prcode_cpm &lt;- bind_cols(df_names, out_data)</code></pre>
</div>
<div id="density-plots-of-cpm-values" class="section level4">
<h4>Density plots of CPM values</h4>
<pre class="r"><code>tidy_cpm &lt;- total_counts_prcode_cpm %&gt;% 
  gather(key = &quot;sampleID&quot;, value = &quot;intensity&quot;, -gene_id) 

dplot1 &lt;- ggplot(tidy_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;) +
  xlim(-5, 100) +
  ggtitle(&quot;Density plot of CPM for NCI DLBCL&quot;)
dplot1</code></pre>
<pre><code>Warning: Removed 920063 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/preprocessing_count_data.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dplot2 &lt;- ggplot(tidy_cpm, aes(log(intensity))) +
  geom_density() +
  theme(legend.position = &quot;none&quot;) +
  xlim(-20, 20) +
  ggtitle(&quot;Density plot of log(CPM) for NCI DLBCL&quot;)
dplot2</code></pre>
<pre><code>Warning: Removed 1402676 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/preprocessing_count_data.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="filter-by-expression-levels" class="section level4">
<h4>Filter by expression levels</h4>
<pre class="r"><code># move gene names to rownames
totcounts_prcode_cpm_matrix &lt;- total_counts_prcode_cpm %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;gene_id&quot;)
totcounts_prcode_cpm_matrix[1:5, 1:5]</code></pre>
<pre><code>                   nci_dlbcl_1 nci_dlbcl_2 nci_dlbcl_3 nci_dlbcl_4
ENSG00000000003.13   2.1464827    4.206426    2.416628    2.456197
ENSG00000000005.5    0.0420879    0.000000    0.000000    0.000000
ENSG00000000419.11  45.7495433   73.986992   26.459434   40.456209
ENSG00000000457.12   7.2391182    6.799428   28.452712   27.505350
ENSG00000000460.15   6.8182392   23.797997   18.856757   16.990390
                   nci_dlbcl_5
ENSG00000000003.13 10.72324364
ENSG00000000005.5   0.02082183
ENSG00000000419.11 41.81023927
ENSG00000000457.12 23.19552119
ENSG00000000460.15 18.01088495</code></pre>
<pre class="r"><code># filter out genes that aren&#39;t at least expressed greater than 1 in at least 45 cases (45 being about the size of the smallest group)
total_cpm_stats &lt;- data.frame(
  total = apply(totcounts_prcode_cpm_matrix, 1, function(x) sum(x &gt; 1, na.rm = TRUE)))
keep &lt;- which(total_cpm_stats$total &gt;= 50) 

dim(totcounts_prcode_cpm_matrix)</code></pre>
<pre><code>[1] 19814   481</code></pre>
<pre class="r"><code>total_cpm_filtered = totcounts_prcode_cpm_matrix[keep,]
dim(total_cpm_filtered)</code></pre>
<pre><code>[1] 14575   481</code></pre>
</div>
<div id="normalize-gene-expression-distributions" class="section level4">
<h4>Normalize gene expression distributions</h4>
<p>Normalisation by the method of trimmed mean of M-values (TMM) is performed using the calcNormFactors function in edgeR. The normalisation factors calculated here are used as a scaling factor for the library sizes.</p>
<pre class="r"><code># get normalization factors
norm_factors &lt;- calcNormFactors(total_cpm_filtered, method = &quot;TMM&quot;)

# apply factor to each column
total_cpm_norm &lt;- map2_dfc(total_cpm_filtered, norm_factors, `*`)
total_cpm_norm &lt;- as.data.frame(total_cpm_norm)
rownames(total_cpm_norm) &lt;- rownames(total_cpm_filtered)
total_cpm_norm[1:5, 1:5]</code></pre>
<pre><code>                   nci_dlbcl_1 nci_dlbcl_2 nci_dlbcl_3 nci_dlbcl_4
ENSG00000000003.13    1.567512    4.081409    2.815118    2.796125
ENSG00000000419.11   33.409514   71.788078   30.822463   46.055177
ENSG00000000457.12    5.286510    6.597347   33.144422   31.311974
ENSG00000000460.15    4.979155   23.090714   21.966142   19.341788
ENSG00000000938.11   25.418278   32.036269   50.507743   92.110354
                   nci_dlbcl_5
ENSG00000000003.13    13.12324
ENSG00000000419.11    51.16789
ENSG00000000457.12    28.38697
ENSG00000000460.15    22.04195
ENSG00000000938.11    61.13136</code></pre>
</div>
<div id="log-transformation" class="section level4">
<h4>Log transformation</h4>
<pre class="r"><code>total_log_cpm_filtered_norm &lt;- log2(total_cpm_norm)
total_log_cpm_filtered_norm[1:5, 1:5]</code></pre>
<pre><code>                   nci_dlbcl_1 nci_dlbcl_2 nci_dlbcl_3 nci_dlbcl_4
ENSG00000000003.13   0.6484762    2.029067    1.493196    1.483429
ENSG00000000419.11   5.0621871    6.165672    4.945910    5.525291
ENSG00000000457.12   2.4023156    2.721886    5.050694    4.968643
ENSG00000000460.15   2.3159009    4.529241    4.457210    4.273649
ENSG00000000938.11   4.6677944    5.001634    5.658433    6.525291
                   nci_dlbcl_5
ENSG00000000003.13    3.714052
ENSG00000000419.11    5.677167
ENSG00000000457.12    4.827157
ENSG00000000460.15    4.462180
ENSG00000000938.11    5.933841</code></pre>
<pre class="r"><code>total_log_cpm_filtered_tbl &lt;- total_log_cpm_filtered_norm %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  as_tibble() %&gt;% 
  print()</code></pre>
<pre><code># A tibble: 14,575 x 482
   gene  nci_dlbcl_1 nci_dlbcl_2 nci_dlbcl_3 nci_dlbcl_4 nci_dlbcl_5
   &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;
 1 ENSG…       0.648        2.03        1.49        1.48        3.71
 2 ENSG…       5.06         6.17        4.95        5.53        5.68
 3 ENSG…       2.40         2.72        5.05        4.97        4.83
 4 ENSG…       2.32         4.53        4.46        4.27        4.46
 5 ENSG…       4.67         5.00        5.66        6.53        5.93
 6 ENSG…       4.44         5.66        5.50        5.28        5.70
 7 ENSG…       4.70         4.62        6.08        5.41        5.79
 8 ENSG…       3.94         5.23        6.04        5.85        6.15
 9 ENSG…       4.78         6.46        5.84        6.07        5.73
10 ENSG…       2.30         3.35        4.15        3.35        3.91
# … with 14,565 more rows, and 476 more variables: nci_dlbcl_6 &lt;dbl&gt;,
#   nci_dlbcl_7 &lt;dbl&gt;, nci_dlbcl_8 &lt;dbl&gt;, nci_dlbcl_9 &lt;dbl&gt;,
#   nci_dlbcl_10 &lt;dbl&gt;, nci_dlbcl_11 &lt;dbl&gt;, nci_dlbcl_12 &lt;dbl&gt;,
#   nci_dlbcl_13 &lt;dbl&gt;, nci_dlbcl_14 &lt;dbl&gt;, nci_dlbcl_15 &lt;dbl&gt;,
#   nci_dlbcl_16 &lt;dbl&gt;, nci_dlbcl_17 &lt;dbl&gt;, nci_dlbcl_18 &lt;dbl&gt;,
#   nci_dlbcl_19 &lt;dbl&gt;, nci_dlbcl_20 &lt;dbl&gt;, nci_dlbcl_21 &lt;dbl&gt;,
#   nci_dlbcl_22 &lt;dbl&gt;, nci_dlbcl_23 &lt;dbl&gt;, nci_dlbcl_24 &lt;dbl&gt;,
#   nci_dlbcl_25 &lt;dbl&gt;, nci_dlbcl_26 &lt;dbl&gt;, nci_dlbcl_27 &lt;dbl&gt;,
#   nci_dlbcl_28 &lt;dbl&gt;, nci_dlbcl_29 &lt;dbl&gt;, nci_dlbcl_30 &lt;dbl&gt;,
#   nci_dlbcl_31 &lt;dbl&gt;, nci_dlbcl_32 &lt;dbl&gt;, nci_dlbcl_33 &lt;dbl&gt;,
#   nci_dlbcl_34 &lt;dbl&gt;, nci_dlbcl_35 &lt;dbl&gt;, nci_dlbcl_36 &lt;dbl&gt;,
#   nci_dlbcl_37 &lt;dbl&gt;, nci_dlbcl_38 &lt;dbl&gt;, nci_dlbcl_39 &lt;dbl&gt;,
#   nci_dlbcl_40 &lt;dbl&gt;, nci_dlbcl_41 &lt;dbl&gt;, nci_dlbcl_42 &lt;dbl&gt;,
#   nci_dlbcl_43 &lt;dbl&gt;, nci_dlbcl_44 &lt;dbl&gt;, nci_dlbcl_45 &lt;dbl&gt;,
#   nci_dlbcl_46 &lt;dbl&gt;, nci_dlbcl_47 &lt;dbl&gt;, nci_dlbcl_48 &lt;dbl&gt;,
#   nci_dlbcl_49 &lt;dbl&gt;, nci_dlbcl_50 &lt;dbl&gt;, nci_dlbcl_51 &lt;dbl&gt;,
#   nci_dlbcl_52 &lt;dbl&gt;, nci_dlbcl_53 &lt;dbl&gt;, nci_dlbcl_54 &lt;dbl&gt;,
#   nci_dlbcl_55 &lt;dbl&gt;, nci_dlbcl_56 &lt;dbl&gt;, nci_dlbcl_57 &lt;dbl&gt;,
#   nci_dlbcl_58 &lt;dbl&gt;, nci_dlbcl_59 &lt;dbl&gt;, nci_dlbcl_60 &lt;dbl&gt;,
#   nci_dlbcl_61 &lt;dbl&gt;, nci_dlbcl_62 &lt;dbl&gt;, nci_dlbcl_63 &lt;dbl&gt;,
#   nci_dlbcl_64 &lt;dbl&gt;, nci_dlbcl_65 &lt;dbl&gt;, nci_dlbcl_66 &lt;dbl&gt;,
#   nci_dlbcl_67 &lt;dbl&gt;, nci_dlbcl_68 &lt;dbl&gt;, nci_dlbcl_69 &lt;dbl&gt;,
#   nci_dlbcl_70 &lt;dbl&gt;, nci_dlbcl_71 &lt;dbl&gt;, nci_dlbcl_72 &lt;dbl&gt;,
#   nci_dlbcl_73 &lt;dbl&gt;, nci_dlbcl_74 &lt;dbl&gt;, nci_dlbcl_75 &lt;dbl&gt;,
#   nci_dlbcl_76 &lt;dbl&gt;, nci_dlbcl_77 &lt;dbl&gt;, nci_dlbcl_78 &lt;dbl&gt;,
#   nci_dlbcl_79 &lt;dbl&gt;, nci_dlbcl_80 &lt;dbl&gt;, nci_dlbcl_81 &lt;dbl&gt;,
#   nci_dlbcl_82 &lt;dbl&gt;, nci_dlbcl_83 &lt;dbl&gt;, nci_dlbcl_84 &lt;dbl&gt;,
#   nci_dlbcl_85 &lt;dbl&gt;, nci_dlbcl_86 &lt;dbl&gt;, nci_dlbcl_87 &lt;dbl&gt;,
#   nci_dlbcl_88 &lt;dbl&gt;, nci_dlbcl_89 &lt;dbl&gt;, nci_dlbcl_90 &lt;dbl&gt;,
#   nci_dlbcl_91 &lt;dbl&gt;, nci_dlbcl_92 &lt;dbl&gt;, nci_dlbcl_93 &lt;dbl&gt;,
#   nci_dlbcl_94 &lt;dbl&gt;, nci_dlbcl_95 &lt;dbl&gt;, nci_dlbcl_96 &lt;dbl&gt;,
#   nci_dlbcl_97 &lt;dbl&gt;, nci_dlbcl_98 &lt;dbl&gt;, nci_dlbcl_99 &lt;dbl&gt;,
#   nci_dlbcl_100 &lt;dbl&gt;, nci_dlbcl_101 &lt;dbl&gt;, nci_dlbcl_102 &lt;dbl&gt;,
#   nci_dlbcl_103 &lt;dbl&gt;, nci_dlbcl_104 &lt;dbl&gt;, nci_dlbcl_105 &lt;dbl&gt;, …</code></pre>
<pre class="r"><code>tidy_log_cpm &lt;- total_log_cpm_filtered_tbl %&gt;% 
  gather(key = &quot;sampleID&quot;, value = &quot;intensity&quot;, -gene) %&gt;% 
  print()</code></pre>
<pre><code># A tibble: 7,010,575 x 3
   gene               sampleID    intensity
   &lt;chr&gt;              &lt;chr&gt;           &lt;dbl&gt;
 1 ENSG00000000003.13 nci_dlbcl_1     0.648
 2 ENSG00000000419.11 nci_dlbcl_1     5.06 
 3 ENSG00000000457.12 nci_dlbcl_1     2.40 
 4 ENSG00000000460.15 nci_dlbcl_1     2.32 
 5 ENSG00000000938.11 nci_dlbcl_1     4.67 
 6 ENSG00000000971.14 nci_dlbcl_1     4.44 
 7 ENSG00000001036.12 nci_dlbcl_1     4.70 
 8 ENSG00000001084.9  nci_dlbcl_1     3.94 
 9 ENSG00000001167.13 nci_dlbcl_1     4.78 
10 ENSG00000001460.16 nci_dlbcl_1     2.30 
# … with 7,010,565 more rows</code></pre>
<pre class="r"><code>dplot3 &lt;- ggplot(tidy_log_cpm, aes(intensity)) +
  geom_density() +
  theme(legend.position = &quot;none&quot;) +
  xlim(-20, 20) +
  ggtitle(&quot;Final density plot of log(CPM) for NCI DLBCL&quot;)
dplot3</code></pre>
<pre><code>Warning: Removed 42970 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/preprocessing_count_data.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="write-out-for-later-use" class="section level2">
<h2>Write out for later use</h2>
<pre class="r"><code>write_csv(total_log_cpm_filtered_norm, &quot;output/expr_matrix.csv&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.4

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] GSVA_1.30.0     edgeR_3.24.3    limma_3.38.3    forcats_0.4.0  
 [5] stringr_1.4.0   dplyr_0.8.1     purrr_0.3.2     readr_1.3.1    
 [9] tidyr_0.8.3     tibble_2.1.3    ggplot2_3.1.1   tidyverse_1.2.1

loaded via a namespace (and not attached):
 [1] nlme_3.1-140         bitops_1.0-6         fs_1.3.1            
 [4] lubridate_1.7.4      bit64_0.9-7          RColorBrewer_1.1-2  
 [7] httr_1.4.0           rprojroot_1.3-2      tools_3.5.3         
[10] backports_1.1.4      utf8_1.1.4           R6_2.4.0            
[13] DBI_1.0.0            lazyeval_0.2.2       BiocGenerics_0.28.0 
[16] colorspace_1.4-1     withr_2.1.2          tidyselect_0.2.5    
[19] bit_1.1-14           compiler_3.5.3       git2r_0.25.2        
[22] graph_1.60.0         cli_1.1.0            rvest_0.3.4         
[25] Biobase_2.42.0       xml2_1.2.0           labeling_0.3        
[28] scales_1.0.0         digest_0.6.19        rmarkdown_1.13      
[31] pkgconfig_2.0.2      htmltools_0.3.6      rlang_0.3.4         
[34] readxl_1.3.1         rstudioapi_0.10      RSQLite_2.1.1       
[37] shiny_1.3.2          generics_0.0.2       jsonlite_1.6        
[40] RCurl_1.95-4.12      magrittr_1.5         Rcpp_1.0.1          
[43] munsell_0.5.0        S4Vectors_0.20.1     fansi_0.4.0         
[46] stringi_1.4.3        whisker_0.3-2        yaml_2.2.0          
[49] plyr_1.8.4           grid_3.5.3           blob_1.1.1          
[52] parallel_3.5.3       promises_1.0.1       crayon_1.3.4        
[55] lattice_0.20-38      haven_2.1.0          annotate_1.60.1     
[58] hms_0.4.2            locfit_1.5-9.1       zeallot_0.1.0       
[61] knitr_1.23           pillar_1.4.1         geneplotter_1.60.0  
[64] stats4_3.5.3         XML_3.98-1.20        glue_1.3.1          
[67] evaluate_0.14        modelr_0.1.4         vctrs_0.1.0         
[70] httpuv_1.5.1         cellranger_1.1.0     gtable_0.3.0        
[73] assertthat_0.2.1     xfun_0.7             mime_0.6            
[76] xtable_1.8-4         broom_0.5.2          later_0.8.0         
[79] shinythemes_1.1.2    AnnotationDbi_1.44.0 memoise_1.1.0       
[82] IRanges_2.16.0       workflowr_1.4.0      GSEABase_1.44.0     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
