<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-06-13" />

<title>Preprocessing NCI Count Data from GDC</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Preprocessing NCI Count Data from GDC</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-06-13</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190613code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190613)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190613code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190613)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetree42da9d5893be09775a582da78b8bec151d4ea9bftargetblank42da9d5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/42da9d5893be09775a582da78b8bec151d4ea9bf" target="_blank">42da9d5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetree42da9d5893be09775a582da78b8bec151d4ea9bftargetblank42da9d5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/enrichment.Rmd
    Untracked:  analysis/newgs.Rmd
    Untracked:  code/gsva1.r
    Untracked:  code/gsva1.sh
    Untracked:  code/run_gsva1.pbs
    Untracked:  data/NIH_PDL1_amp_cases.csv
    Untracked:  data/NIH_PDL1_nonamp_cases.csv
    Untracked:  data/aliquot.tsv
    Untracked:  data/c2_human_immune_genesets.gmt
    Untracked:  data/dlbcl_total_immune_gset_v1_results.rds
    Untracked:  data/gdc_clinical_data.json
    Untracked:  data/gdc_files_and_case_ids.json
    Untracked:  data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Untracked:  data/gset_ids_complete.rds
    Untracked:  data/htseq_counts/
    Untracked:  data/sample.tsv
    Untracked:  data/temp.dgelist_limma.rds
    Untracked:  output/dlbcl_expr_matrix.csv
    Untracked:  output/duke_expression_set_cleaned_log.rds
    Untracked:  output/duke_expressionset.rds
    Untracked:  output/expr_matrix.csv
    Untracked:  output/nci_dlbcl_annotation.csv
    Untracked:  output/nci_dlbcl_unprocessed_counts.csv
    Untracked:  output/nci_expressionset.rds
    Untracked:  output/total_immune_gset_v1.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/42da9d5893be09775a582da78b8bec151d4ea9bf/analysis/preprocessing_count_data.Rmd" target="_blank">42da9d5</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-28
</td>
<td>
major overhaul of Duke count processing, updates of NCI pheno data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/768753ff0d3080e2b6170b9487510dbbd1aea778/docs/preprocessing_count_data.html" target="_blank">768753f</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/5ae4f4765761d496ad037a59b6cb502764cd2893/analysis/preprocessing_count_data.Rmd" target="_blank">5ae4f47</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
working preprocessing notebook
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Clear workspace</p>
<pre class="r"><code>rm(list = ls())</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(&quot;tidyverse&quot;)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.1.1     ✔ purrr   0.3.2
✔ tibble  2.1.3     ✔ dplyr   0.8.1
✔ tidyr   0.8.3     ✔ stringr 1.4.0
✔ readr   1.3.1     ✔ forcats 0.4.0</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(&quot;edgeR&quot;)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(&quot;limma&quot;)
library(&quot;Biobase&quot;)</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following object is masked from &#39;package:limma&#39;:

    plotMA</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind,
    colMeans, colnames, colSums, dirname, do.call, duplicated,
    eval, evalq, Filter, Find, get, grep, grepl, intersect,
    is.unsorted, lapply, lengths, Map, mapply, match, mget, order,
    paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind,
    Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which, which.max,
    which.min</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<p>Read in data</p>
<pre class="r"><code>total_counts &lt;- read_csv(&quot;output/nci_dlbcl_unprocessed_counts.csv&quot;)</code></pre>
<pre><code>Parsed with column specification:
cols(
  .default = col_double(),
  gene = col_character()
)</code></pre>
<pre><code>See spec(...) for full column specifications.</code></pre>
<p>Read in lookup table for features, gencode v22 (used by GDC to label features)</p>
<pre class="r"><code>gencode_gtf &lt;- read_tsv(&quot;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&quot;)</code></pre>
<pre><code>Parsed with column specification:
cols(
  gene_id = col_character(),
  gene_type = col_character(),
  gene_status = col_character(),
  gene_name = col_character(),
  level = col_double(),
  havana_gene = col_character()
)</code></pre>
<pre><code>Warning: 10327 parsing failures.
  row col  expected    actual                                                        file
 5304  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
12445  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
12913  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
13082  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
13310  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
..... ... ......... ......... ...........................................................
See problems(...) for more details.</code></pre>
</div>
<div id="normalization-filtering-and-transformation" class="section level2">
<h2>Normalization, filtering and transformation</h2>
<p>Preprocessing RNAseq count data following methods outlined here: <a href="https://f1000research.com/articles/5-1408/v3" class="uri">https://f1000research.com/articles/5-1408/v3</a></p>
<div id="filter-for-protein-coding-genes" class="section level4">
<h4>Filter for protein coding genes</h4>
<pre class="r"><code># filter for protein coding genes
total_counts_prcode &lt;- total_counts %&gt;%
  dplyr::rename(gene_id = gene) %&gt;%
  left_join(gencode_gtf, by = &quot;gene_id&quot;) %&gt;%
  dplyr::filter(gene_type == &quot;protein_coding&quot;) %&gt;%
  dplyr::select(-gene_name, -gene_type, -gene_status, -level, -havana_gene) %&gt;%
  dplyr::select(gene_id, everything())
total_counts_prcode[1:5, 1:5]</code></pre>
<pre><code># A tibble: 5 x 5
  gene_id            DLBCL11667 DLBCL10501 DLBCL10954 DLBCL10984
  &lt;chr&gt;                   &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;
1 ENSG00000000003.13         51         73        137        121
2 ENSG00000000005.5           1          0          0          0
3 ENSG00000000419.11       1087       1284       1500       1993
4 ENSG00000000457.12        172        118       1613       1355
5 ENSG00000000460.15        162        413       1069        837</code></pre>
<pre class="r"><code>nrow(total_counts)</code></pre>
<pre><code>[1] 60488</code></pre>
<pre class="r"><code>nrow(total_counts_prcode)</code></pre>
<pre><code>[1] 19814</code></pre>
</div>
<div id="correct-for-library-size-convert-to-cpm" class="section level4">
<h4>Correct for library size: convert to CPM</h4>
<pre class="r"><code># normalize rows by log cpm using EdgeR
df_data &lt;- total_counts_prcode %&gt;% 
  dplyr::select(-gene_id) %&gt;% as.matrix()
df_names &lt;- total_counts_prcode %&gt;% dplyr::select(gene_id) 
out_data &lt;- cpm(df_data, log = FALSE) %&gt;% as_tibble()
total_counts_prcode_cpm &lt;- bind_cols(df_names, out_data)</code></pre>
</div>
<div id="density-plots-of-cpm-values" class="section level4">
<h4>Density plots of CPM values</h4>
<pre class="r"><code>tidy_cpm &lt;- total_counts_prcode_cpm %&gt;% 
  gather(key = &quot;sampleID&quot;, value = &quot;intensity&quot;, -gene_id) 

dplot1 &lt;- ggplot(tidy_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;) +
  xlim(-5, 100) +
  ggtitle(&quot;Density plot of CPM for NCI DLBCL&quot;)
dplot1</code></pre>
<pre><code>Warning: Removed 920063 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/preprocessing_count_data.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/768753ff0d3080e2b6170b9487510dbbd1aea778/docs/figure/preprocessing_count_data.Rmd/unnamed-chunk-7-1.png" target="_blank">768753f</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dplot2 &lt;- ggplot(tidy_cpm, aes(log(intensity))) +
  geom_density() +
  theme(legend.position = &quot;none&quot;) +
  xlim(-20, 20) +
  ggtitle(&quot;Density plot of log(CPM) for NCI DLBCL&quot;)
dplot2</code></pre>
<pre><code>Warning: Removed 1402676 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/preprocessing_count_data.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/768753ff0d3080e2b6170b9487510dbbd1aea778/docs/figure/preprocessing_count_data.Rmd/unnamed-chunk-7-2.png" target="_blank">768753f</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="filter-by-expression-levels" class="section level4">
<h4>Filter by expression levels</h4>
<pre class="r"><code># move gene names to rownames
totcounts_prcode_cpm_matrix &lt;- total_counts_prcode_cpm %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;gene_id&quot;)
totcounts_prcode_cpm_matrix[1:5, 1:5]</code></pre>
<pre><code>                   DLBCL11667 DLBCL10501 DLBCL10954 DLBCL10984  DLBCL11206
ENSG00000000003.13  2.1464827   4.206426   2.416628   2.456197 10.72324364
ENSG00000000005.5   0.0420879   0.000000   0.000000   0.000000  0.02082183
ENSG00000000419.11 45.7495433  73.986992  26.459434  40.456209 41.81023927
ENSG00000000457.12  7.2391182   6.799428  28.452712  27.505350 23.19552119
ENSG00000000460.15  6.8182392  23.797997  18.856757  16.990390 18.01088495</code></pre>
<pre class="r"><code>dim(totcounts_prcode_cpm_matrix)</code></pre>
<pre><code>[1] 19814   481</code></pre>
<pre class="r"><code># apply hard cutoffs 
# filter out genes that aren&#39;t at least expressed greater than 1 in at least half of cases
# set filter fraction
min_sm_frac = 0.5
filter_frac = min_sm_frac * ncol(totcounts_prcode_cpm_matrix)
filter_frac</code></pre>
<pre><code>[1] 240.5</code></pre>
<pre class="r"><code>total_cpm_stats &lt;- data.frame(
  total = apply(totcounts_prcode_cpm_matrix, 
                1, 
               function(x) {sum(x &gt; 1, na.rm = TRUE) } ))
keep &lt;- which(total_cpm_stats$total &gt;= filter_frac)

#---------------------
#### DECIDED NOT TO USE
## apply soft cutoffs
# set CPM threshold based on library size
# cpm_thres = 10 / min(colSums(totcounts_prcode_cpm_matrix)) * 10^6
# set minimum sample size
# keep &lt;- rowSums(cpm(totcounts_prcode_cpm_matrix) &gt; cpm_thres) # &gt;= as.integer(ncol(totcounts_prcode_cpm_matrix) / 2)
#---------------------

# apply keep criteria and check results
dim(totcounts_prcode_cpm_matrix)</code></pre>
<pre><code>[1] 19814   481</code></pre>
<pre class="r"><code>total_cpm_filtered = totcounts_prcode_cpm_matrix[keep,]
dim(total_cpm_filtered)</code></pre>
<pre><code>[1] 12858   481</code></pre>
</div>
<div id="density-plots-of-filtered-cpm-values" class="section level4">
<h4>Density plots of filtered CPM values</h4>
<pre class="r"><code>tidy_cpm_filtered &lt;- total_cpm_filtered %&gt;% 
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  gather(key = &quot;sampleID&quot;, value = &quot;intensity&quot;, -gene_id) 

dplot1 &lt;- ggplot(tidy_cpm_filtered, aes(intensity)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;) +
  xlim(-5, 100) +
  ggtitle(&quot;Density plot of CPM for NCI DLBCL&quot;)
dplot1</code></pre>
<pre><code>Warning: Removed 918433 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/preprocessing_count_data.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dplot2 &lt;- ggplot(tidy_cpm_filtered, aes(log(intensity))) +
  geom_density() +
  theme(legend.position = &quot;none&quot;) +
  xlim(-20, 20) +
  ggtitle(&quot;Density plot of log(CPM) for NCI DLBCL&quot;)
dplot2</code></pre>
<pre><code>Warning: Removed 3117 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/preprocessing_count_data.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="normalize-gene-expression-distributions" class="section level4">
<h4>Normalize gene expression distributions</h4>
<p>Normalisation by the method of trimmed mean of M-values (TMM) is performed using the calcNormFactors function in edgeR. The normalisation factors calculated here are used as a scaling factor for the library sizes.</p>
<pre class="r"><code># get normalization factors
norm_factors &lt;- calcNormFactors(total_cpm_filtered, method = &quot;TMM&quot;)

# apply factor to each column
total_cpm_norm &lt;- map2_dfc(total_cpm_filtered, norm_factors, `*`)
total_cpm_norm &lt;- as.data.frame(total_cpm_norm)
rownames(total_cpm_norm) &lt;- rownames(total_cpm_filtered)
total_cpm_norm[1:5, 1:5]</code></pre>
<pre><code>                   DLBCL11667 DLBCL10501 DLBCL10954 DLBCL10984 DLBCL11206
ENSG00000000003.13   1.526803   4.079266   2.715566   2.765948   13.13607
ENSG00000000419.11  32.541866  71.750371  29.732473  45.558137   51.21792
ENSG00000000457.12   5.149219   6.593881  31.972319  30.974047   28.41472
ENSG00000000460.15   4.849846  23.078585  21.189342  19.133046   22.06350
ENSG00000000938.11  24.758163  32.019441  48.721612  91.116274   61.19113</code></pre>
</div>
<div id="log-transformation" class="section level4">
<h4>Log transformation</h4>
<pre class="r"><code># introduce offset to prevent -Inf
offset_total_cpm_norm &lt;- total_cpm_norm + 0.5
offset_total_cpm_norm[1:5, 1:5]</code></pre>
<pre><code>                   DLBCL11667 DLBCL10501 DLBCL10954 DLBCL10984 DLBCL11206
ENSG00000000003.13   2.026803   4.579266   3.215566   3.265948   13.63607
ENSG00000000419.11  33.041866  72.250371  30.232473  46.058137   51.71792
ENSG00000000457.12   5.649219   7.093881  32.472319  31.474047   28.91472
ENSG00000000460.15   5.349846  23.578585  21.689342  19.633046   22.56350
ENSG00000000938.11  25.258163  32.519441  49.221612  91.616274   61.69113</code></pre>
<pre class="r"><code># log transformation
total_log_cpm_filtered_norm &lt;- log2(offset_total_cpm_norm)
total_log_cpm_filtered_norm[1:5, 1:5]</code></pre>
<pre><code>                   DLBCL11667 DLBCL10501 DLBCL10954 DLBCL10984 DLBCL11206
ENSG00000000003.13   1.019206   2.195116   1.685073   1.707502   3.769356
ENSG00000000419.11   5.046223   6.174933   4.918027   5.525384   5.692592
ENSG00000000457.12   2.498051   2.826575   5.021139   4.976091   4.853732
ENSG00000000460.15   2.419497   4.559405   4.438914   4.295212   4.495919
ENSG00000000938.11   4.658678   5.023231   5.621220   6.517532   5.946991</code></pre>
<p>Final density plot</p>
<pre class="r"><code>total_log_cpm_filtered_tbl &lt;- total_log_cpm_filtered_norm %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  as_tibble()

tidy_log_cpm &lt;- total_log_cpm_filtered_tbl %&gt;% 
  gather(key = &quot;sampleID&quot;, value = &quot;intensity&quot;, -gene)

dplot3 &lt;- ggplot(tidy_log_cpm, aes(intensity)) +
  geom_density() +
  theme(legend.position = &quot;none&quot;) +
  xlim(-20, 20) +
  ggtitle(&quot;Final density plot of log(CPM) for NCI DLBCL&quot;)
dplot3</code></pre>
<p><img src="figure/preprocessing_count_data.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="add-in-phenotype-data" class="section level2">
<h2>Add in phenotype data</h2>
<p>Read in file and do some prelim cleaning</p>
<pre class="r"><code>nci_phenotype_data &lt;- read_csv(&quot;output/nci_dlbcl_annotation.csv&quot;)</code></pre>
<pre><code>Parsed with column specification:
cols(
  .default = col_character(),
  days_to_last_follow_up = col_double(),
  updated_datetime = col_datetime(format = &quot;&quot;),
  age_at_diagnosis = col_double(),
  created_datetime = col_datetime(format = &quot;&quot;),
  demographic.updated_datetime = col_datetime(format = &quot;&quot;),
  demographic.created_datetime = col_datetime(format = &quot;&quot;),
  size = col_double()
)</code></pre>
<pre><code>See spec(...) for full column specifications.</code></pre>
<pre class="r"><code>nci_phenotype_data</code></pre>
<pre><code># A tibble: 481 x 67
   case_submitter_… sample case_id pdl1_status filename sample_id
   &lt;chr&gt;            &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;       &lt;chr&gt;    &lt;chr&gt;    
 1 DLBCL11667       nci_d… 3e164d… non-amplif… b384270… 608e5a37…
 2 DLBCL10501       nci_d… 678acb… non-amplif… ab3f958… ed5c027b…
 3 DLBCL10954       nci_d… c1b07b… non-amplif… 92fbf68… e5341594…
 4 DLBCL10984       nci_d… 15b380… low_amplif… dc44d3b… d5ef7640…
 5 DLBCL11206       nci_d… 7ac79d… non-amplif… 717d067… c39bb161…
 6 DLBCL10959       nci_d… 94d33a… low_amplif… 9765c55… 2654c7f3…
 7 DLBCL10985       nci_d… fa7ae8… non-amplif… 93f0175… 0c677fef…
 8 DLBCL11533       nci_d… 18134e… non-amplif… cd23ffe… 6fb786ee…
 9 DLBCL11519       nci_d… 3fe9ec… low_amplif… 03cb176… 72f51808…
10 DLBCL10921       nci_d… b31ef3… non-amplif… bd2bf78… d0a34b13…
# … with 471 more rows, and 61 more variables: sample_submitter_id &lt;chr&gt;,
#   project_id &lt;chr&gt;, sample_type_id &lt;chr&gt;,
#   time_between_excision_and_freezing &lt;chr&gt;, oct_embedded &lt;chr&gt;,
#   tumor_code_id &lt;chr&gt;, intermediate_dimension &lt;chr&gt;, is_ffpe &lt;chr&gt;,
#   pathology_report_uuid &lt;chr&gt;, tumor_descriptor &lt;chr&gt;,
#   sample_type &lt;chr&gt;, distance_normal_to_tumor &lt;chr&gt;,
#   biospecimen_anatomic_site &lt;chr&gt;, state.x &lt;chr&gt;,
#   diagnosis_pathologically_confirmed &lt;chr&gt;, current_weight &lt;chr&gt;,
#   composition &lt;chr&gt;, time_between_clamping_and_freezing &lt;chr&gt;,
#   distributor_reference &lt;chr&gt;, shortest_dimension &lt;chr&gt;,
#   method_of_sample_procurement &lt;chr&gt;, tumor_code &lt;chr&gt;,
#   passage_count &lt;chr&gt;, tissue_type &lt;chr&gt;, biospecimen_laterality &lt;chr&gt;,
#   days_to_sample_procurement &lt;chr&gt;, freezing_method &lt;chr&gt;,
#   preservation_method &lt;chr&gt;, growth_rate &lt;chr&gt;,
#   days_to_collection &lt;chr&gt;, catalog_reference &lt;chr&gt;,
#   initial_weight &lt;chr&gt;, longest_dimension &lt;chr&gt;, submitter_id &lt;chr&gt;,
#   morphology &lt;chr&gt;, days_to_last_follow_up &lt;dbl&gt;,
#   updated_datetime &lt;dttm&gt;, tumor_stage &lt;chr&gt;, age_at_diagnosis &lt;dbl&gt;,
#   created_datetime &lt;dttm&gt;, diagnosis_id &lt;chr&gt;,
#   tissue_or_organ_of_origin &lt;chr&gt;, ann_arbor_clinical_stage &lt;chr&gt;,
#   progression_or_recurrence &lt;chr&gt;, last_known_disease_status &lt;chr&gt;,
#   primary_diagnosis &lt;chr&gt;, tumor_grade &lt;chr&gt;,
#   site_of_resection_or_biopsy &lt;chr&gt;,
#   demographic.updated_datetime &lt;dttm&gt;,
#   demographic.created_datetime &lt;dttm&gt;, demographic.gender &lt;chr&gt;,
#   demographic.state &lt;chr&gt;, demographic.submitter_id &lt;chr&gt;,
#   demographic.race &lt;chr&gt;, demographic.ethnicity &lt;chr&gt;,
#   demographic.vital_status &lt;chr&gt;, demographic.demographic_id &lt;chr&gt;,
#   id &lt;chr&gt;, md5 &lt;chr&gt;, size &lt;dbl&gt;, state.y &lt;chr&gt;</code></pre>
<pre class="r"><code># clean up variables
nci_pheno_data &lt;- nci_phenotype_data %&gt;%
  dplyr::select(case_submitter_id,
                pdl1_status, 
                -sample, 
                case_id, 
                filename,
                -sample_id,
                -sample_submitter_id,
                project_id, 
                -sample_type_id,
                -time_between_excision_and_freezing,
                -oct_embedded, 
                -tumor_code_id,
                -intermediate_dimension,
                -is_ffpe,
                -pathology_report_uuid,
                -sample_type,
                -distance_normal_to_tumor,
                -biospecimen_anatomic_site,
                -state.x,
                -diagnosis_pathologically_confirmed,
                -current_weight,
                -composition,
                -time_between_clamping_and_freezing,
                -distributor_reference,
                -shortest_dimension,
                -method_of_sample_procurement,
                -tumor_code,
                -passage_count,
                -tissue_type,
                -biospecimen_laterality,
                -days_to_sample_procurement,
                -freezing_method,
                -preservation_method,
                -growth_rate,
                -days_to_collection,
                -catalog_reference,
                -initial_weight,
                -longest_dimension,
                -submitter_id,
                -morphology,
                days_to_last_follow_up,
                updated_datetime,
                tumor_stage,
                age_at_diagnosis,
                -created_datetime,
                -diagnosis_id,
                tissue_or_organ_of_origin,
                ann_arbor_clinical_stage,
                progression_or_recurrence,
                last_known_disease_status,
                primary_diagnosis,
                tumor_grade,
                -site_of_resection_or_biopsy,
                -demographic.updated_datetime,
                -demographic.created_datetime,
                demographic.gender,
                demographic.submitter_id,
                demographic.race,
                demographic.ethnicity,
                demographic.vital_status,
                -demographic.demographic_id,
                -id,
                -md5,
                size,
                -state.y
                ) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;case_submitter_id&quot;)

str(nci_pheno_data)</code></pre>
<pre><code>&#39;data.frame&#39;:   481 obs. of  20 variables:
 $ pdl1_status              : chr  &quot;non-amplified&quot; &quot;non-amplified&quot; &quot;non-amplified&quot; &quot;low_amplified&quot; ...
 $ case_id                  : chr  &quot;3e164d5d-afa0-4e96-b177-ea9fdf69da7a&quot; &quot;678acbbd-3f1b-4464-ba21-a59606cf19ae&quot; &quot;c1b07bfa-d234-42d6-8a41-97b7a7924f0d&quot; &quot;15b380e1-d220-42b9-8613-18fcefccb0a4&quot; ...
 $ filename                 : chr  &quot;b384270b-0315-49c6-bb46-d155d8d41fec.htseq_counts.txt.gz&quot; &quot;ab3f9584-f27f-4d6e-a5de-ac0d0c20de13.htseq_counts.txt.gz&quot; &quot;92fbf689-0c26-4664-9dde-c41534f7b390.htseq_counts.txt.gz&quot; &quot;dc44d3ba-5c4d-40a1-b8a2-f27b88325d0b.htseq_counts.txt.gz&quot; ...
 $ project_id               : chr  &quot;NCICCR-DLBCL&quot; &quot;NCICCR-DLBCL&quot; &quot;NCICCR-DLBCL&quot; &quot;NCICCR-DLBCL&quot; ...
 $ days_to_last_follow_up   : num  633 674 425 NA 2299 ...
 $ updated_datetime         : POSIXct, format: &quot;2018-08-31 20:09:50&quot; &quot;2018-08-31 20:09:50&quot; ...
 $ tumor_stage              : chr  &quot;IPI:1&quot; &quot;IPI:1&quot; &quot;IPI:13&quot; &quot;IPI:12&quot; ...
 $ age_at_diagnosis         : num  17520 15330 17155 27375 27740 ...
 $ tissue_or_organ_of_origin: chr  &quot;Lymph node, NOS&quot; &quot;Lymph node, NOS&quot; &quot;Lymph node, NOS&quot; &quot;Lymph node, NOS&quot; ...
 $ ann_arbor_clinical_stage : chr  &quot;Stage III&quot; &quot;Stage II&quot; &quot;Stage I&quot; &quot;Stage I&quot; ...
 $ progression_or_recurrence: chr  &quot;not reported&quot; &quot;not reported&quot; &quot;not reported&quot; &quot;not reported&quot; ...
 $ last_known_disease_status: chr  &quot;not reported&quot; &quot;not reported&quot; &quot;not reported&quot; &quot;not reported&quot; ...
 $ primary_diagnosis        : chr  &quot;Diffuse large B-cell lymphoma, NOS&quot; &quot;Diffuse large B-cell lymphoma, NOS&quot; &quot;Diffuse large B-cell lymphoma, NOS&quot; &quot;Diffuse large B-cell lymphoma, NOS&quot; ...
 $ tumor_grade              : chr  &quot;not reported&quot; &quot;not reported&quot; &quot;not reported&quot; &quot;not reported&quot; ...
 $ demographic.gender       : chr  &quot;female&quot; &quot;male&quot; &quot;male&quot; &quot;female&quot; ...
 $ demographic.submitter_id : chr  &quot;DLBCL11667-demographic&quot; &quot;DLBCL10501-demographic&quot; &quot;DLBCL10954-demographic&quot; &quot;DLBCL10984-demographic&quot; ...
 $ demographic.race         : chr  &quot;not reported&quot; &quot;not reported&quot; &quot;not reported&quot; &quot;not reported&quot; ...
 $ demographic.ethnicity    : chr  &quot;not reported&quot; &quot;not reported&quot; &quot;not reported&quot; &quot;not reported&quot; ...
 $ demographic.vital_status : chr  &quot;Dead&quot; &quot;Alive&quot; &quot;Dead&quot; &quot;Alive&quot; ...
 $ size                     : num  235439 232896 248096 245979 246899 ...
 - attr(*, &quot;spec&quot;)=
  .. cols(
  ..   case_submitter_id = col_character(),
  ..   sample = col_character(),
  ..   case_id = col_character(),
  ..   pdl1_status = col_character(),
  ..   filename = col_character(),
  ..   sample_id = col_character(),
  ..   sample_submitter_id = col_character(),
  ..   project_id = col_character(),
  ..   sample_type_id = col_character(),
  ..   time_between_excision_and_freezing = col_character(),
  ..   oct_embedded = col_character(),
  ..   tumor_code_id = col_character(),
  ..   intermediate_dimension = col_character(),
  ..   is_ffpe = col_character(),
  ..   pathology_report_uuid = col_character(),
  ..   tumor_descriptor = col_character(),
  ..   sample_type = col_character(),
  ..   distance_normal_to_tumor = col_character(),
  ..   biospecimen_anatomic_site = col_character(),
  ..   state.x = col_character(),
  ..   diagnosis_pathologically_confirmed = col_character(),
  ..   current_weight = col_character(),
  ..   composition = col_character(),
  ..   time_between_clamping_and_freezing = col_character(),
  ..   distributor_reference = col_character(),
  ..   shortest_dimension = col_character(),
  ..   method_of_sample_procurement = col_character(),
  ..   tumor_code = col_character(),
  ..   passage_count = col_character(),
  ..   tissue_type = col_character(),
  ..   biospecimen_laterality = col_character(),
  ..   days_to_sample_procurement = col_character(),
  ..   freezing_method = col_character(),
  ..   preservation_method = col_character(),
  ..   growth_rate = col_character(),
  ..   days_to_collection = col_character(),
  ..   catalog_reference = col_character(),
  ..   initial_weight = col_character(),
  ..   longest_dimension = col_character(),
  ..   submitter_id = col_character(),
  ..   morphology = col_character(),
  ..   days_to_last_follow_up = col_double(),
  ..   updated_datetime = col_datetime(format = &quot;&quot;),
  ..   tumor_stage = col_character(),
  ..   age_at_diagnosis = col_double(),
  ..   created_datetime = col_datetime(format = &quot;&quot;),
  ..   diagnosis_id = col_character(),
  ..   tissue_or_organ_of_origin = col_character(),
  ..   ann_arbor_clinical_stage = col_character(),
  ..   progression_or_recurrence = col_character(),
  ..   last_known_disease_status = col_character(),
  ..   primary_diagnosis = col_character(),
  ..   tumor_grade = col_character(),
  ..   site_of_resection_or_biopsy = col_character(),
  ..   demographic.updated_datetime = col_datetime(format = &quot;&quot;),
  ..   demographic.created_datetime = col_datetime(format = &quot;&quot;),
  ..   demographic.gender = col_character(),
  ..   demographic.state = col_character(),
  ..   demographic.submitter_id = col_character(),
  ..   demographic.race = col_character(),
  ..   demographic.ethnicity = col_character(),
  ..   demographic.vital_status = col_character(),
  ..   demographic.demographic_id = col_character(),
  ..   id = col_character(),
  ..   md5 = col_character(),
  ..   size = col_double(),
  ..   state.y = col_character()
  .. )</code></pre>
<pre class="r"><code>nci_pheno_data[1:5,1:5]</code></pre>
<pre><code>             pdl1_status                              case_id
DLBCL11667 non-amplified 3e164d5d-afa0-4e96-b177-ea9fdf69da7a
DLBCL10501 non-amplified 678acbbd-3f1b-4464-ba21-a59606cf19ae
DLBCL10954 non-amplified c1b07bfa-d234-42d6-8a41-97b7a7924f0d
DLBCL10984 low_amplified 15b380e1-d220-42b9-8613-18fcefccb0a4
DLBCL11206 non-amplified 7ac79db4-b307-46be-ba2c-b1731c7f97f0
                                                           filename
DLBCL11667 b384270b-0315-49c6-bb46-d155d8d41fec.htseq_counts.txt.gz
DLBCL10501 ab3f9584-f27f-4d6e-a5de-ac0d0c20de13.htseq_counts.txt.gz
DLBCL10954 92fbf689-0c26-4664-9dde-c41534f7b390.htseq_counts.txt.gz
DLBCL10984 dc44d3ba-5c4d-40a1-b8a2-f27b88325d0b.htseq_counts.txt.gz
DLBCL11206 717d067b-5cea-407d-9546-72fe4e4feba5.htseq_counts.txt.gz
             project_id days_to_last_follow_up
DLBCL11667 NCICCR-DLBCL                    633
DLBCL10501 NCICCR-DLBCL                    674
DLBCL10954 NCICCR-DLBCL                    425
DLBCL10984 NCICCR-DLBCL                     NA
DLBCL11206 NCICCR-DLBCL                   2299</code></pre>
</div>
<div id="write-out-for-later-use" class="section level2">
<h2>Write out for later use</h2>
<p>Following ExpressionSet vignette in Biobase package</p>
<pre class="r"><code># make expression set
# first need variable annotation
v_metadata &lt;- data.frame(
  labelDescription = colnames(nci_pheno_data),
  row.names = colnames(nci_pheno_data))

# make annotated phenoData
phenoData &lt;- new(&quot;AnnotatedDataFrame&quot;, 
                 data = nci_pheno_data, 
                 varMetadata = v_metadata)
phenoData</code></pre>
<pre><code>An object of class &#39;AnnotatedDataFrame&#39;
  rowNames: DLBCL11667 DLBCL10501 ... DLBCL11665 (481 total)
  varLabels: pdl1_status case_id ... size (20 total)
  varMetadata: labelDescription</code></pre>
<pre class="r"><code># convert expression data to matrix format
nci_exprs &lt;- total_log_cpm_filtered_norm %&gt;% as.matrix()

# create global annotation
annotation &lt;- as.character(&quot;RNAseq counts originally published by the Staudt lab at NCI in Schmitz et al NEJM 2018. Raw counts downloaded from GDC as HTseq counts, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed.&quot;)

# assemble expressionset
nci_es &lt;- ExpressionSet(
  assayData = nci_exprs,
  phenoData = phenoData,
  annotation = annotation)
nci_es</code></pre>
<pre><code>ExpressionSet (storageMode: lockedEnvironment)
assayData: 12858 features, 481 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: DLBCL11667 DLBCL10501 ... DLBCL11665 (481 total)
  varLabels: pdl1_status case_id ... size (20 total)
  varMetadata: labelDescription
featureData: none
experimentData: use &#39;experimentData(object)&#39;
Annotation: RNAseq counts originally published by the Staudt lab at NCI in Schmitz et al NEJM 2018. Raw counts downloaded from GDC as HTseq counts, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed. </code></pre>
<pre class="r"><code>saveRDS(nci_es, &quot;output/nci_expressionset.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.4

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] Biobase_2.42.0      BiocGenerics_0.28.0 edgeR_3.24.3       
 [4] limma_3.38.3        forcats_0.4.0       stringr_1.4.0      
 [7] dplyr_0.8.1         purrr_0.3.2         readr_1.3.1        
[10] tidyr_0.8.3         tibble_2.1.3        ggplot2_3.1.1      
[13] tidyverse_1.2.1    

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5 locfit_1.5-9.1   xfun_0.7         haven_2.1.0     
 [5] lattice_0.20-38  vctrs_0.1.0      colorspace_1.4-1 generics_0.0.2  
 [9] htmltools_0.3.6  yaml_2.2.0       utf8_1.1.4       rlang_0.3.4     
[13] pillar_1.4.1     glue_1.3.1       withr_2.1.2      modelr_0.1.4    
[17] readxl_1.3.1     plyr_1.8.4       munsell_0.5.0    gtable_0.3.0    
[21] workflowr_1.4.0  cellranger_1.1.0 rvest_0.3.4      evaluate_0.14   
[25] labeling_0.3     knitr_1.23       fansi_0.4.0      broom_0.5.2     
[29] Rcpp_1.0.1       scales_1.0.0     backports_1.1.4  jsonlite_1.6    
[33] fs_1.3.1         hms_0.4.2        digest_0.6.19    stringi_1.4.3   
[37] grid_3.5.3       rprojroot_1.3-2  cli_1.1.0        tools_3.5.3     
[41] magrittr_1.5     lazyeval_0.2.2   zeallot_0.1.0    crayon_1.3.4    
[45] whisker_0.3-2    pkgconfig_2.0.2  xml2_1.2.0       lubridate_1.7.4 
[49] assertthat_0.2.1 rmarkdown_1.13   httr_1.4.0       rstudioapi_0.10 
[53] R6_2.4.0         nlme_3.1-140     git2r_0.25.2     compiler_3.5.3  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
