<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-07-20" />

<title>sparse_clustering</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">sparse_clustering</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-07-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190613code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190613)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190613code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190613)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetree3b9db7d138cc2bb849acc70f8218adc98120712atargetblank3b9db7da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/3b9db7d138cc2bb849acc70f8218adc98120712a" target="_blank">3b9db7d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetree3b9db7d138cc2bb849acc70f8218adc98120712atargetblank3b9db7da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    NCI_mutations_vs_deconvolution.csv
    Ignored:    data/.DS_Store
    Ignored:    data/1-s2.0-S1074761316304320-mmc2.xlsx
    Ignored:    data/12864_2005_498_MOESM2_ESM.xls
    Ignored:    data/41591_2018_45_MOESM5_ESM.xlsx
    Ignored:    data/CCLE_sample_info_file_2012-10-18.txt
    Ignored:    data/NIH_PDL1_amp_cases.csv
    Ignored:    data/NIH_PDL1_nonamp_cases.csv
    Ignored:    data/PDL1dlbcl.rnaseq.kallisto.raw.txi.coding.cpm1.TMM.logCPM.sm24.csv
    Ignored:    data/PanImmune_GeneSet_Definitions.xlsx
    Ignored:    data/aliquot.tsv
    Ignored:    data/azizi_curated.xlsx
    Ignored:    data/c2_human_immune_genesets.gmt
    Ignored:    data/dlbcl_total_immune_gset_v1_results.rds
    Ignored:    data/ds_JCO.18.01583-2.xlsx
    Ignored:    data/duke_study_driver_mutations_top150_maf.csv
    Ignored:    data/final_analysis_set.maf
    Ignored:    data/gdc_clinical_data.json
    Ignored:    data/gdc_files_and_case_ids.json
    Ignored:    data/gencode.v22.annotation.gtf
    Ignored:    data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Ignored:    data/genesets/
    Ignored:    data/gset_ids_complete.rds
    Ignored:    data/histo_metadata_dlbcl_landscape.csv
    Ignored:    data/journal.pone.0088309.s001.XLS
    Ignored:    data/mmc1.xlsx
    Ignored:    data/nejmoa1801445_appendix_2.xlsx
    Ignored:    data/reactome_imm_signaling.tsv
    Ignored:    data/sample.tsv
    Ignored:    data/sig.rda
    Ignored:    data/sig_genes.txt
    Ignored:    data/temp.dgelist_edger.rds
    Ignored:    data/temp.dgelist_limma.rds
    Ignored:    data/uchi_metadata.csv
    Ignored:    logs/
    Ignored:    output/.DS_Store
    Ignored:    output/NCI_mutations_vs_deconvolution.csv
    Ignored:    output/clust1_vs_others.txt
    Ignored:    output/clust1_vs_others_dge.rnk
    Ignored:    output/clust2_vs_others_dge.rnk
    Ignored:    output/clust3_vs_others_dge.rnk
    Ignored:    output/clust4_vs_others_dge.rnk
    Ignored:    output/cluster1_dge.rds
    Ignored:    output/cluster2_dge.rds
    Ignored:    output/cluster3_dge.rds
    Ignored:    output/cluster4_dge.rds
    Ignored:    output/combined_clin_data.xlsx
    Ignored:    output/combined_dlbcl_expr_matrix.csv
    Ignored:    output/combined_es_gsva2_4cluster.rds
    Ignored:    output/combined_es_gsva4_4cluster.rds
    Ignored:    output/combined_expressionset.rds
    Ignored:    output/combined_tpm_matrix.csv
    Ignored:    output/counts_expressionset_for_limma.rds
    Ignored:    output/dlbcl_expr_matrix.csv
    Ignored:    output/dlbcl_total_immune_gset_v2_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v3_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v4_results.rds
    Ignored:    output/duke_expression_set_cleaned_log.rds
    Ignored:    output/duke_expressionset.rds
    Ignored:    output/duke_expressionset_pext.rds
    Ignored:    output/duke_extended_pheno_data.csv
    Ignored:    output/expr_matrix.csv
    Ignored:    output/gsea_res_cluster1.csv
    Ignored:    output/gset_ids_complete_2.rds
    Ignored:    output/gset_ids_complete_3.rds
    Ignored:    output/gset_ids_complete_4.rds
    Ignored:    output/gsva_v2_4clust_assign.rds
    Ignored:    output/gsva_v3_4clust_assign.rds
    Ignored:    output/gsva_v4_4clust_assign.rds
    Ignored:    output/limma_voom_elist_1.rds
    Ignored:    output/merged_raw_counts.csv
    Ignored:    output/nci_dlbcl_annotation.csv
    Ignored:    output/nci_dlbcl_unprocessed_counts.csv
    Ignored:    output/nci_expressionset.rds
    Ignored:    output/nci_expressionset_pext.rds
    Ignored:    output/nci_extended_pheno_data.csv
    Ignored:    output/networkanal3_clust1.txt
    Ignored:    output/networkanal3_clust2.txt
    Ignored:    output/networkanal3_clust3.txt
    Ignored:    output/networkanal3_clust4.txt
    Ignored:    output/total_immune_gset_v1.rds
    Ignored:    output/tpm_cyt_expressionset.rds
    Ignored:    output/tpm_expressionset.rds
    Ignored:    output/uchi_cluster_assignments.csv
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.pvals.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019_RAW.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.pvals.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819_RAW.txt
    Ignored:    output/xcell_input_matrix.csv

Untracked files:
    Untracked:  NCI_mutations_vs_deconvolution.xls
    Untracked:  analysis/correlation_clusters.Rmd
    Untracked:  batch1_cell_counts.pdf
    Untracked:  data/cibersort/
    Untracked:  data/gistic_results_nci/
    Untracked:  es_10k_hgnc/
    Untracked:  output/ASH_ORAL_figures.png
    Untracked:  output/ASH_ORAL_figures/
    Untracked:  output/Rplot.pdf
    Untracked:  output/Rplot01.pdf
    Untracked:  output/batch1_cell_seg_df.rds
    Untracked:  output/batch1_summary_cell_seg_df.rds
    Untracked:  output/colored_hclust_plot_by_group.pdf
    Untracked:  output/colored_hclust_plot_by_samples.pdf
    Untracked:  output/combined_counts_dge.rds
    Untracked:  output/combined_expr_matrix_for_signature_unprocessed.rmd
    Untracked:  output/dgelist_pheno_data.rds
    Untracked:  output/dlbcl_final_signature_matrix.tsv
    Untracked:  output/dlbcl_gsva5_4cluster_hcpc.rds
    Untracked:  output/dlbcl_gsva5_4cluster_pca.rds
    Untracked:  output/dlbcl_uchi_batch1_mixture.tsv
    Untracked:  output/duke_dlbcl_mixture.tsv
    Untracked:  output/gse_69030_probes.tsv
    Untracked:  output/hclust_labeled_by_cell_type.pdf
    Untracked:  output/mif_pilot_data_plot.pdf
    Untracked:  output/nci_dlbcl_mixture.tsv
    Untracked:  output/phenodata_for_signature_matrix.rmd
    Untracked:  output/reference_matrix_file.tsv
    Untracked:  output/table1.html
    Untracked:  output/uchi_deconvolution_batch_1.rds
    Untracked:  output/uchi_pheno_clust
    Untracked:  output/var_contribution_axis2.pdf
    Untracked:  output/venn.png
    Untracked:  output/venn.png.2020-02-23_11-17-47.log
    Untracked:  output/venn.png.2020-02-23_11-17-56.log
    Untracked:  output/venn.png.2020-02-23_11-20-44.log
    Untracked:  preprocessed_data_for_signature_matrix_es.rds

Unstaged changes:
    Modified:   .gitignore
    Deleted:    Rplot.pdf
    Deleted:    Rplot001.png
    Deleted:    Rplot002.png
    Deleted:    Rplot01.pdf
    Modified:   analysis/cyt.Rmd
    Modified:   analysis/deconvolution.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/newgs.Rmd
    Modified:   analysis/predictor.Rmd
    Modified:   analysis/preprocessing_count_data.Rmd
    Deleted:    data/DLBCL_histo/JG55/HE/14-1464/14-1464_Scan1_annotations.xml.lock
    Modified:   output/design_for_limma.rds
    Modified:   output/dgelist_for_limma.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/3b9db7d138cc2bb849acc70f8218adc98120712a/analysis/sparse_clustering.Rmd" target="_blank">3b9db7d</a>
</td>
<td>
mleukam
</td>
<td>
2020-05-09
</td>
<td>
wflow_publish(c(“analysis/merge.Rmd”, “analysis/mif.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/a78a45cffe9d483012683d912ca3ab8f19d2000b/analysis/sparse_clustering.Rmd" target="_blank">a78a45c</a>
</td>
<td>
mleukam
</td>
<td>
2019-10-12
</td>
<td>
archives dump from local machine - near final gsva v4 cluster model
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/dlbcl_landscape/cc1e45ead669caefaae22546681be422e0a9db2a/docs/sparse_clustering.html" target="_blank">cc1e45e</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/79273ad25d94576529a78b683d26c0239d0f5c42/analysis/sparse_clustering.Rmd" target="_blank">79273ad</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-20
</td>
<td>
sparse hierarchical clustering
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<pre class="r"><code># Clear the workspace
rm(list = ls())</code></pre>
<pre class="r"><code>knitr::opts_chunk$set(eval = FALSE)</code></pre>
<pre class="r"><code># store base R plotting parameters at startup for later reset
.pardefault &lt;- par(no.readonly = T)

# set seed
set.seed(818)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(&quot;tidyverse&quot;)
library(&quot;sparcl&quot;)
library(&quot;Biobase&quot;)
library(&quot;factoextra&quot;)
library(&quot;FactoMineR&quot;)
library(&quot;viridis&quot;)
library(&quot;ggdark&quot;)
library(&quot;jcolors&quot;)
library(&quot;M3C&quot;)
library(&quot;caTools&quot;)
library(&quot;ggConvexHull&quot;)
library(&quot;ggsci&quot;)
library(&quot;ggpubr&quot;)
library(&quot;viridis&quot;)
library(&quot;Rtsne&quot;)</code></pre>
<p>note: after using ggdark, need to restore defaults of fill/color with: “invert_geom_defaults()”</p>
<p>Read in combined expression set (for pheno data)</p>
<pre class="r"><code>combined_es &lt;- readRDS(&quot;output/combined_expressionset.rds&quot;)</code></pre>
<p>Select gene sets from initial library</p>
<pre class="r"><code>gset_ids_complete_4 &lt;- readRDS(&quot;output/gset_ids_complete_4.rds&quot;)
gset_ids_complete_4

names.initial &lt;- names(gset_ids_complete_4) %&gt;%
  enframe() %&gt;%
  dplyr::select(gs_name = value) %&gt;%
  print()

# remove B-cell related gene set names and gene sets used for descriptions of clusters only
# review using View function
# review unpublished uchi_t_inflamed_gs
remove.names &lt;- c(&quot;cyt_gs&quot;, &quot;dhit_gs&quot;, &quot;imsig_bcells_gs&quot;, &quot;thor_bcell_mg_igj&quot;, &quot;thor_bcell_receptors_score&quot;, &quot;thor_b_cell_pca_16704732&quot;, &quot;thor_bcell_21978456&quot;,     &quot;thor_gp2_immune_bcell_score&quot;, &quot;uchi_t_inflamed_gs&quot;)

names.handfiltered &lt;- names.initial %&gt;%
  dplyr::filter(!gs_name %in% remove.names) %&gt;%
  pull(gs_name)
head(names.handfiltered)

gene_sets_1 &lt;- enframe(gset_ids_complete_4) %&gt;%
  dplyr::rename(gs_name_1 = name, genes_1 = value) %&gt;%
  print()

gene_sets_2 &lt;- enframe(gset_ids_complete_4) %&gt;%
  dplyr::rename(gs_name_2 = name, genes_2 = value)

# make a table of pairwise comparisons for gene set names
comb &lt;- combs(names.handfiltered, 2) %&gt;%
  as.data.frame() %&gt;%
  as_tibble() %&gt;%
  dplyr::rename(gs_name_1 = V1, gs_name_2 = V2) %&gt;%
  left_join(gene_sets_1, by = &quot;gs_name_1&quot;) %&gt;%
  left_join(gene_sets_2, by = &quot;gs_name_2&quot;) %&gt;%
      print()

# define function to find overlap of gene set names
overlappy &lt;- function(genes_1, genes_2){
  prct &lt;- (2 * length(intersect(genes_1, genes_2))) / (length(genes_1) + length(genes_2))
  prct
}

comb_prct &lt;- comb %&gt;% 
  mutate(prct = map2_dbl(genes_1, genes_2, overlappy)) %&gt;%
  arrange(desc(prct)) %&gt;%
  print(n = 100)
hist(comb_prct$prct)

# get a list of gene sets to cut
# will cut gene sets with 2/3 genes shared or more
# 17 gene sets meet the criteria
# since column choice randomly selected, will cut column two
gs_cut &lt;- comb_prct %&gt;%
  dplyr::filter(prct &gt; 0.666) %&gt;%
  pull(gs_name_2) %&gt;%
  print()

names.lessoverlap &lt;- names.handfiltered %&gt;%
  enframe() %&gt;%
  dplyr::select(gs_name = value) %&gt;%
  dplyr::filter(!gs_name %in% gs_cut) %&gt;%
  pull(gs_name)
print(names.lessoverlap)
length(names.lessoverlap)
# n = 121

filt_gs &lt;- enframe(gset_ids_complete_4) %&gt;%
  dplyr::rename(gs_name = name, genes = value) %&gt;%
  dplyr::filter(gs_name %in% names.lessoverlap) %&gt;%
  print()
filt_gs_list &lt;- as.list(filt_gs$genes)
names(filt_gs_list) &lt;- as.character(filt_gs$gs_name)
head(filt_gs_list)
  
raw_score_matrix &lt;- readRDS(&quot;output/dlbcl_total_immune_gset_v4_results.rds&quot;) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gs_names&quot;)
raw_score_matrix[1:5, 1:5]
dim(raw_score_matrix)

score_matrix &lt;- raw_score_matrix %&gt;%
  as_tibble() %&gt;%
  dplyr::filter(gs_names %in% names.lessoverlap) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;gs_names&quot;) %&gt;%
  t()
score_matrix[1:5, 1:5]
dim(score_matrix)</code></pre>
<p>Get total number of genes</p>
<pre class="r"><code># get a table of final genes in each gene set by filtered gene set name
filt_genes &lt;- enframe(gset_ids_complete_4) %&gt;%
  dplyr::rename(gs_name = name, genes = value) %&gt;%
  dplyr::filter(gs_name %in% names.handfiltered) %&gt;%
  dplyr::filter(!gs_name %in% gs_cut) %&gt;%
  print()

# vector of all genes
genes &lt;- filt_genes %&gt;%
  pull(genes) %&gt;%
  as_vector()

# get number of unique genes
genes %&gt;%
  unique() %&gt;%
  length()

# frequency table
freq_tab &lt;- filt_genes %&gt;%
  unnest() %&gt;%
  group_by(genes) %&gt;%
  count(genes) %&gt;%
  arrange(desc(n)) %&gt;%
  mutate(prop = n / 121) %&gt;%
  print()

# mean frequency
freq_tab %&gt;% pull(n) %&gt;% as.numeric() %&gt;% mean()

# get hugo symbols for genes
# read in conversion table for gene IDs
gencode_gtf &lt;- read_tsv(&quot;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&quot;) %&gt;%
  print()
freq_tab_hugo &lt;- freq_tab %&gt;%
  dplyr::rename(gene_id = genes) %&gt;%
  left_join(gencode_gtf) %&gt;%
  dplyr::select(gene_name, n, prop, gene_id) %&gt;%
  print()</code></pre>
</div>
<div id="hierarchical-sparse-clustering-httpsrdrr.iocransparclmanhierarchicalsparsecluster.html" class="section level1">
<h1>Hierarchical sparse clustering: <a href="https://rdrr.io/cran/sparcl/man/HierarchicalSparseCluster.html" class="uri">https://rdrr.io/cran/sparcl/man/HierarchicalSparseCluster.html</a></h1>
<p>Manipulating an hclust object <a href="https://stat.ethz.ch/R-manual/R-patched/library/stats/html/hclust.html" class="uri">https://stat.ethz.ch/R-manual/R-patched/library/stats/html/hclust.html</a></p>
<pre class="r"><code>set.seed(818)

# cases in rows and variables in columns
score_matrix[1:5, 1:5]
dim(score_matrix)

# Do tuning parameter selection for sparse hierarchical clustering
perm.out &lt;- HierarchicalSparseCluster.permute(score_matrix, 
                                              wbounds = c(1.5,2:6), 
                                              nperms = 5)
print(perm.out)
plot(perm.out)
str(perm.out)

# Perform sparse hierarchical clustering
sparsehc &lt;- HierarchicalSparseCluster(dists = perm.out$dists, 
                                      wbound = perm.out$bestw, 
                                      method = &quot;complete&quot;)
par(mfrow = c(1, 2))
plot(sparsehc)
sparsehc
str(sparsehc)

# Get cluster assignments
memb &lt;- cutree(sparsehc$hc, k = 4)
summary(as.factor(memb))
memb_df &lt;- enframe(memb) %&gt;%
  select(sp_clust = value) %&gt;%
  print()

# Add cluster assignments to phenoData
pheno_data &lt;- pData(combined_es) %&gt;% 
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  bind_cols(memb_df) %&gt;%
  dplyr::select(sample_id, sp_clust, everything()) %&gt;%
  print()

# Get names of nonzero weighted gene sets
gs_coef &lt;- sparsehc$ws %&gt;% as_tibble() %&gt;% dplyr::rename(Wj = V1)
retained_gene_sets &lt;- enframe(colnames(score_matrix)) %&gt;% 
  bind_cols(gs_coef) %&gt;%
  dplyr::rename(gs = value, index = name) %&gt;%
  filter(Wj != 0) %&gt;%
  arrange(desc(Wj)) %&gt;%
  print(n = nrow(.))</code></pre>
<pre class="r"><code># reset plot parameters
par(.pardefault)

# plot dendogram
fviz_dend(sparsehc$hc, 
          k = 4,
          palette = &quot;nejm&quot;,
          show_labels = FALSE, 
          main = &quot;Sparse Hierarchical Cluster Assignments, k = 4&quot;,
          rect = TRUE,
          rect_lty = 2,
          rect_fill = FALSE,
          lower_rect = -0.0001)</code></pre>
<pre class="r"><code># filter gene sets
# make vector of retained gene set names
keep_gs &lt;- retained_gene_sets$gs

# use vector to filter gene sets
score_mat_filt &lt;- score_matrix %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  dplyr::select(sample_id, one_of(keep_gs)) %&gt;%
  # add cluster assignment
  left_join(pheno_data, by = &quot;sample_id&quot;) %&gt;%
  dplyr::select(sample_id, sp_clust, one_of(keep_gs)) %&gt;%
  mutate(sp_clust = as.factor(sp_clust)) %&gt;%
  print()

sp_pca &lt;- PCA(score_mat_filt[3:59],
    scale.unit = TRUE,
    graph = FALSE)

fviz_pca_ind(sp_pca,
             label = &quot;none&quot;,
             pointshape = 19,
             habillage = score_mat_filt$sp_clust,
             legend.title = &quot;Clusters&quot;) + 
  dark_theme_gray() +
  scale_color_jcolors(palette = &quot;pal3&quot;) +
  scale_fill_jcolors(palette = &quot;pal3&quot;) +
  geom_convexhull(alpha = 0.3, 
                  aes(fill = score_mat_filt$sp_clust),
                  show.legend = FALSE,
                  color = &quot;#76767666&quot;)

fviz_pca_var(sp_pca)</code></pre>
<p>Visualization with tSNE</p>
<p><a href="https://www.r-bloggers.com/playing-with-dimensions-from-clustering-pca-t-sne-to-carl-sagan/" class="uri">https://www.r-bloggers.com/playing-with-dimensions-from-clustering-pca-t-sne-to-carl-sagan/</a></p>
<pre class="r"><code>set.seed(818)

## create score matrix in proper format
score_data &lt;- score_mat_filt %&gt;% 
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;sample_id&quot;) %&gt;%
  dplyr::select(-sp_clust)
score_data[1:5, 1:5]

## get tSNE model
tsne_model_1 &lt;- Rtsne(score_data, 
     dims = 2, 
     initial_dims = 57,
     perplexity = 200,
     verbose = TRUE,
     theta = 0.2,
     pca = TRUE,
     normalize = FALSE
     )

## get the two dimensional matrix for plotting
d_tsne_1 &lt;- as.data.frame(tsne_model_1$Y) 
rownames(d_tsne_1) &lt;- rownames(score_data)

## add groups from sparse clustering
d_tsne_1$cluster &lt;- score_mat_filt$sp_clust
head(d_tsne_1)

## plotting the results
ggplot(d_tsne_1, aes(x = V1, y = V2, color = cluster)) +  
  geom_point() +
  xlab(&quot;&quot;) + ylab(&quot;&quot;) +
  ggtitle(&quot;t-SNE&quot;) +
  theme_classic2() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank()) +
  scale_color_d3() +
  geom_convexhull(aes(fill = cluster), 
                  alpha = 0.1,
                  show.legend = FALSE)</code></pre>
<p>Visualization with UMAP</p>
<pre class="r"><code>library(umap)
sp_clust_umap &lt;- umap(score_data)

## get the two dimensional matrix for plotting
d_umap_1 &lt;- as.data.frame(sp_clust_umap$layout)
head(d_umap_1)

## add back individual gene set expression for exploratory plotting
d_umap_1_gs &lt;- d_umap_1 %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  left_join(score_mat_filt) %&gt;%
  print()

# plot UMAP with cluster labels
ggplot(d_umap_1_gs, aes(x = V1, 
                     y = V2,
                     color = sp_clust)) +  
  geom_point() +
  geom_convexhull(alpha = 0.1,
                  show.legend = FALSE) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;) +
  ggtitle(&quot;UMAP with cluster labels&quot;) +
  theme_classic2() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank()) +
  labs(color = &quot;Cluster&quot;) +
  scale_color_d3()

# plot UMAP with CYT score
cyt_gs &lt;- raw_score_matrix %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;gs_names&quot;) %&gt;%
  as.matrix() %&gt;%
  t() %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  dplyr::select(sample_id, cyt_gs) %&gt;%
  as_tibble() %&gt;%
  print()

d_umap_1_gs_cyt &lt;- d_umap_1_gs %&gt;%
  left_join(cyt_gs) %&gt;%
  print()

ggplot(d_umap_1_gs_cyt, aes(x = V1, 
                     y = V2,
                     color = cyt_gs)) +  
  geom_point() +
  geom_convexhull(aes(fill = sp_clust), 
                  alpha = 0.1,
                  show.legend = FALSE) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;) +
  ggtitle(&quot;UMAP and CYT GSVA score&quot;) +
  theme_classic2() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank()) +
  labs(color = &quot;Cluster&quot;) +
  scale_color_viridis()

## DAP12 gene set
ggplot(d_umap_1_gs_cyt, aes(x = V1, 
                     y = V2,
                     color = thor_dap12_data)) +  
  geom_point() +
  geom_convexhull(aes(fill = sp_clust), 
                  alpha = 0.1,
                  show.legend = FALSE) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;) +
  ggtitle(&quot;UMAP and DAP12 score&quot;) +
  theme_classic2() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank()) +
  labs(color = &quot;Cluster&quot;) +
  scale_color_viridis()

## CD68 gene set
ggplot(d_umap_1_gs_cyt, aes(x = V1, 
                     y = V2,
                     color = thor_cd68)) +  
  geom_point() +
  geom_convexhull(aes(fill = sp_clust), 
                  alpha = 0.1,
                  show.legend = FALSE) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;) +
  ggtitle(&quot;UMAP and CD68 score&quot;) +
  theme_classic2() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank()) +
  labs(color = &quot;Cluster&quot;) +
  scale_color_viridis()

## Imsig macrophages
ggplot(d_umap_1_gs_cyt, aes(x = V1, 
                     y = V2,
                     color = imsig_macrophages_gs)) +  
  geom_point() +
  geom_convexhull(aes(fill = sp_clust), 
                  alpha = 0.1,
                  show.legend = FALSE) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;) +
  ggtitle(&quot;UMAP and Imsig Mac score&quot;) +
  theme_classic2() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank()) +
  labs(color = &quot;Cluster&quot;) +
  scale_color_viridis()

## Trem1
ggplot(d_umap_1_gs_cyt, aes(x = V1, 
                     y = V2,
                     color = thor_trem1_data)) +  
  geom_point() +
  geom_convexhull(aes(fill = sp_clust), 
                  alpha = 0.1,
                  show.legend = FALSE) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;) +
  ggtitle(&quot;UMAP and Trem1 GSVA score&quot;) +
  theme_classic2() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank()) +
  labs(color = &quot;Cluster&quot;) +
  scale_color_viridis()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
