<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-07-11" />

<title>Visualizing results of GSVA with version 4 gene lists</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immune Landscape of DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Visualizing results of GSVA with version 4 gene lists</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-07-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dlbcl_landscape/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190613code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190613)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190613code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190613)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetree3b9db7d138cc2bb849acc70f8218adc98120712atargetblank3b9db7da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/dlbcl_landscape/tree/3b9db7d138cc2bb849acc70f8218adc98120712a" target="_blank">3b9db7d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamdlbcllandscapetree3b9db7d138cc2bb849acc70f8218adc98120712atargetblank3b9db7da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    NCI_mutations_vs_deconvolution.csv
    Ignored:    data/.DS_Store
    Ignored:    data/1-s2.0-S1074761316304320-mmc2.xlsx
    Ignored:    data/12864_2005_498_MOESM2_ESM.xls
    Ignored:    data/41591_2018_45_MOESM5_ESM.xlsx
    Ignored:    data/CCLE_sample_info_file_2012-10-18.txt
    Ignored:    data/NIH_PDL1_amp_cases.csv
    Ignored:    data/NIH_PDL1_nonamp_cases.csv
    Ignored:    data/PDL1dlbcl.rnaseq.kallisto.raw.txi.coding.cpm1.TMM.logCPM.sm24.csv
    Ignored:    data/PanImmune_GeneSet_Definitions.xlsx
    Ignored:    data/aliquot.tsv
    Ignored:    data/azizi_curated.xlsx
    Ignored:    data/c2_human_immune_genesets.gmt
    Ignored:    data/dlbcl_total_immune_gset_v1_results.rds
    Ignored:    data/ds_JCO.18.01583-2.xlsx
    Ignored:    data/duke_study_driver_mutations_top150_maf.csv
    Ignored:    data/final_analysis_set.maf
    Ignored:    data/gdc_clinical_data.json
    Ignored:    data/gdc_files_and_case_ids.json
    Ignored:    data/gencode.v22.annotation.gtf
    Ignored:    data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Ignored:    data/genesets/
    Ignored:    data/gset_ids_complete.rds
    Ignored:    data/histo_metadata_dlbcl_landscape.csv
    Ignored:    data/journal.pone.0088309.s001.XLS
    Ignored:    data/mmc1.xlsx
    Ignored:    data/nejmoa1801445_appendix_2.xlsx
    Ignored:    data/reactome_imm_signaling.tsv
    Ignored:    data/sample.tsv
    Ignored:    data/sig.rda
    Ignored:    data/sig_genes.txt
    Ignored:    data/temp.dgelist_edger.rds
    Ignored:    data/temp.dgelist_limma.rds
    Ignored:    data/uchi_metadata.csv
    Ignored:    logs/
    Ignored:    output/.DS_Store
    Ignored:    output/NCI_mutations_vs_deconvolution.csv
    Ignored:    output/clust1_vs_others.txt
    Ignored:    output/clust1_vs_others_dge.rnk
    Ignored:    output/clust2_vs_others_dge.rnk
    Ignored:    output/clust3_vs_others_dge.rnk
    Ignored:    output/clust4_vs_others_dge.rnk
    Ignored:    output/cluster1_dge.rds
    Ignored:    output/cluster2_dge.rds
    Ignored:    output/cluster3_dge.rds
    Ignored:    output/cluster4_dge.rds
    Ignored:    output/combined_clin_data.xlsx
    Ignored:    output/combined_dlbcl_expr_matrix.csv
    Ignored:    output/combined_es_gsva2_4cluster.rds
    Ignored:    output/combined_es_gsva4_4cluster.rds
    Ignored:    output/combined_expressionset.rds
    Ignored:    output/combined_tpm_matrix.csv
    Ignored:    output/counts_expressionset_for_limma.rds
    Ignored:    output/dlbcl_expr_matrix.csv
    Ignored:    output/dlbcl_total_immune_gset_v2_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v3_results.rds
    Ignored:    output/dlbcl_total_immune_gset_v4_results.rds
    Ignored:    output/duke_expression_set_cleaned_log.rds
    Ignored:    output/duke_expressionset.rds
    Ignored:    output/duke_expressionset_pext.rds
    Ignored:    output/duke_extended_pheno_data.csv
    Ignored:    output/expr_matrix.csv
    Ignored:    output/gsea_res_cluster1.csv
    Ignored:    output/gset_ids_complete_2.rds
    Ignored:    output/gset_ids_complete_3.rds
    Ignored:    output/gset_ids_complete_4.rds
    Ignored:    output/gsva_v2_4clust_assign.rds
    Ignored:    output/gsva_v3_4clust_assign.rds
    Ignored:    output/gsva_v4_4clust_assign.rds
    Ignored:    output/limma_voom_elist_1.rds
    Ignored:    output/merged_raw_counts.csv
    Ignored:    output/nci_dlbcl_annotation.csv
    Ignored:    output/nci_dlbcl_unprocessed_counts.csv
    Ignored:    output/nci_expressionset.rds
    Ignored:    output/nci_expressionset_pext.rds
    Ignored:    output/nci_extended_pheno_data.csv
    Ignored:    output/networkanal3_clust1.txt
    Ignored:    output/networkanal3_clust2.txt
    Ignored:    output/networkanal3_clust3.txt
    Ignored:    output/networkanal3_clust4.txt
    Ignored:    output/total_immune_gset_v1.rds
    Ignored:    output/tpm_cyt_expressionset.rds
    Ignored:    output/tpm_expressionset.rds
    Ignored:    output/uchi_cluster_assignments.csv
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.pvals.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019.txt
    Ignored:    output/xCell_combined_tpm_matrix_xCell_2225072019_RAW.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.pvals.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819.txt
    Ignored:    output/xCell_xcell_input_matrix_xCell_1039071819_RAW.txt
    Ignored:    output/xcell_input_matrix.csv

Untracked files:
    Untracked:  NCI_mutations_vs_deconvolution.xls
    Untracked:  analysis/correlation_clusters.Rmd
    Untracked:  batch1_cell_counts.pdf
    Untracked:  data/cibersort/
    Untracked:  data/gistic_results_nci/
    Untracked:  es_10k_hgnc/
    Untracked:  output/ASH_ORAL_figures.png
    Untracked:  output/ASH_ORAL_figures/
    Untracked:  output/Rplot.pdf
    Untracked:  output/Rplot01.pdf
    Untracked:  output/batch1_cell_seg_df.rds
    Untracked:  output/batch1_summary_cell_seg_df.rds
    Untracked:  output/colored_hclust_plot_by_group.pdf
    Untracked:  output/colored_hclust_plot_by_samples.pdf
    Untracked:  output/combined_counts_dge.rds
    Untracked:  output/combined_expr_matrix_for_signature_unprocessed.rmd
    Untracked:  output/dgelist_pheno_data.rds
    Untracked:  output/dlbcl_final_signature_matrix.tsv
    Untracked:  output/dlbcl_gsva5_4cluster_hcpc.rds
    Untracked:  output/dlbcl_gsva5_4cluster_pca.rds
    Untracked:  output/dlbcl_uchi_batch1_mixture.tsv
    Untracked:  output/duke_dlbcl_mixture.tsv
    Untracked:  output/gse_69030_probes.tsv
    Untracked:  output/hclust_labeled_by_cell_type.pdf
    Untracked:  output/mif_pilot_data_plot.pdf
    Untracked:  output/nci_dlbcl_mixture.tsv
    Untracked:  output/phenodata_for_signature_matrix.rmd
    Untracked:  output/reference_matrix_file.tsv
    Untracked:  output/table1.html
    Untracked:  output/uchi_deconvolution_batch_1.rds
    Untracked:  output/uchi_pheno_clust
    Untracked:  output/var_contribution_axis2.pdf
    Untracked:  output/venn.png
    Untracked:  output/venn.png.2020-02-23_11-17-47.log
    Untracked:  output/venn.png.2020-02-23_11-17-56.log
    Untracked:  output/venn.png.2020-02-23_11-20-44.log
    Untracked:  preprocessed_data_for_signature_matrix_es.rds

Unstaged changes:
    Modified:   .gitignore
    Deleted:    Rplot.pdf
    Deleted:    Rplot001.png
    Deleted:    Rplot002.png
    Deleted:    Rplot01.pdf
    Modified:   analysis/cyt.Rmd
    Modified:   analysis/deconvolution.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/newgs.Rmd
    Modified:   analysis/predictor.Rmd
    Modified:   analysis/preprocessing_count_data.Rmd
    Deleted:    data/DLBCL_histo/JG55/HE/14-1464/14-1464_Scan1_annotations.xml.lock
    Modified:   output/design_for_limma.rds
    Modified:   output/dgelist_for_limma.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/3b9db7d138cc2bb849acc70f8218adc98120712a/analysis/visualizing_gsva_scores_4.Rmd" target="_blank">3b9db7d</a>
</td>
<td>
mleukam
</td>
<td>
2020-05-09
</td>
<td>
wflow_publish(c(“analysis/merge.Rmd”, “analysis/mif.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/a78a45cffe9d483012683d912ca3ab8f19d2000b/analysis/visualizing_gsva_scores_4.Rmd" target="_blank">a78a45c</a>
</td>
<td>
mleukam
</td>
<td>
2019-10-12
</td>
<td>
archives dump from local machine - near final gsva v4 cluster model
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/dlbcl_landscape/blob/01fb3c107e629cfc19b059394907946e892ab03a/analysis/visualizing_gsva_scores_4.Rmd" target="_blank">01fb3c1</a>
</td>
<td>
mleukam
</td>
<td>
2019-07-16
</td>
<td>
updates of analysis pages without building html pages - for record keeping and safekeeping
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Clear the workspace</p>
<pre class="r"><code>rm(list = ls())</code></pre>
<pre class="r"><code>knitr::opts_chunk$set(eval = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(&quot;factoextra&quot;)
library(&quot;FactoMineR&quot;)
library(&quot;cluster&quot;)
library(&quot;ggpubr&quot;)
library(&quot;ggsci&quot;)
library(&quot;Biobase&quot;)
library(&quot;ComplexHeatmap&quot;)
library(&quot;clustertend&quot;)
library(&quot;NbClust&quot;)
library(&quot;jcolors&quot;)
library(&quot;ggdark&quot;)
library(&quot;viridis&quot;)
library(&quot;ggConvexHull&quot;)
library(&quot;tidyverse&quot;)
library(&quot;broom&quot;)</code></pre>
<p>NB: need to transfer scripts to <code>/code/</code>: * run_gsva4.pbs * gsva4.sh * gsva4.r</p>
<pre class="r"><code>gs_matrix &lt;- readRDS(&quot;output/dlbcl_total_immune_gset_v4_results.rds&quot;)
gs_matrix[, 1:5]
str(gs_matrix)

combined_es &lt;- readRDS(&quot;output/combined_expressionset.rds&quot;)
pheno_data &lt;- pData(combined_es)</code></pre>
</div>
<div id="quick-checks" class="section level1">
<h1>Quick Checks</h1>
<div id="does-this-clustering-separate-pd-l1amp-cases" class="section level4">
<h4>Does this clustering separate PD-L1amp cases?</h4>
<p>PD-L1 is a known immune inflamed phenotype. Can this clustering method show some grouping of PD-L1 distinct from the background?</p>
<pre class="r"><code>str(pheno_data)
summary(as.factor(pheno_data$pdl1_status))
pheno_data_tbl &lt;- as.data.frame(pheno_data) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  mutate(pdl1_status = replace_na(pdl1_status, &quot;not_assessed&quot;)) %&gt;%
  mutate(pdl1_status = ifelse(pdl1_status == &quot;non-amplified&quot;, 
                              &quot;non_amplified&quot;, pdl1_status))

summary(as.factor(pheno_data_tbl$pdl1_status))

group_table &lt;- pheno_data_tbl %&gt;%
  dplyr::select(sample_id, pdl1_status)

# merge group data into gsva matrix
pdl1_comb_exprs &lt;- as.data.frame(t(gs_matrix)) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  left_join(group_table) %&gt;%
  dplyr::select(sample_id, pdl1_status, everything()) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;sample_id&quot;) %&gt;%
  mutate(pdl1_status = as.factor(pdl1_status))</code></pre>
</div>
<div id="pca-plot-of-results" class="section level4">
<h4>PCA plot of results</h4>
<pre class="r"><code># remove low amplified and samples not assessed
nrow(pdl1_comb_exprs)
pdl1_comb_exprs_filt &lt;- pdl1_comb_exprs %&gt;%
  dplyr::filter(pdl1_status %in% c(&quot;high_amplified&quot;, &quot;non_amplified&quot;))
nrow(pdl1_comb_exprs_filt)

# get PCA values
dlbcl_pca &lt;- PCA(pdl1_comb_exprs_filt[-1], scale.unit = TRUE, graph = FALSE)

# get eigenvalues
eig_val &lt;- get_eigenvalue(dlbcl_pca)
eig_val
fviz_eig(dlbcl_pca, addlabels = TRUE, ylim = c(0, 50))
ind &lt;- get_pca_ind(dlbcl_pca)

pdl1_status &lt;- pdl1_comb_exprs_filt$pdl1_status

# visualize PCA
fviz_pca_ind(dlbcl_pca,
             label = &quot;none&quot;,
             habillage = pdl1_comb_exprs_filt$pdl1_status,
             palette = &quot;simpsons&quot;,
             addEllipses = TRUE,
             ggtheme = dark_theme_gray()) +
  theme(axis.line = element_line(color = &quot;#767676&quot;)) +
  geom_hline(yintercept = 0, color = &quot;white&quot;, alpha = 0.3) +
  geom_vline(xintercept = 0, color = &quot;white&quot;, alpha = 0.3) +
  ggtitle(&quot;Combined Thorsson and new gene sets \nPD-L1amp vs nonamp, NCI data only&quot;)

fviz_pca_var(dlbcl_pca, col.var = &quot;contrib&quot;,
             gradient.cols = &quot;tron&quot;,
             label = &quot;none&quot;,
             ggtheme = dark_theme_gray()) + 
  theme(axis.line = element_line(color = &quot;#767676&quot;)) +
  geom_hline(yintercept = 0, color = &quot;#76767666&quot;) +
  geom_vline(xintercept = 0, color = &quot;#76767666&quot;)

# visualize PCA -- white
fviz_pca_ind(dlbcl_pca,
             label = &quot;none&quot;,
             habillage = pdl1_comb_exprs_filt$pdl1_status,
             palette = &quot;nejm&quot;,
             addEllipses = TRUE,
             ggtheme = theme_classic()) +
  theme(axis.line = element_line(color = &quot;#767676&quot;)) +
  geom_hline(yintercept = 0, color = &quot;black&quot;, alpha = 0.3) +
  geom_vline(xintercept = 0, color = &quot;black&quot;, alpha = 0.3) +
  ggtitle(&quot;Combined Thorsson and new gene sets \nPD-L1amp vs nonamp, NCI data only&quot;)

fviz_pca_var(dlbcl_pca, col.var = &quot;contrib&quot;,
             gradient.cols = &quot;tron&quot;,
             label = &quot;none&quot;,
             ggtheme = theme_classic()) + 
  theme(axis.line = element_line(color = &quot;#767676&quot;)) +
  geom_hline(yintercept = 0, color = &quot;#76767666&quot;) +
  geom_vline(xintercept = 0, color = &quot;#76767666&quot;)</code></pre>
</div>
<div id="is-there-a-batch-effect-by-source-of-data" class="section level4">
<h4>Is there a batch effect by source of data?</h4>
<p>Set up Duke/NCI groups</p>
<pre class="r"><code>str(pheno_data)
summary(as.factor(pheno_data$source))
group_table &lt;- as.data.frame(pheno_data) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  dplyr::select(sample_id, source)

# merge group data into gsva matrix
dukenci_comb_exprs &lt;- as.data.frame(t(gs_matrix)) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  left_join(group_table) %&gt;%
  dplyr::select(sample_id, source, everything()) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;sample_id&quot;) %&gt;%
  mutate(source = as.factor(source))</code></pre>
</div>
<div id="pca-plot-of-results-1" class="section level4">
<h4>PCA Plot of results</h4>
<pre class="r"><code># get PCA values
dlbcl_pca &lt;- PCA(dukenci_comb_exprs[-1], scale.unit = TRUE, graph = FALSE)

# get eigenvalues
eig_val &lt;- get_eigenvalue(dlbcl_pca)
eig_val
fviz_eig(dlbcl_pca, addlabels = TRUE, ylim = c(0, 50))
ind &lt;- get_pca_ind(dlbcl_pca)

pdl1_status &lt;- dukenci_comb_exprs$pdl1_status

# visualize PCA
fviz_pca_ind(dlbcl_pca,
             label = &quot;none&quot;,
             habillage = dukenci_comb_exprs$source,
             palette = &quot;tron&quot;,
             addEllipses = TRUE,
             ggtheme = dark_theme_gray()) +
  theme(axis.line = element_line(color = &quot;#767676&quot;)) +
  geom_hline(yintercept = 0, color = &quot;white&quot;, alpha = 0.3) +
  geom_vline(xintercept = 0, color = &quot;white&quot;, alpha = 0.3) +
  ggtitle(&quot;Combined Thorsson and new gene sets \nDuke vs NCI data sources&quot;)

fviz_pca_var(dlbcl_pca, col.var = &quot;contrib&quot;,
             gradient.cols = &quot;tron&quot;,
             ggtheme = dark_theme_gray()) +
  geom_hline(yintercept = 0, color = &quot;white&quot;, alpha = 0.3) +
  geom_vline(xintercept = 0, color = &quot;white&quot;, alpha = 0.3)

# Contributions of variables to PC1
fviz_contrib(dlbcl_pca, choice = &quot;var&quot;, axes = 1, top = 15) + 
  dark_theme_classic() +
  theme(axis.text.x = element_text(angle = 80, hjust = 1))

# Contributions of variables to PC2
fviz_contrib(dlbcl_pca, choice = &quot;var&quot;, axes = 2, top = 15) +
  dark_theme_classic() +
  theme(axis.text.x = element_text(angle = 80, hjust = 1))

# visualize PCA -- white
fviz_pca_ind(dlbcl_pca,
             label = &quot;none&quot;,
             habillage = dukenci_comb_exprs$source,
             palette = &quot;nejm&quot;,
             addEllipses = TRUE,
             ggtheme = theme_classic()) +
  theme(axis.line = element_line(color = &quot;#767676&quot;)) +
  geom_hline(yintercept = 0, color = &quot;black&quot;, alpha = 0.3) +
  geom_vline(xintercept = 0, color = &quot;black&quot;, alpha = 0.3) +
  ggtitle(&quot;Combined Thorsson and new gene sets \nDuke vs NCI data sources&quot;)

fviz_pca_var(dlbcl_pca, col.var = &quot;contrib&quot;,
             gradient.cols = &quot;tron&quot;,
             label = &quot;none&quot;,
             ggtheme = theme_classic()) +
  geom_hline(yintercept = 0, color = &quot;black&quot;, alpha = 0.3) +
  geom_vline(xintercept = 0, color = &quot;black&quot;, alpha = 0.3)</code></pre>
</div>
<div id="heatmap-of-raw-gsva-results" class="section level4">
<h4>Heatmap of raw GSVA results</h4>
<pre class="r"><code>ht &lt;- Heatmap(gs_matrix, name = &quot;GSVA Score&quot;, 
                row_title = &quot;Gene Sets&quot;,
                show_row_names = TRUE,
                column_title = paste0(&quot;GSVA scores for Duke and NCI samples&quot;), 
                show_column_names = FALSE,
                column_title_side = &quot;bottom&quot;)
ht</code></pre>
</div>
</div>
<div id="clustering" class="section level1">
<h1>Clustering</h1>
<div id="determine-suitability-of-data-for-clustering" class="section level4">
<h4>Determine suitability of data for clustering</h4>
<p>First visualize similarity among observations with distance matrix</p>
<pre class="r"><code># create pearson correlation matrix
sample_matrix &lt;- t(gs_matrix)
dlbcl_dist &lt;- get_dist(sample_matrix, 
                       stand = TRUE,
                       method = &quot;pearson&quot;)</code></pre>
<p>Plot results</p>
<pre class="r"><code>fviz_dist(dlbcl_dist, show_labels = FALSE) +
  dark_theme_gray() +
  scale_fill_viridis() +
  theme(axis.ticks = element_blank(), axis.text = element_blank())

ggsave(&quot;docs/assets/fvis_dist_1.png&quot;, 
       device = png(),
       width = 6.67, 
       height = 6.67, 
       units = &quot;in&quot;)</code></pre>
<p><img src="assets/fvis_dist_1.png" /></p>
<p>Get Hopkins statistic of clusterability</p>
<pre class="r"><code># Compute Hopkins statistic 
# Below 0.5 is considered &quot;clusterable&quot;
dlbcl_tend &lt;- hopkins(gs_matrix, n = nrow(gs_matrix) - 1, header = FALSE)
dlbcl_tend</code></pre>
</div>
<div id="filter-out-highly-correlated-gene-sets" class="section level4">
<h4>Filter out highly correlated gene sets</h4>
<pre class="r"><code>gs_cor &lt;- cor(gs_matrix, method = c(&quot;pearson&quot;))
library(corrplot)
source(&quot;http://www.sthda.com/upload/rquery_cormat.r&quot;)</code></pre>
<pre class="r"><code># view gene set correlation heatmap
cortables &lt;- rquery.cormat(t(gs_matrix), graphType = &quot;heatmap&quot;)</code></pre>
<pre class="r"><code># reformat gene set correlation table and view
corflat &lt;- rquery.cormat(t(gs_matrix), type = &quot;flatten&quot;, graph = FALSE)
corflat_tbl &lt;- corflat$r %&gt;% as_tibble() %&gt;% arrange(desc(cor)) %&gt;% print(n = 30)

# 22 gene sets 0.96 correlation and above. 
removesets &lt;- c(&quot;thor_module11_prolif_score&quot;, 
                &quot;thor_ifn_21978456&quot;,
                &quot;thor_mhc.i_19272155&quot;,
                &quot;thor_t_cell_pca_16704732&quot;,
                &quot;thor_icr_score&quot;,
                &quot;beijing_cd8_c1_lef1_gs&quot;,
                &quot;thor_minterferon_cluster_21214954&quot;,
                &quot;thor_g_hla-dpa1&quot;,
                &quot;thor_apm2&quot;,
                &quot;thor_tcclassii_score&quot;,
                &quot;thor_ifit3&quot;, 
                &quot;beijing_cd4_c3_gnly_gs&quot;,
                &quot;thor_b_cell_pca_16704732&quot;,
                &quot;thor_interferon_cluster_21214954&quot;,
                &quot;thor_interferon_19272155&quot;,
                &quot;thor_immune_cell_cluster_21214954&quot;,
                &quot;thor_lck_19272155&quot;,
                &quot;thor_module4_tcellbcell_score&quot;)

# 11 gene sets meet the criteria
nrow(gs_matrix)
gs_matrix_filtered_df &lt;- gs_matrix %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene_set&quot;) %&gt;%
  dplyr::filter(!gene_set %in% removesets)
nrow(gs_matrix_filtered_df)

sample_matrix_filtered &lt;- gs_matrix_filtered_df %&gt;%
  column_to_rownames(var = &quot;gene_set&quot;) %&gt;%
  as.matrix() %&gt;%
  t()
sample_matrix_filtered[1:5, 1:5]</code></pre>
<p>Repeat similarity among observations with distance matrix</p>
<pre class="r"><code># create pearson correlation matrix
dlbcl_dist &lt;- get_dist(sample_matrix_filtered, 
                       stand = TRUE, 
                       method = &quot;pearson&quot;)</code></pre>
<pre class="r"><code># plot results
fviz_dist(dlbcl_dist, show_labels = FALSE) +
  dark_theme_gray() +
  scale_fill_viridis() +
  theme(axis.ticks = element_blank(), axis.text = element_blank())

ggsave(&quot;docs/assets/fvis_dist_2.png&quot;, 
       device = png(),
       width = 6.67, 
       height = 6.67, 
       units = &quot;in&quot;)</code></pre>
<p><img src="assets/fvis_dist_2.png" /></p>
<p>Repeat Hopkins statistic of clusterability</p>
<pre class="r"><code># Compute Hopkins statistic for iris dataset
# Below 0.5 is considered &quot;clusterable&quot;
dlbcl_tend &lt;- hopkins(sample_matrix_filtered,
                      n = nrow(gs_matrix_filtered_df) - 1, 
                      header = FALSE)
dlbcl_tend</code></pre>
<p>Determine optimal number of clusters</p>
<pre class="r"><code># Method 1: Elbow Method Using FactoExtra 
fviz_nbclust(sample_matrix_filtered, 
             FUNcluster = kmeans, 
             method = &quot;wss&quot;) 

# Method 2: Silhouette in FactoExtra
fviz_nbclust(sample_matrix_filtered, 
             kmeans, method = &quot;silhouette&quot;) + 
  theme_classic()

# GAP STAT Using Facto Extra 
fviz_nbclust(sample_matrix_filtered, 
             kmeans, nstart = 25, 
             method = &quot;gap_stat&quot;, 
             nboot = 500) + #nboot is # of bootstrap sample - must be included
  labs(subtitle = &quot;Gap Statistic Method&quot;)

# summary method: NbClust
nb &lt;- NbClust(sample_matrix_filtered, 
              distance = &quot;euclidean&quot;, 
              min.nc = 2, 
              max.nc = 10, 
              method = &quot;kmeans&quot;)</code></pre>
</div>
<div id="clustering-1" class="section level4">
<h4>Clustering</h4>
<p>Divide into groups with HCPC</p>
<pre class="r"><code>set.seed(818)
dlbcl_pca &lt;- PCA(sample_matrix_filtered, ncp = 10, graph = FALSE)
dlbcl_hcpc &lt;- HCPC(dlbcl_pca, min = 4, max = 20, graph.scale = &quot;sqrt-inertia&quot;, graph = TRUE)</code></pre>
<p>Write out models for later prediction</p>
<pre class="r"><code>saveRDS(dlbcl_pca, &quot;output/dlbcl_gsva4_4cluster_pca.rds&quot;)
saveRDS(dlbcl_hcpc, &quot;output/dlbcl_gsva4_4cluster_hcpc.rds&quot;)</code></pre>
<pre class="r"><code>fviz_dend(dlbcl_hcpc, 
          k = 4,
          type = &quot;phylogenic&quot;,
          palette = &quot;nejm&quot;,
          show_labels = FALSE, 
          rect = TRUE,
          rect_border = &quot;black&quot;,
          rect_lty = 3,
          rect_fill = FALSE, 
          lower_rect = -1,
          phylo_layout = &quot;layout_as_tree&quot;
          )</code></pre>
<p>Visualize samples with cluster label on PCA plot</p>
<pre class="r"><code>fviz_cluster(dlbcl_hcpc,
             geom = &quot;point&quot;,
             show.clust.cent = TRUE,  
             main = &quot;Immune Response Clusters on GSVA Principal Components&quot;, 
             pointshape = 19,
             theme_classic2()
             ) +
  scale_fill_d3() +
  scale_color_d3() +
  theme_classic2() +
  geom_hline(yintercept = 0, linetype = 3) +
  geom_vline(xintercept = 0, linetype = 3) +
  theme(legend.position = &quot;none&quot;)

#500 x 700

# ASH ORAL plot
clust_pca &lt;- fviz_cluster(dlbcl_hcpc, 
             geom = &quot;point&quot;,
             show.clust.cent = FALSE,  
             main = &quot;Immune Response Clusters&quot;,
             pointshape = 19
             ) + 
  theme_classic2() +
  xlab(&quot;PC1 (44.1% of variance)&quot;) +
  ylab(&quot;PC2 (8.8% of variance)&quot;) +
  scale_fill_d3() +
  scale_color_d3()

ggsave(&quot;output/ASH_ORAL_figures/clust_pca.png&quot;, plot = clust_pca, height = 4, width = 6, units = &quot;in&quot;)</code></pre>
<p>Compare cluster assignments at the individual level.</p>
<pre class="r"><code>gsva2_assgn &lt;- readRDS(&quot;output/gsva_v2_4clust_assign.rds&quot;) %&gt;%
  dplyr::rename(gsva2_clust = clust) %&gt;%
  mutate(gsva2_clust = as.numeric(gsva2_clust)) %&gt;%
  print()

gsva3_assgn &lt;- readRDS(&quot;output/gsva_v3_4clust_assign.rds&quot;) %&gt;%
  mutate(gsva3_clust = as.numeric(gsva3_clust)) %&gt;%
  print()

gsva4_assgn &lt;- dlbcl_hcpc$data.clust %&gt;% 
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble %&gt;%
  dplyr::select(sample_id, gsva4_clust = clust) %&gt;%
  mutate(gsva4_clust = as.numeric(gsva4_clust)) %&gt;%
  print()

merge_assgn &lt;- gsva2_assgn %&gt;%
  left_join(gsva3_assgn) %&gt;%
  left_join(gsva4_assgn) %&gt;%
  print()

cor(merge_assgn$gsva2_clust, merge_assgn$gsva4_clust)
cor(merge_assgn$gsva3_clust, merge_assgn$gsva4_clust)

merge_assgn_24 &lt;- merge_assgn %&gt;% 
  mutate(concordance = ifelse(gsva2_clust == gsva4_clust, 1, 0))
table(merge_assgn_24$concordance)

merge_assgn_34 &lt;- merge_assgn %&gt;% 
  mutate(concordance = ifelse(gsva3_clust == gsva4_clust, 1, 0))
table(merge_assgn_34$concordance)</code></pre>
</div>
<div id="exploration-and-sanity-checks" class="section level2">
<h2>Exploration and sanity checks</h2>
<p>Data cleaning and formatting</p>
<pre class="r"><code># Get dataframe with gene set score matrix by sample and cluster assignment as the last column
clust_assign &lt;- dlbcl_hcpc$data.clust %&gt;% 
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble %&gt;%
  print()

# save cluster assignments for comparison in other notebooks
gsva_v4_4clust_assign &lt;- clust_assign %&gt;%
  dplyr::select(sample_id, gsva2_clust = clust)
saveRDS(gsva_v4_4clust_assign, &quot;output/gsva_v4_4clust_assign.rds&quot;)

# get phenotype data
pheno_data &lt;- pData(combined_es) %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  as_tibble() %&gt;%
  print()

# combined tables
pheno_clust_data &lt;- clust_assign %&gt;%
  left_join(pheno_data, by = &quot;sample_id&quot;) %&gt;%
  dplyr::select(-case_id, -filename, -demographic.submitter_id, demographic.race, demographic.ethnicity, size) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;sample_id&quot;)
pheno_clust_data[1:10, 1:4]

# confirm last column of gene set expression matrix
pheno_clust_data[1:5,125:127]
ncol(pheno_clust_data)
# columns 1-125 are gene set expression matrix, columns 126-195 are phenotype data</code></pre>
<div id="get-pd-l1-for-evaluable-cases" class="section level4">
<h4>Get PD-L1 for evaluable cases</h4>
<pre class="r"><code>pdl1 &lt;- pheno_clust_data %&gt;% 
  dplyr::select(source, clust, pdl1_status) %&gt;%
  dplyr::filter(source == &quot;nci&quot;) %&gt;%
  group_by(clust, pdl1_status) %&gt;%
  drop_na(pdl1_status) %&gt;%
  tally() %&gt;%
  spread(key = pdl1_status, value = n) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;clust&quot;) %&gt;%
  print()

fisher.test(pdl1, simulate.p.value = TRUE, B = 1e6)

percentages &lt;- pdl1 %&gt;%
  mutate(percent = high_amplified / (low_amplified + `non-amplified`)) %&gt;%
  print()

c2v3 &lt;- pdl1 %&gt;%
  dplyr::slice(2:3) %&gt;%
  print()

fisher.test(c2v3, simulate.p.value = TRUE, B = 1e6)

c1v4 &lt;- pdl1 %&gt;%
  dplyr::slice(1,4) %&gt;%
  print()

fisher.test(c1v4, simulate.p.value = TRUE, B = 1e6)</code></pre>
<p>Plots Source for overriding shape change with habillage argument: <a href="https://github.com/kassambara/factoextra/issues/20" class="uri">https://github.com/kassambara/factoextra/issues/20</a></p>
<p>Instructions for convex hull geom: <a href="https://github.com/cmartin/ggConvexHull" class="uri">https://github.com/cmartin/ggConvexHull</a></p>
<pre class="r"><code># convert NAs in pdl1_amp variable
pheno_clust_data$pdl1_status &lt;- pheno_clust_data$pdl1_status %&gt;%
  replace_na(&quot;not_assessed&quot;) %&gt;%
  str_replace_all(&quot;non-amplified&quot;, &quot;non_amplified&quot;) %&gt;%
  as.factor()
summary(pheno_clust_data$pdl1_status)

# make variable for pdl1_amp vs all others
pheno_clust_data$amp_or_not &lt;- fct_recode(pheno_clust_data$pdl1_status,
             &quot;PDL1amp&quot; = &quot;high_amplified&quot;,
             &quot;other&quot; = &quot;low_amplified&quot;,
             &quot;other&quot; = &quot;non_amplified&quot;,
             &quot;other&quot; = &quot;not_assessed&quot;)
summary(pheno_clust_data$amp_or_not)

# make factor for alpha
pheno_clust_data$alpha_var &lt;- fct_recode(pheno_clust_data$amp_or_not,
             &quot;1&quot; = &quot;PDL1amp&quot;,
             &quot;0.2&quot; = &quot;other&quot;) %&gt;%
  as.character() %&gt;%
  as.numeric()
summary(pheno_clust_data$alpha_var)

### LIGHT PLOT
dlbcl_pca &lt;- PCA(pheno_clust_data[, 1:125], graph = FALSE)

mytitle &lt;- expression(paste(italic(&quot;PD-L1&quot;), &quot; amplified cases by cluster&quot;))

pdl1_pca1 &lt;- fviz_pca_ind(dlbcl_pca,
             label = &quot;none&quot;,
             habillage = pheno_clust_data$amp_or_not,
             pointshape = 19,
             alpha.ind = pheno_clust_data$alpha_var,
             show.legend.text = FALSE,
             palette = &quot;d3&quot;,
             addEllipses = FALSE,
             mean.point = FALSE,
             ggtheme = theme_classic2()
             ) +
  geom_convexhull(alpha = 0.3, 
                  aes(fill = pheno_clust_data$clust),
                  show.legend = FALSE,
                  color = &quot;#76767666&quot;) +
  xlab(&quot;PC1 (44.1% of variance)&quot;) +
  ylab(&quot;PC2 (8.8% of variance)&quot;) +
  ggtitle(mytitle)
pdl1_pca1

### DARK PLOT
# get colors
mypal = pal_tron()(6)

dlbcl_pca &lt;- PCA(pheno_clust_data[, 1:125], graph = FALSE)
pdl1_pca &lt;- fviz_pca_ind(dlbcl_pca,
             label = &quot;none&quot;,
             habillage = pheno_clust_data$amp_or_not,
             pointshape = 19,
             alpha.ind = pheno_clust_data$alpha_var,
             show.legend.text = FALSE,
             palette = &quot;ucsc&quot;,
             addEllipses = FALSE,
             ggtheme = dark_theme_gray()
             ) +
  geom_convexhull(alpha = 0.3, 
                  aes(fill = pheno_clust_data$clust),
                  show.legend = FALSE,
                  color = &quot;#76767666&quot;) +
  ggtitle(&quot;Gene set PCA: PDL1 high amp by cluster&quot;)
pdl1_pca

## plot for abstract
invert_geom_defaults()

mytitle &lt;- expression(paste(italic(&quot;PD-L1&quot;), &quot; amplified cases by cluster&quot;))

pdl1_pca2 &lt;- fviz_pca_ind(dlbcl_pca,
             label = &quot;none&quot;,
             habillage = pheno_clust_data$amp_or_not,
             pointshape = 19,
             alpha.ind = pheno_clust_data$alpha_var,
             show.legend.text = FALSE,
             palette = &quot;nejm&quot;,
             addEllipses = FALSE,
             ggtheme = theme_classic2()
             ) +
  geom_convexhull(alpha = 0.0, 
                  aes(fill = pheno_clust_data$clust),
                  show.legend = FALSE,
                  color = &quot;black&quot;) +
  xlab(&quot;PC1 (44.1% of variance)&quot;) +
  ylab(&quot;PC2 (8.8% of variance)&quot;) +
  ggtitle(mytitle)
pdl1_pca2

ggsave(&quot;output/ASH_ORAL_figures/pdl1.png&quot;, plot = pdl1_pca2, width = 6, height = 4, units = &quot;in&quot;)

# get counts by cluster
table(pheno_clust_data$amp_or_not, pheno_clust_data$clust)

# plot cytotoxic GSVA score on combined GSVA PCA plot
dlbcl_pca &lt;- PCA(pheno_clust_data[, 1:125], graph = FALSE)
fviz_pca_ind(dlbcl_pca, 
             col.ind = pheno_clust_data$cyt_gs,
             label = &quot;none&quot;,
             addEllipses = FALSE,
             ggtheme = dark_theme_gray()) +
  labs(color = &quot;CYT GSVA score&quot;) +
  geom_convexhull(alpha = 0.0, 
                  aes(fill = pheno_clust_data$clust),
                  show.legend = FALSE,
                  color = &quot;white&quot;) +
  ggtitle(&quot;Gene set PCA: CYT GSVA by cluster&quot;) +
  scale_color_viridis()

# plot DHIT score on combined GSVA PCA plot
dlbcl_pca &lt;- PCA(pheno_clust_data[, 1:125], graph = FALSE)
fviz_pca_ind(dlbcl_pca, 
             col.ind = pheno_clust_data$dhit_gs,
             label = &quot;none&quot;,
             addEllipses = FALSE,
             ggtheme = dark_theme_gray()) +
  labs(color = &quot;DHIT GSVA score&quot;) +
  geom_convexhull(alpha = 0.0, 
                  aes(fill = pheno_clust_data$clust),
                  show.legend = FALSE,
                  color = &quot;white&quot;) +
  ggtitle(&quot;Gene set PCA: DHIT GSVA score by cluster&quot;) +
  scale_color_viridis()</code></pre>
</div>
<div id="abc-vs-gcb" class="section level4">
<h4>ABC vs GCB</h4>
<pre class="r"><code># check categorical variable
summary(as.factor(pheno_clust_data$abc_gcb_rna))

# get PCA values
dlbcl_pca &lt;- PCA(pheno_clust_data[, 1:125], graph = FALSE)

# dark plot
fviz_pca_ind(dlbcl_pca, 
             col.ind = pheno_clust_data$abc_gcb_rna,
             pointshape = 19,
             label = &quot;none&quot;,
             addEllipses = FALSE,
             ggtheme = dark_theme_gray()) +
  labs(color = &quot;Cell of Origin&quot;) +
  geom_convexhull(alpha = 0.0, 
                  aes(fill = pheno_clust_data$clust),
                  show.legend = FALSE,
                  color = &quot;white&quot;) +
  ggtitle(&quot;Gene set PCA: Cell of origin by cluster&quot;) +
  scale_color_jcolors(palette = &quot;pal3&quot;)

invert_geom_defaults()

# light plot
fviz_pca_ind(dlbcl_pca, 
             col.ind = pheno_clust_data$abc_gcb_rna,
             pointshape = 19,
             label = &quot;none&quot;,
             addEllipses = FALSE,
             ggtheme = theme_classic2()) +
  labs(color = &quot;Cell of Origin&quot;) +
  geom_convexhull(alpha = 0.0, 
                  aes(fill = pheno_clust_data$clust),
                  show.legend = FALSE,
                  color = &quot;black&quot;) +
  ggtitle(&quot;Gene set PCA: Cell of origin by cluster&quot;) +
  scale_color_nejm()

# variable plot

fviz_pca_var(dlbcl_pca, col.var = &quot;contrib&quot;,
             gradient.cols = &quot;tron&quot;,
             label = &quot;none&quot;,
             ggtheme = dark_theme_gray()) +
  geom_hline(yintercept = 0, color = &quot;white&quot;, alpha = 0.3) +
  geom_vline(xintercept = 0, color = &quot;white&quot;, alpha = 0.3)

# Contributions of variables to PC1
fviz_contrib(dlbcl_pca, choice = &quot;var&quot;, axes = 1, top = 15) + 
  dark_theme_classic() +
  theme(axis.text.x = element_text(angle = 80, hjust = 1))

# Contributions of variables to PC2
fviz_contrib(dlbcl_pca, choice = &quot;var&quot;, axes = 2, top = 15) +
  dark_theme_classic() +
  theme(axis.text.x = element_text(angle = 80, hjust = 1))

# Get counts
coo &lt;- table(pheno_clust_data$abc_gcb_rna, pheno_clust_data$clust)
coo</code></pre>
<p>Ratios | COO | clust 1 | clust 2 | clust 3 | clust 4 | |——-|———–|———–|———–|———–| | ABC | 0.47 | 0.45 | 0.38 | 0.47 | |——-|———–|———–|———–|———–| | GCB | 0.34 | 0.40 | 0.43 | 0.33 | |——-|———–|———–|———–|———–| | Unc. | 0.19 | 0.15 | 0.19 | 0.20 | |——-|———–|———–|———–|———–|</p>
<pre class="r"><code># reformat data for comparisons and plotting
coo_data &lt;- pheno_clust_data %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;sample_id&quot;) %&gt;%
  dplyr::select(sample_id, abc_gcb_rna, clust) %&gt;%
  mutate(abc_abc_rna = as.factor(abc_gcb_rna),
         clust = as.factor(clust)) %&gt;%
  as_tibble()
coo_data

# make table for comparison of proportions
coo_table &lt;- coo_data %&gt;% 
  count(clust, abc_gcb_rna) %&gt;% 
  tidyr::spread(key = clust, value = n) %&gt;% 
  as.data.frame() %&gt;% 
  column_to_rownames(var = &quot;abc_gcb_rna&quot;) %&gt;%
  as.data.frame() %&gt;% 
  rownames_to_column(var = &quot;coo&quot;) %&gt;% 
  dplyr::rename(cold = `1`, 
                intermediate_t = `2`, 
                intermediate_m = `3`, 
                inflamed = `4`) %&gt;%
  mutate(tot = cold + intermediate_t + intermediate_m + inflamed) %&gt;% 
  mutate(not_cold = tot - cold) %&gt;% 
  mutate(not_intermediate_t = tot - intermediate_t) %&gt;% 
  mutate(not_intermediate_m = tot - intermediate_m) %&gt;%
  mutate(not_inflamed = tot - inflamed)

# test for differences in cluster 1 proportions
clust1 &lt;- coo_table %&gt;%
  dplyr::select(coo, cold, not_cold) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;coo&quot;) %&gt;%
  as.matrix() %&gt;%
  print()

clust1_p &lt;- tidy(chisq.test(clust1)) %&gt;%
  print()

clust1_prop &lt;- clust1 %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;coo&quot;) %&gt;%
  as_tibble() %&gt;% 
  mutate(proportion = cold / (cold + not_cold)) %&gt;%
  mutate(clust = rep(&quot;1&quot;, nrow(.))) %&gt;%
  print()

# test for differences in cluster 2 proportions
clust2 &lt;- coo_table %&gt;%
  dplyr::select(coo, intermediate_t, not_intermediate_t) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;coo&quot;) %&gt;%
  as.matrix() %&gt;%
  print()

clust2_p &lt;- tidy(chisq.test(clust2)) %&gt;%
  print()

clust2_prop &lt;- clust2 %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;coo&quot;) %&gt;%
  as_tibble() %&gt;% 
  mutate(proportion = intermediate_t / (intermediate_t + not_intermediate_t)) %&gt;%
  mutate(clust = rep(&quot;2&quot;, nrow(.))) %&gt;%
  print()

# test for differences in cluster 3 proportions
clust3 &lt;- coo_table %&gt;%
  dplyr::select(coo, intermediate_m, not_intermediate_m) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;coo&quot;) %&gt;%
  as.matrix() %&gt;%
  print()

clust3_p &lt;- tidy(chisq.test(clust3)) %&gt;%
  print()

clust3_prop &lt;- clust3 %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;coo&quot;) %&gt;%
  as_tibble() %&gt;% 
  mutate(proportion = intermediate_m / (intermediate_m + not_intermediate_m)) %&gt;%
  mutate(clust = rep(&quot;3&quot;, nrow(.))) %&gt;%
  print()

# test for differences in cluster 4 proportions
clust4 &lt;- coo_table %&gt;%
  dplyr::select(coo, inflamed, not_inflamed) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;coo&quot;) %&gt;%
  as.matrix() %&gt;%
  print()

clust4_p &lt;- tidy(chisq.test(clust4)) %&gt;%
  print()

clust4_prop &lt;- clust4 %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;coo&quot;) %&gt;%
  as_tibble() %&gt;% 
  mutate(proportion = inflamed / (inflamed + not_inflamed)) %&gt;%
  mutate(clust = rep(&quot;4&quot;, nrow(.))) %&gt;%
  print()

coo_totals &lt;- coo_data %&gt;%
  dplyr::group_by(clust, abc_gcb_rna) %&gt;%
  summarize(n = n()) %&gt;%
  summarize(sum = sum(n)) %&gt;%
  print()

coo_sum_prop &lt;- coo_data %&gt;%
  dplyr::group_by(clust, abc_gcb_rna) %&gt;%
  summarize(n = n()) %&gt;%
  ungroup() %&gt;%
  left_join(coo_totals) %&gt;%
  mutate(proportion = n / sum) %&gt;%
  mutate(percent = round((proportion * 100), 0)) %&gt;%
  mutate(percent = paste0(percent, &quot;%&quot;)) %&gt;%
  mutate(clust = as.factor(clust)) %&gt;%
  mutate(clust = fct_recode(clust, &quot;Cold&quot; = &quot;1&quot;,
                            &quot;Intermediate-T&quot; = &quot;2&quot;,
                            &quot;Intermediate-M&quot; = &quot;3&quot;,
                            &quot;Inflamed&quot; = &quot;4&quot;)) %&gt;%
  mutate(percent_n = paste0(&quot;n = &quot;, n, &quot;\n&quot;, &quot;(&quot;, percent, &quot;)&quot;)) %&gt;%
  dplyr::rename(coo = abc_gcb_rna) %&gt;%
  print()

# make data frame containing all p-values
pvals &lt;- bind_rows(clust1_p, clust2_p, clust3_p, clust4_p) %&gt;%
  bind_cols(clust = c(&quot;Cold&quot;, &quot;Intermediate-T&quot;, &quot;Intermediate-M&quot;, &quot;Inflamed&quot;)) %&gt;%
  mutate(p.value = round(p.value, 3)) %&gt;%
  print()
               
# light plot
coo_bar &lt;- ggplot(coo_sum_prop, 
                  aes(x = coo,
                      y = n,
                      fill = coo)) +
  geom_bar(stat = &quot;identity&quot;) +
  facet_wrap(~ clust) +
  theme_classic2() +
  scale_color_nejm() +
  scale_fill_nejm() +
  labs(title = &quot;Cell of origin by cluster&quot;,
       fill = &quot;Cell of Origin&quot;,
       x = &quot;&quot;,
       y = &quot;Number of cases&quot;) +
  geom_text(aes(label = percent_n), 
            vjust = 1.2, 
            color = &quot;white&quot;, 
            size = 3.3) +
  geom_text(data = pvals, 
            aes(y = 185, 
                x = 1, 
                label = paste0(&quot;p = &quot;, p.value), 
                group = clust), 
                inherit.aes = FALSE)
  
coo_bar 

# dark plot
coo_bar &lt;- ggplot(coo_sum, 
                  aes(x = abc_gcb_rna,
                      y = n,
                      fill = abc_gcb_rna)) +
  geom_bar(stat = &quot;identity&quot;) +
  facet_wrap(~ clust) +
  dark_theme_gray() +
  scale_color_jcolors(palette = &quot;pal3&quot;) +
  scale_fill_jcolors(palette = &quot;pal3&quot;) +
  labs(title = &quot;Counts of COO Subtype by Cluster&quot;,
       fill = &quot;Cell of Origin&quot;,
       x = &quot;&quot;,
       y = &quot;count (n)&quot;)
  
coo_bar

invert_geom_defaults()

# manual color assignment
mypal = pal_d3(alpha = 1)(4)
names(mypal) &lt;- c(&quot;Cold&quot;, &quot;Intermediate-T&quot;, &quot;Intermediate-M&quot;, &quot;Inflamed&quot;)

# facet by cell of origin
coo_bar2 &lt;- ggplot(coo_sum_prop, 
                  aes(x = clust,
                      y = proportion,
                      fill = clust)) +
  geom_bar(stat = &quot;identity&quot;) +
  facet_wrap(~ coo) +
  theme_classic2() +
  scale_color_manual(values = mypal) +
  scale_fill_manual(values = mypal) +
  labs(title = &quot;Cell of origin subtypes are evenly distributed in all immune clusters&quot;,
       fill = &quot;Cluster&quot;,
       x = &quot;&quot;,
       y = &quot;Proportion of each cluster&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 14)) +
  theme(strip.text.x = element_text(size = 14, color = &quot;black&quot;, face = &quot;bold.italic&quot;
        )) +
  theme(legend.position = &quot;none&quot;)
  
ggsave(&quot;output/ASH_ORAL_figures/coo.png&quot;, coo_bar2, height = 4.5, width = 7, units = &quot;in&quot;)</code></pre>
</div>
</div>
<div id="survival-analysis-per-cluster-assignment" class="section level2">
<h2>Survival analysis per cluster assignment</h2>
<pre class="r"><code>library(&quot;survival&quot;)
library(&quot;survminer&quot;)

pheno_clust_data_surv &lt;- pheno_clust_data %&gt;%
  drop_na(os_years) %&gt;%
  drop_na(os_status) %&gt;%
  mutate(five_yr_os = ifelse(os_years &gt; 5, 5, os_years)) %&gt;%
  mutate(five_yr_status = ifelse(os_years &gt; 5, 0, os_status)) %&gt;%
  mutate(clust = fct_recode(clust, &quot;Cold&quot; = &quot;1&quot;,
                            &quot;Intermediate-T&quot; = &quot;2&quot;,
                            &quot;Intermediate-M&quot; = &quot;3&quot;,
                            &quot;Inflamed&quot; = &quot;4&quot;))
surv_clust &lt;- Surv(pheno_clust_data_surv$five_yr_os,
                   pheno_clust_data_surv$five_yr_status)

fit1 &lt;- survfit(surv_clust ~ clust, 
                data = pheno_clust_data_surv)

survdiff(surv_clust ~ clust, 
                data = pheno_clust_data_surv)

plot1 &lt;- ggsurvplot(fit1, 
                    data = pheno_clust_data_surv, 
                    pval = TRUE, 
                    pval.method = TRUE,
                    risk.table = TRUE,
                    palette = &quot;d3&quot;,
                    tables.height = 0.3,
                    ggtheme = theme_classic2(),
                    title = &quot;Overall survival with R-CHOP by immune response cluster&quot;,
                    legend.title = &quot;Cluster&quot;, 
                    legend = &quot;none&quot;,
                    tables.y.text = FALSE,
                    ylab = &quot;Proportion alive&quot;,
                    xlab = &quot;Time in years&quot;)
plot1

fit2 &lt;- survfit(surv_clust ~ clust + abc_gcb_rna, 
                data = pheno_clust_data)

fit2_sum &lt;- surv_summary(fit2)

plot2 &lt;- ggsurvplot(fit2, data = pheno_clust_data, pval = FALSE)
plot2$plot + 
  facet_wrap(~ clust)</code></pre>
<pre class="r"><code>## Write out new expressionset
pheno_d &lt;- pheno_clust_data
pheno_d[1:5, 1:5]

v_metadata &lt;- data.frame(
  labelDescription = colnames(pheno_d),
  row.names = colnames(pheno_d) )
  
phenoData &lt;- new(&quot;AnnotatedDataFrame&quot;, 
                 data = pheno_d, 
                 varMetadata = v_metadata)

exprs_matrix &lt;- exprs(combined_es)
str(exprs_matrix)

annotation &lt;- as.character(&quot;Duke RNAseq counts derived from raw sequence data published by the Dave lab at Duke in Reddy et al Cell 2017. Quantified with Kallisto with reverse strand parameter, collected with tximport, filtered for only protein-coding genes, selected for cases with expression in &gt;12,000 genes, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed. Batch corrected using ComBat by sequencing lane. NCI RNAseq counts originally published by the Staudt lab at NCI in Schmitz et al NEJM 2018. Raw counts downloaded from GDC as HTseq counts, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed. Batch corrected for NCI vs Duke source using ComBat. 125-gene GSVA v4 gene set expression matrix in columns 1:125 of pData, columns 126:197 are phenotype data including 4-cluster model assignments in column `clust`&quot;)

# assemble expressionset
combined_es_4cluster &lt;- ExpressionSet(
  assayData = exprs_matrix,
  phenoData = phenoData,
  annotation = annotation)
combined_es_4cluster

# write out new expressionset
saveRDS(combined_es_4cluster, &quot;output/combined_es_gsva4_4cluster.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
