---
title: "Tumor-intrinsic mutations"
author: "mleukam"
date: "2019-07-13"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Setup

Clear the workspace
```{r}
rm(list = ls())
```

```{r}
knitr::opts_chunk$set(eval = FALSE)
```

Load packages
```{r}
library(maftools)
library(tidyverse)
library(RColorBrewer)
library(knitr)
library(ggpubr)
library(ggsci)
library(Biobase)
library(BiocGenerics)
library(parallel)
```

## Read in and clean data

Read in NCI mutation data
```{r}
# read in significance rank list from MutSig2CV
mutsig_siggenes <- read_delim("data/sig_genes.txt", "\t")

# read in final analysis MAF from MutSig2CV
final_analysis_set <- read_delim(
  "data/final_analysis_set.maf", "\t")

# repair nonstandard variant classification
final_analysis_set$Variant_Classification <- str_replace(final_analysis_set$Variant_Classification, "Start_Codon_Del", "Translation_Start_Site")

final_analysis_set$Variant_Classification <- str_replace(final_analysis_set$Variant_Classification, "Stop_Codon_Ins", "Nonstop_Mutation")

final_analysis_set <- as.data.frame(final_analysis_set)

nci_mut_by_sample <- final_analysis_set %>%
  dplyr::select(sample_id = Tumor_Sample_Barcode, Hugo_Symbol) %>%
  mutate(Hugo_Symbol = ifelse(Hugo_Symbol == "MLL2", "KMT2D", Hugo_Symbol)) %>%
  as_tibble() %>%
  print()
  
```

Read in Duke mutation data
```{r}
duke_mut_raw <- read_csv("data/duke_study_driver_mutations_top150_maf.csv")

duke_mut_by_sample <- duke_mut_raw %>%
  dplyr::select(Tumor_Sample_Barcode, Hugo_Symbol) %>%
  tidyr::separate(Tumor_Sample_Barcode, 
                  into = c(NA, NA, "sample_id"), 
                  sep = "_")  %>%
  mutate(sample_id = paste0("duke_", sample_id)) %>%
  arrange(sample_id) %>%
  as_tibble() %>%
  mutate(Hugo_Symbol = ifelse(Hugo_Symbol == "MLL2", "KMT2D", Hugo_Symbol)) %>%
  print()
```

#### Combine mutation data

```{r}
combined_mut <- nci_mut_by_sample %>%
  bind_rows(duke_mut_by_sample) %>%
  print()

# remove duplicate mutations in the same patient and same gene
unique_mut <- combined_mut %>%
  distinct(sample_id, Hugo_Symbol, .keep_all = TRUE) %>%
  print()

nrow(combined_mut)
nrow(unique_mut)
```

#### Read in deconvolution data
```{r}
nci_decon <- read_csv("data/cibersort/NCI/abs_cell_number_estimate_022320/CIBERSORTx_Job2_Results.csv")

duke_decon <- read_csv("data/cibersort/Duke/abs_cell_number_estimate_022320/CIBERSORTx_Job3_Results.csv")

combined_decon <- nci_decon %>%
  bind_rows(duke_decon) %>%
  print()
```

Read in cluster assignment and combine
```{r}
gsva4_es <- readRDS("output/combined_es_gsva4_4cluster.rds")

clust_assgn <- pData(gsva4_es) %>% 
  as.data.frame() %>%
  rownames_to_column(var = "sample_id") %>%
  dplyr::select(sample_id, clust) %>%
  as_tibble() %>%
  print()

# get the total n
clust_assgn %>% distinct(sample_id) %>% nrow()

mut_clust <- unique_mut %>%
  left_join(clust_assgn) %>%
  print()
```

#### Mutation counts by cluster

Get mutation counts per gene for each cluster and combine back to a single table
```{r}
# get number of cases in the cluster
mut_clust %>% dplyr::filter(clust == "1") %>% distinct(sample_id) %>% nrow()
mut_clust %>% dplyr::filter(clust == "2") %>% distinct(sample_id) %>% nrow()
mut_clust %>% dplyr::filter(clust == "3") %>% distinct(sample_id) %>% nrow()
mut_clust %>% dplyr::filter(clust == "4") %>% distinct(sample_id) %>% nrow()

# get counts by gene for clusters and compare each cluster for significance against pooled counts from each other cluster

# first get mutation counts for each gene per cluster and create column for counts of unmutated genes
cluster_1 <- mut_clust %>%
  dplyr::filter(clust == "1") %>%
  dplyr::select(-clust) %>%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %>%
  count(Hugo_Symbol) %>%
  dplyr::rename(c1_mut = n) %>%
  mutate(c1_nonmut = 352 - c1_mut) %>%
  print()

cluster_2 <- mut_clust %>%
  dplyr::filter(clust == "2") %>%
  dplyr::select(-clust) %>%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %>%
  count(Hugo_Symbol) %>%
  dplyr::rename(c2_mut = n) %>%
  mutate(c2_nonmut = 277 - c2_mut) %>%
  print() 

cluster_3 <- mut_clust %>%
  dplyr::filter(clust == "3") %>%
  dplyr::select(-clust) %>%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %>%
  count(Hugo_Symbol) %>%
  dplyr::rename(c3_mut = n) %>%
  mutate(c3_nonmut = 222 - c3_mut) %>%
  print() 

cluster_4 <- mut_clust %>%
  dplyr::filter(clust == "4") %>%
  dplyr::select(-clust) %>%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %>%
  count(Hugo_Symbol) %>%
  dplyr::rename(c4_mut = n) %>%
  mutate(c4_nonmut = 317 - c4_mut) %>%
  print() 

mut_counts_by_cluster <- cluster_1 %>%
  full_join(cluster_2) %>%
  full_join(cluster_3) %>%
  full_join(cluster_4) %>%
  replace_na(list(c1_mut = 0, c2_mut = 0, c3_mut = 0, c4_mut = 0)) %>%
  replace_na(list(c1_nonmut = 0, c2_nonmut = 0, c3_nonmut = 0, c4_nonmut = 0)) %>%
  arrange(desc(c4_mut)) %>%
  print()

mut_counts_matrix <- as.data.frame(mut_counts_by_cluster) %>%
  column_to_rownames(var = "Hugo_Symbol") %>%
  as.matrix()

mut_counts_matrix[1:5, ]
mut_counts <- mut_counts_matrix[, c(1,3,5,7)]
  
# filter for genes without any meaningful number of total mutations
# filter out genes that have a mutant allele fraction < 0.05
cutoff <- 1189 * 0.05

mut_counts_filtered <- mut_counts_by_cluster %>%
  mutate(mut_totals = c1_mut + c2_mut + c3_mut + c4_mut) %>%
  filter(mut_totals > cutoff) %>%
  print()
```

## Test pairwise significance

Significance testing per for each cluster vs pool of other clusters

https://stackoverflow.com/questions/41507230/fisher-test-on-multiple-data-entries-and-extract-results-in-separate-table/41512086#41512086?newreg=fbae75b1bdbc4796b925c53ca1e2f132

https://blog.rstudio.com/2016/02/02/tidyr-0-4-0/

https://jennybc.github.io/purrr-tutorial/index.html

```{r}
clust1_vs_pool <- mut_counts_filtered %>%
  mutate(pool_mut = c2_mut + c3_mut + c4_mut) %>%
  mutate(pool_nonmut = c2_nonmut + c3_nonmut + c4_nonmut) %>%
  dplyr::select(Hugo_Symbol, c1_mut, c1_nonmut, pool_mut, pool_nonmut) %>%
  print()

sig <- clust1_vs_pool %>%
  nest(-Hugo_Symbol) %>%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %>% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %>%
  mutate(stats = map(fisher, ~broom::glance(.x))) %>%
  unnest(stats) %>%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %>%
  arrange(p.value)

clust_1_summary <- sig %>%
  left_join(clust1_vs_pool) %>%
  mutate(p_adj = p.adjust(p.value, method = "BH")) %>%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %>%
  mutate(c1_rate = c1_mut / (c1_mut + c1_nonmut)) %>%
  print()

clust_1_plotdata <- clust_1_summary %>%
  mutate(pool_rate = pool_mut / pool_nonmut) %>%
  select(Hugo_Symbol, c1_rate, pool_rate, p_adj) %>%
  pivot_longer(cols = c(c1_rate, pool_rate),
               names_to = "c1_vs_pool", 
               values_to = "mut_rate") %>%
  dplyr::filter(p_adj < 0.26)
clust_1_plotdata

clust_1_plot <- ggplot(data = clust_1_plotdata,
                       aes(x = Hugo_Symbol, y = mut_rate, group = c1_vs_pool, fill = c1_vs_pool)) +
  geom_col(position = "dodge")
clust_1_plot
```

```{r}
clust2_vs_pool <- mut_counts_filtered %>%
  mutate(pool_mut = c1_mut + c3_mut + c4_mut) %>%
  mutate(pool_nonmut = c1_nonmut + c3_nonmut + c4_nonmut) %>%
  dplyr::select(Hugo_Symbol, c2_mut, c2_nonmut, pool_mut, pool_nonmut) %>%
  print()

sig <- clust2_vs_pool %>%
  nest(-Hugo_Symbol) %>%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %>% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %>%
  mutate(stats = map(fisher, ~broom::glance(.x))) %>%
  unnest(stats) %>%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %>%
  arrange(p.value)

clust_2_summary <- sig %>%
  left_join(clust2_vs_pool) %>%
  mutate(p_adj = p.adjust(p.value, method = "BH")) %>%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %>%
  print()
```

```{r}
clust3_vs_pool <- mut_counts_filtered %>%
  mutate(pool_mut = c1_mut + c2_mut + c4_mut) %>%
  mutate(pool_nonmut = c1_nonmut + c2_nonmut + c4_nonmut) %>%
  dplyr::select(Hugo_Symbol, c3_mut, c3_nonmut, pool_mut, pool_nonmut) %>%
  print()

sig <- clust3_vs_pool %>%
  nest(-Hugo_Symbol) %>%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %>% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %>%
  mutate(stats = map(fisher, ~broom::glance(.x))) %>%
  unnest(stats) %>%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %>%
  arrange(p.value)

clust_3_summary <- sig %>%
  left_join(clust3_vs_pool) %>%
  mutate(p_adj = p.adjust(p.value, method = "BH")) %>%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %>%
  print()
```

```{r}
clust4_vs_pool <- mut_counts_filtered %>%
  mutate(pool_mut = c1_mut + c2_mut + c3_mut) %>%
  mutate(pool_nonmut = c1_nonmut + c2_nonmut + c3_nonmut) %>%
  dplyr::select(Hugo_Symbol, c4_mut, c4_nonmut, pool_mut, pool_nonmut) %>%
  print()

sig <- clust4_vs_pool %>%
  nest(-Hugo_Symbol) %>%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %>% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %>%
  mutate(stats = map(fisher, ~broom::glance(.x))) %>%
  unnest(stats) %>%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %>%
  arrange(p.value)

clust_4_summary <- sig %>%
  left_join(clust4_vs_pool) %>%
  mutate(p_adj = p.adjust(p.value, method = "BH")) %>%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %>%
  print()
```

## Test pairwise mutations of pre-selected genes
```{r}
# EZH2 across all clusters

# get column of mutants by cluster
ezh2_mut <- mut_counts_by_cluster %>% 
  dplyr::filter(Hugo_Symbol %in% c("EZH2")) %>%
  dplyr::select(c1 = c1_mut, c2 = c2_mut, c3 = c3_mut, c4 = c4_mut, -Hugo_Symbol) %>%
  as.matrix() %>%
  t() %>%
  print()

colnames(ezh2_mut) <- "ezh2_mut"
ezh2_mut

# get column of non-mutants by cluster
ezh2_nonmut <- mut_counts_by_cluster %>% 
  dplyr::filter(Hugo_Symbol %in% c("EZH2")) %>%
  dplyr::select(c1 = c1_nonmut, c2 = c2_nonmut, c3 = c3_nonmut, c4 = c4_nonmut, -Hugo_Symbol) %>%
  as.matrix() %>%
  t() %>%
  print()

colnames(ezh2_nonmut) <- "ezh2_nonmut"
ezh2_nonmut

# combine
ezh2 <- cbind(ezh2_mut, ezh2_nonmut)
ezh2

# test significance
fisher.test(ezh2, simulate.p.value = TRUE)

# TNFAIP3 across all clusters

# get column of mutants by cluster
tnfaip3_mut <- mut_counts_by_cluster %>% 
  dplyr::filter(Hugo_Symbol %in% c("TNFAIP3")) %>%
  dplyr::select(c1 = c1_mut, c2 = c2_mut, c3 = c3_mut, c4 = c4_mut, -Hugo_Symbol) %>%
  as.matrix() %>%
  t() %>%
  print()

colnames(tnfaip3_mut) <- "tnfaip3_mut"
tnfaip3_mut

# get column of non-mutants by cluster
tnfaip3_nonmut <- mut_counts_by_cluster %>% 
  dplyr::filter(Hugo_Symbol %in% c("TNFAIP3")) %>%
  dplyr::select(c1 = c1_nonmut, c2 = c2_nonmut, c3 = c3_nonmut, c4 = c4_nonmut, -Hugo_Symbol) %>%
  as.matrix() %>%
  t() %>%
  print()

colnames(tnfaip3_nonmut) <- "tnfaip3_nonmut"
tnfaip3_nonmut

# combine
tnfaip3 <- cbind(tnfaip3_mut, tnfaip3_nonmut)
tnfaip3

# test significance
fisher.test(tnfaip3, simulate.p.value = TRUE)

# get column of mutants by cluster
ezh2_mut <- mut_counts_by_cluster %>% 
  dplyr::filter(Hugo_Symbol %in% c("EZH2")) %>%
  dplyr::select(c1 = c1_mut, c2 = c2_mut, c3 = c3_mut, c4 = c4_mut, -Hugo_Symbol) %>%
  as.matrix() %>%
  t() %>%
  print()

colnames(ezh2_mut) <- "ezh2_mut"
ezh2_mut

# get column of non-mutants by cluster
ezh2_nonmut <- mut_counts_by_cluster %>% 
  dplyr::filter(Hugo_Symbol %in% c("EZH2")) %>%
  dplyr::select(c1 = c1_nonmut, c2 = c2_nonmut, c3 = c3_nonmut, c4 = c4_nonmut, -Hugo_Symbol) %>%
  as.matrix() %>%
  t() %>%
  print()

colnames(ezh2_nonmut) <- "ezh2_nonmut"
ezh2_nonmut

# combine
ezh2 <- cbind(ezh2_mut, ezh2_nonmut)
ezh2

# test significance
fisher.test(ezh2, simulate.p.value = TRUE)
```

```{r}
# Test significance in hottest and coldest clusters compared to each other
clust4_vs_clust1 <- mut_counts_filtered %>%
  dplyr::select(Hugo_Symbol, c4_mut, c4_nonmut, c1_mut, c1_nonmut) %>%
  print()

sig <- clust4_vs_clust1 %>%
  nest(-Hugo_Symbol) %>%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %>% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %>%
  mutate(stats = map(fisher, ~broom::glance(.x))) %>%
  unnest(stats) %>%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %>%
  arrange(p.value)

clust4_clust1_summary <- sig %>%
  left_join(clust4_vs_clust1) %>%
  mutate(p_adj = p.adjust(p.value, method = "BH")) %>%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %>%
  arrange(p_adj) %>%
  print()

clust_14_plotdata <- clust4_clust1_summary %>%
  mutate(c4_rate = c4_mut / (c4_nonmut + c4_mut)) %>%
  mutate(c1_rate = c1_mut / (c1_nonmut + c1_mut) ) %>%
  mutate(difference = (c4_rate - c1_rate)^2) %>%
  arrange(desc(difference)) %>%
  slice(1:10) %>%
  select(Hugo_Symbol, c1_rate, c4_rate, difference) %>%
  mutate(Hugo_Symbol = as_factor(Hugo_Symbol)) %>%
  mutate(Hugo_Symbol = fct_reorder(Hugo_Symbol, difference, .desc = TRUE)) %>%
  pivot_longer(cols = c(c1_rate, c4_rate),
               names_to = "c1_vs_c4", 
               values_to = "mut_rate") %>%
  mutate(c1_vs_c4 = as.factor(c1_vs_c4)) %>%
  mutate(c1_vs_c4 = fct_recode(c1_vs_c4, "Inflamed" = "c4_rate",
                               "Cold" = "c1_rate")) %>%
  mutate(c1_vs_c4 = fct_relevel(c1_vs_c4, "Cold", "Inflamed")) %>%
  print()

clust_14_plot <- ggplot(data = clust_14_plotdata,
                       aes(x = Hugo_Symbol, 
                           y = mut_rate, 
                           group = c1_vs_c4, 
                           fill = c1_vs_c4)) +
  geom_col(position = "dodge") +
  theme_classic2() +
  scale_fill_d3(name = "Cluster") +
  xlab("") +
  ylab("Mutation Rate")
clust_14_plot

# Test significance in intermedite clusters compared to each other
clust2_vs_clust3 <- mut_counts_filtered %>%
  dplyr::select(Hugo_Symbol, c2_mut, c2_nonmut, c3_mut, c3_nonmut) %>%
  print()

sig <- clust2_vs_clust3 %>%
  nest(-Hugo_Symbol) %>%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %>% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %>%
  mutate(stats = map(fisher, ~broom::glance(.x))) %>%
  unnest(stats) %>%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %>%
  arrange(p.value)

clust2_clust3_summary <- sig %>%
  left_join(clust2_vs_clust3) %>%
  mutate(p_adj = p.adjust(p.value, method = "BH")) %>%
  dplyr::select(Hugo_Symbol, p_adj, p = p.value, everything()) %>%
  print()

clust_23_plotdata <- clust2_clust3_summary %>%
  mutate(c2_rate = c2_mut / (c2_nonmut + c2_mut) ) %>%
  mutate(c3_rate = c3_mut / (c3_nonmut + c3_mut) ) %>%
  mutate(difference = (c3_rate - c2_rate)^2) %>%
  arrange(desc(difference)) %>%
  slice(1:10) %>%
  select(Hugo_Symbol, c2_rate, c3_rate, difference) %>%
  mutate(Hugo_Symbol = as_factor(Hugo_Symbol)) %>%
  mutate(Hugo_Symbol = fct_reorder(Hugo_Symbol, difference, .desc = TRUE)) %>%
  pivot_longer(cols = c(c2_rate, c3_rate),
               names_to = "c2_vs_c3", 
               values_to = "mut_rate") %>%
  mutate(c2_vs_c3 = as.factor(c2_vs_c3)) %>%
  mutate(c2_vs_c3 = fct_recode(c2_vs_c3, "Intermediate-T" = "c2_rate",
                               "Intermediate-M" = "c3_rate")) %>%
  mutate(c2_vs_c3 = fct_relevel(c2_vs_c3, "Intermediate-T", "Intermediate-M")) %>%
  print()

clust_23_plot <- ggplot(data = clust_23_plotdata,
                       aes(x = Hugo_Symbol, 
                           y = mut_rate, 
                           group = c2_vs_c3, 
                           fill = c2_vs_c3)) +
  geom_col(position = "dodge") +
  theme_classic2() +
  scale_fill_d3(name = "Cluster") +
  xlab("") +
  ylab("Mutation Rate")
clust_23_plot

```

## g:Profiler input

Get list of genes mutated over allele fraction of 0.05 for each individual cluster for input into g:profiler ordered enrichment test

Rank by pairwise p-value and select only those significant at p < 0.05 in univariate analysis for further investigation. Rank by mutant allele frequency to use with g:Profiler "ordered query"

```{r}
cluster_1_maf <- mut_clust %>%
  dplyr::filter(clust == "1") %>%
  dplyr::select(-clust) %>%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %>%
  count(Hugo_Symbol) %>%
  dplyr::rename(c1_mut = n) %>%
  mutate(MAF = c1_mut/352) %>%
  arrange(desc(MAF)) %>%
  left_join(mut_counts_by_cluster) %>%
  # filter out MAF with at least 1% MAF
  dplyr::filter(MAF > 0.01) %>%
  print()
  
clust1_vs_pool <- cluster_1_maf %>%
  mutate(pool_mut = c2_mut + c3_mut + c4_mut) %>%
  mutate(pool_nonmut = c2_nonmut + c3_nonmut + c4_nonmut) %>%
  dplyr::select(Hugo_Symbol, c1_mut, c1_nonmut, pool_mut, pool_nonmut) %>%
  print()

clust_1_ranked <- clust1_vs_pool %>%
  nest(data = c(c1_mut, c1_nonmut, pool_mut, pool_nonmut)) %>%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %>% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %>%
  mutate(stats = map(fisher, ~broom::glance(.x))) %>%
  unnest(stats) %>%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %>%
  # dplyr::filter(p.value < 0.05) %>%
  left_join(cluster_1_maf) %>%
  arrange(desc(MAF)) %>%
  dplyr::select(Hugo_Symbol) %>%
  print()

#-----------------------------------
cluster_2_maf <- mut_clust %>%
  dplyr::filter(clust == "2") %>%
  dplyr::select(-clust) %>%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %>%
  count(Hugo_Symbol) %>%
  dplyr::rename(c2_mut = n) %>%
  mutate(MAF = c2_mut/277) %>%
  arrange(desc(MAF)) %>%
  left_join(mut_counts_by_cluster) %>%
  # filter out MAF with at least 1% MAF
  dplyr::filter(MAF > 0.01) %>%
  print()
  
clust2_vs_pool <- cluster_2_maf %>%
  mutate(pool_mut = c1_mut + c3_mut + c4_mut) %>%
  mutate(pool_nonmut = c1_nonmut + c3_nonmut + c4_nonmut) %>%
  dplyr::select(Hugo_Symbol, c2_mut, c2_nonmut, pool_mut, pool_nonmut) %>%
  print()

clust_2_ranked <- clust2_vs_pool %>%
  nest(data = c(c2_mut, c2_nonmut, pool_mut, pool_nonmut)) %>%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %>% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %>%
  mutate(stats = map(fisher, ~broom::glance(.x))) %>%
  unnest(stats) %>%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %>%
  # dplyr::filter(p.value < 0.05) %>%
  left_join(cluster_2_maf) %>%
  arrange(desc(MAF)) %>%
  dplyr::select(Hugo_Symbol) %>%
  print()

#-----------------------------------
cluster_3_maf <- mut_clust %>%
  dplyr::filter(clust == "3") %>%
  dplyr::select(-clust) %>%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %>%
  count(Hugo_Symbol) %>%
  dplyr::rename(c3_mut = n) %>%
  mutate(MAF = c3_mut/222) %>%
  arrange(desc(MAF)) %>%
  left_join(mut_counts_by_cluster) %>%
  # filter out MAF with at least 1% MAF
  dplyr::filter(MAF > 0.01) %>%
  print()
  
clust3_vs_pool <- cluster_3_maf %>%
  mutate(pool_mut = c2_mut + c1_mut + c4_mut) %>%
  mutate(pool_nonmut = c2_nonmut + c1_nonmut + c4_nonmut) %>%
  dplyr::select(Hugo_Symbol, c3_mut, c3_nonmut, pool_mut, pool_nonmut) %>%
  print()

clust_3_ranked <- clust3_vs_pool %>%
  nest(data = c(c3_mut, c3_nonmut, pool_mut, pool_nonmut)) %>%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %>% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %>%
  mutate(stats = map(fisher, ~broom::glance(.x))) %>%
  unnest(stats) %>%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %>%
  # dplyr::filter(p.value < 0.05) %>%
  left_join(cluster_3_maf) %>%
  arrange(desc(MAF)) %>%
  dplyr::select(Hugo_Symbol) %>%
  print()

#-----------------------------------
cluster_4_maf <- mut_clust %>%
  dplyr::filter(clust == "4") %>%
  dplyr::select(-clust) %>%
  mutate(sample_id = as.factor(sample_id),
         Hugo_Symbol = as.factor(Hugo_Symbol)) %>%
  count(Hugo_Symbol) %>%
  dplyr::rename(c4_mut = n) %>%
  mutate(MAF = c4_mut/317) %>%
  arrange(desc(MAF)) %>%
  left_join(mut_counts_by_cluster) %>%
  # filter out MAF with at least 1% MAF
  dplyr::filter(MAF > 0.01) %>%
  print()
  
clust4_vs_pool <- cluster_4_maf %>%
  mutate(pool_mut = c2_mut + c1_mut + c3_mut) %>%
  mutate(pool_nonmut = c2_nonmut + c1_nonmut + c3_nonmut) %>%
  dplyr::select(Hugo_Symbol, c4_mut, c4_nonmut, pool_mut, pool_nonmut) %>%
  print()

clust_4_ranked <- clust4_vs_pool %>%
  nest(data = c(c4_mut, c4_nonmut, pool_mut, pool_nonmut)) %>%
  mutate(matrix = map(data, ~matrix(unlist(.x), nrow = 2))) %>% 
  mutate(fisher = map(matrix, ~fisher.test(.x))) %>%
  mutate(stats = map(fisher, ~broom::glance(.x))) %>%
  unnest(stats) %>%
  dplyr::select(Hugo_Symbol, p.value, odds = estimate) %>%
  # dplyr::filter(p.value < 0.05) %>%
  left_join(cluster_4_maf) %>%
  arrange(desc(MAF)) %>%
  dplyr::select(Hugo_Symbol) %>%
  print()

#-----------------------------------

mutations <- list(c1 = pull(clust_1_ranked, Hugo_Symbol), c2 = pull(clust_2_ranked, Hugo_Symbol), c3 = pull(clust_3_ranked, Hugo_Symbol), c4 = pull(clust_4_ranked, Hugo_Symbol))

library(gprofiler2)
gostres <- gost(query = mutations,
                organism = "hsapiens", 
                ordered_query = TRUE, 
                multi_query = TRUE, 
                significant = TRUE, 
                exclude_iea = FALSE, 
                measure_underrepresentation = FALSE, 
                evcodes = FALSE, 
                user_threshold = 0.05, 
                correction_method = "g_SCS", 
                domain_scope = "annotated", 
                custom_bg = NULL, 
                numeric_ns = "", 
                sources = NULL, 
                as_short_link = TRUE)
gostres


```

## Display mutation density versus PCA from GSVA v4/v5
```{r}
dlbcl_pca <- readRDS("output/dlbcl_gsva5_4cluster_pca.rds")
dlbcl_hcpc <- readRDS("output/dlbcl_gsva5_4cluster_hcpc.rds")

hcpc_clust <- dlbcl_hcpc$data.clust %>%
  as.data.frame() %>%
  rownames_to_column(var = "sample_id") %>%
  as_tibble() %>%
  dplyr::select(sample_id, clust)

# get coordinates for each sample and merge with total mutation data
# merge with cluster assignment
pca_ind_coord <- dlbcl_pca$ind$coord %>%
  as.data.frame() %>%
  rownames_to_column(var = "sample_id") %>%
  as_tibble() %>%
  left_join(unique_mut) %>%
  left_join(hcpc_clust) %>%
  print()

pca_ind_coord %>%
  group_by(Hugo_Symbol) %>%
  summarize(mean_dim1 = mean(Dim.1), n = n()) %>%
  dplyr::filter(n > 25) %>%
  arrange(desc(mean_dim1))

pca_ind_coord %>%
  group_by(Hugo_Symbol) %>%
  summarize(mean_dim2 = mean(Dim.2), n = n()) %>%
  dplyr::filter(n > 25) %>%
  arrange(desc(mean_dim2))
```

MYD88
```{r}
pca_ind_coord %>%
  dplyr::filter(Hugo_Symbol == "MYD88") %>%
  ggplot(aes(x = Dim.1, y = Dim.2)) + 
    geom_bin2d(bins = 8) +
    scale_fill_continuous(type = "viridis") +
    theme_classic2()

pca_ind_coord %>%
  dplyr::filter(Hugo_Symbol == "MYD88") %>%
  ggplot(aes(x = Dim.1)) +
  geom_density() +
  geom_vline(xintercept = 0) +
  geom_vline(aes(xintercept = mean(Dim.1)), 
             color = "blue", 
             linetype = "dashed", 
             size = 1)
```

MYC
```{r}
pca_ind_coord %>%
  dplyr::filter(Hugo_Symbol == "MYC") %>%
  ggplot(aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 12) +
    scale_fill_continuous(type = "viridis") +
    theme_classic2()

pca_ind_coord %>%
  dplyr::filter(Hugo_Symbol == "EZH2") %>%
  ggplot(aes(x = Dim.1)) +
  geom_density() +
  geom_vline(xintercept = 0) +
  geom_vline(aes(xintercept = mean(Dim.1)), 
             color = "blue", 
             linetype = "dashed", 
             size = 1) +
  theme_classic2() +
  ylab("Mutation frequency")
```

TP53
```{r}
tp53_ind_coord <- pca_ind_coord %>%
  dplyr::filter(Hugo_Symbol == "TP53")

ggplot(data = tp53_ind_coord,
       aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 12) +
    scale_fill_continuous(type = "viridis") +
    theme_classic2()

ggplot(data = tp53_ind_coord,
       aes(x = Dim.1)) +
  geom_density() +
  geom_vline(xintercept = 0) +
  geom_vline(data = tp53_ind_coord,
             aes(xintercept = mean(Dim.1)), 
             color = "blue", 
             linetype = "dashed", 
             size = 1) +
  theme_classic2() +
  ylab("Mutation frequency") +
  geom_density(data = tp53_ind_coord, aes(x = Dim.2)) +
  geom_vline(data = tp53_ind_coord,
             aes(xintercept = mean(Dim.1)), 
             color = "red", 
             linetype = "dashed", 
             size = 1)
```

SOCS1
```{r}
socs1_ind_coord <- pca_ind_coord %>%
  dplyr::filter(Hugo_Symbol == "SOCS1")

foxo1_ind_coord <- pca_ind_coord %>%
  dplyr::filter(Hugo_Symbol == "FOXO1")

ggplot(data = socs1_ind_coord,
       aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 12) +
    scale_fill_continuous(type = "viridis") +
    theme_classic2()

ggplot(data = foxo1_ind_coord,
       aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 10) +
    scale_fill_continuous(type = "viridis") +
    theme_classic2()

ggplot(data = socs1_ind_coord,
       aes(x = Dim.1)) +
  geom_density(color = "blue") +
  geom_vline(xintercept = 0) +
  geom_vline(data = socs1_ind_coord,
             aes(xintercept = mean(Dim.1)), 
             color = "blue", 
             linetype = "dashed", 
             size = 1) +
  theme_classic2() +
  ylab("Mutation frequency") +
  geom_density(data = foxo1_ind_coord, aes(x = Dim.1), color = "red") +
  geom_vline(data = foxo1_ind_coord,
             aes(xintercept = mean(Dim.1)), 
             color = "red", 
             linetype = "dashed", 
             size = 1)

jak1_ind_coord <- pca_ind_coord %>%
  dplyr::filter(Hugo_Symbol == "JAK1")

s1pr2_ind_coord <- pca_ind_coord %>%
  dplyr::filter(Hugo_Symbol == "S1PR2")

ggplot(data = jak1_ind_coord,
       aes(x = Dim.2)) +
  geom_density(color = "red") +
  geom_vline(xintercept = 0) +
  geom_vline(data = jak1_ind_coord,
             aes(xintercept = mean(Dim.2)), 
             color = "red", 
             linetype = "dashed", 
             size = 1) +
  theme_classic2() +
  ylab("Mutation frequency") +
  geom_density(data = s1pr2_ind_coord, aes(x = Dim.2), color = "blue") +
  geom_vline(data = s1pr2_ind_coord,
             aes(xintercept = mean(Dim.2)), 
             color = "blue", 
             linetype = "dashed", 
             size = 1)

ggplot(data = s1pr2_ind_coord,
       aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 5) +
    scale_fill_continuous(type = "viridis") +
    theme_classic2() +
    geom_hline(yintercept = 0, linetype = "dashed", color = "grey", size = 1)

ggplot(data = jak1_ind_coord,
       aes(x = Dim.1, y = Dim.2)) + 
    geom_hex(bins = 5) +
    scale_fill_continuous(type = "viridis") +
    theme_classic2() +
  geom_hline(yintercept = 0, linetype = "dashed", color = "grey", size = 1)

```

SOCS1, oncogene or tumor suppressor: https://www.sciencedirect.com/science/article/abs/pii/S1043466616300059
