---
title: "correlation_clusters"
author: "mleukam"
date: "2019-12-20"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Setup

Clear workspace
```{r}
rm(list = ls())
```

```{r}
knitr::opts_chunk$set(eval = FALSE)
```

Load packages
```{r}
# basics
library(parallel)

# tidyverse
library(tidyverse)

# clustering
library(MineICA)
library(fastICA)

# database
library(RSQLite)
library(AnnotationForge)
library(human.db0)
library(org.Hs.eg.db)
```

## ICA decomposition

Following MineICA vignette:
https://www.bioconductor.org/packages/release/bioc/vignettes/MineICA/inst/doc/MineICA.pdf

Merged and normalized expressionset from [Merge notebook](merge.html) combined with pheno data including cluster assignment from [visualizing_gsva_scores_4 notebook](visualizing_gsva_scores_4.Rmd)

```{r}
combined_es <- readRDS("output/combined_es_gsva4_4cluster.rds")

# we restrict the data to the 10,000 features with the highest IQR
es_10k <- selectFeatures_IQR(combined_es, 10000)

# change rownames to hugo symbols
# read in conversion table for gene IDs
gencode_gtf <- read_tsv("data/gencode.v22.primary_assembly.annotation.gtf.geneinfo")
expr_rnames <- exprs(es_10k) %>% 
  as.data.frame() %>%
  rownames_to_column(var = "gene_id") %>%
  as_tibble() %>%
  left_join(gencode_gtf) %>%
  dplyr::select(gene_name, everything()) %>%
  drop_na(gene_name) %>%
  dplyr::select(-gene_id, -gene_type, -gene_status, -havana_gene) %>%
  pull(gene_name)
head(expr_rnames)
length(expr_rnames)
# get matrix
expr_matrix <- exprs(es_10k)
# add rownames
rownames(expr_matrix) <- expr_rnames
# collapse identical gene symbols
expr_matrix <- avereps(expr_matrix)
dim(expr_matrix)

expr_matrix[1:5, 1:5]

#--------------------------------------------------------------------
## remake expression set
#--------------------------------------------------------------------

# first need variable annotation
v_metadata <- colnames(pData(combined_es)) %>% enframe() %>% 
  dplyr::select(variables = value) %>% 
  mutate(annotation = variables) %>%
  as.data.frame() %>%
  column_to_rownames(var = "variables")

# make pheno data into annotated data frame
phenoData <- new("AnnotatedDataFrame", 
                 data = pData(combined_es), 
                 varMetadata = v_metadata)

# create global annotation
annotation <- as.character("Duke RNAseq counts derived from raw sequence data published by the Dave lab at Duke in Reddy et al Cell 2017. Quantified with Kallisto with reverse strand parameter, collected with tximport, filtered for only protein-coding genes, selected for cases with expression in >12,000 genes, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed. Batch corrected using ComBat by sequencing lane. NCI RNAseq counts originally published by the Staudt lab at NCI in Schmitz et al NEJM 2018. Raw counts downloaded from GDC as HTseq counts, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed. Batch corrected for NCI vs Duke source using ComBat. 125-gene GSVA v4 gene set expression matrix in columns 1:125 of pData, columns 126:197 are phenotype data including 4-cluster model assignments in column `clust`. This version is filtered to include only the 10,000 features with the highest IQR. The gene names were converted to gene symbols and 2 shared gene names were collapsed with avereps")

# assemble expressionset
es_10k_hgnc <- ExpressionSet(
  assayData = expr_matrix,
  phenoData = phenoData,
  annotation = annotation)
es_10k_hgnc 
```

The ICA algorithm fastICA, is implemented in R and may be run with function runICA. FastICA relies on random initializations and the estimated components may vary between iterations. A way to alleviate this problem is to run fastICA several times, cluster the estimates, and use as the final estimates the centrotypes of the clusters. This strategy is proposed in the matlab package icasso Johan Himberg et al. [2004]. The function clusterFastICARuns implements this strategy and can be used to estimate the components.

```{r}
set.seed(313)

library(parallel)
# Random initializations are used for each iteration of FastICA
# Estimates are clustered using hierarchical clustering with average linkage
res <- clusterFastICARuns(X = exprs(es_10k_hgnc), 
                          nbComp = 8, 
                          alg.type = "deflation", 
                          nbIt = 10,
                          funClus = "Mclust",
                          method = "average")
str(res)
```

Before building an IcaSet instance, we need to create a MineICAParams instance that will contain
a few parameters used during the analysis of the ICA decomposition.
You need to specify the directory where you would like to put the outputs of the analysis (slot
resPath), the threshold applied to the projection values used to select the contributing elements
(slot selCutoff), and the threshold you would like to use for statistical significance (slot pvalCutoff).

Make annotation database notes here:
https://bioconductor.org/packages/release/bioc/vignettes/AnnotationForge/inst/doc/SQLForge.pdf
```{r}
## build params
params <- buildMineICAParams(resPath = "es_10k_hgnc/", selCutoff = 3, pvalCutoff = 0.05)

## define the reference samples if any, here no normal sample is available
refSamples_es_10k <- character(0)

# The function buildIcaSet encompasses the step of feature annotation. During the annotation step (either performed using the annotation package or biomaRt) if several features are available for a same gene, the median value across those features is attributed to the gene. Data can also be provided at the final annotation level (e.g dat and S are already indexed by gene ids), in that case please use alreadyAnnot=TRUE in the function buildIcaSet so that no annotation will be performed. We can now build the object icaSetMainz with help of function buildIcaSet:
resBuild <- buildIcaSet(params = params, 
                        A = data.frame(res$A), 
                        S = data.frame(res$S),
                        dat = data.frame(exprs(es_10k_hgnc)),
                        pData = pData(es_10k),
                        alreadyAnnot = TRUE,
                        refSamples = refSamples_es_10k)

icaSet10k <- resBuild$icaSet
icaSet10k
params <- resBuild$params
params
featureNames(icaSet10k)[1:5]
geneNames(icaSet10k)[1:5]
```

Select the contributing features or genes: When applying ICA decomposition to genomic data, for example here gene expression data, the distribution of the gene projections on the ICs is expected to be super-Gaussian: a large portion of genes follows a (super-)Gaussian centered at zero and a small portion belongs to an outgrowth located on the right and/or on the left of the distribution. In order to select the elements belonging to this outgrowth, we used the conventional way based on a threshold. The thresholds can typically be 3 or 4 standard deviations from the mean. We refer to the resulting selected genes as the “contributing genes”.

```{r}
## Extract the contributing genes
contrib <- selectContrib(icaSet10k, cutoff = 3, level = "genes")

## Show the first contributing genes of the fomponents
sort(abs(contrib[[1]]),decreasing = TRUE)[1:10]
sort(abs(contrib[[2]]),decreasing = TRUE)[1:10]
sort(abs(contrib[[3]]),decreasing = TRUE)[1:10]
sort(abs(contrib[[4]]),decreasing = TRUE)[1:10]
sort(abs(contrib[[5]]),decreasing = TRUE)[1:10]

## select the annotations of interest
varLabels(icaSet10k)

# restrict the phenotype data to the variables of interest
keepVar <- c("clust", "source", "abc_gcb_rna", "amp_or_not")
# specify the variables that should be treated as character
icaSet10k$clust <- as.character(icaSet10k$clust)
icaSet10k$source <- as.character(icaSet10k$source)
icaSet10k$abc_gcb_rna <- as.character(icaSet10k$abc_gcb_rna)
icaSet10k$amp_or_not <- as.character(icaSet10k$amp_or_not)

## Run the analysis of the ICA decomposition
# only enrichment in KEGG gene sets are tested
runAn(params = params, 
      icaSet = icaSet10k, 
      keepVar = keepVar, 
      dbGOstats = "KEGG",
      funClus = "Mclust")
```
