---
title: "cyt"
author: "mleukam"
date: "2019-07-29"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Setup

Clear the workspace
```{r}
rm(list = ls())
```

```{r}
# load packages
library(tidyverse)
library(Biobase)
library(psych)
```

Read in data
Matrix of counts in TPM created for deconvolution analysis in [deconvolution notebook](deconvolution.html)

```{r}
tpm_es <- readRDS("output/tpm_expressionset.rds")
```

```{r}
tpm <- exprs(tpm_es)
tpm[1:5, 1:5]

# to calculate CYT score, take geometric mean of TPM for PRF1 and GZMA
geomean <- function(x, na.rm = TRUE, nan.rm = FALSE, eta = NA_real_) {
  nan.count <- is.nan(x) %>%
    sum()
  na.count <- is.na(x) %>%
    sum()
  value.count <- !is.na(x) %>%
    sum()
  case_when(
    #Handle cases when there are negative values, all values are missing, or
    #missing values are not tolerated.
    (nan.count > 0 & !nan.rm) | any(x < 0, na.rm = TRUE) ~ NaN,
    (na.count > 0 & !na.rm) | value.count == 0 ~ NA_real_,

    #Handle cases when non-missing values are either all positive or all zero.
    #In these cases the eta parameter is irrelevant and therefore ignored.
    all(x > 0, na.rm = TRUE) ~ exp(mean(log(x), na.rm = TRUE)),
    all(x == 0, na.rm = TRUE) ~ 0,

    #All remaining cases are cases when there are a mix of positive and zero values.
    #By default, we do not use an artificial constant or propagate zeros.
    is.na(eta) ~ exp(sum(log(x[x > 0]), na.rm = TRUE) / value.count),
    eta > 0 ~ exp(mean(log(x + eta), na.rm = TRUE)) - eta,
    TRUE ~ 0 #only propagate zeroes when eta is set to 0 (or less than 0)
  )
}

cyt_data <- tpm %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_name") %>%
  as_tibble() %>%
  dplyr::filter(gene_name %in% c("PRF1", "GZMA")) %>%
  dplyr::select(-gene_name) %>%
  as.list() %>%
  print()

CYT = map_dbl(cyt_data, geomean)
```

Update pheno data
```{r}
pheno_data <- pData(tpm_es) %>%
  rownames_to_column(var = "sample_id") %>%
  as_tibble() %>%
  add_column(CYT) %>%
  dplyr::select(sample_id, CYT, everything()) %>%
  as.data.frame() %>%
  column_to_rownames(var = "sample_id")

pheno_data[1:5, 1:5]
```

Remake expression set with cyt scores
```{r}
# use updated pheno data frame 
v_metadata <- data.frame(
  labelDescription = colnames(pheno_data),
  row.names = colnames(pheno_data) )
  
phenoData <- new("AnnotatedDataFrame", 
                 data = pheno_data, 
                 varMetadata = v_metadata)
phenoData

annotation <- as.character("Duke RNAseq counts derived from raw sequence data published by the Dave lab at Duke in Reddy et al Cell 2017. Quantified with Kallisto with reverse strand parameter, collected with tximport, filtered for only protein-coding genes, selected for cases with expression in >12,000 genes, converted to TPM using gene lengths from Gencode v22 GTF file. NCI RNAseq counts originally published by the Staudt lab at NCI in Schmitz et al NEJM 2018. Raw counts downloaded from GDC as HTseq counts, Converted to TPM using gene lengths from Gencode v22 GTF file. PHENOTYPE DATA NOW ALSO INCLUDES CYT SCORE")

# assemble expressionset
tpm_cyt_es <- ExpressionSet(
  assayData = tpm,
  phenoData = phenoData,
  annotation = annotation)
tpm_cyt_es
```

```{r}
saveRDS(tpm_cyt_es, "output/tpm_cyt_expressionset.rds")
```
