---
title: "deconvolution"
author: "mleukam"
date: "2019-07-11"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Setup
Clear the workspace
```{r}
rm(list = ls())
```

Load packages
```{r}
library("tidyverse")
library("edgeR")
library("limma")
library("Biobase")
library("preprocessCore")
library("ggsci")
library("ggdark")
library("viridis")
library("ggpubr")
library("GenomicFeatures")
```

## Prepare data (TPM)

Read in data
```{r}
# Combined duke+nci es with gsva4, 4-cluster model phenodata and log-transformed, normalized read counts
combined_es <- readRDS("output/combined_es_gsva4_4cluster.rds")
```

Plan for data prep to TPM:

1. Start with raw counts (HTSeq or Kallisto for example)
2. Convert to TPM

From xCell:

> xCell uses the expression levels ranking and not the actual values, thus normalization does not have an effect, however normalizing to gene length (RPKM/FPKM/TPM/RSEM) is required.

I will not normalize across lane or source, and TMM normalization will be skipped. I will filter for protein coding genes but not for lowly expressed genes since that may affect the relative ranking.

## NCI 

Read in raw counts
```{r}
total_counts <- read_csv("output/nci_dlbcl_unprocessed_counts.csv")

# read in conversion table for gene IDs
gencode_gtf <- read_tsv("data/gencode.v22.primary_assembly.annotation.gtf.geneinfo")
```

#### Convert to gene name (expected input)

From xCell:

> The expression matrix should be a matrix with genes in rows and samples in columns. The rownames should be gene symbols. If the data contains non-unique gene symbols, rows with same gene symbols will be averaged.

```{r}
# filter for protein coding genes and convert ot hugo ID (gene symbol)
total_counts_prcode <- total_counts %>%
  dplyr::rename(gene_id = gene) %>%
  left_join(gencode_gtf, by = "gene_id") %>%
  dplyr::filter(gene_type == "protein_coding") %>%
  dplyr::select(-gene_type, -gene_status, -level, -havana_gene) %>%
  dplyr::select(gene_name, gene_id, everything())
total_counts_prcode

nrow(total_counts)
nrow(total_counts_prcode)
```

#### Convert to TPM

Original full GTF file for gencode v22 downloaded from: https://www.gencodegenes.org/human/release_22.html

Change of function name: https://support.bioconductor.org/p/100757/

TPM is very similar to RPKM and FPKM. The only difference is the order of operations. Here’s how you calculate TPM:

1. Divide the read counts by the length of each gene in kilobases. This gives you reads per kilobase (RPK).
2. Count up all the RPK values in a sample and divide this number by 1,000,000. This is your “per million” scaling factor.
3. Divide the RPK values by the “per million” scaling factor. This gives you TPM.

https://www.rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/

```{r}
gencode_v22_annotation <- read_delim("data/gencode.v22.annotation.gtf", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 5)
```

```{r eval=FALSE}
###
# Note - geting exonic gene sizes takes a while (~10+ mins on laptop)
txdb <- makeTxDbFromGFF("data/gencode.v22.annotation.gtf", format = "gtf")

# collect the exons per gene id
exons_list_per_gene <- exonsBy(txdb, by = "gene")

# then for each gene, reduce all the exons to a set of non overlapping exons, calculate their lengths (widths) and sum then
exonic_gene_sizes <- lapply(exons_list_per_gene, function(x){
  sum(width(reduce(x)))
  })

saveRDS(exonic_gene_sizes, "output/gencode_v22_exonic_gene_sizes.rds")
```


```{r}
# use exonic gene sizes to get reads per kilobase or RPK matrix
exonic_gene_sizes <- readRDS("output/gencode_v22_exonic_gene_sizes.rds")
exonic_gene_sizes[1:3]

eff_length <- exonic_gene_sizes %>%
  enframe() %>%
  mutate(length = unlist(value)) %>%
  mutate(length = length/1000) %>%
  dplyr::select(gene_id = name, length) %>%
  print()

tc_length <- total_counts_prcode %>% 
  left_join(eff_length) %>%
  dplyr::select(gene_name, length, everything()) %>%
  print()

count_matrix <- tc_length %>%
  as.data.frame() %>%
  dplyr::select(-gene_name, -length) %>%
  column_to_rownames(var = "gene_id") %>%
  as.matrix()

count_matrix[1:5, 1:5]
str(count_matrix)

rpk <- count_matrix / tc_length$length
rpk[1:5, 1:5]
```

```{r}
# get a "per million scaling factor" for each sample
per_million_scaling_factor <- colSums(rpk) / 1e6
per_million_scaling_factor[1:5]
scale_list <- as.list(per_million_scaling_factor)

# apply the scaling factor to each sample
rpk_list <- as.list(as.data.frame(rpk))

tpm <- map2_df(rpk_list, scale_list, function(list, divisor){
  list / divisor
  }) %>%
  as.data.frame()

rownames(tpm) <- rownames(count_matrix)

nci_tpm <- tpm %>%
  rownames_to_column(var = "gene_id") %>%
  as_tibble() %>%
  print()
```

## Duke

Read in raw counts for Duke samples as dgelist
```{r}
dgelist <- readRDS("data/temp.dgelist_edger.rds")
str(dgelist)
```

Convert GENCODE v29 to v22 gene ids to match GDC standard
```{r}
# get expression matrix
expr_duke <- dgelist[[1]]
expr_duke[1:5, 1:5]
col__names <- colnames(expr_duke)
col_names_rep <- str_split(col__names, pattern = "\\.", simplify = TRUE) %>% 
  as_tibble() %>%
  pull(V1) %>%
  paste0("duke_", .) %>%
  print()
colnames(expr_duke) <- col_names_rep
expr_duke[1:5, 1:5]

# gencode v29 was used to generate rownames of expr_duke matrix
# GDC standard is gencode v22, which is source of NCI data
# will convert to gencode v22 to match rows of NCI data
geneID_split <- str_split(gencode_gtf$gene_id, pattern = "\\.", simplify = TRUE) %>% 
  as_tibble() %>%
  dplyr::select(geneID_split = V1)

gencode_gtf_split <- gencode_gtf %>% 
  bind_cols(geneID_split) %>%
  dplyr::select(gene_id_v22 = gene_id, geneID_split, everything()) %>%
  print()

row__names <- rownames(expr_duke) %>%
  enframe() %>%
  dplyr::select(geneID_split = value) %>%
  print()

# check how many gene ids are missing in v22 that are named in v29
setdiff(row__names$geneID_split, gencode_gtf_split$geneID_split)
# approx 1900 (previous checks show none after filtering for protein coding genes)

# merge gene name data with expression matrix
expr_duke_df <- expr_duke %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_id_v29") %>%
  as_tibble() %>%
  mutate(geneID_split = str_replace(gene_id_v29, pattern = "\\.*", "")) %>%
  dplyr::select(gene_id_v29, geneID_split, everything()) %>%
  left_join(gencode_gtf_split) %>%
  dplyr::select(gene_id_v29, gene_id_v22, geneID_split, gene_type, everything()) %>%
  dplyr::select(-gene_status, -level, -havana_gene) %>%
  print()

# check how many gene ids are missing from version 22  
summary(is.na(expr_duke_df$gene_id_v22))

# check the gene types in the expression matrix
summary(as.factor(expr_duke_df$gene_type))
```

#### Filter for protein coding
```{r}
# filter for protein coding genes only
expr_duke_filtered <- expr_duke_df %>% 
  dplyr::filter(gene_type == "protein_coding")
  
# any missing gene ids also protein coding genes?
summary(is.na(expr_duke_filtered$gene_id_v22))
# NO

expr_duke_protcoding <- expr_duke_filtered %>%
  dplyr::select(-gene_id_v29, -geneID_split, -gene_type) %>%
  dplyr::select(gene_id = gene_id_v22, gene_name, everything()) %>%
  print()
```

#### Subset cases based on protein coding library coverage

Following recommendation in methods section of Reddy et al Cell 2017 (original data source). They reported including only those libraries containing at least 12,000 protein coding genes.

```{r}
gene_coverage_threshold <- 12000

ncol(expr_duke_protcoding)
expr_duke_subset <- expr_duke_protcoding[, colSums(expr_duke_protcoding != 0) > gene_coverage_threshold]

expr_duke_subset[1:5, 1:5]
ncol(expr_duke_subset)
```

#### Convert to TPM
TPM is very similar to RPKM and FPKM. The only difference is the order of operations. Here’s how you calculate TPM:

1. Divide the read counts by the length of each gene in kilobases. This gives you reads per kilobase (RPK).
2. Count up all the RPK values in a sample and divide this number by 1,000,000. This is your “per million” scaling factor.
3. Divide the RPK values by the “per million” scaling factor. This gives you TPM.

https://www.rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/

Exonic gene size object created above for NCI samples

```{r}
exonic_gene_sizes <- readRDS("output/gencode_v22_exonic_gene_sizes.rds")
exonic_gene_sizes[1:3]

eff_length <- exonic_gene_sizes %>%
  enframe() %>%
  mutate(length = unlist(value)) %>%
  mutate(length = length/1000) %>%
  dplyr::select(gene_id = name, length) %>%
  print()

tc_length <- expr_duke_subset %>% 
  left_join(eff_length) %>%
  dplyr::select(gene_name, length, everything()) %>%
  print()

count_matrix <- tc_length %>%
  as.data.frame() %>%
  dplyr::select(-gene_name, -length) %>%
  column_to_rownames(var = "gene_id") %>%
  as.matrix()

count_matrix[1:5, 1:5]
str(count_matrix)

rpk <- count_matrix / tc_length$length
rpk[1:5, 1:5]

per_million_scaling_factor <- colSums(rpk) / 1e6
per_million_scaling_factor[1:5]
scale_list <- as.list(per_million_scaling_factor)

rpk_list <- as.list(as.data.frame(rpk))

tpm <- map2_df(rpk_list, scale_list, function(list, divisor){
  list / divisor
  }) %>%
  as.data.frame()

rownames(tpm) <- rownames(count_matrix)

duke_tpm <- tpm %>%
  rownames_to_column(var = "gene_id") %>%
  as_tibble() %>%
  print()
```

## Merge

```{r}
# messy coding - changed variable names here...
duke_tpm_df <- duke_tpm
nci_tpm_df <- nci_tpm

# filter rownames to only keep intersecting rows
duke_genes <- duke_tpm_df %>% pull(gene_id)
length(duke_genes)
nci_genes <- nci_tpm_df %>% pull(gene_id)
length(nci_genes)

combined_genes <- intersect(duke_genes, nci_genes)
length(combined_genes)
head(combined_genes)

duke_tpm_filt <- duke_tpm_df %>%
  dplyr::filter(gene_id %in% combined_genes)
nrow(duke_tpm_filt)

nci_tpm_filt <- nci_tpm_df %>%
  dplyr::filter(gene_id %in% combined_genes)
nrow(nci_tpm_filt)

comb_tpm <- nci_tpm_filt %>%
  left_join(duke_tpm_filt, by = "gene_id") %>%
  left_join(gencode_gtf) %>%
  dplyr::select(-gene_id, -gene_type, -gene_status, -level, -havana_gene) %>%
  dplyr::select(gene_name, everything()) %>%
  drop_na(gene_name)

tpm_mat <- comb_tpm[2:1190]

tpm_mat_names <- pull(comb_tpm, gene_name)

tpm_mat <- as.matrix(tpm_mat)
rownames(tpm_mat) <- tpm_mat_names
tpm_mat[1:5, 1:5]

# average over shared gene names
combined_tpm <- avereps(tpm_mat, ID = rownames(tpm_mat))
dim(combined_tpm)
combined_tpm[1:10, 1:5]
```

#### Reformat pheno data to match

```{r}
pheno_data_tbl <- pData(combined_es) %>%
  as.data.frame() %>%
  rownames_to_column(var = "sample_id") %>%
  dplyr::select(sample_id, everything()) %>%
  as_tibble() %>%
  print()

samplenames <- pheno_data_tbl %>% pull(sample_id)
length(samplenames)
columnnames <- colnames(combined_tpm)
length(columnnames)
identical(samplenames, columnnames)

pheno_data <- pheno_data_tbl %>%
  as.data.frame() %>%
  column_to_rownames(var = "sample_id")
```

Recreate expression set for TPM and write out CSV for xCell
```{r}
# use updated pheno data frame 
v_metadata <- data.frame(
  labelDescription = colnames(pheno_data),
  row.names = colnames(pheno_data) )
  
phenoData <- new("AnnotatedDataFrame", 
                 data = pheno_data, 
                 varMetadata = v_metadata)
phenoData

annotation <- as.character("Duke RNAseq counts derived from raw sequence data published by the Dave lab at Duke in Reddy et al Cell 2017. Quantified with Kallisto with reverse strand parameter, collected with tximport, filtered for only protein-coding genes, selected for cases with expression in >12,000 genes, converted to TPM using gene lengths from Gencode v22 GTF file. NCI RNAseq counts originally published by the Staudt lab at NCI in Schmitz et al NEJM 2018. Raw counts downloaded from GDC as HTseq counts, Converted to TPM using gene lengths from Gencode v22 GTF file.")

# assemble expressionset
tpm_es <- ExpressionSet(
  assayData = combined_tpm,
  phenoData = phenoData,
  annotation = annotation)
tpm_es
```

```{r}
saveRDS(tpm_es, "output/tpm_expressionset.rds")
write.csv(combined_tpm, "output/combined_tpm_matrix.csv")
```

## xCell results

Submitted to web portal 7/18/2019

```{r}
xc_results <- read_delim("output/xCell_combined_tpm_matrix_xCell_2225072019.txt", "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  dplyr::rename(cell_type = X1) %>% 
  as.data.frame() %>%
  column_to_rownames(var = "cell_type") %>%
  as.matrix %>%
  t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "sample_id") %>%
  as_tibble() %>%
  print()

# read in cluster assignments
gsva4_es <- readRDS("output/combined_es_gsva4_4cluster.rds")
clust_assgn <- pData(gsva4_es) %>%
  as.data.frame() %>%
  rownames_to_column(var = "sample_id") %>%
  dplyr::select(sample_id, clust) %>%
  as_tibble() %>%
  print()

# add to dataframe
xc_results_clust <- xc_results %>%
  left_join(clust_assgn) %>%
  dplyr::select(sample_id, clust, everything()) %>%
  print()

# quick look at cytotoxic T-cells
xc_results_clust %>%
  group_by(clust) %>%
  summarize(mean(`CD8+ T-cells`))

# transform data for boxplot
xc_tidy <- xc_results_clust %>%
  gather(key = "cell_type", value = "xc_score", -clust, -sample_id) %>%
  mutate(cell_type = as.factor(cell_type)) %>%
  mutate(clust = as.factor(clust)) %>%
  print()

# select cell types of interest
keepers <- c("CD8+ T-cells", "Th1 cells", "Th2 cells", "Tregs", "Macrophages", "Macrophages M1", "Macrophages M2", "DC", "Eosinophils")

xc_tidy_filt <- xc_tidy %>%
  dplyr::filter(cell_type %in% keepers) %>%
  mutate(cell_type = fct_drop(cell_type)) %>%
  mutate(cell_type = fct_relevel(cell_type, 
                                "Th1 cells" = "Th1 cells",
                                "Th2 cells" = "Th2 cells", 
                                "CD8+ T-cells" = "CD8+ T-cells",
                                "Tregs" = "Tregs", 
                                "Macrophages" = "Macrophages", 
                                "Macrophages M1" = "Macrophages M1", 
                                "Macrophages M2" = "Macrophages M2", 
                                "DC" = "DC", 
                                "Eosinophils" = "Eosinophils")) %>%
  mutate(clust = fct_recode(clust, "Cold  (n = 361)" = "1",
                            "Intermediate-T  (n = 283)" = "2",
                            "Intermediate-M  (n = 225)" = "3",
                            "Inflamed  (n = 320)" = "4"))

# boxplot
total_boxplot <- ggplot(xc_tidy_filt, 
                        aes(x = cell_type, y = xc_score)) +
  geom_boxplot(aes(fill = cell_type)) +
  facet_wrap("clust") +
  ylab("xCell scores") +
  xlab("Cell type") +
  ggtitle("Immune Cell Deconvolution of DLBCL Transcriptomes (n = 1189)") +
  theme_classic2() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(fill = "Selected Cell Types")

total_boxplot

keepers2 <- c("CD8+ T-cells", "Th1 cells", "Th2 cells", "Tregs", "Macrophages", "Macrophages M1")

xc_tidy_filt2 <- xc_tidy_filt %>%
  dplyr::filter(cell_type %in% keepers2) %>%
  mutate(cell_type = fct_drop(cell_type)) %>%
  mutate(clust = fct_recode(clust, "Cold" = "Cold  (n = 361)",
                            "Intermediate-T" = "Intermediate-T  (n = 283)",
                            "Intermediate-M" = "Intermediate-M  (n = 225)",
                            "Inflamed" = "Inflamed  (n = 320)"))

# boxplot
cells_boxplot <- ggplot(xc_tidy_filt2, 
                        aes(x = clust, y = xc_score)) +
  geom_boxplot(aes(fill = clust)) +
  facet_grid(cols = vars(cell_type)) +
  ylab("xCell scores") +
  xlab("") +
  ggtitle("Immune Cell Deconvolution of DLBCL Transcriptomes (n = 1189)") +
  theme_classic2() +
  scale_fill_d3() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(fill = "Cluster") +
  stat_compare_means(aes(label = paste0("p = ", ..p.format..)),
                     method = "anova") +
  theme(axis.text = element_text(size = 10))

cells_boxplot
# 1100 x 500
```

#### Additional plots

Tips on ANOVA: http://www.sthda.com/english/wiki/one-way-anova-test-in-r
Tips on significance bars: https://www.r-bloggers.com/add-p-values-and-significance-levels-to-ggplots/

if needed: invert_geom_defaults()

```{r}
xc_tidy_cd8 <- xc_tidy %>%
  dplyr::filter(cell_type == "CD8+ T-cells") %>%
  mutate(clust = fct_recode(clust,
                            "cold" = "1",
                            "intermediate-T" = "2",
                            "intermediate-M" = "3",
                            "inflamed" = "4")) %>%
  print()

# boxplot
cd8_boxplot <- ggplot(xc_tidy_filt, aes(x = clust, y = xc_score)) +
  geom_boxplot(aes(fill = clust)) +
  ylab("xCell scores") +
  xlab("Cluster") +
  ggtitle("Immune Cell Deconvolution of CD8 T-cells") +
  dark_theme_gray() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

cd8_boxplot

# significance testing
group_by(xc_tidy_cd8, clust) %>%
  summarise(
    count = n(),
    mean = mean(xc_score, na.rm = TRUE),
    sd = sd(xc_score, na.rm = TRUE)
  )

# Compute the analysis of variance
compare_means(xc_score ~ clust, 
              data = xc_tidy_cd8,
              method = "anova")

my_comparisons <- list(c("1", "4"), 
                       c("2", "4"), 
                       c("3", "4"))

# boxplot with significance testing
cd8_boxplot_sig <- ggplot(xc_tidy_cd8, aes(x = clust, y = xc_score)) +
  geom_boxplot(aes(fill = clust)) +
  ylab("xCell scores") +
  xlab("") +
  ggtitle("Immune Cell Deconvolution of CD8 T-cells") +
  theme_classic2() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = rel(1.4))) +
  stat_compare_means(method = "anova", label.y = -0.05, label.x = 1.5, size = 5)

cd8_boxplot_sig

```

```{r}
xc_tidy_mac <- xc_tidy %>%
  dplyr::filter(cell_type == "Macrophages") %>%
  mutate(clust = fct_recode(clust,
                            "cold" = "1",
                            "intermediate-T" = "2",
                            "intermediate-M" = "3",
                            "inflamed" = "4")) %>%
  print()

# significance testing
group_by(xc_tidy_mac, clust) %>%
  summarise(
    count = n(),
    mean = mean(xc_score, na.rm = TRUE),
    sd = sd(xc_score, na.rm = TRUE)
  )

# Compute the analysis of variance
compare_means(xc_score ~ clust, 
              data = xc_tidy_mac,
              method = "anova")

# boxplot with significance testing
mac_boxplot_sig <- ggplot(xc_tidy_mac, aes(x = clust, y = xc_score)) +
  geom_boxplot(aes(fill = clust)) +
  ylab("xCell scores") +
  xlab("") +
  ggtitle("Immune Cell Deconvolution of Total Macrophages") +
  theme_classic2() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = rel(1.4))) +
  stat_compare_means(method = "anova", label.y = -0.05, label.x = 1.5, size = 5)

mac_boxplot_sig

```

```{r}
xc_tidy_dc <- xc_tidy %>%
  dplyr::filter(cell_type == "DC")

# significance testing
group_by(xc_tidy_dc, clust) %>%
  summarise(
    count = n(),
    mean = mean(xc_score, na.rm = TRUE),
    sd = sd(xc_score, na.rm = TRUE)
  )

# Compute the analysis of variance
compare_means(xc_score ~ clust, 
              data = xc_tidy_dc,
              method = "anova")

# boxplot with significance testing
dc_boxplot_sig <- ggplot(xc_tidy_dc, aes(x = clust, y = xc_score)) +
  geom_boxplot(aes(fill = clust)) +
  ylab("xCell scores") +
  xlab("Cluster") +
  ggtitle("Immune Cell Deconvolution of Dendritic Cells") +
  dark_theme_gray() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  stat_compare_means(label = "p.signif", 
                     method = "t.test",
                     ref.group = ".all.",
                     label.y = 0.3) +
  geom_hline(yintercept = median(xc_tidy_dc$xc_score), 
             linetype = 2) +
  stat_compare_means(method = "anova", label.y = -0.02)

dc_boxplot_sig
```

```{r}
xc_tidy_m1 <- xc_tidy %>%
  dplyr::filter(cell_type == "Macrophages M1")

# significance testing
group_by(xc_tidy_m1, clust) %>%
  summarise(
    count = n(),
    mean = mean(xc_score, na.rm = TRUE),
    sd = sd(xc_score, na.rm = TRUE)
  )

# Compute the analysis of variance
compare_means(xc_score ~ clust, 
              data = xc_tidy_m1,
              method = "anova")

# boxplot with significance testing
m1_boxplot_sig <- ggplot(xc_tidy_m1, aes(x = clust, y = xc_score)) +
  geom_boxplot(aes(fill = clust)) +
  ylab("xCell scores") +
  xlab("Cluster") +
  ggtitle("Immune Cell Deconvolution of M1 Macrophages") +
  dark_theme_gray() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  stat_compare_means(label = "p.signif", 
                     method = "t.test",
                     ref.group = ".all.",
                     label.y = 0.4) +
  geom_hline(yintercept = median(xc_tidy_m1$xc_score), 
             linetype = 2) +
  stat_compare_means(method = "anova", label.y = -0.02)

m1_boxplot_sig
```

```{r}
xc_tidy_m2 <- xc_tidy %>%
  dplyr::filter(cell_type == "Macrophages M2") %>%
  mutate(clust = fct_recode(clust,
                            "cold" = "1",
                            "intermediate-T" = "2",
                            "intermediate-M" = "3",
                            "inflamed" = "4"))

# significance testing
group_by(xc_tidy_m2, clust) %>%
  summarise(
    count = n(),
    mean = mean(xc_score, na.rm = TRUE),
    sd = sd(xc_score, na.rm = TRUE)
  )

# Compute the analysis of variance
compare_means(xc_score ~ clust, 
              data = xc_tidy_m2,
              method = "anova")

# boxplot with significance testing
m2_boxplot_sig <- ggplot(xc_tidy_m2, aes(x = clust, y = xc_score)) +
  geom_boxplot(aes(fill = clust)) +
  ylab("xCell scores") +
  xlab("Cluster") +
  ggtitle("Immune Cell Deconvolution of M2 Macrophages") +
  theme_classic2() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  stat_compare_means(method = "anova", label.y = -0.02)

m2_boxplot_sig
```

```{r}
xc_tidy_th2 <- xc_tidy %>%
  dplyr::filter(cell_type == "Th2 cells")

# significance testing
group_by(xc_tidy_th2, clust) %>%
  summarise(
    count = n(),
    mean = mean(xc_score, na.rm = TRUE),
    sd = sd(xc_score, na.rm = TRUE)
  )

# Compute the analysis of variance
compare_means(xc_score ~ clust, 
              data = xc_tidy_th2,
              method = "anova")

# boxplot with significance testing
th2_boxplot_sig <- ggplot(xc_tidy_th2, aes(x = clust, y = xc_score)) +
  geom_boxplot(aes(fill = clust)) +
  ylab("xCell scores") +
  xlab("Cluster") +
  ggtitle("Immune Cell Deconvolution of Th2 Cells") +
  dark_theme_gray() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  stat_compare_means(label = "p.signif", 
                     method = "t.test",
                     ref.group = ".all.",
                     label.y = 1.15) +
  geom_hline(yintercept = median(xc_tidy_th2$xc_score), 
             linetype = 2) +
  stat_compare_means(method = "anova", label.y = -0.02)

th2_boxplot_sig
```

```{r}
xc_tidy_th1 <- xc_tidy %>%
  dplyr::filter(cell_type == "Th2 cells") %>%
  mutate(clust = fct_recode(clust,
                            "cold" = "1",
                            "intermediate-T" = "2",
                            "intermediate-M" = "3",
                            "inflamed" = "4"))

# significance testing
group_by(xc_tidy_th1, clust) %>%
  summarise(
    count = n(),
    mean = mean(xc_score, na.rm = TRUE),
    sd = sd(xc_score, na.rm = TRUE)
  )

# Compute the analysis of variance
compare_means(xc_score ~ clust, 
              data = xc_tidy_th1,
              method = "anova")

# boxplot with significance testing
th1_boxplot_sig <- ggplot(xc_tidy_th1, aes(x = clust, y = xc_score)) +
  geom_boxplot(aes(fill = clust)) +
  ylab("xCell scores") +
  xlab("") +
  ggtitle("Immune Cell Deconvolution of Th1 Cells") +
  theme_classic2() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = rel(1.4))) +
  stat_compare_means(method = "anova", label.y = -0.05, label.x = 1.5, size = 5)

th1_boxplot_sig
```

```{r}
xc_tidy_treg <- xc_tidy %>%
  dplyr::filter(cell_type == "Tregs")

# significance testing
group_by(xc_tidy_treg, clust) %>%
  summarise(
    count = n(),
    mean = mean(xc_score, na.rm = TRUE),
    sd = sd(xc_score, na.rm = TRUE)
  )

# Compute the analysis of variance
compare_means(xc_score ~ clust, 
              data = xc_tidy_treg,
              method = "anova")

# boxplot with significance testing
treg_boxplot_sig <- ggplot(xc_tidy_treg, aes(x = clust, y = xc_score)) +
  geom_boxplot(aes(fill = clust)) +
  ylab("xCell scores") +
  xlab("Cluster") +
  ggtitle("Immune Cell Deconvolution of Treg Cells") +
  dark_theme_gray() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  stat_compare_means(label = "p.signif", 
                     method = "t.test",
                     ref.group = ".all.",
                     label.y = 1.15) +
  geom_hline(yintercept = median(xc_tidy_treg$xc_score), 
             linetype = 2) +
  stat_compare_means(method = "anova", label.y = -0.02)

treg_boxplot_sig
```

#### Comparisons
```{r}
clust_23_means <- xc_tidy %>%
  dplyr::filter(clust %in% c("2", "3")) %>%
  group_by(cell_type, clust) %>%
  summarize(
    count = n(),
    mean = mean(xc_score, na.rm = TRUE),
    sd = sd(xc_score, na.rm = TRUE)
  ) %>%
  print()

clust_2_means <- clust_23_means %>%
  dplyr::filter(clust == 2) %>%
  dplyr::rename(clust_2_count = count, 
                clust_2_mean = mean,
                clust_2_sd = sd) %>%
  dplyr::select(-clust) %>%
  print()

clust_3_means <- clust_23_means %>%
  dplyr::filter(clust == 3) %>%
  dplyr::rename(clust_3_count = count, 
                clust_3_mean = mean,
                clust_3_sd = sd) %>%
  dplyr::select(-clust) %>%
  print()

clust_2vs3_means <- clust_2_means %>%
  left_join(clust_3_means, by = "cell_type") %>%
  print()

clust_23 <- xc_tidy %>%
  dplyr::filter(clust %in% c("2", "3")) %>%
  group_by(cell_type, clust) %>%
  print()

clust_23_th1_sig <- clust_23 %>%
  dplyr::filter(cell_type == "Th1 cells") %>%
  compare_means(xc_score ~ clust, .) %>%
  print()

clust_23_M1_sig <- clust_23 %>%
  dplyr::filter(cell_type == "Macrophages M1") %>%
  compare_means(xc_score ~ clust, .) %>%
  print()

clust23box <- ggplot(clust_23, aes(x = cell_type, y = xc_score, fill = clust)) +
  geom_boxplot() + 
  ylab("xCell scores") +
  xlab("Cluster") +
  ggtitle("Comparison of all cell types, C2 vs C3") +
  dark_theme_gray() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
clust23box
```

```{r}
clust_14_means <- xc_tidy %>%
  dplyr::filter(clust %in% c("1", "4")) %>%
  group_by(cell_type, clust) %>%
  summarize(
    count = n(),
    mean = mean(xc_score, na.rm = TRUE),
    sd = sd(xc_score, na.rm = TRUE)
  ) %>%
  print()

clust_1_means <- clust_14_means %>%
  dplyr::filter(clust == 1) %>%
  dplyr::rename(clust_1_count = count, 
                clust_1_mean = mean,
                clust_1_sd = sd) %>%
  dplyr::select(-clust) %>%
  print()

clust_4_means <- clust_14_means %>%
  dplyr::filter(clust == 4) %>%
  dplyr::rename(clust_4_count = count, 
                clust_4_mean = mean,
                clust_4_sd = sd) %>%
  dplyr::select(-clust) %>%
  print()

clust_1vs4_means <- clust_1_means %>%
  left_join(clust_4_means, by = "cell_type") %>%
  print()

clust_14 <- xc_tidy %>%
  dplyr::filter(clust %in% c("1", "4")) %>%
  group_by(cell_type, clust) %>%
  print()

clust14box <- ggplot(clust_14, aes(x = cell_type, y = xc_score, fill = clust)) +
  geom_boxplot() + 
  ylab("xCell scores") +
  xlab("Cluster") +
  ggtitle("Comparison of all cell types, C1 vs C4") +
  dark_theme_gray() +
  scale_color_viridis(discrete = TRUE) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
clust14box

## significance testing
# make a list of each of the cell types to be analyzed
cell_types <- unique(clust_14$cell_type) %>%
  as.character() %>%
  as.list() %>%
  print()

# define function to test significant differences in median for each cell type 

```

# Abandoned strategies
Investigated TIMER and MuSiC, but decided not to use in the current analysis. Code is preserved for record-keeping.

## Prepare counts for TIMER
https://cistrome.shinyapps.io/timer/

### Duke data

#### Read in data
```{r eval=FALSE}
total_counts <- read_csv("output/nci_dlbcl_unprocessed_counts.csv")
```

Duke data (raw reads estimated by Kallisto)
See [analysis/dukedata.Rmd](dukedata.html) for creation of DGE list from kallisto output files

See `/code/` for source code of Kallisto analysis from raw sequences

```{r eval=FALSE}
dgelist <- readRDS("data/temp.dgelist_edger.rds")
str(dgelist)
```

read in conversion table for gene IDs
```{r eval=FALSE}
gencode_gtf <- read_tsv("data/gencode.v22.primary_assembly.annotation.gtf.geneinfo")
```

#### Data cleaning

Convert GENCODE v29 to gene symbols for TIMER
```{r eval=FALSE}
# get expression matrix
expr_duke <- dgelist[[1]]
expr_duke[1:5, 1:5]
col__names <- colnames(expr_duke)
col_names_rep <- str_split(col__names, pattern = "\\.", simplify = TRUE) %>% 
  as_tibble() %>%
  pull(V1) %>%
  paste0("duke_", .) %>%
  print()
colnames(expr_duke) <- col_names_rep
expr_duke[1:5, 1:5]

# format gencode table for left join
geneID_split <- str_split(gencode_gtf$gene_id, pattern = "\\.", simplify = TRUE) %>% 
  as_tibble() %>%
  dplyr::select(geneID_split = V1)

gencode_gtf_split <- gencode_gtf %>% 
  bind_cols(geneID_split) %>%
  dplyr::select(gene_id_v22 = gene_id, geneID_split, everything()) %>%
  print()

# left join, select columns, reformat
expr_duke_df <- expr_duke %>%
  as.data.frame() %>%
  rownames_to_column(var = "geneID_split") %>%
  as_tibble() %>%
  left_join(gencode_gtf_split) %>%
  dplyr::select(gene_name, starts_with("duke")) %>%
  drop_na(gene_name) %>%
  as.data.frame()

# merge duplicate entries for genes (average expression)
nrow(expr_duke_df)
expr_duke_data <- as.matrix(expr_duke_df[, 2:772])
rownames(expr_duke_data) <- expr_duke_df[,1]
expr_duke_nodups <- avereps(expr_duke_data)
nrow(expr_duke_nodups)
expr_duke_nodups[1:5, 1:5]
```

Filter samples to match those included in the clustering analysis
```{r eval=FALSE}
duke_es <- readRDS("output/duke_expressionset.rds")
prior_duke_expr <- exprs(duke_es)
samplelist <- c("gene_symbol", colnames(prior_duke_expr))

expr_duke_decon <- expr_duke_nodups %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_symbol") %>%
  as_tibble() %>%
  dplyr::select(one_of(samplelist)) %>%
  column_to_rownames(var = "gene_symbol") %>%
  as.matrix()

dim(expr_duke_decon)
str(expr_duke_decon)
expr_duke_decon[1:5, 1:5]
```

### NCI

Read in NCI raw counts
```{r eval=FALSE}
total_counts <- read_csv("output/nci_dlbcl_unprocessed_counts.csv")
```

Convert gene id to gene symbol, eliminate duplicates
```{r eval=FALSE}
tot_counts_symbol <- total_counts %>%
  dplyr::rename(gene_id = gene) %>%
  left_join(gencode_gtf) %>%
  dplyr::select(gene_name, starts_with("DLBCL")) %>%
  drop_na(gene_name)

tot_counts_matrix <- tot_counts_symbol[, 2:482] %>%
  as.matrix() 
rownames(tot_counts_matrix) <- tot_counts_symbol$gene_name
 
dim(tot_counts_matrix)
str(tot_counts_matrix)
tot_counts_matrix[1:5, 1:5]

# eliminate duplicates by averaging counts
# merge duplicate entries for genes (average expression)
expr_nci_nodups <- avereps(tot_counts_matrix, ID = rownames(tot_counts_matrix))
nrow(expr_nci_nodups)
expr_nci_nodups[1:5, 1:5]
```

### Merge

There are more rows in the NCI dataset after removal of duplicates and missing gene symbols. Will keep only those rows present in both datasets to allow for fair comparison. 

```{r eval=FALSE}
nci_counts_df <- expr_nci_nodups %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_symbol") %>%
  as_tibble()

duke_counts_df <- expr_duke_nodups %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_symbol") %>%
  as_tibble()

merge_counts_df <- nci_counts_df %>%
  inner_join(duke_counts_df) %>%
  print()

## Write out for use in TIMER online application
merge_counts <- merge_counts_df %>%
  as.data.frame() %>%
  column_to_rownames(var = "gene_symbol") %>%
  as.matrix()

write.csv(merge_counts, "output/merged_raw_counts.csv")
```

Limit on TIMER upload to 50 MB, total size of CSV now is ~ 231 MB. Will need to split into parts. Leave a little overlap to correct for any batch effect (would not expect any due to per-sample analysis)

```{r eval=FALSE}
ncol(merge_counts) / 5

mc1 <- merge_counts[, 1:250]
rownames(mc1) <- rownames(merge_counts)
write.csv(mc1, "output/merged_raw_counts_1.csv")
mc2 <- merge_counts[, 240:500]
rownames(mc2) <- rownames(merge_counts)
write.csv(mc2, "output/merged_raw_counts_2.csv")
mc3 <- merge_counts[, 490:650]
rownames(mc3) <- rownames(merge_counts)
write.csv(mc3, "output/merged_raw_counts_3.csv")
mc4 <- merge_counts[, 640:800]
rownames(mc4) <- rownames(merge_counts)
write.csv(mc4, "output/merged_raw_counts_4.csv")
mc5 <- merge_counts[, 790:950]
rownames(mc5) <- rownames(merge_counts)
write.csv(mc5, "output/merged_raw_counts_5.csv")
mc6 <- merge_counts[, 940:1100]
rownames(mc6) <- rownames(merge_counts)
write.csv(mc6, "output/merged_raw_counts_6.csv")
mc7 <- merge_counts[, 1090:1252]
rownames(mc7) <- rownames(merge_counts)
write.csv(mc7, "output/merged_raw_counts_7.csv")
```

### TIMER results
```{r eval=FALSE}
mc1 <- read_csv("output/mc1_score_matrix.csv")
mc2 <- read_csv("output/mc2_score_matrix.csv")
mc3 <- read_csv("output/mc3_score_matrix.csv")
mc4 <- read_csv("output/mc4_score_matrix.csv")
mc5 <- read_csv("output/mc5_score_matrix.csv")
mc6 <- read_csv("output/mc6_score_matrix.csv")
mc7 <- read_csv("output/mc7_score_matrix.csv")

combined_scores <- mc1 %>%
  bind_rows(mc2, mc3, mc4, mc5, mc6, mc7) %>%
  arrange(sampleID) %>%
  as.data.frame()

combined_score_matrix_norm <- combined_scores[, 2:7] %>%
  as.matrix() %>%
  normalize.quantiles()

rownames(combined_score_matrix_norm) <- combined_scores %>% pull(sampleID)
# reviewed matrix using the View function in R studio - batch effects were persistent. Given the effort necessary to correct this issue, I will try another, newer deconvolution method that can process the whole matrix at once.
```

## MuSiC

https://www-nature-com.proxy.uchicago.edu/articles/nmeth.1830
Use non-transformed (raw counts) for deconvolution

http://xuranw.github.io/MuSiC/articles/MuSiC.html

```{r}
# install devtools if necessary
#install.packages('devtools')

# install the MuSiC package
#devtools::install_github('xuranw/MuSiC')

# load
#library(MuSiC)
```
