---
title: "Immune Landscape of DLBCL"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

## Introduction
Welcome to the computational lab notebook for our research project focused on profiling the immune microenvironment of DLBCL using gene expression data and multi-channel immunohistochemistry, then linking clusters of similar immune signatures with mutations in tumors.

[Slides containing work-in-progress figures](https://docs.google.com/presentation/d/1sKzira205EXmpdX3D1h6Hy_67gWcaE9-PLBPEcOIE5g/edit?usp=sharing)

## Background

The over-arching questions are:

1. Are there recurrent immune phenotypes in DLBCL?
2. What is the influence of tumor gene mutations on immune phenotypes in DLBCL?

The primary data sources are:

1. Duke dataset
    * From [Reddy et al Cell. 2017 Oct 5;171(2):481](https://www.ncbi.nlm.nih.gov/pubmed/28985567)
    * Hosted at the European Genome-phenome Archive [EGAD00001003600](https://www.ebi.ac.uk/ega/datasets/EGAD00001003600)
2. NCI dataset 
    * From [Schmitz et al N Engl J Med. 2018 Apr 12;378(15):1396](https://www.ncbi.nlm.nih.gov/pubmed/?term=29641966)  
    * Hosted at the Genomic Data Commons project ID [NCICCR-DLBCL](https://portal.gdc.cancer.gov/projects/NCICCR-DLBCL)

## Aim 1: Identify Recurrent Immune Phenotypes using RNAseq Data

### Prepare RNAseq data for analysis

All steps in data download, cleaning, and analysis are documented in R markdown notebooks listed below or in source code in `/code/` folder in github repo.

* [Download raw RNAseq counts in NCI lymphoma dataset from GDC](gdc_download.html)
* [Preprocessing NCI counts](preprocessing_count_data.html)
* Pseudoalign and count Duke RNAseq sequence data
  * Run on Gardner HPC cluster
  * Source code in `/code/`)
* [Preprocessing Duke counts](dukedata.html)
* [Expanded phenotype data for Duke and NCI datasets](ext_pheno.html)
* [Merge expression and associated clinical/metadata, batch correction](merge.html)

### Gene set variation analysis

#### First try at gene list collection:

Gene lists selected from work-in-progress analyis of TCGA macrophage signatures and from immune-related C2 "curated" gene sets from MSigDB. Also included T-cell inflamed signature from prior publications (Riyue Bao). No additional filtering was done after selection of candidate 49 gene lists. GSVA run on expression data from NCI only to see how well it recapitulates known _PD-L1_ amplified immune phenotype.

* [Data prep for GSVA](enrichment.html)
* GSVA analysis run on Gardner Supercomputer (HPC) Cluster at UChicago
  * Source code in `/code/`
* [Visualizing and clustering GSVA results](clustering_enrichment_scores.html)

There is poor separation, especially on PCA2, of the known immune phenotype _PD-L1_ amplified in NCI data. Contribution of individual gene sets to the PCA axes shows high concentration pointing towards (0,1) on a unit circle and few parallel to the PC2 axis. Will try to find additional gene sets that can separate immune phenotypes better.

#### Second and third gene list collections:

* [New gene set collections](newgs.html)

Downloaded and formatted 83 gene sets first published in Thorsson et al. Also found an additional 60 gene sets from recent publications, especially focused on recent single-cell RNAseq publications of tumor-infiltrating immune cells.

* [Visualizing and clustering GSVA results using recently published gene sets](visualizing_gsva_scores_2.html)
* [Visualizing and clustering GSVA results using Thorsson et al gene sets](visualizing_gsva_scores_3.html)
* [Visualizing and clustering GSVA results using a combination of recently published and Thorsson et al gene sets](visualizing_gsva_scores_4.html)
* [Visualizing and clustering cleaned GSVA results with B-cell and overlapping gene sets removed](visualizing_gsva_scores_5.html)

Both new collections of gene sets perform better than original MSigDB in separating known phenotype PD-L1, and have a broader distribution of vector angles in contributions of each gene set to the principal component axes. A four-cluster model was found to best fit the data in both cases, with 75% identical assignments and 87% Pearson concordance with cluster assignment treated as a numerical value. Several gene sets were removed from Thorsson et al and combined GSVA analyses due to multicollinearity. See [GSVA v3 notebook](visualizing_gsva_scores_3.html) and [GSVA v4 notebook](visualizing_gsva_scores_4.html) for details.

### GSVA Clustering

4-cluster model assignments made using HCPC on the first 10 principal components of GSVA v4 (combination of Thorsson and new gene set) scores. See [notebook]((visualizing_gsva_scores_4.html)) for details. 

Also tested sparse heirarchical clustering to reduce number of contributing gene sets via L1 (Lasso-like) mechanism of iteratively forcing coefficients of each variable toward 0 while optimizing the gap statistic. This resulted in selection of 35 gene sets which could explain a high fraction of the total variability, unfortunately these were predominantly T-cell-related gene sets which were already overrepresented in the original gene set list due to characteristics of T-cells and larger number of studies focused on characterizing T-cells. The overall clustering was poorer, and was essentially a gradient on these closely related T-cell, IFN-gamma axis. This model was discarded in favor of noise reduction with HCPC of the first 10 principal components as described above. 

* [Sparse hierarchical clustering](sparse_clustering.html)

tSNE and other dimensionality-reducing visualization and clustering techniques were compared to the PCA-based method used above, without any significant improvement in interpretibility. These other methods were not pursued; the PCA-based method was carried forward.

* [Visualization and clustering with tSNE](tsne.html)

### Describe the HCPC clusters

#### Differential gene expression

* [Differential gene expression by cluster](cluster_dge.html)
* [Differential gene set expression by cluster](enrichment.html)

#### Deconvolution of immune cell types

* [xCell deconvolution](deconvolution.html)
* [Creation of custom signature matrix](new_signature_matrix.Rmd)
* [CIBERSORTx deconvolution with custom matrix](new_deconvolution_results.html)

#### Clinical data and phenotype

* [Clinical data](table1.html)
* [Cytolytic activity score](cyt.html)

## Aim 2: Link Immune Phenotype Clusters to Tumor-Instrinsic Factors

### Mutation calls

* Somatic mutation pipeline run on HPC cluster. Source code and scripts in [/code/](https://github.com/mleukam/dlbcl_landscape/tree/master/code)

Input data:
 * Duke: somatic mutations in top 150 driver genes (supplemental data from Reddy et al Cell 2017)
 * NCI: non-silent somatic mutations from all samples (supplemental data from Schmitz et al NEJM 2018)

NCI data assessed for likelihood of driver status using MutSig2CV. Only those genes with q score < 0.1 included in downstream analysis.

Mutation data collapsed to one mutation per gene per patient for statistical tests of proportionality.

* [Identify significantly different mutations by cluster](assign_mutations.Rmd)

## Aim 3: Investigate Primary Samples

#### Derive predictor

Derive predictive algorithm using gene expression data for cluster assignment.

* [Prediction algorithm](predictor.html)
* [Multiparameter IHC results](mif.html)

##### Antibody panel:
| Marker | Cell type | Company | Product ID |
|--------|-----------|---------|------------|
| CD4 | Th and Treg  |         |            |
| CD8    | CTL       |         |            |
| FoxP3  | Treg      |         |            |
| CD68   | Macrophage|         |            |
| CD11c  |        DC |         |            |
| PAX5   | B-cell    |         |            |
| DAPI   | all       |         |            |

## Assorted Figures and Other Notebooks

* [ASH 2019 abstract figures](abstract_figure.html)
* [Draft of table 1](table1.html)
