---
title: "Immune Landscape of DLBCL"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

Welcome to the computational lab notebook for our research project focused on profiling the immune microenvironment of DLBCL using gene expression data and multi-channel immunohistochemistry, then linking clusters of similar immune signatures with mutations in tumors.

The over-arching questions are:

1. Are there recurrent immune phenotypes in DLBCL?
2. What is the influence of tumor gene mutations on immune phenotypes in DLBCL?

The primary data sources are:

1. Duke dataset
    * From [Reddy et al Cell. 2017 Oct 5;171(2):481](https://www.ncbi.nlm.nih.gov/pubmed/28985567)
    * Hosted at the European Genome-phenome Archive [EGAD00001003600](https://www.ebi.ac.uk/ega/datasets/EGAD00001003600)
2. NCI dataset 
    * From [Schmitz et al N Engl J Med. 2018 Apr 12;378(15):1396](https://www.ncbi.nlm.nih.gov/pubmed/?term=29641966)  
    * Hosted at the Genomic Data Commons project ID [NCICCR-DLBCL](https://portal.gdc.cancer.gov/projects/NCICCR-DLBCL)

Short-term to-do list:

* New gene lists - re-run GSVA on all samples and re-cluster
* Evaluate new gene lists in NCI data, PD-L1amp vs non
* Inverse correction for tumor purity?
* Download GDC mutation data
* Combined mutation calls pipeline

Major questions:

* How to account for malignant B-cells vs immune response B-cells?

## Aim 1: Identify Recurrent Immune Phenotypes using RNAseq Data

### Prepare RNAseq data for analysis

* [Download raw RNAseq counts in NCI lymphoma dataset from GDC](gdc_download.html)
* [Preprocessing NCI counts](preprocessing_count_data.html)
* Pseudoalign and count Duke RNAseq sequence data
  * Run on Gardner HPC cluster
  * Source code in `/code/`)
* [Preprocessing Duke counts](dukedata.html)
* [Expanded phenotype data for Duke and NCI datasets](ext_pheno.html)
* [Merge expression and associated clinical/metadata, batch correction](merge.html)

### Gene set variation analysis

#### First try at gene list collection:

Gene lists selected from work-in-progress analyis of TCGA macrophage signatures and from immune-related C2 "curated" gene sets from MSigDB. Also included T-cell inflamed signature from prior publications (Riyue Bao). No additional filtering was done after selection of candidate gene lists. GSVA run on expression data from NCI only to see how well it recapitulates known _PD-L1_ amplified immune phenotype.

* [GSVA](enrichment.html)
* [Clustering GSVA results](clustering_enrichment_scores.html)

There is poor separation, especially on PCA2, of the known immune phenotype _PD-L1_ amplified in NCI data.

![](assets/first_gsva_pca_samples.png)

This is because the gene sets are highly clustered in the same axis on the PCA analysis. There is too much co-variation among the immune gene sets I initially selected, and the overall analysis is not able to find meaningul differences in immune phenotype.

![](assets/first_gsva_pca_variables.png)

#### Gene set collection refinement:

We will need better-separating gene sets to proceed.

* [New gene set collection](newgs.html)
* [Visualizing and clustering new GSVA results](clustering_enrichment_scores.html)

Looking at the PCA plot of new immune gene signature enrichment of only PD-L1 amplified and non-amplified (NCI samples only), there is some separation on both PCA1 and PCA2 with a preponderance of samples in the southeast quadrant, though the effect is not marked.

![](assets/second_gsva_pca_individuals.png)

There are still many correlated gene sets in the NCI sample, but a better distribution than previously.

![](assets/second_gsva_pca_variables.png)

This PCA plot collapses immune gene set enrichment in all analyzed samples, and compares all NCI samples (enriched in ABC) to all Duke samples. There is no significant difference in distribution between the NCI and Duke data, and there is good scatter on the two PCA axis (though the contribution of PCA2 is fairly small).
![](assets/second_gsva_pca_individuals_dukenci.png)

The contribution of each of the gene sets to this distribution shows better distribution overall.
![](assets/second_gsva_pca_variables_dukenci.png)

A heatmap of gene set expression by sample is clustered with heirarchical clustering for both samples and gene sets
![](assets/second_gsva_heatmap.png)

## Aim 2: Link Immune Phenotype Clusters to Oncogenes

### Mutation calls

## Aim 3: Visualize the Immune Phenotype with Multi-Channel IHC

## Sub-aims:

#### Sub-aim 1: PD-L1 amplified cohort

#### Sub-aim 2: Additional gene signatures: cytolytic and double hit
