---
title: "Expanded search for immune gene sets"
author: "mleukam"
date: "2019-06-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Data Sources

#### Gene sets from prior publications of immune landscape

Gene sets referenced in [Thorssen et al Cell 2018](https://www.cell.com/immunity/fulltext/S1074-7613(18)30121-3?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1074761318301213%3Fshowall%3Dtrue#secsectitle0095) all obtained from [GDC website here](https://gdc.cancer.gov/about-data/publications/panimmune)

_Gene signatures were downloaded from GDC on 7/5/2019_

Original sources:
* [Wolf et al PLOS One 2014](https://www.ncbi.nlm.nih.gov/pubmed/24516633?dopt=Abstract). Gene sets obtained from xls [File S1](https://doi.org/10.1371/journal.pone.0088309.s001) at PLOS One website
  * Source of interferon signature used in Wolf: [Hu et al Bmc Genomics 2006]()
  * Source of B and T cell signatures used in Wolf: [Palmer et al BMC Genomics 2006](https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-7-115)
* [Chan et al 2013](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002920). Additional support for LYM gene set found  [here](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002920) in TCGA analysis by the same author.
* For the ICR gene sets, three citations are given: ([Galon et al Immunity 2013](https://www.sciencedirect.com/science/article/pii/S1074761313002926), [Bedognetti et al Curr Opin Immunol 2016](https://www.sciencedirect.com/science/article/pii/S0952791516300012#bib0355)), and [Hendrickx et al OncoImmunology 2017].

#### Additional gene sets considered 

The goal is to find new, single-cell derived immune gene sets. A broad literature search, focused on gene signatures linked to known immune phenotypes in oncology found the following sources:

_NB: all of the following signatures were downloaded on 7/1/2019_

* Simple 2 gene set of CYT score: `log-average of GZMA and PRF1 expression in TPMs` from [Rooney et al Cell 2015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4856474/). This has been associated with somatic mutations in 34 genes. 
* DHIT signature as a marker of immune exclusion [Ennishi et al JCO 2019](https://ascopubs.org/doi/full/10.1200/JCO.18.01583) -- ranked gene list in table S9
* Cell-type-specific immune signatures for oncology derived from tissue experiements in [Nirmal et al CIR 2018](http://cancerimmunolres.aacrjournals.org/content/6/11/1388.figures-only), downloaded as RDA file from github: https://github.com/ajitjohnson/imsig/blob/master/data/sig.rda 
* Differentially expressed genes used for immune-related clustering in breast cancer single-cell RNAseq experiments published by MSKCC group in [Azizi et al Cell 2018](https://www.cell.com/cell/fulltext/S0092-8674(18)30723-2#%20). Gene lists downloaded as xlsx file from table S4. 
* Detailed single cell T-cell profiling with RNAseq published in [Guo et al Nature Medicine 2018](https://www.nature.com/articles/s41591-018-0045-3). T-cell cluster representative genes as xls file (supplemental table 3) downloaded from journal site. 
* Treg signatures in NSCLC and CRC https://www.ncbi.nlm.nih.gov/pubmed/27851914 published in [De Simone et al Immunity 2016](https://www.sciencedirect.com/science/article/pii/S1074761316304320?via%3Dihub#app3) 
    * Description of tables here: https://ars.els-cdn.com/content/image/1-s2.0-S1074761316304320-mmc1.pdf. Gene signature in table S4 for Treg signatures and table S5 for immune checkpoint signatures. 
* T-cell inflammation signature from Spranger, Bao, Gajewski (UChicago).
* Macrophage signatures from TCGA macs project (UChicago).
  
## Setup

Clear the workspace
```{r}
rm(list = ls())
```

Load packages
```{r}
library("tidyverse")
library("readxl")
```

Read in Gencode v22 (GDC standard) conversion table with gene symbols

```{r}
gencode_gtf <- read_tsv("data/gencode.v22.primary_assembly.annotation.gtf.geneinfo")
gencode_gtf
```

## GSVA gene sets version 2 (my literature-derived gene sets)

#### Reformat gene sets for GSVA 
The final output needs to be a list of character vectors, each containing a set of Gencode v22 gene ids. Following the order of literature search and downloads above. 

CYT gene set
```{r}
# make list of gene symbols
# convert to gencode v22 gene id
cyt_gs = list(gene_name = c("GZMA", "PRF1")) %>%
  as_tibble() %>%
  left_join(gencode_gtf) %>%
  pull(gene_id) %>%
  print()
```

DHIT (Double hit-like) gene set
```{r}
# read in excel table
ds_JCO_18_01583_2 <- read_excel("data/ds_JCO.18.01583-2.xlsx", 
                                sheet = "Table S9", 
                                skip = 1)
ds_JCO_18_01583_2

# convert to gencode v22 gene id
dhit_gs <- ds_JCO_18_01583_2 %>%
  select(gene_name = `Gene name`) %>%
  left_join(gencode_gtf) %>%
  drop_na(gene_id) %>%
  pull(gene_id) %>%
  print()
```

imsig cell=specific gene set
```{r}
# read in RDA file from imsig github
load("data/sig.rda")
sig[1:10, 1:2]

# convert to named list of gene sets, each a 1-column tibble
sig_list <- as_tibble(sig) %>%
  nest(key = gene) %>%
  as.list() %>%
  print()
sig_dfs <- sig_list$data
names(sig_dfs) <- sig_list$cell
str(sig_dfs)
sig_dfs

sig_gs <- sig_dfs %>%
  map(function(df){dplyr::rename(df, gene_name = gene)}) %>%
  map(function(df){mutate(df, gene_name = as.character(gene_name))}) %>%
  map(function(df){left_join(df, gencode_gtf)}) %>%
  map(function(df){drop_na(df, gene_id)}) %>%
  map(function(df){pull(df, gene_id)}) %>%
  print()

names <- paste0("imsig_", 
                c("bcells", "proliferation", "interferon", "macrophages", "monocytes", "neutrophils", "nkcells", "plasmacells", "translation","tcells"),
                "_gs")
names
names(sig_gs) <- names
sig_gs
```

Azizi et al MSKCC single cell breast cancer profiling paper
```{r}
# read in curated gene sets from figure S4
mskcc_curated <- read_excel("data/azizi_curated.xlsx", 
    sheet = "Tcell Signatures", skip = 2)

mskcc_list <- mskcc_curated %>%
  as.list() %>%
  map(function(x){x[!is.na(x)]}) 

mskcc_gs <- mskcc_list %>%
  map(function(df){enframe(df)}) %>%
  map(function(df){dplyr::select(df, gene_name = value)}) %>%
  map(function(df){left_join(df, gencode_gtf)}) %>%
  map(function(df){drop_na(df, gene_id)}) %>%
  map(function(df){pull(df, gene_id)}) %>%
  print()

names <- paste0("mskcc_",
                c("treg", "cd8activation", "antiinflammatory", "anergy", "proinflammatory", "lipid mediators", "glycolysis", "tcacycle", "pentosephosphatepathway", "glycogenmetabolism", "glucosedeprivation", "m1macrophage", "m2macrophage", "cytolyticeffector", "type1ifn_response", "type2ifn_response", "hypoxia_regulated", "tcellterminal_diff", "cycle_g1s", "cycle_g2m"),
                "_gs")
names
names(mskcc_gs) <- names
mskcc_gs
```

Guo et al T-cell profiling in lung adenocarcinoma
```{r}
#--------------------------------------------
## CD8

beijing_cd8source <- read_excel("data/41591_2018_45_MOESM5_ESM.xlsx", 
    sheet = "CD8", skip = 1) %>%
  dplyr::select(gene_name = `Gene Symbol`, cluster = Cluster) %>%
  nest(key = gene_name) %>%
  drop_na(cluster)

beijing_cd8list <- beijing_cd8source %>%
  as.list() 
beijing_cd8list <- beijing_cd8list$data

names <- beijing_cd8source$cluster
names <- names %>%
  str_replace_all(., "-", "_") %>%
  str_to_lower()
names <- paste0("beijing_", names, "_gs") %>%
  print()
names(beijing_cd8list) <- names
beijing_cd8list

#--------------------------------------------
## CD4

beijing_cd4convsource <- read_excel("data/41591_2018_45_MOESM5_ESM.xlsx", 
    sheet = "CD4 Tconv", skip = 1)  %>%
  dplyr::select(gene_name = `Gene Symbol`, cluster = Cluster) %>%
  nest(key = gene_name) %>%
  drop_na(cluster) %>%
  print()

beijing_cd4convlist <- beijing_cd4convsource %>%
  as.list() 
beijing_cd4convlist <- beijing_cd4convlist$data

names <- beijing_cd4convsource$cluster
names <- names %>%
  str_replace_all(., "-", "_") %>%
  str_to_lower()
names <- paste0("beijing_", names, "_gs") %>%
  print()
names(beijing_cd4convlist) <- names
beijing_cd4convlist

#--------------------------------------------
## TREGS

beijing_treg_source <- read_excel("data/41591_2018_45_MOESM5_ESM.xlsx", 
    sheet = "CD4 Treg", skip = 1) %>%
  dplyr::select(gene_name = `Gene Symbol`, cluster = Cluster) %>%
  nest(key = gene_name) %>%
  drop_na(cluster) %>%
  print()

beijing_treg_list <- beijing_treg_source %>%
  as.list() 
beijing_treg_list <- beijing_treg_list$data

names <- beijing_treg_source$cluster
names <- names %>%
  str_replace_all(., "-", "_") %>%
  str_to_lower()
names <- paste0("beijing_", names, "_gs") %>%
  print()
names(beijing_treg_list) <- names
beijing_treg_list

#--------------------------------------------
## TREGS - TNFPRS +/-

beijing_tnfrsf9_source <- read_excel("data/41591_2018_45_MOESM5_ESM.xlsx", 
    sheet = "CD4-C9-CTLA4 (TNFRSF9+ or -)", 
    skip = 1) %>%
  dplyr::select(gene_name = `Gene Symbol`, cluster = Cluster) %>%
  nest(key = gene_name) %>%
  drop_na(cluster) %>%
  print()

beijing_tnfrsf9_list <- beijing_tnfrsf9_source %>%
  as.list() 
beijing_tnfrsf9_list <- beijing_tnfrsf9_list$data

names <- c("beijing_cd4_c9_ctla4_tnfrsf9_neg_gs", "beijing_cd4_c9_ctla4_tnfrsf9_pos_gs")
names(beijing_tnfrsf9_list) <- names
beijing_tnfrsf9_list

#--------------------------------------------
## Collect to a single list

beijing_list <- c(beijing_cd8list, beijing_cd4convlist, beijing_treg_list, beijing_tnfrsf9_list) %>%
  as.list() %>%
  map(function(df){pull(df, gene_name)})
str(beijing_list)

beijing_gs <- beijing_list %>%
  map(function(df){enframe(df)}) %>%
  map(function(df){dplyr::select(df, gene_name = value)}) %>%
  map(function(df){left_join(df, gencode_gtf)}) %>%
  map(function(df){drop_na(df, gene_id)}) %>%
  map(function(df){pull(df, gene_id)}) %>%
  print()
```

Treg and checkpoint signatures in NSCLC and CRC from De Simone et al Immunity 2016
```{r}
italy_checkpoint <- read_excel("data/1-s2.0-S1074761316304320-mmc2.xlsx", 
    sheet = "Table S5", skip = 1) %>%
  dplyr::select(1) %>%
  dplyr::rename(gene_name = 1) %>%
  separate(gene_name, into = c("gene_name", "alt_name"), sep = " ", remove = TRUE) %>%
  dplyr::select(gene_name) %>%
  as.list() %>%
  print()

italy_treg <- read_excel("data/1-s2.0-S1074761316304320-mmc2.xlsx", 
    sheet = "Table S4", skip = 5) %>% 
  dplyr::select(gene_name = 1) %>%
  as.list() %>%
  print()

italy_list <- c(italy_checkpoint, italy_treg)
names(italy_list) <- c("italy_checkpoint_gs", "italy_treg_gs")
italy_list

italy_gs <- italy_list %>%
  map(function(df){enframe(df)}) %>%
  map(function(df){dplyr::select(df, gene_name = value)}) %>%
  map(function(df){left_join(df, gencode_gtf)}) %>%
  map(function(df){drop_na(df, gene_id)}) %>%
  map(function(df){pull(df, gene_id)}) %>%
  print()

```

Import UChicago gene sets from prior studies
```{r}
uchicago_gs_source <- readRDS("data/gset_ids_complete.rds")
uchicago_gs_mac_up <- uchicago_gs_source[1:7]
uchicago_gs_tinflamed <- uchicago_gs_source[15]
uchicago_gs <- c(uchicago_gs_mac_up, uchicago_gs_tinflamed)
names(uchicago_gs) <- c("uchi_mac_c1_gs", "uchi_mac_c2_gs", "uchi_mac_c3_gs", "uchi_mac_c4_gs", "uchi_mac_c5_gs", "uchi_mac_c6_gs", "uchi_mac_c7_gs", "uchi_t_inflamed_gs")
uchicago_gs
```


#### Assemble list of gene sets
```{r}
gset_ids_complete_2 <- c(list("cyt_gs" = cyt_gs), 
                         list("dhit_gs" = dhit_gs),
                         sig_gs,
                         mskcc_gs,
                         beijing_gs,
                         italy_gs,
                         uchicago_gs)
str(gset_ids_complete_2)
```

#### Write out for GSVA on cluster
```{r}
saveRDS(gset_ids_complete_2, "output/gset_ids_complete_2.rds")
```

## GSVA gene sets version 3 (Thorsson et al)

Read in data
```{r}
thorsson_genesets <- read_excel("data/PanImmune_GeneSet_Definitions.xlsx", 
    sheet = "Genes")

# after review of data in spreadsheet viewer, the first 83 gene sets in column 2 of sheet setSources are the genesets used in Thorsson clustering
thorsson_genesets_names <- read_excel("data/PanImmune_GeneSet_Definitions.xlsx", 
    sheet = "setSources") %>% 
  pull(SetName) %>%
  .[1:83]

# convert to named list of gene sets, each a 1-column tibble
thorsson_genesets_list <- as_tibble(thorsson_genesets) %>%
  nest(key = Gene) %>%
  # keep only the gene sets 
  filter(SetName %in% thorsson_genesets_names) %>%
  as.list() %>%
  print()
thorsson_genesets_df <- thorsson_genesets_list$data
names(thorsson_genesets_df) <- thorsson_genesets_list$SetName
str(thorsson_genesets_df)
thorsson_genesets_df

thor_gs <- thorsson_genesets_df %>%
  map(function(df){dplyr::rename(df, gene_name = Gene)}) %>%
  map(function(df){mutate(df, gene_name = as.character(gene_name))}) %>%
  map(function(df){left_join(df, gencode_gtf)}) %>%
  map(function(df){drop_na(df, gene_id)}) %>%
  map(function(df){pull(df, gene_id)}) %>%
  print()

# reformat names
thornames <- names(thor_gs) %>%
  paste("thor_", .) %>%
  str_to_lower() %>% 
  str_replace_all(" ", "") %>%
  print()
names(thor_gs) <- thornames

str(thor_gs)
```

Missing data:
thor_gp2_immunetcellbcell_score
thor_gp11_immune_ifn
thor_cd103pos_cd103neg_ratio_25446897 -- is this one calculated?

Following links in Wolf et al shows immune ifn score is cited out to [Hu et al BMC genomics 2006](https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-7-96). Genes input manually from list in Figure 2.

```{r}
# note: hypothetical protein FLJ10700 was discontinued by NCBI in 2006
thor_gp11_immune_ifn_df <- c("IFI30", "IFITM1", "ISGF3G", "IFI6", "IFNA2", "STAT1", "ISG20", "CXCL9", "CASP1", "NMI", "LYN", "SOD2") %>%
  enframe() %>%
  dplyr::rename(gene_name = value) %>%
  dplyr::select(gene_name) %>%
  left_join(gencode_gtf) %>%
  drop_na(gene_id) %>%
  pull(gene_id) %>%
  print()

thor_gp11_immune_ifn_gs <- list("thor_gp11_immune_ifn" = thor_gp11_immune_ifn_df)
```

Following links in Wolf et al shows T cell B cell score is cited out to [Palmer et al BMC Genomics 2006](https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-7-115). Additional File 2.

```{r}
bandt_source <- read_excel("data/12864_2005_498_MOESM2_ESM.xls", 
    sheet = "Sheet1")

thor_gp2_immune_bcell <- bandt_source %>%
  dplyr::filter(`Cell Type` == "B") %>%
  dplyr::select(Symbol) %>%
  print()

thor_gp2_immune_tell <- bandt_source %>%
  dplyr::filter(`Cell Type` == "T") %>%
  dplyr::select(Symbol) %>%
  print()

thorsson_bandt_df <- list(thor_gp2_immune_bcell, thor_gp2_immune_tell)

thor_bandt_gs <- thorsson_bandt_df %>%
  map(function(df){dplyr::rename(df, gene_name = Symbol)}) %>%
  map(function(df){mutate(df, gene_name = as.character(gene_name))}) %>%
  map(function(df){left_join(df, gencode_gtf)}) %>%
  map(function(df){drop_na(df, gene_id)}) %>%
  map(function(df){pull(df, gene_id)}) 

names(thor_bandt_gs) <- c("thor_gp2_immune_bcell_score", "thor_gp2_immune_tcell_score")

thor_bandt_gs
```

#### Combine missing data back into gene set list
```{r}
thor_gs_filtered <- thor_gs %>%
  list_modify("thor_gp2_immunetcellbcell_score" = NULL) %>%
  list_modify("thor_gp11_immune_ifn" = NULL) %>%
  list_modify("thor_cd103pos_cd103neg_ratio_25446897" = NULL) %>%
  list_modify("thor_gp2_immune_bcell_score" = NULL) %>%
  list_modify("thor_gp2_immune_tcell_score" = NULL) %>%
  list_modify("thor_gp2_immune_bcell_score" = NULL) %>%
  list_modify("thor_gp2_immune_tcell_score" = NULL) %>%
  list_modify("thor_gp11_immune_ifn" = NULL)
  
thor_gs_complete <- c(thor_gs_filtered, thor_bandt_gs, thor_gp11_immune_ifn_gs)
str(thor_gs_complete)
length(thor_gs_complete)

```

## Write out for later use
```{r}
saveRDS(thor_gs_complete, "output/gset_ids_complete_3.rds")
```

## Combined gene sets for GSVA v4
New literature-based gene sets and gene sets from Thorsson et al. Immunity 2018.

```{r}
gset_ids_complete_4 <- c(gset_ids_complete_2, thor_gs_complete)
str(gset_ids_complete_4)
# 143 gene sets
saveRDS(gset_ids_complete_4, "output/gset_ids_complete_4.rds")
```
