---
title: "Preprocessing NCI Count Data from GDC"
author: "mleukam"
date: "2019-06-13"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Setup

Clear workspace
```{r}
rm(list = ls())
```

Load packages
```{r}
library("tidyverse")
library("edgeR")
library("limma")
library("Biobase")
```

Read in data
```{r}
total_counts <- read_csv("output/nci_dlbcl_unprocessed_counts.csv")
```

Read in lookup table for features, gencode v22 (used by GDC to label features)
```{r}
gencode_gtf <- read_tsv("data/gencode.v22.primary_assembly.annotation.gtf.geneinfo")
```

## Normalization, filtering and transformation

Preprocessing RNAseq count data following methods outlined here: https://f1000research.com/articles/5-1408/v3

#### Filter for protein coding genes
```{r}
# filter for protein coding genes
total_counts_prcode <- total_counts %>%
  dplyr::rename(gene_id = gene) %>%
  left_join(gencode_gtf, by = "gene_id") %>%
  dplyr::filter(gene_type == "protein_coding") %>%
  dplyr::select(-gene_name, -gene_type, -gene_status, -level, -havana_gene) %>%
  dplyr::select(gene_id, everything())
total_counts_prcode[1:5, 1:5]

nrow(total_counts)
nrow(total_counts_prcode)
```

#### Correct for library size: convert to CPM
```{r}
# normalize rows by log cpm using EdgeR
df_data <- total_counts_prcode %>% 
  dplyr::select(-gene_id) %>% as.matrix()
df_names <- total_counts_prcode %>% dplyr::select(gene_id) 
out_data <- cpm(df_data, log = FALSE) %>% as_tibble()
total_counts_prcode_cpm <- bind_cols(df_names, out_data)
```

#### Density plots of CPM values
```{r}
tidy_cpm <- total_counts_prcode_cpm %>% 
  gather(key = "sampleID", value = "intensity", -gene_id) 

dplot1 <- ggplot(tidy_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = "none") +
  xlim(-5, 100) +
  ggtitle("Density plot of CPM for NCI DLBCL")
dplot1

dplot2 <- ggplot(tidy_cpm, aes(log(intensity))) +
  geom_density() +
  theme(legend.position = "none") +
  xlim(-20, 20) +
  ggtitle("Density plot of log(CPM) for NCI DLBCL")
dplot2
```

#### Filter by expression levels
```{r}
# move gene names to rownames
totcounts_prcode_cpm_matrix <- total_counts_prcode_cpm %>%
  as.data.frame() %>%
  column_to_rownames(var = "gene_id")
totcounts_prcode_cpm_matrix[1:5, 1:5]
dim(totcounts_prcode_cpm_matrix)

# apply hard cutoffs 
# filter out genes that aren't at least expressed greater than 1 in at least half of cases
# set filter fraction
min_sm_frac = 0.5
filter_frac = min_sm_frac * ncol(totcounts_prcode_cpm_matrix)
filter_frac
total_cpm_stats <- data.frame(
  total = apply(totcounts_prcode_cpm_matrix, 
                1, 
               function(x) {sum(x > 1, na.rm = TRUE) } ))
keep <- which(total_cpm_stats$total >= filter_frac)

#---------------------
#### DECIDED NOT TO USE
## apply soft cutoffs
# set CPM threshold based on library size
# cpm_thres = 10 / min(colSums(totcounts_prcode_cpm_matrix)) * 10^6
# set minimum sample size
# keep <- rowSums(cpm(totcounts_prcode_cpm_matrix) > cpm_thres) # >= as.integer(ncol(totcounts_prcode_cpm_matrix) / 2)
#---------------------

# apply keep criteria and check results
dim(totcounts_prcode_cpm_matrix)
total_cpm_filtered = totcounts_prcode_cpm_matrix[keep,]
dim(total_cpm_filtered)
```

#### Density plots of filtered CPM values
```{r}
tidy_cpm_filtered <- total_cpm_filtered %>% 
  rownames_to_column(var = "gene_id") %>%
  gather(key = "sampleID", value = "intensity", -gene_id) 

dplot1 <- ggplot(tidy_cpm_filtered, aes(intensity)) +
  geom_density() + 
  theme(legend.position = "none") +
  xlim(-5, 100) +
  ggtitle("Density plot of CPM for NCI DLBCL")
dplot1

dplot2 <- ggplot(tidy_cpm_filtered, aes(log(intensity))) +
  geom_density() +
  theme(legend.position = "none") +
  xlim(-20, 20) +
  ggtitle("Density plot of log(CPM) for NCI DLBCL")
dplot2
```

#### Normalize gene expression distributions

Normalisation by the method of trimmed mean of M-values (TMM) is performed using the calcNormFactors function in edgeR. The normalisation factors calculated here are used as a scaling factor for the library sizes. 
```{r}
# get normalization factors
norm_factors <- calcNormFactors(total_cpm_filtered, method = "TMM")

# apply factor to each column
total_cpm_norm <- map2_dfc(total_cpm_filtered, norm_factors, `*`)
total_cpm_norm <- as.data.frame(total_cpm_norm)
rownames(total_cpm_norm) <- rownames(total_cpm_filtered)
total_cpm_norm[1:5, 1:5]
```

#### Log transformation
```{r}
# introduce offset to prevent -Inf
offset_total_cpm_norm <- total_cpm_norm + 0.5
offset_total_cpm_norm[1:5, 1:5]

# log transformation
total_log_cpm_filtered_norm <- log2(offset_total_cpm_norm)
total_log_cpm_filtered_norm[1:5, 1:5]
```

Final density plot
```{r}
total_log_cpm_filtered_tbl <- total_log_cpm_filtered_norm %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene") %>%
  as_tibble()

tidy_log_cpm <- total_log_cpm_filtered_tbl %>% 
  gather(key = "sampleID", value = "intensity", -gene)

dplot3 <- ggplot(tidy_log_cpm, aes(intensity)) +
  geom_density() +
  theme(legend.position = "none") +
  xlim(-20, 20) +
  ggtitle("Final density plot of log(CPM) for NCI DLBCL")
dplot3
```

## Add in phenotype data

Read in file and do some prelim cleaning
```{r}
nci_phenotype_data <- read_csv("output/nci_dlbcl_annotation.csv")
nci_phenotype_data

# clean up variables
nci_pheno_data <- nci_phenotype_data %>%
  dplyr::select(case_submitter_id,
                pdl1_status, 
                -sample, 
                case_id, 
                filename,
                -sample_id,
                -sample_submitter_id,
                project_id, 
                -sample_type_id,
                -time_between_excision_and_freezing,
                -oct_embedded, 
                -tumor_code_id,
                -intermediate_dimension,
                -is_ffpe,
                -pathology_report_uuid,
                -sample_type,
                -distance_normal_to_tumor,
                -biospecimen_anatomic_site,
                -state.x,
                -diagnosis_pathologically_confirmed,
                -current_weight,
                -composition,
                -time_between_clamping_and_freezing,
                -distributor_reference,
                -shortest_dimension,
                -method_of_sample_procurement,
                -tumor_code,
                -passage_count,
                -tissue_type,
                -biospecimen_laterality,
                -days_to_sample_procurement,
                -freezing_method,
                -preservation_method,
                -growth_rate,
                -days_to_collection,
                -catalog_reference,
                -initial_weight,
                -longest_dimension,
                -submitter_id,
                -morphology,
                days_to_last_follow_up,
                updated_datetime,
                tumor_stage,
                age_at_diagnosis,
                -created_datetime,
                -diagnosis_id,
                tissue_or_organ_of_origin,
                ann_arbor_clinical_stage,
                progression_or_recurrence,
                last_known_disease_status,
                primary_diagnosis,
                tumor_grade,
                -site_of_resection_or_biopsy,
                -demographic.updated_datetime,
                -demographic.created_datetime,
                demographic.gender,
                demographic.submitter_id,
                demographic.race,
                demographic.ethnicity,
                demographic.vital_status,
                -demographic.demographic_id,
                -id,
                -md5,
                size,
                -state.y
                ) %>%
  as.data.frame() %>%
  column_to_rownames(var = "case_submitter_id")

str(nci_pheno_data)
nci_pheno_data[1:5,1:5]
```

## Write out for later use
Following ExpressionSet vignette in Biobase package

```{r}
# make expression set
# first need variable annotation
v_metadata <- data.frame(
  labelDescription = colnames(nci_pheno_data),
  row.names = colnames(nci_pheno_data))

# make annotated phenoData
phenoData <- new("AnnotatedDataFrame", 
                 data = nci_pheno_data, 
                 varMetadata = v_metadata)
phenoData

# convert expression data to matrix format
nci_exprs <- total_log_cpm_filtered_norm %>% as.matrix()

# create global annotation
annotation <- as.character("RNAseq counts originally published by the Staudt lab at NCI in Schmitz et al NEJM 2018. Raw counts downloaded from GDC as HTseq counts, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed.")

# assemble expressionset
nci_es <- ExpressionSet(
  assayData = nci_exprs,
  phenoData = phenoData,
  annotation = annotation)
nci_es
```

```{r}
saveRDS(nci_es, "output/nci_expressionset.rds")
```

