---
title: "Visualizing results of GSVA with version 4 gene lists"
author: "mleukam"
date: "2019-07-11"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Setup

Clear the workspace
```{r}
rm(list = ls())
```

Load packages
```{r}
library("tidyverse")
library("factoextra")
library("FactoMineR")
library("cluster")
library("ggpubr")
library("ggsci")
library("Biobase")
library("ComplexHeatmap")
library("clustertend")
library("NbClust")
library("jcolors")
library("ggdark")
library("viridis")
```

```{r}
library("devtools")
install_github("cmartin/ggConvexHull")
library("ggConvexHull")
```

NB: need to transfer scripts to `/code/`:
* run_gsva4.pbs
* gsva4.sh
* gsva4.r

```{r}
gs_matrix <- readRDS("output/dlbcl_total_immune_gset_v4_results.rds")
gs_matrix[, 1:5]
str(gs_matrix)

combined_es <- readRDS("output/combined_expressionset.rds")
pheno_data <- pData(combined_es)
```

## Set up PD-L1 groups
```{r}
str(pheno_data)
summary(as.factor(pheno_data$pdl1_status))
pheno_data_tbl <- as.data.frame(pheno_data) %>%
  rownames_to_column(var = "sample_id") %>%
  as_tibble() %>%
  mutate(pdl1_status = replace_na(pdl1_status, "not_assessed")) %>%
  mutate(pdl1_status = ifelse(pdl1_status == "non-amplified", 
                              "non_amplified", pdl1_status))

summary(as.factor(pheno_data_tbl$pdl1_status))

group_table <- pheno_data_tbl %>%
  dplyr::select(sample_id, pdl1_status)

# merge group data into gsva matrix
pdl1_comb_exprs <- as.data.frame(t(gs_matrix)) %>%
  rownames_to_column(var = "sample_id") %>%
  as_tibble() %>%
  left_join(group_table) %>%
  dplyr::select(sample_id, pdl1_status, everything()) %>%
  as.data.frame() %>%
  column_to_rownames(var = "sample_id") %>%
  mutate(pdl1_status = as.factor(pdl1_status))
```

#### PCA plot of results
```{r}
# remove low amplified and samples not assessed
nrow(pdl1_comb_exprs)
pdl1_comb_exprs_filt <- pdl1_comb_exprs %>%
  dplyr::filter(pdl1_status %in% c("high_amplified", "non_amplified"))
nrow(pdl1_comb_exprs_filt)

# get PCA values
dlbcl_pca <- PCA(pdl1_comb_exprs_filt[-1], scale.unit = TRUE, graph = FALSE)

# get eigenvalues
eig_val <- get_eigenvalue(dlbcl_pca)
eig_val
fviz_eig(dlbcl_pca, addlabels = TRUE, ylim = c(0, 50))
ind <- get_pca_ind(dlbcl_pca)

pdl1_status <- pdl1_comb_exprs_filt$pdl1_status

# visualize PCA
fviz_pca_ind(dlbcl_pca,
             label = "none",
             habillage = pdl1_comb_exprs_filt$pdl1_status,
             palette = "simpsons",
             addEllipses = TRUE,
             ggtheme = dark_theme_gray()) +
  theme(axis.line = element_line(color = "#767676")) +
  geom_hline(yintercept = 0, color = "white", alpha = 0.3) +
  geom_vline(xintercept = 0, color = "white", alpha = 0.3) +
  ggtitle("Combined Thorsson and new gene sets \nPD-L1amp vs nonamp, NCI data only")

fviz_pca_var(dlbcl_pca, col.var = "contrib",
             gradient.cols = "tron",
             label = "none",
             ggtheme = dark_theme_gray()) + 
  theme(axis.line = element_line(color = "#767676")) +
  geom_hline(yintercept = 0, color = "#76767666") +
  geom_vline(xintercept = 0, color = "#76767666")
```

## Set up Duke/NCI groups
```{r}
str(pheno_data)
summary(as.factor(pheno_data$source))
group_table <- as.data.frame(pheno_data) %>%
  rownames_to_column(var = "sample_id") %>%
  dplyr::select(sample_id, source)

# merge group data into gsva matrix
dukenci_comb_exprs <- as.data.frame(t(gs_matrix)) %>%
  rownames_to_column(var = "sample_id") %>%
  as_tibble() %>%
  left_join(group_table) %>%
  dplyr::select(sample_id, source, everything()) %>%
  as.data.frame() %>%
  column_to_rownames(var = "sample_id") %>%
  mutate(source = as.factor(source))
```

#### PCA Plot of results
```{r}
# get PCA values
dlbcl_pca <- PCA(dukenci_comb_exprs[-1], scale.unit = TRUE, graph = FALSE)

# get eigenvalues
eig_val <- get_eigenvalue(dlbcl_pca)
eig_val
fviz_eig(dlbcl_pca, addlabels = TRUE, ylim = c(0, 50))
ind <- get_pca_ind(dlbcl_pca)

pdl1_status <- dukenci_comb_exprs$pdl1_status

# visualize PCA
fviz_pca_ind(dlbcl_pca,
             label = "none",
             habillage = dukenci_comb_exprs$source,
             palette = "tron",
             addEllipses = TRUE,
             ggtheme = dark_theme_gray()) +
  theme(axis.line = element_line(color = "#767676")) +
  geom_hline(yintercept = 0, color = "white", alpha = 0.3) +
  geom_vline(xintercept = 0, color = "white", alpha = 0.3) +
  ggtitle("Combined Thorsson and new gene sets \nDuke vs NCI data sources")

fviz_pca_var(dlbcl_pca, col.var = "contrib",
             gradient.cols = "tron",
             label = "none",
             ggtheme = dark_theme_gray()) +
  geom_hline(yintercept = 0, color = "white", alpha = 0.3) +
  geom_vline(xintercept = 0, color = "white", alpha = 0.3)

# Contributions of variables to PC1
fviz_contrib(dlbcl_pca, choice = "var", axes = 1, top = 15) + 
  dark_theme_classic() +
  theme(axis.text.x = element_text(angle = 80, hjust = 1))

# Contributions of variables to PC2
fviz_contrib(dlbcl_pca, choice = "var", axes = 2, top = 15) +
  dark_theme_classic() +
  theme(axis.text.x = element_text(angle = 80, hjust = 1))
```

#### Heatmap of raw results
```{r}
ht <- Heatmap(gs_matrix, name = "GSVA Score", 
                row_title = "Gene Sets",
                show_row_names = TRUE,
                column_title = paste0("GSVA scores for Duke and NCI samples"), 
                show_column_names = FALSE,
                column_title_side = "bottom")
ht
```

## Clustering

#### Determine suitability of data for clustering

First visualize similarity among observations with distance matrix
```{r}
# create pearson correlation matrix
sample_matrix <- t(gs_matrix)
dlbcl_dist <- get_dist(sample_matrix, 
                       stand = TRUE, 
                       method = "pearson")
fviz_dist(dlbcl_dist, show_labels = FALSE) +
  dark_theme_gray() +
  scale_fill_viridis() +
  theme(axis.ticks = element_blank(), axis.text = element_blank())
```

Get Hopkins statistic of clusterability
```{r}
# Compute Hopkins statistic for iris dataset
# Below 0.5 is considered "clusterable"
dlbcl_tend <- hopkins(gs_matrix, n = nrow(gs_matrix) - 1, header = FALSE)
dlbcl_tend
```

#### Filter out highly correlated gene sets
```{r}
gs_cor <- cor(gs_matrix, method = c("pearson"))
library(corrplot)
source("http://www.sthda.com/upload/rquery_cormat.r")

# view gene set correlation heatmap
cortables <- rquery.cormat(t(gs_matrix), graphType = "heatmap")

# reformat gene set correlation table and view
corflat <- rquery.cormat(t(gs_matrix), type = "flatten", graph = FALSE)
corflat_tbl <- corflat$r %>% as_tibble() %>% arrange(desc(cor)) %>% print(n = 30)
# 22 gene sets 0.96 correlation and above. 
removesets <- c("thor_module11_prolif_score", 
                "thor_ifn_21978456",
                "thor_mhc.i_19272155",
                "thor_t_cell_pca_16704732",
                "thor_icr_score",
                "beijing_cd8_c1_lef1_gs",
                "thor_minterferon_cluster_21214954",
                "thor_g_hla-dpa1",
                "thor_apm2",
                "thor_tcclassii_score",
                "thor_ifit3", 
                "beijing_cd4_c3_gnly_gs",
                "thor_b_cell_pca_16704732",
                "thor_interferon_cluster_21214954",
                "thor_interferon_19272155",
                "thor_immune_cell_cluster_21214954",
                "thor_lck_19272155",
                "thor_module4_tcellbcell_score")
# 11 gene sets meet the criteria
nrow(gs_matrix)
gs_matrix_filtered_df <- gs_matrix %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_set") %>%
  dplyr::filter(!gene_set %in% removesets)
nrow(gs_matrix_filtered_df)

sample_matrix_filtered <- gs_matrix_filtered_df %>%
  column_to_rownames(var = "gene_set") %>%
  as.matrix() %>%
  t()
sample_matrix_filtered[1:5, 1:5]

```

Repeat similarity among observations with distance matrix
```{r}
# create pearson correlation matrix
dlbcl_dist <- get_dist(sample_matrix_filtered, 
                       stand = TRUE, 
                       method = "pearson")

fviz_dist(dlbcl_dist, show_labels = FALSE) +
  dark_theme_gray() +
  scale_fill_viridis() +
  theme(axis.ticks = element_blank(), axis.text = element_blank())

```

Repeat Hopkins statistic of clusterability
```{r}
# Compute Hopkins statistic for iris dataset
# Below 0.5 is considered "clusterable"
dlbcl_tend <- hopkins(sample_matrix_filtered,
                      n = nrow(gs_matrix_filtered_df) - 1, 
                      header = FALSE)
dlbcl_tend
```

Determine optimal number of clusters

```{r}
# Method 1: Elbow Method Using FactoExtra 
fviz_nbclust(sample_matrix_filtered, 
             FUNcluster = kmeans, 
             method = "wss") 

# Method 2: Silhouette in FactoExtra
fviz_nbclust(sample_matrix_filtered, 
             kmeans, method = "silhouette") + 
  theme_classic()

# GAP STAT Using Facto Extra 
fviz_nbclust(sample_matrix_filtered, 
             kmeans, nstart = 25, 
             method = "gap_stat", 
             nboot = 500) + #nboot is # of bootstrap sample - must be included
  labs(subtitle = "Gap Statistic Method")

# summary method: NbClust
nb <- NbClust(sample_matrix_filtered, 
              distance = "euclidean", 
              min.nc = 2, 
              max.nc = 10, 
              method = "kmeans")
```

#### Clustering

Divide into groups with HCPC
```{r}
dlbcl_pca <- PCA(sample_matrix_filtered, ncp = 10, graph = FALSE)
dlbcl_hcpc <- HCPC(dlbcl_pca, min = 4, max = 20, graph.scale = "sqrt-inertia", graph = TRUE)
```

```{r}
fviz_dend(dlbcl_hcpc, 
          k = 4,
          palette = "nejm",
          show_labels = FALSE, 
          rect = TRUE,
          rect_border = "black",
          rect_lty = 3,
          rect_fill = FALSE, 
          lower_rect = -1,
          phylo_layout = "layout_with_drl"
          )
```

Visualize samples with cluster label on PCA plot
```{r}
fviz_cluster(dlbcl_hcpc,
             geom = "point",
             show.clust.cent = TRUE,  
             main = "Factor map", 
             ggtheme = dark_theme_gray()
             ) +
  scale_color_jcolors(palette = "pal3") +
  scale_fill_jcolors(palette = "pal3")
```

Compare cluster assignments at the individual level.

```{r}
gsva2_assgn <- readRDS("output/gsva_v2_4clust_assign.rds") %>%
  dplyr::rename(gsva2_clust = clust) %>%
  mutate(gsva2_clust = as.numeric(gsva2_clust)) %>%
  print()

gsva3_assgn <- readRDS("output/gsva_v3_4clust_assign.rds") %>%
  mutate(gsva3_clust = as.numeric(gsva3_clust)) %>%
  print()

gsva4_assgn <- dlbcl_hcpc$data.clust %>% 
  rownames_to_column(var = "sample_id") %>%
  as_tibble %>%
  dplyr::select(sample_id, gsva4_clust = clust) %>%
  mutate(gsva4_clust = as.numeric(gsva4_clust)) %>%
  print()

merge_assgn <- gsva2_assgn %>%
  left_join(gsva3_assgn) %>%
  left_join(gsva4_assgn) %>%
  print()

cor(merge_assgn$gsva2_clust, merge_assgn$gsva4_clust)
cor(merge_assgn$gsva3_clust, merge_assgn$gsva4_clust)

merge_assgn_24 <- merge_assgn %>% 
  mutate(concordance = ifelse(gsva2_clust == gsva4_clust, 1, 0))
table(merge_assgn_24$concordance)

merge_assgn_34 <- merge_assgn %>% 
  mutate(concordance = ifelse(gsva3_clust == gsva4_clust, 1, 0))
table(merge_assgn_34$concordance)
```

## Exploration and sanity checks

Data cleaning and formatting
```{r}
# Get dataframe with gene set score matrix by sample and cluster assignment as the last column
clust_assign <- dlbcl_hcpc$data.clust %>% 
  rownames_to_column(var = "sample_id") %>%
  as_tibble %>%
  print()

# save cluster assignments for comparison in other notebooks
gsva_v4_4clust_assign <- clust_assign %>%
  dplyr::select(sample_id, gsva2_clust = clust)
saveRDS(gsva_v4_4clust_assign, "output/gsva_v4_4clust_assign.rds")

# get phenotype data
pheno_data <- pData(combined_es) %>%
  rownames_to_column(var = "sample_id") %>%
  as_tibble() %>%
  print()

# combined tables
pheno_clust_data <- clust_assign %>%
  left_join(pheno_data, by = "sample_id") %>%
  dplyr::select(-case_id, -filename, -demographic.submitter_id, demographic.race, demographic.ethnicity, size) %>%
  as.data.frame() %>%
  column_to_rownames(var = "sample_id")
pheno_clust_data[1:10, 1:4]

# confirm last column of gene set expression matrix
pheno_clust_data[1:5,125:127]
ncol(pheno_clust_data)
# columns 1-125 are gene set expression matrix, columns 126-195 are phenotype data
```

Plots
Source for overriding shape change with habillage argument: https://github.com/kassambara/factoextra/issues/20

Instructions for convex hull geom: https://github.com/cmartin/ggConvexHull

```{r}
# convert NAs in pdl1_amp variable
pheno_clust_data$pdl1_status <- pheno_clust_data$pdl1_status %>%
  replace_na("not_assessed") %>%
  str_replace_all("non-amplified", "non_amplified") %>%
  as.factor()
summary(pheno_clust_data$pdl1_status)

# make variable for pdl1_amp vs all others
pheno_clust_data$amp_or_not <- fct_recode(pheno_clust_data$pdl1_status,
             "PDL1amp" = "high_amplified",
             "other" = "low_amplified",
             "other" = "non_amplified",
             "other" = "not_assessed")
summary(pheno_clust_data$amp_or_not)

# make factor for alpha
pheno_clust_data$alpha_var <- fct_recode(pheno_clust_data$amp_or_not,
             "1" = "PDL1amp",
             "0.2" = "other") %>%
  as.character() %>%
  as.numeric()
summary(pheno_clust_data$alpha_var)

# get colors
mypal = pal_tron()(6)

dlbcl_pca <- PCA(pheno_clust_data[, 1:125], graph = FALSE)
pdl1_pca <- fviz_pca_ind(dlbcl_pca,
             label = "none",
             habillage = pheno_clust_data$amp_or_not,
             pointshape = 19,
             alpha.ind = pheno_clust_data$alpha_var,
             show.legend.text = FALSE,
             palette = "ucsc",
             addEllipses = FALSE,
             ggtheme = dark_theme_gray()
             ) +
  geom_convexhull(alpha = 0.3, 
                  aes(fill = pheno_clust_data$clust),
                  show.legend = FALSE,
                  color = "#76767666") +
  ggtitle("Gene set PCA: PDL1 high amp by cluster")
pdl1_pca

# get counts by cluster
table(pheno_clust_data$amp_or_not, pheno_clust_data$clust)

# plot cytotoxic score on combined GSVA PCA plot
dlbcl_pca <- PCA(pheno_clust_data[, 1:60], graph = FALSE)
fviz_pca_ind(dlbcl_pca, 
             col.ind = pheno_clust_data$cyt_gs,
             label = "none",
             addEllipses = FALSE,
             ggtheme = dark_theme_gray()) +
  labs(color = "CYT GSVA score") +
  geom_convexhull(alpha = 0.0, 
                  aes(fill = pheno_clust_data$clust),
                  show.legend = FALSE,
                  color = "white") +
  ggtitle("Gene set PCA: CYT GSVA by cluster") +
  scale_color_viridis()

# plot cytotoxic score on combined GSVA PCA plot
dlbcl_pca <- PCA(pheno_clust_data[, 1:125], graph = FALSE)
fviz_pca_ind(dlbcl_pca, 
             col.ind = pheno_clust_data$dhit_gs,
             label = "none",
             addEllipses = FALSE,
             ggtheme = dark_theme_gray()) +
  labs(color = "DHIT GSVA score") +
  geom_convexhull(alpha = 0.0, 
                  aes(fill = pheno_clust_data$clust),
                  show.legend = FALSE,
                  color = "white") +
  ggtitle("Gene set PCA: DHIT GSVA score by cluster") +
  scale_color_viridis()

```

#### ABC vs GCB
```{r}
# check categorical variable
summary(as.factor(pheno_clust_data$abc_gcb_rna))

# get PCA values
dlbcl_pca <- PCA(pheno_clust_data[, 1:125], graph = FALSE)

# plot
fviz_pca_ind(dlbcl_pca, 
             col.ind = pheno_clust_data$abc_gcb_rna,
             pointshape = 19,
             label = "none",
             addEllipses = FALSE,
             ggtheme = dark_theme_gray()) +
  labs(color = "Cell of Origin") +
  geom_convexhull(alpha = 0.0, 
                  aes(fill = pheno_clust_data$clust),
                  show.legend = FALSE,
                  color = "white") +
  ggtitle("Gene set PCA: Cell of origin by cluster") +
  scale_color_jcolors(palette = "pal3")

fviz_pca_var(dlbcl_pca, col.var = "contrib",
             gradient.cols = "tron",
             label = "none",
             ggtheme = dark_theme_gray()) +
  geom_hline(yintercept = 0, color = "white", alpha = 0.3) +
  geom_vline(xintercept = 0, color = "white", alpha = 0.3)

# Contributions of variables to PC1
fviz_contrib(dlbcl_pca, choice = "var", axes = 1, top = 15) + 
  dark_theme_classic() +
  theme(axis.text.x = element_text(angle = 80, hjust = 1))

# Contributions of variables to PC2
fviz_contrib(dlbcl_pca, choice = "var", axes = 2, top = 15) +
  dark_theme_classic() +
  theme(axis.text.x = element_text(angle = 80, hjust = 1))

# Get counts
coo <- table(pheno_clust_data$abc_gcb_rna, pheno_clust_data$clust)
coo
```

Ratios
| COO   |  clust 1  |  clust 2  |  clust 3  |  clust 4  |
|-------|-----------|-----------|-----------|-----------|
| ABC   |    0.47   |   0.45    |   0.38    |   0.47    |
|-------|-----------|-----------|-----------|-----------|
| GCB   |    0.34   |    0.40   |   0.43    |   0.33    |
|-------|-----------|-----------|-----------|-----------|
| Unc.  |     0.19  |   0.15    |    0.19   |   0.20    |
|-------|-----------|-----------|-----------|-----------|

```{r}
coo_data <- pheno_clust_data %>%
  as.data.frame() %>%
  rownames_to_column(var = "sample_id") %>%
  dplyr::select(sample_id, abc_gcb_rna, clust) %>%
  mutate(abc_abc_rna = as.factor(abc_gcb_rna),
         clust = as.factor(clust)) %>%
  as_tibble()
coo_data

coo_sum <- coo_data %>%
  count(abc_gcb_rna, clust)

coo_bar <- ggplot(coo_sum, 
                  aes(x = abc_gcb_rna,
                      y = n,
                      fill = abc_gcb_rna)) +
  geom_bar(stat = "identity") +
  facet_wrap(~ clust) +
  dark_theme_gray() +
  scale_color_jcolors(palette = "pal3") +
  scale_fill_jcolors(palette = "pal3") +
  labs(title = "Counts of COO Subtype by Cluster",
       fill = "Cell of Origin",
       x = "",
       y = "count (n)")
  
coo_bar

```

## Survival analysis per cluster assignment
```{r}
library("survival")
library("survminer")

surv_clust <- Surv(time = pheno_clust_data$os_years, pheno_clust_data$os_status)

fit1 <- survfit(surv_clust ~ clust, 
                data = pheno_clust_data)
plot1 <- ggsurvplot(fit1, 
                    data = pheno_clust_data, 
                    pval = TRUE, 
                    pval.method = TRUE)
plot1$plot + 
  dark_theme_gray() +
  scale_color_jcolors(palette = "pal3")


fit2 <- survfit(surv_clust ~ clust + abc_gcb_rna, 
                data = pheno_clust_data)

fit2_sum <- surv_summary(fit2)

plot2 <- ggsurvplot(fit2, data = pheno_clust_data, pval = FALSE)
plot2$plot + 
  facet_wrap(~ clust) + 
  dark_theme_gray() +
  scale_color_viridis(discrete = TRUE)
```

```{r}
## Write out new expressionset
pheno_d <- pheno_clust_data
pheno_d[1:5, 1:5]

v_metadata <- data.frame(
  labelDescription = colnames(pheno_d),
  row.names = colnames(pheno_d) )
  
phenoData <- new("AnnotatedDataFrame", 
                 data = pheno_d, 
                 varMetadata = v_metadata)

exprs_matrix <- exprs(combined_es)
str(exprs_matrix)

annotation <- as.character("Duke RNAseq counts derived from raw sequence data published by the Dave lab at Duke in Reddy et al Cell 2017. Quantified with Kallisto with reverse strand parameter, collected with tximport, filtered for only protein-coding genes, selected for cases with expression in >12,000 genes, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed. Batch corrected using ComBat by sequencing lane. NCI RNAseq counts originally published by the Staudt lab at NCI in Schmitz et al NEJM 2018. Raw counts downloaded from GDC as HTseq counts, converted to CPM, filtered out genes expressed less than 1 in half or more of cases, TMM normalized and log2-transformed. Batch corrected for NCI vs Duke source using ComBat. 125-gene GSVA v4 gene set expression matrix in columns 1:125 of pData, columns 126:197 are phenotype data including 4-cluster model assignments in column `clust`")

# assemble expressionset
combined_es_4cluster <- ExpressionSet(
  assayData = exprs_matrix,
  phenoData = phenoData,
  annotation = annotation)
combined_es_4cluster

# write out new expressionset
saveRDS(combined_es_4cluster, "output/combined_es_gsva4_4cluster.rds")
```


```
